<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7.7 Chemistry-based Sufficient Similarity Analysis | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="7.7 Chemistry-based Sufficient Similarity Analysis | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="7.7 Chemistry-based Sufficient Similarity Analysis | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7.7 Chemistry-based Sufficient Similarity Analysis | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle Roell, Lauren Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-viewing-1.html"/>
<link rel="next" href="creating-a-heat-map-of-the-mixtures-chemistry-data.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TAME Toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#background"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#objectives"><i class="fa fa-check"></i>Objectives</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#module-development-overview"><i class="fa fa-check"></i>Module Development Overview</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#concluding-remarks"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#funding"><i class="fa fa-check"></i>Funding</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#author-affiliations"><i class="fa fa-check"></i>Author Affiliations</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html"><i class="fa fa-check"></i><b>1</b> Introduction to Coding in R</a><ul>
<li class="chapter" data-level="1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html"><i class="fa fa-check"></i><b>1.1</b> General Introduction and Installation of R and RStudio</a><ul>
<li class="chapter" data-level="1.1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html#what-is-r"><i class="fa fa-check"></i><b>1.1.1</b> What is R?</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="downloading-and-installing-r.html"><a href="downloading-and-installing-r.html"><i class="fa fa-check"></i><b>1.2</b> Downloading and Installing R</a></li>
<li class="chapter" data-level="1.3" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html"><i class="fa fa-check"></i><b>1.3</b> Downloading and Installing R Studio</a></li>
<li class="chapter" data-level="1.4" data-path="introduction-to-r-packages.html"><a href="introduction-to-r-packages.html"><i class="fa fa-check"></i><b>1.4</b> Introduction to R Packages</a><ul>
<li class="chapter" data-level="1.4.1" data-path="introduction-to-r-packages.html"><a href="introduction-to-r-packages.html#downloadinginstalling-r-packages"><i class="fa fa-check"></i><b>1.4.1</b> Downloading/Installing R Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="scripting-basics.html"><a href="scripting-basics.html"><i class="fa fa-check"></i><b>1.5</b> Scripting Basics</a><ul>
<li class="chapter" data-level="1.5.1" data-path="scripting-basics.html"><a href="scripting-basics.html#comments"><i class="fa fa-check"></i><b>1.5.1</b> Comments</a></li>
<li class="chapter" data-level="1.5.2" data-path="scripting-basics.html"><a href="scripting-basics.html#starting-code"><i class="fa fa-check"></i><b>1.5.2</b> Starting Code</a></li>
<li class="chapter" data-level="1.5.3" data-path="scripting-basics.html"><a href="scripting-basics.html#setting-your-working-directory"><i class="fa fa-check"></i><b>1.5.3</b> Setting Your Working Directory</a></li>
<li class="chapter" data-level="1.5.4" data-path="scripting-basics.html"><a href="scripting-basics.html#importing-files"><i class="fa fa-check"></i><b>1.5.4</b> Importing Files</a></li>
<li class="chapter" data-level="1.5.5" data-path="scripting-basics.html"><a href="scripting-basics.html#viewing-data"><i class="fa fa-check"></i><b>1.5.5</b> Viewing Data</a></li>
<li class="chapter" data-level="1.5.6" data-path="scripting-basics.html"><a href="scripting-basics.html#exporting-data"><i class="fa fa-check"></i><b>1.5.6</b> Exporting Data</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="concluding-remarks-1.html"><a href="concluding-remarks-1.html"><i class="fa fa-check"></i><b>1.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-organization-basics.html"><a href="data-organization-basics.html"><i class="fa fa-check"></i><b>2</b> Data Organization Basics</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html"><i class="fa fa-check"></i><b>2.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="2.1.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#set-your-working-directory-1"><i class="fa fa-check"></i><b>2.1.1</b> Set your working directory</a></li>
<li class="chapter" data-level="2.1.2" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#importing-example-datasets"><i class="fa fa-check"></i><b>2.1.2</b> Importing example datasets</a></li>
<li class="chapter" data-level="2.1.3" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#viewing-example-datasets"><i class="fa fa-check"></i><b>2.1.3</b> Viewing example datasets</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="merging-data-using-base-r-syntax.html"><a href="merging-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.2</b> Merging Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.3" data-path="filtering-and-subsetting-data-using-base-r-syntax.html"><a href="filtering-and-subsetting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.3</b> Filtering and Subsetting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.4" data-path="melting-and-casting-data-using-base-r-syntax.html"><a href="melting-and-casting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.4</b> Melting and Casting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.5" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html"><i class="fa fa-check"></i><b>2.5</b> Introduction to Tidyverse</a><ul>
<li class="chapter" data-level="2.5.1" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#downloading-and-loading-the-tidyverse-package"><i class="fa fa-check"></i><b>2.5.1</b> Downloading and Loading the Tidyverse Package</a></li>
<li class="chapter" data-level="2.5.2" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#merging-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.2</b> Merging Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.3" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#filtering-and-subsetting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.3</b> Filtering and Subsetting Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.4" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#melting-and-casting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.4</b> Melting and Casting Data using Tidyverse Syntax</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="concluding-remarks-2.html"><a href="concluding-remarks-2.html"><i class="fa fa-check"></i><b>2.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html"><i class="fa fa-check"></i><b>3</b> Finding and Visualizing Data Trend</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html"><i class="fa fa-check"></i><b>3.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="3.1.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html#script-preparations"><i class="fa fa-check"></i><b>3.1.1</b> Script Preparations</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html"><i class="fa fa-check"></i><b>3.2</b> Basic Statistical Tests and Visualizations of Data Trends</a><ul>
<li class="chapter" data-level="3.2.1" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#visualize-and-test-data-for-normality"><i class="fa fa-check"></i><b>3.2.1</b> Visualize and Test Data for Normality</a></li>
<li class="chapter" data-level="3.2.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i><b>3.2.2</b> Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="3.2.3" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i><b>3.2.3</b> Two-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><i class="fa fa-check"></i><b>3.3</b> Regression Modeling and Visualization: Linear and Logistic Regressions</a><ul>
<li class="chapter" data-level="3.3.1" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#linear-regression"><i class="fa fa-check"></i><b>3.3.1</b> Linear Regression</a></li>
<li class="chapter" data-level="3.3.2" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#logistic-regression"><i class="fa fa-check"></i><b>3.3.2</b> Logistic Regression</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="concluding-remarks-3.html"><a href="concluding-remarks-3.html"><i class="fa fa-check"></i><b>3.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html"><i class="fa fa-check"></i><b>4</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-training-module-2.html"><a href="introduction-to-training-module-2.html"><i class="fa fa-check"></i><b>4.1</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="4.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html"><i class="fa fa-check"></i><b>4.2</b> Script Preparations</a><ul>
<li class="chapter" data-level="4.2.1" data-path="script-preparations-1.html"><a href="script-preparations-1.html#formatting-dataframes-for-downstream-visualization-code"><i class="fa fa-check"></i><b>4.2.1</b> Formatting dataframes for downstream visualization code</a></li>
<li class="chapter" data-level="4.2.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html#scaling-dataframes-for-downstream-data-visualizations"><i class="fa fa-check"></i><b>4.2.2</b> Scaling dataframes for downstream data visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html"><i class="fa fa-check"></i><b>4.3</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.3.1" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#density-plot-visualizations"><i class="fa fa-check"></i><b>4.3.1</b> Density Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.2" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#ggally-visualizations"><i class="fa fa-check"></i><b>4.3.2</b> GGally Visualizations</a></li>
<li class="chapter" data-level="4.3.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#boxplot-visualizations"><i class="fa fa-check"></i><b>4.3.3</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="4.3.4" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#correlation-plot-visualizations"><i class="fa fa-check"></i><b>4.3.4</b> Correlation Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.5" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#hierarchical-clustering-visualizations"><i class="fa fa-check"></i><b>4.3.5</b> Hierarchical Clustering Visualizations</a></li>
<li class="chapter" data-level="4.3.6" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#heatmap-visualizations"><i class="fa fa-check"></i><b>4.3.6</b> Heatmap Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="concluding-remarks-4.html"><a href="concluding-remarks-4.html"><i class="fa fa-check"></i><b>4.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i><b>5</b> FAIR Data Management Practices</a><ul>
<li class="chapter" data-level="5.1" data-path="background-on-training-module.html"><a href="background-on-training-module.html"><i class="fa fa-check"></i><b>5.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="5.2" data-path="training-modules-questions.html"><a href="training-modules-questions.html"><i class="fa fa-check"></i><b>5.2</b> Training Module’s <strong>Questions</strong></a></li>
<li class="chapter" data-level="5.3" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html"><i class="fa fa-check"></i><b>5.3</b> Introduction to FAIR</a><ul>
<li class="chapter" data-level="5.3.1" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#with-this-background-we-can-now-answer-question-1"><i class="fa fa-check"></i><b>5.3.1</b> With this background, we can now answer <strong>Question 1</strong>:</a></li>
<li class="chapter" data-level="5.3.2" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#we-can-also-answer-question-2"><i class="fa fa-check"></i><b>5.3.2</b> We can also answer <strong>Question 2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="breaking-down-fair-letter-by-letter.html"><a href="breaking-down-fair-letter-by-letter.html"><i class="fa fa-check"></i><b>5.4</b> Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="5.5" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html"><i class="fa fa-check"></i><b>5.5</b> F (Findable) in FAIR</a><ul>
<li class="chapter" data-level="5.5.1" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html#with-this-information-can-can-now-answer-question-3"><i class="fa fa-check"></i><b>5.5.1</b> With this information, can can now answer <strong>Question 3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="a-accessible-in-fair.html"><a href="a-accessible-in-fair.html"><i class="fa fa-check"></i><b>5.6</b> A (Accessible) in FAIR</a></li>
<li class="chapter" data-level="5.7" data-path="i-interoperable-in-fair.html"><a href="i-interoperable-in-fair.html"><i class="fa fa-check"></i><b>5.7</b> I (Interoperable) in FAIR</a></li>
<li class="chapter" data-level="5.8" data-path="r-reusable-in-fair.html"><a href="r-reusable-in-fair.html"><i class="fa fa-check"></i><b>5.8</b> R (Reusable) in FAIR</a></li>
<li class="chapter" data-level="5.9" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html"><i class="fa fa-check"></i><b>5.9</b> What does this mean to you?</a><ul>
<li class="chapter" data-level="5.9.1" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html#with-this-information-can-can-now-answer-question-4"><i class="fa fa-check"></i><b>5.9.1</b> With this information, can can now answer <strong>Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.10" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html"><i class="fa fa-check"></i><b>5.10</b> Data Repositories for Sharing of Data</a><ul>
<li class="chapter" data-level="5.10.1" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html#with-this-information-can-can-now-answer-question-5"><i class="fa fa-check"></i><b>5.10.1</b> With this information, can can now answer <strong>Question 5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html"><i class="fa fa-check"></i><b>5.11</b> Helpful Resources on FAIR</a><ul>
<li class="chapter" data-level="5.11.1" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-training-resources-on-fair"><i class="fa fa-check"></i><b>5.11.1</b> Additional Training Resources on FAIR</a></li>
<li class="chapter" data-level="5.11.2" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-examples-and-documents-on-fair"><i class="fa fa-check"></i><b>5.11.2</b> Additional Examples and Documents on FAIR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="6" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i><b>6</b> Machine Learning and Predictive Modeling for Environmental Health</a><ul>
<li class="chapter" data-level="6.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html"><i class="fa fa-check"></i><b>6.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="6.1.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#introduction-to-machine-learning-ml-and-predictive-modeling"><i class="fa fa-check"></i><b>6.1.1</b> Introduction to Machine Learning (ML) and Predictive Modeling</a></li>
<li class="chapter" data-level="6.1.2" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#defining-predictive-modeling-in-the-context-of-toxicology-and-environmental-health"><i class="fa fa-check"></i><b>6.1.2</b> Defining predictive modeling in the context of toxicology and environmental health</a></li>
<li class="chapter" data-level="6.1.3" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#distinguish-between-machine-learning-ml-and-traditional-statistical-methods"><i class="fa fa-check"></i><b>6.1.3</b> Distinguish between machine learning (ML) and traditional statistical methods</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="introduction-to-training-module-3.html"><a href="introduction-to-training-module-3.html"><i class="fa fa-check"></i><b>6.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="6.3" data-path="training-modules-environmental-health-questions.html"><a href="training-modules-environmental-health-questions.html"><i class="fa fa-check"></i><b>6.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="6.4" data-path="script-preparations-2.html"><a href="script-preparations-2.html"><i class="fa fa-check"></i><b>6.4</b> Script Preparations</a></li>
<li class="chapter" data-level="6.5" data-path="loading-the-example-dataset.html"><a href="loading-the-example-dataset.html"><i class="fa fa-check"></i><b>6.5</b> Loading the Example Dataset</a></li>
<li class="chapter" data-level="6.6" data-path="data-viewing.html"><a href="data-viewing.html"><i class="fa fa-check"></i><b>6.6</b> Data Viewing</a></li>
<li class="chapter" data-level="6.7" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html"><i class="fa fa-check"></i><b>6.7</b> Evaluating the Original Physicochemical Property Data across Substances</a><ul>
<li class="chapter" data-level="6.7.1" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html#with-these-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>6.7.1</b> With these, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html"><i class="fa fa-check"></i><b>6.8</b> Identifying Clusters of Chemicals through k-means</a><ul>
<li class="chapter" data-level="6.8.1" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html#with-this-we-can-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>6.8.1</b> With this, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html"><i class="fa fa-check"></i><b>6.9</b> Principal Component Analysis (PCA)</a><ul>
<li class="chapter" data-level="6.9.1" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-these-results-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>6.9.1</b> With these results, we can answer <strong>Environmental Health Question #3</strong>:</a></li>
<li class="chapter" data-level="6.9.2" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>6.9.2</b> With this, we can answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="6.9.3" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-5"><i class="fa fa-check"></i><b>6.9.3</b> With this, we can answer <strong>Environmental Health Question #5</strong>:</a></li>
<li class="chapter" data-level="6.9.4" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#we-can-also-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>6.9.4</b> We can also answer <strong>Environmental Health Question #6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="concluding-remarks-5.html"><a href="concluding-remarks-5.html"><i class="fa fa-check"></i><b>6.10</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html"><i class="fa fa-check"></i><b>7</b> Mixtures Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html"><i class="fa fa-check"></i><b>7.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="7.1.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html#introduction-to-mixtures-toxicology-and-in-silico-modeling-to-address-mixtures"><i class="fa fa-check"></i><b>7.1.1</b> Introduction to Mixtures Toxicology and <em>In Silico</em> Modeling to Address Mixtures</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="introduction-to-training-module-4.html"><a href="introduction-to-training-module-4.html"><i class="fa fa-check"></i><b>7.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="7.3" data-path="training-modules-environmental-health-questions-1.html"><a href="training-modules-environmental-health-questions-1.html"><i class="fa fa-check"></i><b>7.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="7.4" data-path="script-preparations-3.html"><a href="script-preparations-3.html"><i class="fa fa-check"></i><b>7.4</b> Script Preparations</a></li>
<li class="chapter" data-level="7.5" data-path="loading-the-example-ginkgo-biloba-datasets.html"><a href="loading-the-example-ginkgo-biloba-datasets.html"><i class="fa fa-check"></i><b>7.5</b> Loading the Example <em>Ginkgo biloba</em> Datasets</a></li>
<li class="chapter" data-level="7.6" data-path="data-viewing-1.html"><a href="data-viewing-1.html"><i class="fa fa-check"></i><b>7.6</b> Data Viewing</a><ul>
<li class="chapter" data-level="7.6.1" data-path="data-viewing-1.html"><a href="data-viewing-1.html#lets-also-see-what-kind-of-data-are-organized-within-the-datasets"><i class="fa fa-check"></i><b>7.6.1</b> Let’s also see what kind of data are organized within the datasets</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>7.7</b> Chemistry-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="7.7.1" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-chemical-profiles"><i class="fa fa-check"></i><b>7.7.1</b> This plot tells us a lot about sample groupings based on chemical profiles!</a></li>
<li class="chapter" data-level="7.7.2" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-questions-1-2"><i class="fa fa-check"></i><b>7.7.2</b> With this, we can answer <strong>Environmental Health Questions 1-2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><i class="fa fa-check"></i><b>7.8</b> Creating a Heat Map of the Mixtures Chemistry Data</a><ul>
<li class="chapter" data-level="7.8.1" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#this-plot-tells-us-a-lot-about-the-individual-chemicals-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>7.8.1</b> This plot tells us a lot about the individual chemicals that differentiate the sample groupings</a></li>
<li class="chapter" data-level="7.8.2" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-this-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>7.8.2</b> With this, we can answer <strong>Environmental Health Question 3</strong>:</a></li>
<li class="chapter" data-level="7.8.3" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#lets-now-revisit-the-pca-plot"><i class="fa fa-check"></i><b>7.8.3</b> Let’s now revisit the PCA plot</a></li>
<li class="chapter" data-level="7.8.4" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-these-plots-side-by-side-we-can-now-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>7.8.4</b> With these plots, side-by-side, we can now answer <strong>Environmental Health Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>7.9</b> Toxicity-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="7.9.1" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-toxicity-profiles"><i class="fa fa-check"></i><b>7.9.1</b> This plot tells us a lot about sample groupings based on toxicity profiles!</a></li>
<li class="chapter" data-level="7.9.2" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-question-5-1"><i class="fa fa-check"></i><b>7.9.2</b> With this, we can answer <em>Environmental Health Question 5</em>:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><i class="fa fa-check"></i><b>7.10</b> Creating a Heat Map of the Mixtures Toxicity Data</a><ul>
<li class="chapter" data-level="7.10.1" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#this-plot-tells-us-a-lot-about-the-individual-genes-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>7.10.1</b> This plot tells us a lot about the individual genes that differentiate the sample groupings</a></li>
<li class="chapter" data-level="7.10.2" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#with-this-we-can-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>7.10.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><i class="fa fa-check"></i><b>7.11</b> Comparing Results from the Chemistry vs. Toxicity Sufficient Similarity Analyses</a><ul>
<li class="chapter" data-level="7.11.1" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html#visual-representation-of-some-of-the-major-grouping-similarities-vs-differences"><i class="fa fa-check"></i><b>7.11.1</b> Visual representation of some of the major grouping similarities vs differences</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="these-plots-can-help-us-answer-environmental-health-question-7.html"><a href="these-plots-can-help-us-answer-environmental-health-question-7.html"><i class="fa fa-check"></i><b>7.12</b> These plots can help us answer <strong>Environmental Health Question 7</strong>:</a></li>
<li class="chapter" data-level="7.13" data-path="concluding-remarks-6.html"><a href="concluding-remarks-6.html"><i class="fa fa-check"></i><b>7.13</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html"><i class="fa fa-check"></i><b>8</b> -Omics Analyses and Systems Biology</a><ul>
<li class="chapter" data-level="8.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html"><i class="fa fa-check"></i><b>8.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="8.1.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i><b>8.1.1</b> Introduction to the Field of “-Omics”</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="introduction-to-the-omics-endpoint-transcriptomics.html"><a href="introduction-to-the-omics-endpoint-transcriptomics.html"><i class="fa fa-check"></i><b>8.2</b> Introduction to the -Omics Endpoint, Transcriptomics</a></li>
<li class="chapter" data-level="8.3" data-path="introduction-to-training-module-5.html"><a href="introduction-to-training-module-5.html"><i class="fa fa-check"></i><b>8.3</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="8.4" data-path="training-modules-environmental-health-questions-2.html"><a href="training-modules-environmental-health-questions-2.html"><i class="fa fa-check"></i><b>8.4</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="8.5" data-path="script-preparations-4.html"><a href="script-preparations-4.html"><i class="fa fa-check"></i><b>8.5</b> Script Preparations</a></li>
<li class="chapter" data-level="8.6" data-path="loading-the-example-transcriptomic-dataset-and-metadata.html"><a href="loading-the-example-transcriptomic-dataset-and-metadata.html"><i class="fa fa-check"></i><b>8.6</b> Loading the Example Transcriptomic Dataset and Metadata</a></li>
<li class="chapter" data-level="8.7" data-path="data-viewing-2.html"><a href="data-viewing-2.html"><i class="fa fa-check"></i><b>8.7</b> Data Viewing</a><ul>
<li class="chapter" data-level="8.7.1" data-path="data-viewing-2.html"><a href="data-viewing-2.html#with-this-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>8.7.1</b> With this, we can answer <strong>Environmental Health Question 1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="formatting-data-for-downstream-statistics.html"><a href="formatting-data-for-downstream-statistics.html"><i class="fa fa-check"></i><b>8.8</b> Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="8.9" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html"><i class="fa fa-check"></i><b>8.9</b> Transcriptomics Data QA/QC</a><ul>
<li class="chapter" data-level="8.9.1" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#background-filter"><i class="fa fa-check"></i><b>8.9.1</b> Background Filter</a></li>
<li class="chapter" data-level="8.9.2" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#sample-filtering"><i class="fa fa-check"></i><b>8.9.2</b> Sample Filtering</a></li>
<li class="chapter" data-level="8.9.3" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#identifying-removing-sample-outliers"><i class="fa fa-check"></i><b>8.9.3</b> Identifying &amp; Removing Sample Outliers</a></li>
<li class="chapter" data-level="8.9.4" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#with-this-we-can-answer-environmental-health-question-2-1"><i class="fa fa-check"></i><b>8.9.4</b> With this, we can answer <strong>Environmental Health Question 2</strong>:</a></li>
<li class="chapter" data-level="8.9.5" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#we-can-also-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>8.9.5</b> We can also answer <strong>Environmental Health Question 3</strong>:</a></li>
<li class="chapter" data-level="8.9.6" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#with-this-we-can-answer-environmental-health-question-4-1"><i class="fa fa-check"></i><b>8.9.6</b> With this, we can answer <strong>Environmental Health Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><i class="fa fa-check"></i><b>8.10</b> Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a><ul>
<li class="chapter" data-level="8.10.1" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-5-2"><i class="fa fa-check"></i><b>8.10.1</b> With this, we can answer <strong>Environmental Health Question 5</strong>:</a></li>
<li class="chapter" data-level="8.10.2" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-6-1"><i class="fa fa-check"></i><b>8.10.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
<li class="chapter" data-level="8.10.3" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-7"><i class="fa fa-check"></i><b>8.10.3</b> With this, we can answer <strong>Environmental Health Question 7</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.11" data-path="visualizing-statistical-results-using-ma-plots.html"><a href="visualizing-statistical-results-using-ma-plots.html"><i class="fa fa-check"></i><b>8.11</b> Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="8.12" data-path="visualizing-statistical-results-using-volcano-plots.html"><a href="visualizing-statistical-results-using-volcano-plots.html"><i class="fa fa-check"></i><b>8.12</b> Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="8.13" data-path="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><i class="fa fa-check"></i><b>8.13</b> Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a><ul>
<li class="chapter" data-level="8.13.1" data-path="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#with-this-we-can-answer-environmental-health-question-8"><i class="fa fa-check"></i><b>8.13.1</b> With this, we can answer <strong>Environmental Health Question 8</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.14" data-path="concluding-remarks-7.html"><a href="concluding-remarks-7.html"><i class="fa fa-check"></i><b>8.14</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i><b>9</b> Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="10" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html"><i class="fa fa-check"></i><b>10</b> Read-Across Toxicity Predictions</a></li>
<li class="chapter" data-level="11" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html"><i class="fa fa-check"></i><b>11</b> Environmental Health Databases: Comparative Toxicogenomics Database</a><ul>
<li class="chapter" data-level="11.1" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html"><i class="fa fa-check"></i><b>11.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="11.1.1" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html#introduction-to-comparative-toxicogenomics-database-ctd"><i class="fa fa-check"></i><b>11.1.1</b> Introduction to Comparative Toxicogenomics Database (CTD)</a></li>
<li class="chapter" data-level="11.1.2" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html#introduction-to-training-module-6"><i class="fa fa-check"></i><b>11.1.2</b> Introduction to Training Module</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="training-modules-environmental-health-questions-3.html"><a href="training-modules-environmental-health-questions-3.html"><i class="fa fa-check"></i><b>11.2</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="11.3" data-path="script-preparations-5.html"><a href="script-preparations-5.html"><i class="fa fa-check"></i><b>11.3</b> Script Preparations</a></li>
<li class="chapter" data-level="11.4" data-path="organizing-example-dataset-from-ctd.html"><a href="organizing-example-dataset-from-ctd.html"><i class="fa fa-check"></i><b>11.4</b> Organizing Example Dataset from CTD</a></li>
<li class="chapter" data-level="11.5" data-path="loading-the-example-ctd-dataset-into-r.html"><a href="loading-the-example-ctd-dataset-into-r.html"><i class="fa fa-check"></i><b>11.5</b> Loading the Example CTD Dataset into R</a></li>
<li class="chapter" data-level="11.6" data-path="data-viewing-3.html"><a href="data-viewing-3.html"><i class="fa fa-check"></i><b>11.6</b> Data Viewing</a></li>
<li class="chapter" data-level="11.7" data-path="filtering-the-data-for-genes-with-altered-expression.html"><a href="filtering-the-data-for-genes-with-altered-expression.html"><i class="fa fa-check"></i><b>11.7</b> Filtering the Data for Genes with Altered Expression</a><ul>
<li class="chapter" data-level="11.7.1" data-path="filtering-the-data-for-genes-with-altered-expression.html"><a href="filtering-the-data-for-genes-with-altered-expression.html#with-this-we-can-answer-environmental-health-question-1-1"><i class="fa fa-check"></i><b>11.7.1</b> With this, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="identifying-genes-under-epigenetic-control.html"><a href="identifying-genes-under-epigenetic-control.html"><i class="fa fa-check"></i><b>11.8</b> Identifying Genes under Epigenetic Control</a></li>
<li class="chapter" data-level="11.9" data-path="method-1-for-list-comparisons-merging.html"><a href="method-1-for-list-comparisons-merging.html"><i class="fa fa-check"></i><b>11.9</b> Method 1 for list comparisons: Merging</a><ul>
<li class="chapter" data-level="11.9.1" data-path="method-1-for-list-comparisons-merging.html"><a href="method-1-for-list-comparisons-merging.html#with-this-we-can-answer-environmental-health-question-2-2"><i class="fa fa-check"></i><b>11.9.1</b> With this, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="11.10" data-path="method-2-for-list-comparisons-intersection.html"><a href="method-2-for-list-comparisons-intersection.html"><i class="fa fa-check"></i><b>11.10</b> Method 2 for list comparisons: Intersection</a></li>
<li class="chapter" data-level="11.11" data-path="concluding-remarks-8.html"><a href="concluding-remarks-8.html"><i class="fa fa-check"></i><b>11.11</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html"><i class="fa fa-check"></i><b>12</b> Environmental Health Databases: Gene Expression Omnibus</a><ul>
<li class="chapter" data-level="12.1" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html"><i class="fa fa-check"></i><b>12.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="12.1.1" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html#introduction-to-the-environmental-health-database-gene-expression-omnibus-geo"><i class="fa fa-check"></i><b>12.1.1</b> Introduction to the Environmental Health Database, Gene Expression Omnibus (GEO)</a></li>
<li class="chapter" data-level="12.1.2" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html#introduction-to-training-module-7"><i class="fa fa-check"></i><b>12.1.2</b> Introduction to Training Module</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="training-modules-environmental-health-questions-4.html"><a href="training-modules-environmental-health-questions-4.html"><i class="fa fa-check"></i><b>12.2</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="12.3" data-path="script-preparations-6.html"><a href="script-preparations-6.html"><i class="fa fa-check"></i><b>12.3</b> Script Preparations</a></li>
<li class="chapter" data-level="12.4" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html"><i class="fa fa-check"></i><b>12.4</b> Loading and Organizing the Example Dataset</a><ul>
<li class="chapter" data-level="12.4.1" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html#manually-downloading-and-uploading-geo-files"><i class="fa fa-check"></i><b>12.4.1</b> 1. Manually Downloading and Uploading GEO Files</a></li>
<li class="chapter" data-level="12.4.2" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html#uploading-and-organizing-geo-files-through-the-geoquery-package"><i class="fa fa-check"></i><b>12.4.2</b> 2. Uploading and Organizing GEO Files through the GEOquery Package</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><i class="fa fa-check"></i><b>12.5</b> Adding Gene Symbol Information to -Omic Data Sets through Platform Annotation Files</a><ul>
<li class="chapter" data-level="12.5.1" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html#with-this-we-can-now-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>12.5.1</b> With this, we can now answer <strong>Environmental Health Question #1</strong>:</a></li>
<li class="chapter" data-level="12.5.2" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html#we-can-also-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>12.5.2</b> We can also answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="visualizing-gene-expression-data-using-boxplots-and-heat-maps.html"><a href="visualizing-gene-expression-data-using-boxplots-and-heat-maps.html"><i class="fa fa-check"></i><b>12.6</b> Visualizing Gene Expression Data using Boxplots and Heat Maps</a></li>
<li class="chapter" data-level="12.7" data-path="boxplot-visualizations-1.html"><a href="boxplot-visualizations-1.html"><i class="fa fa-check"></i><b>12.7</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="12.8" data-path="heatmap-visualizations-1.html"><a href="heatmap-visualizations-1.html"><i class="fa fa-check"></i><b>12.8</b> Heatmap Visualizations</a><ul>
<li class="chapter" data-level="12.8.1" data-path="heatmap-visualizations-1.html"><a href="heatmap-visualizations-1.html#we-can-also-answer-environmental-health-question-3-1"><i class="fa fa-check"></i><b>12.8.1</b> We can also answer <strong>Environmental Health Question #3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="statistical-analyses-to-identify-genes-altered-by-formaldehyde.html"><a href="statistical-analyses-to-identify-genes-altered-by-formaldehyde.html"><i class="fa fa-check"></i><b>12.9</b> Statistical Analyses to Identify Genes altered by Formaldehyde</a></li>
<li class="chapter" data-level="12.10" data-path="interpreting-results.html"><a href="interpreting-results.html"><i class="fa fa-check"></i><b>12.10</b> Interpreting Results</a><ul>
<li class="chapter" data-level="12.10.1" data-path="interpreting-results.html"><a href="interpreting-results.html#with-this-we-can-answer-environmental-health-question-4-2"><i class="fa fa-check"></i><b>12.10.1</b> With this, we can answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="12.10.2" data-path="interpreting-results.html"><a href="interpreting-results.html#with-this-we-can-answer-environmental-health-question-5-3"><i class="fa fa-check"></i><b>12.10.2</b> With this, we can answer <strong>Environmental Health Question #5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="12.11" data-path="concluding-remarks-9.html"><a href="concluding-remarks-9.html"><i class="fa fa-check"></i><b>12.11</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="database-integration-air-quality-study-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-study-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i><b>13</b> Database Integration: Air Quality Study, Mortality, and Environmental Justice Data</a></li>
<li class="chapter" data-level="14" data-path="additional-resources.html"><a href="additional-resources.html"><i class="fa fa-check"></i><b>14</b> Additional Resources</a><ul>
<li class="chapter" data-level="14.1" data-path="r-programming-resources.html"><a href="r-programming-resources.html"><i class="fa fa-check"></i><b>14.1</b> R Programming Resources:</a></li>
<li class="chapter" data-level="14.2" data-path="r-packages-resources.html"><a href="r-packages-resources.html"><i class="fa fa-check"></i><b>14.2</b> R Packages Resources:</a></li>
<li class="chapter" data-level="14.3" data-path="community-discussions-on-r-and-r-packages.html"><a href="community-discussions-on-r-and-r-packages.html"><i class="fa fa-check"></i><b>14.3</b> Community Discussions on R and R Packages:</a></li>
<li class="chapter" data-level="14.4" data-path="data-science-and-statistical-analysis-resources.html"><a href="data-science-and-statistical-analysis-resources.html"><i class="fa fa-check"></i><b>14.4</b> Data Science and Statistical Analysis Resources:</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chemistry-based-sufficient-similarity-analysis" class="section level2">
<h2><span class="header-section-number">7.7</span> Chemistry-based Sufficient Similarity Analysis</h2>
<p>The first method employed in this Sufficient Similarity analysis is Principal Component Analysis (PCA). PCA is a very common dimensionality reduction technique, as detailed in the ‘Machine Learning and Predictive Modeling’ training module.</p>
<p>In summary, PCA finds dimensions (eigenvectors) in the higher dimensional original data that capture as much of the variation as possible, which you can then plot. This allows you to project higher dimensional data, in this case 12 dimensions (representing 12 measured chemicals), in fewer dimensions (we’ll use 2). These dimensions, or components, capture the “essence” of the original dataset.</p>
<p>Before we can run PCA on this chemistry dataset, we first need to scale the data across samples.
We do this here for the chemistry dataset, because we specifically want to evaluate and potentially highlight/emphasize chemicals that may be at relatively low abundance. These low-abundance chemicals may actually be contaminants that drive toxicological effects.</p>
<div id="lets-first-re-save-the-original-chemistry-dataset-to-compare-off-of" class="section level4">
<h4><span class="header-section-number">7.7.0.1</span> Let’s first re-save the original chemistry dataset to compare off of</h4>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb18-1"></a>chem_original &lt;-<span class="st"> </span>chem</span></code></pre></div>
</div>
<div id="and-make-a-scaled-version-to-carry-forward-in-this-analysis" class="section level4">
<h4><span class="header-section-number">7.7.0.2</span> And make a scaled version to carry forward in this analysis</h4>
</div>
<div id="here-we-move-the-sample-column-the-row-names-then-scale-and-center-data" class="section level4">
<h4><span class="header-section-number">7.7.0.3</span> Here, we move the sample column the row names then scale and center data</h4>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb19-1"></a>chem &lt;-<span class="st"> </span>chem <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;Sample&quot;</span>)</span>
<span id="cb19-2"><a href="chemistry-based-sufficient-similarity-analysis.html#cb19-2"></a>chem &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">scale</span>(<span class="kw">as.matrix</span>(chem)))</span></code></pre></div>
</div>
<div id="lets-now-compare-one-of-the-rows-of-data-here-sample-gbe_e-to-see-what-scaling-did" class="section level4">
<h4><span class="header-section-number">7.7.0.4</span> Let’s now compare one of the rows of data (here, sample GbE_E) to see what scaling did:</h4>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb20-1"></a>chem_original[<span class="dv">5</span>,]</span></code></pre></div>
<pre><code>## # A tibble: 1 x 13
##   Sample Bilobalide Ginkgolide_A Ginkgolide_B Ginkgolide_C Ginkgolide_J Rutin
##   &lt;chr&gt;       &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt;
## 1 GbE_E         1.5         2.13         1.46          2.6         1.21  1.21
## # … with 6 more variables: Quercetin &lt;dbl&gt;, Kaempferol &lt;dbl&gt;,
## #   Isorhamnetin &lt;dbl&gt;, Ginkgolic_Acid_C15 &lt;dbl&gt;, Ginkgolic_Acid_C17 &lt;dbl&gt;,
## #   Ginkgotoxin &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb22-1"></a>chem[<span class="dv">5</span>,]</span></code></pre></div>
<pre><code>##       Bilobalide Ginkgolide_A Ginkgolide_B Ginkgolide_C Ginkgolide_J     Rutin
## GbE_E -0.4996768   0.02749315    0.2559249      1.36305    0.3842836 0.5758008
##       Quercetin Kaempferol Isorhamnetin Ginkgolic_Acid_C15 Ginkgolic_Acid_C17
## GbE_E 0.2059607 -0.2420023     -0.61695         -0.2737699         -0.2317415
##       Ginkgotoxin
## GbE_E  -0.8114437</code></pre>
<p>You can see that scaling made the concentrations distributed across each chemical center around 0.</p>
</div>
<div id="now-we-can-run-pca-on-the-scaled-data" class="section level4">
<h4><span class="header-section-number">7.7.0.5</span> Now, we can run PCA on the scaled data</h4>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb24-1"></a>chem_pca &lt;-<span class="st"> </span><span class="kw">princomp</span>(chem)</span></code></pre></div>
</div>
<div id="looking-at-the-scree-plot-we-see-the-first-two-principal-components-capture-most-of-the-variance-in-the-data-64" class="section level4">
<h4><span class="header-section-number">7.7.0.6</span> Looking at the scree plot, we see the first two principal components capture most of the variance in the data (~64%)</h4>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb25-1"></a><span class="kw">fviz_eig</span>(chem_pca)</span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>Here are the resulting PCA scores for each sample, for each principal component (shown here as components 1-12)</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb26-1"></a><span class="kw">head</span>(chem_pca<span class="op">$</span>scores)</span></code></pre></div>
<pre><code>##           Comp.1      Comp.2     Comp.3     Comp.4      Comp.5      Comp.6
## GbE_A -2.6256689 -0.53470486 0.38196672  1.5147505  1.34638144 -0.24020735
## GbE_B -3.7343187 -1.07490206 0.07766108  0.2529618 -0.01731847 -0.02446797
## GbE_C -3.8019563 -1.06232680 0.08335445  0.2588084 -0.01912736 -0.03423438
## GbE_D  0.7175782 -0.09566345 0.90082998 -0.7762090 -0.40507924 -0.78361086
## GbE_E  0.4157644 -0.14794948 1.16402759 -0.4856176 -0.15497152 -0.64206760
## GbE_F -3.7621819 -1.04784203 0.08687721  0.2503188 -0.02195005 -0.04030373
##           Comp.7     Comp.8      Comp.9     Comp.10     Comp.11      Comp.12
## GbE_A -0.8890488  0.8145597  0.15021776 -0.54318277 -0.32353295  0.049538024
## GbE_B  0.3988596 -0.3227102 -0.10344907 -0.12911495  0.11127631  0.008306532
## GbE_C  0.3817061 -0.2788579 -0.13057528 -0.02613584  0.08148776  0.011485316
## GbE_D -1.1916851 -0.4306198  0.08460588  0.26115540  0.01065657 -0.053819603
## GbE_E -1.1412900 -0.5632547  0.12309347 -0.02872126  0.24882196  0.047691048
## GbE_F  0.3948245 -0.3105324 -0.10539998 -0.11015645  0.10607314  0.012066512</code></pre>
<p>And the resulting loading factors of each chemical’s contribution towards each principal component. Results are arranged by a chemical’s contribution to PC1, the component accounting for most of the variation in the data.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb28-1"></a><span class="kw">head</span>(chem_pca<span class="op">$</span>loadings)</span></code></pre></div>
<pre><code>##                 Comp.1      Comp.2     Comp.3      Comp.4      Comp.5
## Bilobalide   0.3260729 0.085117226 0.24541927  0.36968834  0.03575030
## Ginkgolide_A 0.3732546 0.184101347 0.08586157  0.02446723 -0.03299212
## Ginkgolide_B 0.3611463 0.201280307 0.05996588 -0.19928568 -0.05965128
## Ginkgolide_C 0.3616497 0.001365838 0.12157943 -0.24092774 -0.10470516
## Ginkgolide_J 0.2995761 0.178689209 0.09144978  0.35348565  0.47372240
## Rutin        0.1763429 0.012553984 0.59496115 -0.47712128  0.08063637
##                    Comp.6      Comp.7      Comp.8      Comp.9     Comp.10
## Bilobalide    0.004975969  0.55079870  0.35062913  0.23584371  0.25226337
## Ginkgolide_A -0.247529927  0.30284388 -0.30206946 -0.01601813  0.20620248
## Ginkgolide_B -0.068401213 -0.00687696  0.09497565 -0.83752355  0.02062813
## Ginkgolide_C -0.185456111 -0.45474675 -0.34421625  0.32067335  0.45797955
## Ginkgolide_J -0.062170758 -0.55785820  0.36066655  0.03820900 -0.14858754
## Rutin         0.517141873  0.05060803  0.03111005  0.15083257 -0.27469825
##                  Comp.11     Comp.12
## Bilobalide    0.29295985  0.23837797
## Ginkgolide_A -0.60099943 -0.41172878
## Ginkgolide_B  0.24851189  0.06938891
## Ginkgolide_C  0.28752202  0.17463609
## Ginkgolide_J -0.22581449 -0.03024110
## Rutin        -0.09884752 -0.04092322</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb30-1"></a>loadings &lt;-<span class="st"> </span><span class="kw">as.data.frame.matrix</span>(chem_pca<span class="op">$</span>loadings)  <span class="co"># pulling the chemical-specific loadings into a separate matrix</span></span>
<span id="cb30-2"><a href="chemistry-based-sufficient-similarity-analysis.html#cb30-2"></a>loadings <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(Comp<span class="fl">.1</span>))  <span class="co"># sorting the loadings from highest to lowest for component #1</span></span></code></pre></div>
<pre><code>##                        Comp.1       Comp.2      Comp.3      Comp.4       Comp.5
## Quercetin           0.3801459 -0.001945021 -0.19193647  0.04697879  0.007656212
## Ginkgolide_A        0.3732546  0.184101347  0.08586157  0.02446723 -0.032992122
## Ginkgolide_C        0.3616497  0.001365838  0.12157943 -0.24092774 -0.104705164
## Ginkgolide_B        0.3611463  0.201280307  0.05996588 -0.19928568 -0.059651275
## Bilobalide          0.3260729  0.085117226  0.24541927  0.36968834  0.035750299
## Kaempferol          0.3001354 -0.085004317 -0.29667523 -0.15611039 -0.655825688
## Ginkgolide_J        0.2995761  0.178689209  0.09144978  0.35348565  0.473722400
## Isorhamnetin        0.2740348 -0.075119327 -0.29665890  0.39008587 -0.060880190
## Rutin               0.1763429  0.012553984  0.59496115 -0.47712128  0.080636368
## Ginkgotoxin         0.1675373 -0.034318656 -0.56232119 -0.47935782  0.551341021
## Ginkgolic_Acid_C15 -0.1201265  0.667543042 -0.11607308 -0.05802100 -0.085384063
## Ginkgolic_Acid_C17 -0.1418140  0.653142232 -0.09559022 -0.02189315 -0.084006824
##                          Comp.6      Comp.7      Comp.8      Comp.9     Comp.10
## Quercetin          -0.249799236  0.13263831 -0.30855302  0.18612332 -0.72929122
## Ginkgolide_A       -0.247529927  0.30284388 -0.30206946 -0.01601813  0.20620248
## Ginkgolide_C       -0.185456111 -0.45474675 -0.34421625  0.32067335  0.45797955
## Ginkgolide_B       -0.068401213 -0.00687696  0.09497565 -0.83752355  0.02062813
## Bilobalide          0.004975969  0.55079870  0.35062913  0.23584371  0.25226337
## Kaempferol          0.050018874 -0.12464461  0.51491286  0.16055155 -0.07828551
## Ginkgolide_J       -0.062170758 -0.55785820  0.36066655  0.03820900 -0.14858754
## Isorhamnetin        0.730543567 -0.06658953 -0.34052044 -0.10456587  0.10158173
## Rutin               0.517141873  0.05060803  0.03111005  0.15083257 -0.27469825
## Ginkgotoxin         0.092918281  0.20744490  0.16087302  0.11145659  0.17463719
## Ginkgolic_Acid_C15  0.057775937  0.03440329 -0.13417826  0.06575084 -0.07780733
## Ginkgolic_Acid_C17  0.133278823 -0.05017155  0.08415192  0.16593739  0.03564092
##                         Comp.11      Comp.12
## Quercetin           0.273123642  0.008854815
## Ginkgolide_A       -0.600999427 -0.411728782
## Ginkgolide_C        0.287522018  0.174636086
## Ginkgolide_B        0.248511890  0.069388910
## Bilobalide          0.292959851  0.238377968
## Kaempferol         -0.211380567 -0.020939233
## Ginkgolide_J       -0.225814490 -0.030241100
## Isorhamnetin        0.002690835 -0.006305513
## Rutin              -0.098847524 -0.040923217
## Ginkgotoxin        -0.005807642  0.016904160
## Ginkgolic_Acid_C15 -0.285797465  0.633437667
## Ginkgolic_Acid_C17  0.383124914 -0.577639931</code></pre>
<p>These resulting loading factors allow us to identify which constituents (of the 12 total) contribute to the principal components explaining data variabilities. For instance, we can see here that:<br />
<strong>Quercetin</strong> is listed at the top, with the largest loading value for principal component 1. Thus, Quercetin represents the constituents that contributes to the overall variability in the dataset to the greatest extent.<br />
The next three chemicals are all <strong>Ginkgolide</strong> constituents, followed by <strong>Bilobalide</strong> and <strong>Kaempferol</strong>, and so forth.</p>
<p>If we look at principal component 2, we can now see a different set of chemicals contributing to the variability captured in this component:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb32-1"></a>loadings <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(Comp<span class="fl">.2</span>))  <span class="co"># sorting the loadings from highest to lowest for component #2</span></span></code></pre></div>
<pre><code>##                        Comp.1       Comp.2      Comp.3      Comp.4       Comp.5
## Ginkgolic_Acid_C15 -0.1201265  0.667543042 -0.11607308 -0.05802100 -0.085384063
## Ginkgolic_Acid_C17 -0.1418140  0.653142232 -0.09559022 -0.02189315 -0.084006824
## Ginkgolide_B        0.3611463  0.201280307  0.05996588 -0.19928568 -0.059651275
## Ginkgolide_A        0.3732546  0.184101347  0.08586157  0.02446723 -0.032992122
## Ginkgolide_J        0.2995761  0.178689209  0.09144978  0.35348565  0.473722400
## Bilobalide          0.3260729  0.085117226  0.24541927  0.36968834  0.035750299
## Rutin               0.1763429  0.012553984  0.59496115 -0.47712128  0.080636368
## Ginkgolide_C        0.3616497  0.001365838  0.12157943 -0.24092774 -0.104705164
## Quercetin           0.3801459 -0.001945021 -0.19193647  0.04697879  0.007656212
## Ginkgotoxin         0.1675373 -0.034318656 -0.56232119 -0.47935782  0.551341021
## Isorhamnetin        0.2740348 -0.075119327 -0.29665890  0.39008587 -0.060880190
## Kaempferol          0.3001354 -0.085004317 -0.29667523 -0.15611039 -0.655825688
##                          Comp.6      Comp.7      Comp.8      Comp.9     Comp.10
## Ginkgolic_Acid_C15  0.057775937  0.03440329 -0.13417826  0.06575084 -0.07780733
## Ginkgolic_Acid_C17  0.133278823 -0.05017155  0.08415192  0.16593739  0.03564092
## Ginkgolide_B       -0.068401213 -0.00687696  0.09497565 -0.83752355  0.02062813
## Ginkgolide_A       -0.247529927  0.30284388 -0.30206946 -0.01601813  0.20620248
## Ginkgolide_J       -0.062170758 -0.55785820  0.36066655  0.03820900 -0.14858754
## Bilobalide          0.004975969  0.55079870  0.35062913  0.23584371  0.25226337
## Rutin               0.517141873  0.05060803  0.03111005  0.15083257 -0.27469825
## Ginkgolide_C       -0.185456111 -0.45474675 -0.34421625  0.32067335  0.45797955
## Quercetin          -0.249799236  0.13263831 -0.30855302  0.18612332 -0.72929122
## Ginkgotoxin         0.092918281  0.20744490  0.16087302  0.11145659  0.17463719
## Isorhamnetin        0.730543567 -0.06658953 -0.34052044 -0.10456587  0.10158173
## Kaempferol          0.050018874 -0.12464461  0.51491286  0.16055155 -0.07828551
##                         Comp.11      Comp.12
## Ginkgolic_Acid_C15 -0.285797465  0.633437667
## Ginkgolic_Acid_C17  0.383124914 -0.577639931
## Ginkgolide_B        0.248511890  0.069388910
## Ginkgolide_A       -0.600999427 -0.411728782
## Ginkgolide_J       -0.225814490 -0.030241100
## Bilobalide          0.292959851  0.238377968
## Rutin              -0.098847524 -0.040923217
## Ginkgolide_C        0.287522018  0.174636086
## Quercetin           0.273123642  0.008854815
## Ginkgotoxin        -0.005807642  0.016904160
## Isorhamnetin        0.002690835 -0.006305513
## Kaempferol         -0.211380567 -0.020939233</code></pre>
<p>With <strong>Ginkgolic Acids</strong> listed first here.</p>
</div>
<div id="we-can-also-visualize-sample-groupings-based-on-these-principal-components-1-2" class="section level4">
<h4><span class="header-section-number">7.7.0.7</span> We can also visualize sample groupings based on these principal components 1 &amp; 2</h4>
</div>
<div id="to-view-the-pca-plot" class="section level4">
<h4><span class="header-section-number">7.7.0.8</span> To view the PCA plot:</h4>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-1"></a><span class="co"># First pull the percent variation captured by each component</span></span>
<span id="cb34-2"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-2"></a>pca_percent &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="dv">100</span><span class="op">*</span>chem_pca<span class="op">$</span>sdev<span class="op">^</span><span class="dv">2</span><span class="op">/</span><span class="kw">sum</span>(chem_pca<span class="op">$</span>sdev<span class="op">^</span><span class="dv">2</span>),<span class="dv">1</span>)</span>
<span id="cb34-3"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-3"></a></span>
<span id="cb34-4"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-4"></a><span class="co"># Then make a dataframe for the PCA plot generation script using first three components</span></span>
<span id="cb34-5"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-5"></a>pca_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">PC1 =</span> chem_pca<span class="op">$</span>scores[,<span class="dv">1</span>], <span class="dt">PC2 =</span> chem_pca<span class="op">$</span>scores[,<span class="dv">2</span>])</span>
<span id="cb34-6"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-6"></a></span>
<span id="cb34-7"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-7"></a><span class="co"># Plot this dataframe</span></span>
<span id="cb34-8"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-8"></a>chem_pca_plt &lt;-<span class="st"> </span><span class="kw">ggplot</span>(pca_df, <span class="kw">aes</span>(PC1,PC2))<span class="op">+</span></span>
<span id="cb34-9"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-9"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">size=</span><span class="fl">0.3</span>)<span class="op">+</span></span>
<span id="cb34-10"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-10"></a><span class="st">  </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> <span class="dv">0</span>, <span class="dt">size=</span><span class="fl">0.3</span>)<span class="op">+</span></span>
<span id="cb34-11"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-11"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;deepskyblue3&quot;</span>) <span class="op">+</span></span>
<span id="cb34-12"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-12"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">rownames</span>(pca_df)), <span class="dt">fontface=</span><span class="st">&quot;bold&quot;</span>, <span class="dt">position=</span><span class="kw">position_jitter</span>(<span class="dt">width=</span><span class="fl">0.4</span>,<span class="dt">height=</span><span class="fl">0.4</span>))<span class="op">+</span></span>
<span id="cb34-13"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-13"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">x=</span><span class="kw">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>,pca_percent[<span class="dv">1</span>],<span class="st">&quot;%)&quot;</span>), <span class="dt">y=</span><span class="kw">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>,pca_percent[<span class="dv">2</span>],<span class="st">&quot;%)&quot;</span>))<span class="op">+</span></span>
<span id="cb34-14"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-14"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;GbE Sample PCA by Chemistry Profiles&quot;</span>)</span>
<span id="cb34-15"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-15"></a></span>
<span id="cb34-16"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-16"></a></span>
<span id="cb34-17"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-17"></a><span class="co"># Changing the colors of the titles and axis text</span></span>
<span id="cb34-18"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-18"></a>chem_pca_plt &lt;-<span class="st"> </span>chem_pca_plt <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">plot.title=</span><span class="kw">element_text</span>(<span class="dt">color=</span><span class="st">&quot;deepskyblue3&quot;</span>, <span class="dt">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb34-19"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-19"></a>                    <span class="dt">axis.title.x=</span><span class="kw">element_text</span>(<span class="dt">color=</span><span class="st">&quot;deepskyblue3&quot;</span>, <span class="dt">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb34-20"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-20"></a>                    <span class="dt">axis.title.y=</span><span class="kw">element_text</span>(<span class="dt">color=</span><span class="st">&quot;deepskyblue3&quot;</span>, <span class="dt">face=</span><span class="st">&quot;bold&quot;</span>))</span>
<span id="cb34-21"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-21"></a></span>
<span id="cb34-22"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-22"></a><span class="co"># Viewing this resulting plot</span></span>
<span id="cb34-23"><a href="chemistry-based-sufficient-similarity-analysis.html#cb34-23"></a>chem_pca_plt</span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-21-1.png" width="768" /></p>
</div>
<div id="this-plot-tells-us-a-lot-about-sample-groupings-based-on-chemical-profiles" class="section level3">
<h3><span class="header-section-number">7.7.1</span> This plot tells us a lot about sample groupings based on chemical profiles!</h3>
</div>
<div id="with-this-we-can-answer-environmental-health-questions-1-2" class="section level3">
<h3><span class="header-section-number">7.7.2</span> With this, we can answer <strong>Environmental Health Questions 1-2</strong>:</h3>
<div id="based-on-the-chemical-analysis-which-ginkgo-biloba-extract-looks-the-most-different" class="section level4">
<h4><span class="header-section-number">7.7.2.1</span> (1) Based on the chemical analysis, which <em>Ginkgo biloba</em> extract looks the most different?</h4>
</div>
<div id="answer-gbe_g" class="section level4">
<h4><span class="header-section-number">7.7.2.2</span> <em>Answer: GbE_G</em></h4>
</div>
<div id="when-viewing-the-variability-between-chemical-profiles-how-many-groupings-of-potentially-sufficiently-similar-ginkgo-biloba-samples-do-you-see" class="section level4">
<h4><span class="header-section-number">7.7.2.3</span> (2) When viewing the variability between chemical profiles, how many groupings of potentially ‘sufficiently similar’ <em>Ginkgo biloba</em> samples do you see?</h4>
</div>
<div id="answer-approximately-4-though-could-argue-1-1-bottom-left-group-bottom-right-group-and-two-completely-separate-samples-of-gbe_g-and-gbe_n" class="section level4">
<h4><span class="header-section-number">7.7.2.4</span> <em>Answer: Approximately 4 (though could argue +1/-1): bottom left group; bottom right group; and two completely separate samples of GbE_G and GbE_N</em></h4>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-viewing-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
