<!DOCTYPE html>
<html lang="" xml:lang="">
<head>
  
  <!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-M60LEM76PT"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-M60LEM76PT');
</script>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.1 Comparative Toxicogenomics Database | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="3.1 Comparative Toxicogenomics Database | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="3.1 Comparative Toxicogenomics Database | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.1 Comparative Toxicogenomics Database | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle R. Roell, Lauren E. Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/Favicon_v1.png" type="image/x-icon" />
<link rel="prev" href="read-across-toxicity-predictions.html"/>
<link rel="next" href="gene-expression-omnibus.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/Data Training Icon for Top Left Bookdown_v2.png"><br></a></li>

<li class="divider"></li>
<li><a href="index.html#preface">Preface<span></span></a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science<span></span></b></span></li>
<li><a href="introduction-to-coding-in-r.html#introduction-to-coding-in-r">1.1 Introduction to Coding in R<span></span></a>
<ul>
<li><a href="introduction-to-coding-in-r.html#introduction-to-training-module">Introduction to Training Module<span></span></a></li>
<li><a href="introduction-to-coding-in-r.html#r-packages">R Packages<span></span></a></li>
<li><a href="introduction-to-coding-in-r.html#scripting-basics">Scripting Basics<span></span></a></li>
<li><a href="introduction-to-coding-in-r.html#concluding-remarks-1">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="data-organization-basics.html#data-organization-basics">1.2 Data Organization Basics<span></span></a>
<ul>
<li><a href="data-organization-basics.html#introduction-to-training-module-1">Introduction to Training Module<span></span></a></li>
<li><a href="data-organization-basics.html#data-manipulation-using-base-r">Data Manipulation using Base R<span></span></a></li>
<li><a href="data-organization-basics.html#introduction-to-tidyverse">Introduction to Tidyverse<span></span></a></li>
<li><a href="data-organization-basics.html#concluding-remarks-2">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="finding-and-visualizing-data-trends.html#finding-and-visualizing-data-trends">1.3 Finding and Visualizing Data Trends<span></span></a>
<ul>
<li><a href="finding-and-visualizing-data-trends.html#introduction-to-training-module-2">Introduction to Training Module<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#basic-data-analysis">Basic Data Analysis<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#regression-modeling">Regression Modeling<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#categorical-data-analysis">Categorical Data Analysis<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#concluding-remarks-3">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="high-dimensional-data-visualizations.html#high-dimensional-data-visualizations">1.4 High-Dimensional Data Visualizations<span></span></a>
<ul>
<li><a href="high-dimensional-data-visualizations.html#introduction-to-training-module-3">Introduction to Training Module<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#high-dimensional-data-visualizations-1">High-Dimensional Data Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#density-plot-visualizations">Density Plot Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#ggally-visualizations">GGally Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#boxplot-visualizations">Boxplot Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#correlation-plot-visualizations">Correlation Plot Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#hierarchical-clustering-visualizations">Hierarchical Clustering Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#heat-map-visualizations">Heat Map Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#concluding-remarks-4">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="fair-data-management-practices.html#fair-data-management-practices">1.5 FAIR Data Management Practices<span></span></a>
<ul>
<li><a href="fair-data-management-practices.html#introduction-to-training-module-4">Introduction to Training Module<span></span></a></li>
<li><a href="fair-data-management-practices.html#introduction-to-fair">Introduction to FAIR<span></span></a></li>
<li><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter">Breaking Down FAIR, Letter-by-Letter<span></span></a></li>
<li><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data">Data Repositories for Sharing of Data<span></span></a></li>
<li><a href="fair-data-management-practices.html#helpful-resources-on-fair">Helpful Resources on FAIR<span></span></a></li>
<li><a href="fair-data-management-practices.html#concluding-remarks-5">Concluding Remarks<span></span></a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Chemical-<br>Biological Analyses <br>and Predictive Modeling<span></span></b></span></li>
<li><a href="dose-response-modeling.html#dose-response-modeling">2.1 Dose-Response Modeling<span></span></a>
<ul>
<li><a href="dose-response-modeling.html#introduction-to-training-module-5">Introduction to Training Module<span></span></a></li>
<li><a href="dose-response-modeling.html#plotting-data-in-dose-response">Plotting Data in Dose-Response<span></span></a></li>
<li><a href="dose-response-modeling.html#dose-response-curve-fitting">Dose-Response Curve Fitting<span></span></a></li>
<li><a href="dose-response-modeling.html#comparing-curve-fits">Comparing Curve Fits<span></span></a></li>
<li><a href="dose-response-modeling.html#curve-fitting-example">Curve Fitting Example<span></span></a></li>
<li><a href="dose-response-modeling.html#deriving-benchmark-dose">Deriving Benchmark Dose<span></span></a></li>
<li><a href="dose-response-modeling.html#concluding-remarks-6">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="machine-learning-and-predictive-modeling.html#machine-learning-and-predictive-modeling">2.2 Machine Learning and Predictive Modeling<span></span></a>
<ul>
<li><a href="machine-learning-and-predictive-modeling.html#introduction-to-training-module-6">Introduction to Training Module<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#k-means-analysis">K-means Analysis<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#principal-component-analysis">Principal Component Analysis<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#combining-k-means-with-pca">Combining K-Means with PCA<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#concluding-remarks-7">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="mixtures-analyses.html#mixtures-analyses">2.3 Mixtures Analyses<span></span></a>
<ul>
<li><a href="mixtures-analyses.html#introduction-to-training-module-7">Introduction to Training Module<span></span></a></li>
<li><a href="mixtures-analyses.html#chemistry-based-approach">Chemistry-based Approach<span></span></a></li>
<li><a href="mixtures-analyses.html#toxicity-based-approach">Toxicity-based Approach<span></span></a></li>
<li><a href="mixtures-analyses.html#comparing-results">Comparing Results<span></span></a></li>
<li><a href="mixtures-analyses.html#concluding-remarks-8">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="omics-analyses-and-systems-biology.html#omics-analyses-and-systems-biology">2.4 -Omics Analyses and Systems Biology<span></span></a>
<ul>
<li><a href="omics-analyses-and-systems-biology.html#introduction-to-training-module-8">Introduction to Training Module<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#transcriptomics-data-qaqc">Transcriptomics Data QA/QC<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#statistical-analysis-of-gene-expression">Statistical Analysis of Gene Expression<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#ma-plots">MA Plots<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#volcano-plots">Volcano Plots<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#pathway-enrichment-analysis">Pathway Enrichment Analysis<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#concluding-remarks-9">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="toxicokinetic-modeling.html#toxicokinetic-modeling">2.5 Toxicokinetic Modeling<span></span></a>
<ul>
<li><a href="toxicokinetic-modeling.html#introduction-to-training-module-9">Introduction to Training Module<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk">Data and Models used in Toxicokinetic Modeling (TK)<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#calculating-steady-state-concentration">Calculating steady-state concentration<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#reverse-toxicokinetics">Reverse Toxicokinetics<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#monte-carlo-approach">Monte Carlo Approach<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#chemical-specific-example">Chemical-Specific Example<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#calculating-bioactivity-exposure-ratios-bers">Calculating Bioactivity-Exposure Ratios (BERs)<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#concluding-remarks-10">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="read-across-toxicity-predictions.html#read-across-toxicity-predictions">2.6 Read-Across Toxicity Predictions<span></span></a>
<ul>
<li><a href="read-across-toxicity-predictions.html#introduction-to-training-module-10">Introduction to Training Module<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#read-across-example-analysis">Read-Across Example Analysis<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#calculating-chemical-similarities">Calculating Chemical Similarities<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#chemical-read-across-to-predict-acute-toxicity">Chemical Read-Across to Predict Acute Toxicity<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#concluding-remarks-11">Concluding Remarks<span></span></a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Environmental <br>Health Database Mining<span></span></b></span></li>
<li><a href="comparative-toxicogenomics-database.html#comparative-toxicogenomics-database">3.1 Comparative Toxicogenomics Database<span></span></a>
<ul>
<li><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-11">Introduction to Training Module<span></span></a></li>
<li><a href="comparative-toxicogenomics-database.html#ctd-data-in-r">CTD Data in R<span></span></a></li>
<li><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control">Identifying Genes under Epigenetic Control<span></span></a></li>
<li><a href="comparative-toxicogenomics-database.html#concluding-remarks-12">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="gene-expression-omnibus.html#gene-expression-omnibus">3.2 Gene Expression Omnibus<span></span></a>
<ul>
<li><a href="gene-expression-omnibus.html#introduction-to-training-module-12">Introduction to Training Module<span></span></a></li>
<li><a href="gene-expression-omnibus.html#geo-data-in-r">GEO Data in R<span></span></a></li>
<li><a href="gene-expression-omnibus.html#visualizing-data">Visualizing Data<span></span></a></li>
<li><a href="gene-expression-omnibus.html#statistical-analyses">Statistical Analyses<span></span></a></li>
<li><a href="gene-expression-omnibus.html#concluding-remarks-13">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#database-integration-air-quality-mortality-and-environmental-justice-data">3.3 Database Integration: Air Quality, Mortality, and Environmental Justice Data<span></span></a>
<ul>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-13">Introduction to Training Module<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#population-weighted-vs-unweighted-exposures">Population-weighted vs Unweighted Exposures<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#regression-modeling-1">Regression Modeling<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations">Environmental Justice Considerations<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-14">Concluding Remarks<span></span></a></li>
</ul></li>
<li class="part"><span><b>Additional Resources<span></span></b></span></li>
<li><a href="resources.html#resources">Resources<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="comparative-toxicogenomics-database" class="section level1 hasAnchor">
<h1>3.1 Comparative Toxicogenomics Database<a href="comparative-toxicogenomics-database.html#comparative-toxicogenomics-database" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Lauren E. Koval, Dr. Kyle R. Roell, and Dr. Julia E. Rager</p>
<p>Fall 2021</p>
<div id="background-on-training-module" class="section level4 hasAnchor">
<h4>Background on Training Module<a href="comparative-toxicogenomics-database.html#background-on-training-module" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
<div id="introduction-to-comparative-toxicogenomics-database-ctd" class="section level4 hasAnchor">
<h4>Introduction to Comparative Toxicogenomics Database (CTD)<a href="comparative-toxicogenomics-database.html#introduction-to-comparative-toxicogenomics-database-ctd" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>CTD is a publicly available, online database that provides manually curated information about chemical-gene/protein interactions, chemical-disease and gene-disease relationships. CTD also recently incorporated curation of exposure data and chemical-phenotype relationships.</p>
<p>CTD is located at: <a href="http://ctdbase.org/" class="uri">http://ctdbase.org/</a>. Here is a screenshot of the CTD homepage (as of August 5, 2021):
<img src="_book/TAME_Toolkit_files/figure-html/Module3_1_CTD_homepage.jpg" width="750" style="display: block; margin: auto;" /></p>
</div>
<div id="introduction-to-training-module-11" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="comparative-toxicogenomics-database.html#introduction-to-training-module-11" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this activity, we will be using CTD to access and download data to perform data organization and analysis as an applications-based example towards environmental health research. This activity represents a demonstration of basic data manipulation, filtering, and organization steps in R, while highlighting the utility of CTD to identify novel genomic/epigenomic relationships to environmental exposures. Example visualizations are also included in this training module’s script, providing visualizations of gene list comparison results.</p>
<div id="training-modules-environmental-health-questions-6" class="section level4 hasAnchor">
<h4>Training Module’s <strong>Environmental Health Questions</strong><a href="comparative-toxicogenomics-database.html#training-modules-environmental-health-questions-6" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This training module was specifically developed to answer the following environmental health questions:</p>
<ol style="list-style-type: decimal">
<li>Which genes show altered expression in response to arsenic exposure?</li>
<li>Of the genes showing altered expression, which may be under epigenetic control?</li>
</ol>
</div>
<div id="script-preparations-8" class="section level4 hasAnchor">
<h4>Script Preparations<a href="comparative-toxicogenomics-database.html#script-preparations-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
<div id="cleaning-the-global-environment-8" class="section level4 hasAnchor">
<h4>Cleaning the global environment<a href="comparative-toxicogenomics-database.html#cleaning-the-global-environment-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb510"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb510-1"><a href="comparative-toxicogenomics-database.html#cb510-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span></code></pre></div>
</div>
<div id="installing-required-r-packages-8" class="section level4 hasAnchor">
<h4>Installing required R packages<a href="comparative-toxicogenomics-database.html#installing-required-r-packages-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If you already have these packages installed, you can skip this step, or you can run the below code which checks installation status for you.</p>
<div class="sourceCode" id="cb511"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb511-1"><a href="comparative-toxicogenomics-database.html#cb511-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;tidyverse&quot;</span>))</span>
<span id="cb511-2"><a href="comparative-toxicogenomics-database.html#cb511-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;tidyverse&quot;</span>)</span>
<span id="cb511-3"><a href="comparative-toxicogenomics-database.html#cb511-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;VennDiagram&quot;</span>))</span>
<span id="cb511-4"><a href="comparative-toxicogenomics-database.html#cb511-4" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;VennDiagram&quot;</span>)</span>
<span id="cb511-5"><a href="comparative-toxicogenomics-database.html#cb511-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;grid&quot;</span>))</span>
<span id="cb511-6"><a href="comparative-toxicogenomics-database.html#cb511-6" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;grid&quot;</span>)</span></code></pre></div>
</div>
<div id="loading-r-packages-required-for-this-session-7" class="section level4 hasAnchor">
<h4>Loading R packages required for this session<a href="comparative-toxicogenomics-database.html#loading-r-packages-required-for-this-session-7" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb512"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb512-1"><a href="comparative-toxicogenomics-database.html#cb512-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb512-2"><a href="comparative-toxicogenomics-database.html#cb512-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(VennDiagram)</span>
<span id="cb512-3"><a href="comparative-toxicogenomics-database.html#cb512-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(grid)</span></code></pre></div>
</div>
<div id="set-your-working-directory-10" class="section level4 hasAnchor">
<h4>Set your working directory<a href="comparative-toxicogenomics-database.html#set-your-working-directory-10" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb513"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb513-1"><a href="comparative-toxicogenomics-database.html#cb513-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="ctd-data-in-r" class="section level2 hasAnchor">
<h2>CTD Data in R<a href="comparative-toxicogenomics-database.html#ctd-data-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="organizing-example-dataset-from-ctd" class="section level4 hasAnchor">
<h4>Organizing Example Dataset from CTD<a href="comparative-toxicogenomics-database.html#organizing-example-dataset-from-ctd" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>CTD requires manual querying of its database, outside of the R scripting environment. Because of this, let’s first manually pull the data we need for this example analysis. We can answer both of the example questions by pulling all chemical-gene relationship data for arsenic, which we can do by following the below steps:</p>
<p>Navigate to the main CTD website: <a href="http://ctdbase.org/" class="uri">http://ctdbase.org/</a>.</p>
<p>Select at the top, ‘Search’ -&gt; ‘Chemical-Gene Interactions’.
<br>
<img src="_book/TAME_Toolkit_files/figure-html/Module3_1_CTD_Step1.jpg" width="630" style="display: block; margin: auto;" /></p>
<p><br></p>
<p>Select to query all chemical-gene interaction data for arsenic.
<br>
<img src="_book/TAME_Toolkit_files/figure-html/Module3_1_CTD_Step2.jpg" width="600" style="display: block; margin: auto;" />
<br></p>
<p>Note that there are lots of results, represented by many many rows of data! Scroll to the bottom of the webpage and select to download as ‘CSV’.
<br>
<img src="_book/TAME_Toolkit_files/figure-html/Module3_1_CTD_Step3.jpg" width="600" style="display: block; margin: auto;" /></p>
<p><br></p>
<p>This is the file that we can now use to import into the R environment and analyze!
Note that the data pulled here represent data available on August 1, 2021</p>
</div>
<div id="loading-the-example-ctd-dataset-into-r" class="section level4 hasAnchor">
<h4>Loading the Example CTD Dataset into R<a href="comparative-toxicogenomics-database.html#loading-the-example-ctd-dataset-into-r" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
<div id="read-in-the-csv-file-of-the-results-from-ctd-query" class="section level4 hasAnchor">
<h4>Read in the csv file of the results from CTD query<a href="comparative-toxicogenomics-database.html#read-in-the-csv-file-of-the-results-from-ctd-query" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb514"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb514-1"><a href="comparative-toxicogenomics-database.html#cb514-1" aria-hidden="true" tabindex="-1"></a>ctd <span class="ot">=</span> <span class="fu">read_csv</span>(<span class="st">&quot;Module3_1/Module3_1_CTDOutput_ArsenicGene_Interactions.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="data-viewing-5" class="section level4 hasAnchor">
<h4>Data Viewing<a href="comparative-toxicogenomics-database.html#data-viewing-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Let’s first see how many rows and columns of data this file contains:</p>
<div class="sourceCode" id="cb515"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb515-1"><a href="comparative-toxicogenomics-database.html#cb515-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ctd)</span></code></pre></div>
<pre><code>## [1] 6280    9</code></pre>
<p>This dataset includes 6280 observations (represented by rows) linking arsenic exposure to gene-level alterations
With information spanning across 9 columns</p>
<p>Let’s also see what kind of data are organized within the columns:</p>
<div class="sourceCode" id="cb517"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb517-1"><a href="comparative-toxicogenomics-database.html#cb517-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ctd)</span></code></pre></div>
<pre><code>## [1] &quot;Chemical Name&quot;       &quot;Chemical ID&quot;         &quot;CAS RN&quot;             
## [4] &quot;Gene Symbol&quot;         &quot;Gene ID&quot;             &quot;Interaction&quot;        
## [7] &quot;Interaction Actions&quot; &quot;Reference Count&quot;     &quot;Organism Count&quot;</code></pre>
<div class="sourceCode" id="cb519"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb519-1"><a href="comparative-toxicogenomics-database.html#cb519-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Viewing the first five rows of data, across all 9 columns</span></span>
<span id="cb519-2"><a href="comparative-toxicogenomics-database.html#cb519-2" aria-hidden="true" tabindex="-1"></a>ctd[<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] </span></code></pre></div>
<pre><code>## # A tibble: 9 × 5
##   `Chemical Name` `Chemical ID` `CAS RN`  `Gene Symbol` `Gene ID`
##   &lt;chr&gt;           &lt;chr&gt;         &lt;chr&gt;     &lt;chr&gt;             &lt;dbl&gt;
## 1 Arsenic         D001151       7440-38-2 AACSP1           729522
## 2 Arsenic         D001151       7440-38-2 AADACL2          344752
## 3 Arsenic         D001151       7440-38-2 AAGAB             79719
## 4 Arsenic         D001151       7440-38-2 AAK1              22848
## 5 Arsenic         D001151       7440-38-2 AAMDC             28971
## 6 Arsenic         D001151       7440-38-2 AAR2              25980
## 7 Arsenic         D001151       7440-38-2 AASS              10157
## 8 Arsenic         D001151       7440-38-2 ABCA1                19
## 9 Arsenic         D001151       7440-38-2 ABCA12            26154</code></pre>
</div>
<div id="filtering-data-for-genes-with-altered-expression" class="section level4 hasAnchor">
<h4>Filtering Data for Genes with Altered Expression<a href="comparative-toxicogenomics-database.html#filtering-data-for-genes-with-altered-expression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To identify genes with altered expression in association with arsenic, we can leverage the results of our CTD query and filter this dataset to include only the rows that contain the term “expression” in the “Interaction Actions” column.</p>
<div class="sourceCode" id="cb521"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb521-1"><a href="comparative-toxicogenomics-database.html#cb521-1" aria-hidden="true" tabindex="-1"></a>exp_filt <span class="ot">=</span> ctd <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;expression&quot;</span>, <span class="st">`</span><span class="at">Interaction Actions</span><span class="st">`</span>))</span></code></pre></div>
<p>We now have 2586 observations, representing instances of arsenic exposure causing a changes in a target gene’s expression levels.</p>
<div class="sourceCode" id="cb522"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb522-1"><a href="comparative-toxicogenomics-database.html#cb522-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(exp_filt)</span></code></pre></div>
<pre><code>## [1] 2586    9</code></pre>
<p>Let’s see how many unique genes this represents:</p>
<div class="sourceCode" id="cb524"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb524-1"><a href="comparative-toxicogenomics-database.html#cb524-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(exp_filt<span class="sc">$</span><span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>))</span></code></pre></div>
<pre><code>## [1] 1878</code></pre>
<p>This reflects 1878 unique genes that show altered expression in association with arsenic.</p>
<p>Let’s make a separate dataframe that includes only the unique genes, based on the “Gene Symbol” column.</p>
<div class="sourceCode" id="cb526"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb526-1"><a href="comparative-toxicogenomics-database.html#cb526-1" aria-hidden="true" tabindex="-1"></a>exp_genes <span class="ot">=</span> exp_filt <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(<span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>, <span class="at">.keep_all=</span><span class="cn">TRUE</span>)</span>
<span id="cb526-2"><a href="comparative-toxicogenomics-database.html#cb526-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb526-3"><a href="comparative-toxicogenomics-database.html#cb526-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing columns besides gene identifier</span></span>
<span id="cb526-4"><a href="comparative-toxicogenomics-database.html#cb526-4" aria-hidden="true" tabindex="-1"></a>exp_genes <span class="ot">=</span> exp_genes[,<span class="dv">4</span>] </span>
<span id="cb526-5"><a href="comparative-toxicogenomics-database.html#cb526-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb526-6"><a href="comparative-toxicogenomics-database.html#cb526-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Viewing the first 10 genes listed</span></span>
<span id="cb526-7"><a href="comparative-toxicogenomics-database.html#cb526-7" aria-hidden="true" tabindex="-1"></a>exp_genes[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,] </span></code></pre></div>
<pre><code>## # A tibble: 10 × 1
##    `Gene Symbol`
##    &lt;chr&gt;        
##  1 AADACL2      
##  2 AAK1         
##  3 AASS         
##  4 ABCA12       
##  5 ABCC1        
##  6 ABCC2        
##  7 ABCC3        
##  8 ABCC4        
##  9 ABCG4        
## 10 ABHD12B</code></pre>
<p>This now provides us a list of 1878 genes showing altered expression in association with arsenic.</p>
<div id="technical-notes-on-running-the-distinct-function-within-tidyverse" class="section level5 hasAnchor">
<h5>Technical notes on running the distinct function within tidyverse:<a href="comparative-toxicogenomics-database.html#technical-notes-on-running-the-distinct-function-within-tidyverse" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>By default, the distinct function keeps the first instance of a duplicated value. This does have implications if the rest of the values in the rows differ. You will only retain the data associated with the first instance of the duplicated value (which is why we just retained the gene column here). It may be useful to first find the rows with the duplicate value and verify that results are as you would expect before removing observations. For example, in this dataset, expression levels can increase or decrease. If you were looking for just increases in expression, and there were genes that showed increased and decreased expression across different samples, using the distinct function just on “Gene Symbol” would not give you the results you wanted. If the first instance of the gene symbol noted decreased expression, that gene would not be returned in the results even though it might be one you would want. For this example case, we only care about expression change, regardless of direction, so this is not an issue. The distinct function can also take multiple columns to consider jointly as the value to check for duplicates if you are concerned about this.</p>
<p><br></p>
<!-- #### With this, we can answer **Environmental Health Question #1**: -->
<!-- #### (1) Which genes show altered expression in response to arsenic exposure? -->
<!-- #### *Answer: This list of 1878 genes have published evidence supporting their altered expression levels associated with arsenic exposure.* -->
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 1</strong>:</i>
Which genes show altered expression in response to arsenic exposure?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: This list of 1878 genes have published evidence supporting their altered expression levels associated with arsenic exposure.</p>
</div>
<p><br></p>
</div>
</div>
</div>
<div id="identifying-genes-under-epigenetic-control" class="section level2 hasAnchor">
<h2>Identifying Genes under Epigenetic Control<a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For this dataset, let’s focus on gene-level methylation as a marker of epigenetic regulation. Let’s return to our main dataframe, representing the results of the CTD query, and filter these results for only the rows that contain the term “methylation” in the “Interaction Actions” column.</p>
<div class="sourceCode" id="cb528"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb528-1"><a href="comparative-toxicogenomics-database.html#cb528-1" aria-hidden="true" tabindex="-1"></a>met_filt <span class="ot">=</span> ctd <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;methylation&quot;</span>,<span class="st">`</span><span class="at">Interaction Actions</span><span class="st">`</span>))</span></code></pre></div>
<p>We now have 3211 observations, representing instances of arsenic exposure causing a changes in a target gene’s methylation levels.</p>
<div class="sourceCode" id="cb529"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb529-1"><a href="comparative-toxicogenomics-database.html#cb529-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(met_filt)</span></code></pre></div>
<pre><code>## [1] 3211    9</code></pre>
<p>Let’s see how many unique genes this represents.</p>
<div class="sourceCode" id="cb531"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb531-1"><a href="comparative-toxicogenomics-database.html#cb531-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(met_filt<span class="sc">$</span><span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>))</span></code></pre></div>
<pre><code>## [1] 3142</code></pre>
<p>This reflects 3142 unique genes that show altered methylation in association with arsenic</p>
<p>Let’s make a separate dataframe that includes only the unique genes, based on the “Gene Symbol” column.</p>
<div class="sourceCode" id="cb533"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb533-1"><a href="comparative-toxicogenomics-database.html#cb533-1" aria-hidden="true" tabindex="-1"></a>met_genes <span class="ot">=</span> met_filt <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(<span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>, <span class="at">.keep_all=</span><span class="cn">TRUE</span>)</span>
<span id="cb533-2"><a href="comparative-toxicogenomics-database.html#cb533-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-3"><a href="comparative-toxicogenomics-database.html#cb533-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing columns besides gene identifier</span></span>
<span id="cb533-4"><a href="comparative-toxicogenomics-database.html#cb533-4" aria-hidden="true" tabindex="-1"></a>met_genes <span class="ot">=</span> met_genes[,<span class="dv">4</span>] </span></code></pre></div>
<p>This now provides us a list of 3142 genes showing altered methylation in association with arsenic.</p>
<p>With this list of genes with altered methylation, we can now compare it to previous list of genes with altered expression to yeild our final list of genes of interest. To achieve this last step, we present two different methods to carry out list comparisons below.</p>
<div id="method-1-for-list-comparisons-merging" class="section level4 hasAnchor">
<h4>Method 1 for list comparisons: Merging<a href="comparative-toxicogenomics-database.html#method-1-for-list-comparisons-merging" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Merge the expression results with the methylation resuts on the Gene Symbol column found in both datasets.</p>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="comparative-toxicogenomics-database.html#cb534-1" aria-hidden="true" tabindex="-1"></a>merge_df <span class="ot">=</span> <span class="fu">merge</span>(exp_genes, met_genes, <span class="at">by =</span> <span class="st">&quot;Gene Symbol&quot;</span>)</span></code></pre></div>
<p>We end up with 315 rows reflecting the 315 genes that show altered expression and altered methylation</p>
<p>Let’s view these genes:</p>
<div class="sourceCode" id="cb535"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb535-1"><a href="comparative-toxicogenomics-database.html#cb535-1" aria-hidden="true" tabindex="-1"></a>merge_df[<span class="dv">1</span><span class="sc">:</span><span class="dv">315</span>,]</span></code></pre></div>
<pre><code>##   [1] &quot;ABCC4&quot;     &quot;ABHD17A&quot;   &quot;ABLIM2&quot;    &quot;ACAD9&quot;     &quot;ACKR2&quot;     &quot;ACP3&quot;     
##   [7] &quot;ADAMTS1&quot;   &quot;AFF1&quot;      &quot;AGO2&quot;      &quot;ALDH3B2&quot;   &quot;ANPEP&quot;     &quot;AOPEP&quot;    
##  [13] &quot;AP3D1&quot;     &quot;APBB3&quot;     &quot;APP&quot;       &quot;AQP1&quot;      &quot;ARF1&quot;      &quot;ARID5B&quot;   
##  [19] &quot;AS3MT&quot;     &quot;ASAP1&quot;     &quot;ATF2&quot;      &quot;ATG7&quot;      &quot;ATP6V1C2&quot;  &quot;ATXN1&quot;    
##  [25] &quot;ATXN7&quot;     &quot;BACH1&quot;     &quot;BCAR1&quot;     &quot;BCL2&quot;      &quot;BCL6&quot;      &quot;BDNF&quot;     
##  [31] &quot;BECN1&quot;     &quot;BMI1&quot;      &quot;BMPR1A&quot;    &quot;C1GALT1C1&quot; &quot;C1S&quot;       &quot;C2CD3&quot;    
##  [37] &quot;CAMP&quot;      &quot;CARD18&quot;    &quot;CASP8&quot;     &quot;CASTOR1&quot;   &quot;CBR4&quot;      &quot;CBS&quot;      
##  [43] &quot;CCDC68&quot;    &quot;CCL14&quot;     &quot;CCL20&quot;     &quot;CCL24&quot;     &quot;CCR2&quot;      &quot;CD2&quot;      
##  [49] &quot;CD27&quot;      &quot;CD40&quot;      &quot;CDC42&quot;     &quot;CDH1&quot;      &quot;CDK2&quot;      &quot;CDK4&quot;     
##  [55] &quot;CDK5&quot;      &quot;CDK6&quot;      &quot;CDKN1B&quot;    &quot;CDKN2A&quot;    &quot;CELF1&quot;     &quot;CENPM&quot;    
##  [61] &quot;CEP72&quot;     &quot;CERK&quot;      &quot;CES4A&quot;     &quot;CFAP300&quot;   &quot;CHORDC1&quot;   &quot;CLEC4D&quot;   
##  [67] &quot;CLIC5&quot;     &quot;CMBL&quot;      &quot;CNTNAP2&quot;   &quot;CRCP&quot;      &quot;CREBBP&quot;    &quot;CUX2&quot;     
##  [73] &quot;CYP1B1&quot;    &quot;CYP26B1&quot;   &quot;CYP2U1&quot;    &quot;DAPK1&quot;     &quot;DAXX&quot;      &quot;DCAF7&quot;    
##  [79] &quot;DDB2&quot;      &quot;DHX32&quot;     &quot;DLK1&quot;      &quot;DNMT1&quot;     &quot;DSG1&quot;      &quot;DYNC2I2&quot;  
##  [85] &quot;ECHS1&quot;     &quot;EDAR&quot;      &quot;EFCAB2&quot;    &quot;EHMT2&quot;     &quot;EML2&quot;      &quot;EPHA1&quot;    
##  [91] &quot;EPHA2&quot;     &quot;EPM2AIP1&quot;  &quot;ERBB4&quot;     &quot;ERCC2&quot;     &quot;ERN2&quot;      &quot;ESR1&quot;     
##  [97] &quot;ETFB&quot;      &quot;ETFDH&quot;     &quot;F3&quot;        &quot;FAM25A&quot;    &quot;FAM43A&quot;    &quot;FAM50B&quot;   
## [103] &quot;FAM53C&quot;    &quot;FAS&quot;       &quot;FBLN2&quot;     &quot;FBXO32&quot;    &quot;FGF2&quot;      &quot;FGFR3&quot;    
## [109] &quot;FGGY&quot;      &quot;FOSB&quot;      &quot;FPR2&quot;      &quot;FTH1P3&quot;    &quot;FTL&quot;       &quot;GAK&quot;      
## [115] &quot;GAS1&quot;      &quot;GFRA1&quot;     &quot;GGACT&quot;     &quot;GLI2&quot;      &quot;GLI3&quot;      &quot;GNPDA1&quot;   
## [121] &quot;GOLGA4&quot;    &quot;GSTM3&quot;     &quot;GTSE1&quot;     &quot;H2AC6&quot;     &quot;H6PD&quot;      &quot;HAPLN2&quot;   
## [127] &quot;HCRT&quot;      &quot;HDAC4&quot;     &quot;HGF&quot;       &quot;HLA-DQA1&quot;  &quot;HOTAIR&quot;    &quot;HSD17B2&quot;  
## [133] &quot;HSPA1B&quot;    &quot;HSPA1L&quot;    &quot;HYAL1&quot;     &quot;IER3&quot;      &quot;IFNAR2&quot;    &quot;IFNG&quot;     
## [139] &quot;IGF1&quot;      &quot;IKBKB&quot;     &quot;IL10&quot;      &quot;IL16&quot;      &quot;IL1R1&quot;     &quot;IL1RAP&quot;   
## [145] &quot;IL20RA&quot;    &quot;INPP4B&quot;    &quot;IRF1&quot;      &quot;ITGA8&quot;     &quot;ITGAM&quot;     &quot;ITGB1&quot;    
## [151] &quot;JMJD6&quot;     &quot;JUP&quot;       &quot;KCNQ1&quot;     &quot;KEAP1&quot;     &quot;KLC1&quot;      &quot;KLHL21&quot;   
## [157] &quot;KRT1&quot;      &quot;KRT18&quot;     &quot;KRT27&quot;     &quot;LAMB1&quot;     &quot;LCE2B&quot;     &quot;LEPR&quot;     
## [163] &quot;LGALS7&quot;    &quot;LMF1&quot;      &quot;LMNA&quot;      &quot;LRP8&quot;      &quot;LRRC20&quot;    &quot;MALAT1&quot;   
## [169] &quot;MAOA&quot;      &quot;MAP2&quot;      &quot;MAP2K6&quot;    &quot;MAP3K8&quot;    &quot;MAPT&quot;      &quot;MARVELD3&quot; 
## [175] &quot;MBNL2&quot;     &quot;MEF2C&quot;     &quot;MEG3&quot;      &quot;MGMT&quot;      &quot;MICB&quot;      &quot;MLC1&quot;     
## [181] &quot;MLH1&quot;      &quot;MMP19&quot;     &quot;MOSMO&quot;     &quot;MPG&quot;       &quot;MRAP&quot;      &quot;MSH2&quot;     
## [187] &quot;MSI2&quot;      &quot;MT1M&quot;      &quot;MTOR&quot;      &quot;MUC1&quot;      &quot;MYH14&quot;     &quot;MYL9&quot;     
## [193] &quot;MYRIP&quot;     &quot;NCL&quot;       &quot;NEBL&quot;      &quot;NEDD4&quot;     &quot;NES&quot;       &quot;NEU1&quot;     
## [199] &quot;NFE2L2&quot;    &quot;NLRP3&quot;     &quot;NOS2&quot;      &quot;NPM1&quot;      &quot;NRF1&quot;      &quot;NRG1&quot;     
## [205] &quot;NRP2&quot;      &quot;NTM&quot;       &quot;NUAK2&quot;     &quot;NUP62CL&quot;   &quot;OASL&quot;      &quot;OCLN&quot;     
## [211] &quot;OSBPL5&quot;    &quot;PALS1&quot;     &quot;PCSK6&quot;     &quot;PDZD2&quot;     &quot;PECAM1&quot;    &quot;PFKFB3&quot;   
## [217] &quot;PGAP2&quot;     &quot;PGK1&quot;      &quot;PIAS1&quot;     &quot;PLA2G4D&quot;   &quot;PLCD1&quot;     &quot;PLEC&quot;     
## [223] &quot;PLEKHA6&quot;   &quot;PLEKHG3&quot;   &quot;PPFIA4&quot;    &quot;PPFIBP2&quot;   &quot;PPTC7&quot;     &quot;PRDX1&quot;    
## [229] &quot;PRKCQ&quot;     &quot;PRMT6&quot;     &quot;PRR5L&quot;     &quot;PRSS3&quot;     &quot;PTGS2&quot;     &quot;PTPRE&quot;    
## [235] &quot;PVT1&quot;      &quot;PYROXD2&quot;   &quot;RAB11FIP3&quot; &quot;RAMP1&quot;     &quot;RAP1GAP2&quot;  &quot;RAPGEF1&quot;  
## [241] &quot;RASAL2&quot;    &quot;RELCH&quot;     &quot;RGMA&quot;      &quot;RHEBL1&quot;    &quot;RHOH&quot;      &quot;RIPOR1&quot;   
## [247] &quot;RNF213&quot;    &quot;RNF216&quot;    &quot;ROBO1&quot;     &quot;S100P&quot;     &quot;S1PR1&quot;     &quot;SBF1&quot;     
## [253] &quot;SBNO2&quot;     &quot;SCGB3A1&quot;   &quot;SCHIP1&quot;    &quot;SELENOW&quot;   &quot;SEMA5B&quot;    &quot;SGMS1&quot;    
## [259] &quot;SH2B2&quot;     &quot;SKP2&quot;      &quot;SLC22A5&quot;   &quot;SLC44A2&quot;   &quot;SLC6A6&quot;    &quot;SNCA&quot;     
## [265] &quot;SNHG32&quot;    &quot;SNX1&quot;      &quot;SORL1&quot;     &quot;SPHK1&quot;     &quot;SPINK1&quot;    &quot;SPSB1&quot;    
## [271] &quot;SPTBN1&quot;    &quot;SQSTM1&quot;    &quot;SRGAP1&quot;    &quot;SSU72&quot;     &quot;STAT3&quot;     &quot;STK17B&quot;   
## [277] &quot;STX1A&quot;     &quot;STX3&quot;      &quot;SULT2B1&quot;   &quot;TCEA3&quot;     &quot;TERT&quot;      &quot;TGFB1&quot;    
## [283] &quot;TGFB3&quot;     &quot;TGFBR2&quot;    &quot;THNSL2&quot;    &quot;TIMP2&quot;     &quot;TLR10&quot;     &quot;TMEM86A&quot;  
## [289] &quot;TNFRSF10B&quot; &quot;TNFRSF10D&quot; &quot;TNFRSF1B&quot;  &quot;TNFSF10&quot;   &quot;TNNC2&quot;     &quot;TP53&quot;     
## [295] &quot;TRIB1&quot;     &quot;TRNP1&quot;     &quot;TSC22D3&quot;   &quot;TSLP&quot;      &quot;TXNRD1&quot;    &quot;UAP1&quot;     
## [301] &quot;UBE2J2&quot;    &quot;ULK1&quot;      &quot;USP36&quot;     &quot;VAV3&quot;      &quot;VWF&quot;       &quot;WDR26&quot;    
## [307] &quot;WDR55&quot;     &quot;WNK1&quot;      &quot;WWTR1&quot;     &quot;XDH&quot;       &quot;ZBTB25&quot;    &quot;ZEB1&quot;     
## [313] &quot;ZNF200&quot;    &quot;ZNF267&quot;    &quot;ZNF696&quot;</code></pre>
<!-- #### With this, we can answer **Environmental Health Question #2**: -->
<!-- #### (2) Of the genes showing altered expression, which may be under epigenetic control? -->
<!-- #### *Answer: We identified 315 genes with altered expression resulting from arsenic exposure, that also demonstrate epigenetic modifications from arsenic. These genes include many high interest molecules involved in regulating cell health, including several cyclin dependent kinases (e.g., CDK2, CDK4, CDK5, CDK6), molecules involved in oxidative stress (e.g., FOSB, NOS2), and cytokines involved in inflammatory response pathways (e.g., IFNG, IL10, IL16, IL1R1, IR1RAP, TGFB1, TGFB3).* -->
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 2</strong>:</i>
Of the genes showing altered expression, which may be under epigenetic control?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: We identified 315 genes with altered expression resulting from arsenic exposure, that also demonstrate epigenetic modifications from arsenic. These genes include many high interest molecules involved in regulating cell health, including several cyclin dependent kinases (e.g., CDK2, CDK4, CDK5, CDK6), molecules involved in oxidative stress (e.g., FOSB, NOS2), and cytokines involved in inflammatory response pathways (e.g., IFNG, IL10, IL16, IL1R1, IR1RAP, TGFB1, TGFB3).</p>
</div>
</div>
<div id="method-2-for-list-comparisons-intersection" class="section level4 hasAnchor">
<h4>Method 2 for list comparisons: Intersection<a href="comparative-toxicogenomics-database.html#method-2-for-list-comparisons-intersection" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>For further training, shown here is another method for pulling this list of interest, through the use of the ‘intersection’ function.</p>
<p>Obtain a list of the overlapping genes in the overall expression results and the methylation results.</p>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="comparative-toxicogenomics-database.html#cb537-1" aria-hidden="true" tabindex="-1"></a>inxn <span class="ot">=</span> <span class="fu">intersect</span>(exp_filt<span class="sc">$</span><span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>,met_filt<span class="sc">$</span><span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>)</span></code></pre></div>
<p>Again, we end up with a list of 315 unique genes that show altered expression and altered methylation.</p>
<p>This list can be viewed on its own or converted to a dataframe (df).</p>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="comparative-toxicogenomics-database.html#cb538-1" aria-hidden="true" tabindex="-1"></a>inxn_df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">genes=</span>inxn)</span></code></pre></div>
<p>This list can also be conveniently used to filter the original query results.</p>
<div class="sourceCode" id="cb539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb539-1"><a href="comparative-toxicogenomics-database.html#cb539-1" aria-hidden="true" tabindex="-1"></a>inxn_df_all_data <span class="ot">=</span> ctd <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span> <span class="sc">%in%</span> inxn)</span></code></pre></div>
<p>Note that in this last case, the same 315 genes are present, but this time the results contain all records from the original query results, hence the 875 rows (875 records observations reflecting the 315 genes).</p>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="comparative-toxicogenomics-database.html#cb540-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">unique</span>(<span class="fu">sort</span>(inxn_df_all_data<span class="sc">$</span><span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>))<span class="sc">==</span><span class="fu">sort</span>(merge_df<span class="sc">$</span><span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>))</span></code></pre></div>
<pre><code>##    Mode    TRUE 
## logical     315</code></pre>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="comparative-toxicogenomics-database.html#cb542-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(inxn_df_all_data)</span></code></pre></div>
<pre><code>## [1] 875   9</code></pre>
<p>Visually we can represent this as a Venn diagram. Here, we use the <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-35">“VennDiagram”</a> R package.</p>
<div class="sourceCode" id="cb544"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb544-1"><a href="comparative-toxicogenomics-database.html#cb544-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the data we previously used for intersection in the venn diagram function</span></span>
<span id="cb544-2"><a href="comparative-toxicogenomics-database.html#cb544-2" aria-hidden="true" tabindex="-1"></a>venn.plt <span class="ot">=</span> <span class="fu">venn.diagram</span>(</span>
<span id="cb544-3"><a href="comparative-toxicogenomics-database.html#cb544-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> <span class="fu">list</span>(exp_filt<span class="sc">$</span><span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>, met_filt<span class="sc">$</span><span class="st">`</span><span class="at">Gene Symbol</span><span class="st">`</span>),</span>
<span id="cb544-4"><a href="comparative-toxicogenomics-database.html#cb544-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">category.names =</span> <span class="fu">c</span>(<span class="st">&quot;Altered Expression&quot;</span> , <span class="st">&quot;Altered Methylation&quot;</span>),</span>
<span id="cb544-5"><a href="comparative-toxicogenomics-database.html#cb544-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">filename =</span> <span class="st">&quot;venn_diagram.png&quot;</span>,</span>
<span id="cb544-6"><a href="comparative-toxicogenomics-database.html#cb544-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb544-7"><a href="comparative-toxicogenomics-database.html#cb544-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Change font size, type, and position</span></span>
<span id="cb544-8"><a href="comparative-toxicogenomics-database.html#cb544-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">cat.cex =</span> <span class="fl">1.15</span>,</span>
<span id="cb544-9"><a href="comparative-toxicogenomics-database.html#cb544-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">cat.fontface =</span> <span class="st">&quot;bold&quot;</span>,</span>
<span id="cb544-10"><a href="comparative-toxicogenomics-database.html#cb544-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">cat.default.pos =</span> <span class="st">&quot;outer&quot;</span>,</span>
<span id="cb544-11"><a href="comparative-toxicogenomics-database.html#cb544-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">cat.pos =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">27</span>, <span class="dv">27</span>),</span>
<span id="cb544-12"><a href="comparative-toxicogenomics-database.html#cb544-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">cat.dist =</span> <span class="fu">c</span>(<span class="fl">0.055</span>, <span class="fl">0.055</span>),</span>
<span id="cb544-13"><a href="comparative-toxicogenomics-database.html#cb544-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb544-14"><a href="comparative-toxicogenomics-database.html#cb544-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Change color of ovals</span></span>
<span id="cb544-15"><a href="comparative-toxicogenomics-database.html#cb544-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">col=</span><span class="fu">c</span>(<span class="st">&quot;#440154ff&quot;</span>, <span class="st">&#39;#21908dff&#39;</span>),</span>
<span id="cb544-16"><a href="comparative-toxicogenomics-database.html#cb544-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">fill =</span> <span class="fu">c</span>(<span class="fu">alpha</span>(<span class="st">&quot;#440154ff&quot;</span>,<span class="fl">0.3</span>), <span class="fu">alpha</span>(<span class="st">&#39;#21908dff&#39;</span>,<span class="fl">0.3</span>)),</span>
<span id="cb544-17"><a href="comparative-toxicogenomics-database.html#cb544-17" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="03-Chapter3_files/figure-html/print-venn-1.png" width="672" /></p>
</div>
</div>
<div id="concluding-remarks-12" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="comparative-toxicogenomics-database.html#concluding-remarks-12" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In conclusion, we identified 315 genes that show altered expression in response to arsenic exposure that may be under epigenetic control. These genes represent critical mediators of oxidative stress and inflammation, among other important cellular processes. Results yielded an important list of genes representing potential targets for further evaluation, to better understand mechanism of environmental exposure-induced disease. Together, this example highlights the utility of CTD to address environmental health research questions.</p>
<p>For more information, see the recently updated primary CTD publication:</p>
<ul>
<li>Davis AP, Grondin CJ, Johnson RJ, Sciaky D, Wiegers J, Wiegers TC, Mattingly CJ. Comparative Toxicogenomics Database (CTD): update 2021. Nucleic Acids Res. 2021 Jan 8;49(D1):D1138-D1143. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/33068428/">33068428</a>.</li>
</ul>
<p>Additional case studies relevant to environmental health research include the following:</p>
<ul>
<li><p>An example publication leveraging CTD findings to identify mechanisms of metals-induced birth defects: Ahir BK, Sanders AP, Rager JE, Fry RC. Systems biology and birth defects prevention: blockade of the glucocorticoid receptor prevents arsenic-induced birth defects. Environ Health Perspect. 2013 Mar;121(3):332-8. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/23458687/">23458687</a>.</p></li>
<li><p>An example publication leveraging CTD to help fill data gaps on data poor chemicals, in combination with ToxCast/Tox21 data streams, to elucidate environmental influences on disease pathways: Kosnik MB, Planchart A, Marvel SW, Reif DM, Mattingly CJ. Integration of curated and high-throughput screening data to elucidate environmental influences on disease pathways. Comput Toxicol. 2019 Nov;12:100094. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/31453412/">31453412</a>.</p></li>
<li><p>An example publication leveraging CTD to extract chemical-disease relationships used to derive new chemical risk values, with the goal of prioritizing connections between environmental factors, genetic variants, and human diseases: Kosnik MB, Reif DM. Determination of chemical-disease risk values to prioritize connections between environmental factors, genetic variants, and human diseases. Toxicol Appl Pharmacol. 2019 Sep 15;379:114674. <a href="https://pubmed.ncbi.nlm.nih.gov/31323264/">PMID: 31323264</a>.</p></li>
</ul>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="read-across-toxicity-predictions.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gene-expression-omnibus.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
