<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Controlling for Sources of Sample Heterogeneity | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="Chapter 9 Controlling for Sources of Sample Heterogeneity | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Controlling for Sources of Sample Heterogeneity | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Controlling for Sources of Sample Heterogeneity | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle Roell, Lauren Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="transcriptomics-data-qaqc.html"/>
<link rel="next" href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TAME Toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#background"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#objectives"><i class="fa fa-check"></i>Objectives</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#module-development-overview"><i class="fa fa-check"></i>Module Development Overview</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#concluding-remarks"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#funding"><i class="fa fa-check"></i>Funding</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#author-affiliations"><i class="fa fa-check"></i>Author Affiliations</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html"><i class="fa fa-check"></i><b>1</b> Introduction to Coding in R</a><ul>
<li class="chapter" data-level="1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html"><i class="fa fa-check"></i><b>1.1</b> General Introduction and Installation of R and RStudio</a><ul>
<li class="chapter" data-level="1.1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html#what-is-r"><i class="fa fa-check"></i><b>1.1.1</b> What is R?</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="downloading-and-installing-r.html"><a href="downloading-and-installing-r.html"><i class="fa fa-check"></i><b>1.2</b> Downloading and Installing R</a></li>
<li class="chapter" data-level="1.3" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html"><i class="fa fa-check"></i><b>1.3</b> Downloading and Installing R Studio</a></li>
<li class="chapter" data-level="1.4" data-path="introduction-to-r-packages.html"><a href="introduction-to-r-packages.html"><i class="fa fa-check"></i><b>1.4</b> Introduction to R Packages</a><ul>
<li class="chapter" data-level="1.4.1" data-path="introduction-to-r-packages.html"><a href="introduction-to-r-packages.html#downloadinginstalling-r-packages"><i class="fa fa-check"></i><b>1.4.1</b> Downloading/Installing R Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="scripting-basics.html"><a href="scripting-basics.html"><i class="fa fa-check"></i><b>1.5</b> Scripting Basics</a><ul>
<li class="chapter" data-level="1.5.1" data-path="scripting-basics.html"><a href="scripting-basics.html#comments"><i class="fa fa-check"></i><b>1.5.1</b> Comments</a></li>
<li class="chapter" data-level="1.5.2" data-path="scripting-basics.html"><a href="scripting-basics.html#starting-code"><i class="fa fa-check"></i><b>1.5.2</b> Starting Code</a></li>
<li class="chapter" data-level="1.5.3" data-path="scripting-basics.html"><a href="scripting-basics.html#setting-your-working-directory"><i class="fa fa-check"></i><b>1.5.3</b> Setting Your Working Directory</a></li>
<li class="chapter" data-level="1.5.4" data-path="scripting-basics.html"><a href="scripting-basics.html#importing-files"><i class="fa fa-check"></i><b>1.5.4</b> Importing Files</a></li>
<li class="chapter" data-level="1.5.5" data-path="scripting-basics.html"><a href="scripting-basics.html#viewing-data"><i class="fa fa-check"></i><b>1.5.5</b> Viewing Data</a></li>
<li class="chapter" data-level="1.5.6" data-path="scripting-basics.html"><a href="scripting-basics.html#exporting-data"><i class="fa fa-check"></i><b>1.5.6</b> Exporting Data</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="concluding-remarks-1.html"><a href="concluding-remarks-1.html"><i class="fa fa-check"></i><b>1.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-organization-basics.html"><a href="data-organization-basics.html"><i class="fa fa-check"></i><b>2</b> Data Organization Basics</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html"><i class="fa fa-check"></i><b>2.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="2.1.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#set-your-working-directory-1"><i class="fa fa-check"></i><b>2.1.1</b> Set your working directory</a></li>
<li class="chapter" data-level="2.1.2" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#importing-example-datasets"><i class="fa fa-check"></i><b>2.1.2</b> Importing example datasets</a></li>
<li class="chapter" data-level="2.1.3" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#viewing-example-datasets"><i class="fa fa-check"></i><b>2.1.3</b> Viewing example datasets</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="merging-data-using-base-r-syntax.html"><a href="merging-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.2</b> Merging Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.3" data-path="filtering-and-subsetting-data-using-base-r-syntax.html"><a href="filtering-and-subsetting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.3</b> Filtering and Subsetting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.4" data-path="melting-and-casting-data-using-base-r-syntax.html"><a href="melting-and-casting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.4</b> Melting and Casting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.5" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html"><i class="fa fa-check"></i><b>2.5</b> Introduction to Tidyverse</a><ul>
<li class="chapter" data-level="2.5.1" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#downloading-and-loading-the-tidyverse-package"><i class="fa fa-check"></i><b>2.5.1</b> Downloading and Loading the Tidyverse Package</a></li>
<li class="chapter" data-level="2.5.2" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#merging-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.2</b> Merging Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.3" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#filtering-and-subsetting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.3</b> Filtering and Subsetting Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.4" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#melting-and-casting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.4</b> Melting and Casting Data using Tidyverse Syntax</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="concluding-remarks-2.html"><a href="concluding-remarks-2.html"><i class="fa fa-check"></i><b>2.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html"><i class="fa fa-check"></i><b>3</b> Finding and Visualizing Data Trend</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html"><i class="fa fa-check"></i><b>3.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="3.1.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html#script-preparations"><i class="fa fa-check"></i><b>3.1.1</b> Script Preparations</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html"><i class="fa fa-check"></i><b>3.2</b> Basic Statistical Tests and Visualizations of Data Trends</a><ul>
<li class="chapter" data-level="3.2.1" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#visualize-and-test-data-for-normality"><i class="fa fa-check"></i><b>3.2.1</b> Visualize and Test Data for Normality</a></li>
<li class="chapter" data-level="3.2.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i><b>3.2.2</b> Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="3.2.3" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i><b>3.2.3</b> Two-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><i class="fa fa-check"></i><b>3.3</b> Regression Modeling and Visualization: Linear and Logistic Regressions</a><ul>
<li class="chapter" data-level="3.3.1" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#linear-regression"><i class="fa fa-check"></i><b>3.3.1</b> Linear Regression</a></li>
<li class="chapter" data-level="3.3.2" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#logistic-regression"><i class="fa fa-check"></i><b>3.3.2</b> Logistic Regression</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="concluding-remarks-3.html"><a href="concluding-remarks-3.html"><i class="fa fa-check"></i><b>3.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html"><i class="fa fa-check"></i><b>4</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-training-module-2.html"><a href="introduction-to-training-module-2.html"><i class="fa fa-check"></i><b>4.1</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="4.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html"><i class="fa fa-check"></i><b>4.2</b> Script Preparations</a><ul>
<li class="chapter" data-level="4.2.1" data-path="script-preparations-1.html"><a href="script-preparations-1.html#formatting-dataframes-for-downstream-visualization-code"><i class="fa fa-check"></i><b>4.2.1</b> Formatting dataframes for downstream visualization code</a></li>
<li class="chapter" data-level="4.2.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html#scaling-dataframes-for-downstream-data-visualizations"><i class="fa fa-check"></i><b>4.2.2</b> Scaling dataframes for downstream data visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html"><i class="fa fa-check"></i><b>4.3</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.3.1" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#density-plot-visualizations"><i class="fa fa-check"></i><b>4.3.1</b> Density Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.2" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#ggally-visualizations"><i class="fa fa-check"></i><b>4.3.2</b> GGally Visualizations</a></li>
<li class="chapter" data-level="4.3.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#boxplot-visualizations"><i class="fa fa-check"></i><b>4.3.3</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="4.3.4" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#correlation-plot-visualizations"><i class="fa fa-check"></i><b>4.3.4</b> Correlation Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.5" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#hierarchical-clustering-visualizations"><i class="fa fa-check"></i><b>4.3.5</b> Hierarchical Clustering Visualizations</a></li>
<li class="chapter" data-level="4.3.6" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#heatmap-visualizations"><i class="fa fa-check"></i><b>4.3.6</b> Heatmap Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="concluding-remarks-4.html"><a href="concluding-remarks-4.html"><i class="fa fa-check"></i><b>4.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i><b>5</b> FAIR Data Management Practices</a><ul>
<li class="chapter" data-level="5.1" data-path="background-on-training-module.html"><a href="background-on-training-module.html"><i class="fa fa-check"></i><b>5.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="5.2" data-path="training-modules-questions.html"><a href="training-modules-questions.html"><i class="fa fa-check"></i><b>5.2</b> Training Module’s <strong>Questions</strong></a></li>
<li class="chapter" data-level="5.3" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html"><i class="fa fa-check"></i><b>5.3</b> Introduction to FAIR</a><ul>
<li class="chapter" data-level="5.3.1" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#with-this-background-we-can-now-answer-question-1"><i class="fa fa-check"></i><b>5.3.1</b> With this background, we can now answer <strong>Question 1</strong>:</a></li>
<li class="chapter" data-level="5.3.2" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#we-can-also-answer-question-2"><i class="fa fa-check"></i><b>5.3.2</b> We can also answer <strong>Question 2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="breaking-down-fair-letter-by-letter.html"><a href="breaking-down-fair-letter-by-letter.html"><i class="fa fa-check"></i><b>5.4</b> Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="5.5" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html"><i class="fa fa-check"></i><b>5.5</b> F (Findable) in FAIR</a><ul>
<li class="chapter" data-level="5.5.1" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html#with-this-information-can-can-now-answer-question-3"><i class="fa fa-check"></i><b>5.5.1</b> With this information, can can now answer <strong>Question 3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="a-accessible-in-fair.html"><a href="a-accessible-in-fair.html"><i class="fa fa-check"></i><b>5.6</b> A (Accessible) in FAIR</a></li>
<li class="chapter" data-level="5.7" data-path="i-interoperable-in-fair.html"><a href="i-interoperable-in-fair.html"><i class="fa fa-check"></i><b>5.7</b> I (Interoperable) in FAIR</a></li>
<li class="chapter" data-level="5.8" data-path="r-reusable-in-fair.html"><a href="r-reusable-in-fair.html"><i class="fa fa-check"></i><b>5.8</b> R (Reusable) in FAIR</a></li>
<li class="chapter" data-level="5.9" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html"><i class="fa fa-check"></i><b>5.9</b> What does this mean to you?</a><ul>
<li class="chapter" data-level="5.9.1" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html#with-this-information-can-can-now-answer-question-4"><i class="fa fa-check"></i><b>5.9.1</b> With this information, can can now answer <strong>Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.10" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html"><i class="fa fa-check"></i><b>5.10</b> Data Repositories for Sharing of Data</a><ul>
<li class="chapter" data-level="5.10.1" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html#with-this-information-can-can-now-answer-question-5"><i class="fa fa-check"></i><b>5.10.1</b> With this information, can can now answer <strong>Question 5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html"><i class="fa fa-check"></i><b>5.11</b> Helpful Resources on FAIR</a><ul>
<li class="chapter" data-level="5.11.1" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-training-resources-on-fair"><i class="fa fa-check"></i><b>5.11.1</b> Additional Training Resources on FAIR</a></li>
<li class="chapter" data-level="5.11.2" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-examples-and-documents-on-fair"><i class="fa fa-check"></i><b>5.11.2</b> Additional Examples and Documents on FAIR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="6" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i><b>6</b> Machine Learning and Predictive Modeling for Environmental Health</a><ul>
<li class="chapter" data-level="6.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html"><i class="fa fa-check"></i><b>6.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="6.1.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#introduction-to-machine-learning-ml-and-predictive-modeling"><i class="fa fa-check"></i><b>6.1.1</b> Introduction to Machine Learning (ML) and Predictive Modeling</a></li>
<li class="chapter" data-level="6.1.2" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#defining-predictive-modeling-in-the-context-of-toxicology-and-environmental-health"><i class="fa fa-check"></i><b>6.1.2</b> Defining predictive modeling in the context of toxicology and environmental health</a></li>
<li class="chapter" data-level="6.1.3" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#distinguish-between-machine-learning-ml-and-traditional-statistical-methods"><i class="fa fa-check"></i><b>6.1.3</b> Distinguish between machine learning (ML) and traditional statistical methods</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="introduction-to-training-module-3.html"><a href="introduction-to-training-module-3.html"><i class="fa fa-check"></i><b>6.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="6.3" data-path="training-modules-environmental-health-questions.html"><a href="training-modules-environmental-health-questions.html"><i class="fa fa-check"></i><b>6.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="6.4" data-path="script-preparations-2.html"><a href="script-preparations-2.html"><i class="fa fa-check"></i><b>6.4</b> Script Preparations</a></li>
<li class="chapter" data-level="6.5" data-path="loading-the-example-dataset.html"><a href="loading-the-example-dataset.html"><i class="fa fa-check"></i><b>6.5</b> Loading the Example Dataset</a></li>
<li class="chapter" data-level="6.6" data-path="data-viewing.html"><a href="data-viewing.html"><i class="fa fa-check"></i><b>6.6</b> Data Viewing</a></li>
<li class="chapter" data-level="6.7" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html"><i class="fa fa-check"></i><b>6.7</b> Evaluating the Original Physicochemical Property Data across Substances</a><ul>
<li class="chapter" data-level="6.7.1" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html#with-these-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>6.7.1</b> With these, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html"><i class="fa fa-check"></i><b>6.8</b> Identifying Clusters of Chemicals through k-means</a><ul>
<li class="chapter" data-level="6.8.1" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html#with-this-we-can-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>6.8.1</b> With this, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html"><i class="fa fa-check"></i><b>6.9</b> Principal Component Analysis (PCA)</a><ul>
<li class="chapter" data-level="6.9.1" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-these-results-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>6.9.1</b> With these results, we can answer <strong>Environmental Health Question #3</strong>:</a></li>
<li class="chapter" data-level="6.9.2" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>6.9.2</b> With this, we can answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="6.9.3" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-5"><i class="fa fa-check"></i><b>6.9.3</b> With this, we can answer <strong>Environmental Health Question #5</strong>:</a></li>
<li class="chapter" data-level="6.9.4" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#we-can-also-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>6.9.4</b> We can also answer <strong>Environmental Health Question #6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="concluding-remarks-5.html"><a href="concluding-remarks-5.html"><i class="fa fa-check"></i><b>6.10</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html"><i class="fa fa-check"></i><b>7</b> Mixtures Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html"><i class="fa fa-check"></i><b>7.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="7.1.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html#introduction-to-mixtures-toxicology-and-in-silico-modeling-to-address-mixtures"><i class="fa fa-check"></i><b>7.1.1</b> Introduction to Mixtures Toxicology and <em>In Silico</em> Modeling to Address Mixtures</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="introduction-to-training-module-4.html"><a href="introduction-to-training-module-4.html"><i class="fa fa-check"></i><b>7.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="7.3" data-path="training-modules-environmental-health-questions-1.html"><a href="training-modules-environmental-health-questions-1.html"><i class="fa fa-check"></i><b>7.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="7.4" data-path="script-preparations-3.html"><a href="script-preparations-3.html"><i class="fa fa-check"></i><b>7.4</b> Script Preparations</a></li>
<li class="chapter" data-level="7.5" data-path="loading-the-example-ginkgo-biloba-datasets.html"><a href="loading-the-example-ginkgo-biloba-datasets.html"><i class="fa fa-check"></i><b>7.5</b> Loading the Example <em>Ginkgo biloba</em> Datasets</a></li>
<li class="chapter" data-level="7.6" data-path="data-viewing-1.html"><a href="data-viewing-1.html"><i class="fa fa-check"></i><b>7.6</b> Data Viewing</a><ul>
<li class="chapter" data-level="7.6.1" data-path="data-viewing-1.html"><a href="data-viewing-1.html#lets-also-see-what-kind-of-data-are-organized-within-the-datasets"><i class="fa fa-check"></i><b>7.6.1</b> Let’s also see what kind of data are organized within the datasets</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>7.7</b> Chemistry-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="7.7.1" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-chemical-profiles"><i class="fa fa-check"></i><b>7.7.1</b> This plot tells us a lot about sample groupings based on chemical profiles!</a></li>
<li class="chapter" data-level="7.7.2" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-questions-1-2"><i class="fa fa-check"></i><b>7.7.2</b> With this, we can answer <strong>Environmental Health Questions 1-2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><i class="fa fa-check"></i><b>7.8</b> Creating a Heat Map of the Mixtures Chemistry Data</a><ul>
<li class="chapter" data-level="7.8.1" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#this-plot-tells-us-a-lot-about-the-individual-chemicals-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>7.8.1</b> This plot tells us a lot about the individual chemicals that differentiate the sample groupings</a></li>
<li class="chapter" data-level="7.8.2" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-this-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>7.8.2</b> With this, we can answer <strong>Environmental Health Question 3</strong>:</a></li>
<li class="chapter" data-level="7.8.3" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#lets-now-revisit-the-pca-plot"><i class="fa fa-check"></i><b>7.8.3</b> Let’s now revisit the PCA plot</a></li>
<li class="chapter" data-level="7.8.4" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-these-plots-side-by-side-we-can-now-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>7.8.4</b> With these plots, side-by-side, we can now answer <strong>Environmental Health Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>7.9</b> Toxicity-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="7.9.1" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-toxicity-profiles"><i class="fa fa-check"></i><b>7.9.1</b> This plot tells us a lot about sample groupings based on toxicity profiles!</a></li>
<li class="chapter" data-level="7.9.2" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-question-5-1"><i class="fa fa-check"></i><b>7.9.2</b> With this, we can answer <em>Environmental Health Question 5</em>:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><i class="fa fa-check"></i><b>7.10</b> Creating a Heat Map of the Mixtures Toxicity Data</a><ul>
<li class="chapter" data-level="7.10.1" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#this-plot-tells-us-a-lot-about-the-individual-genes-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>7.10.1</b> This plot tells us a lot about the individual genes that differentiate the sample groupings</a></li>
<li class="chapter" data-level="7.10.2" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#with-this-we-can-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>7.10.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><i class="fa fa-check"></i><b>7.11</b> Comparing Results from the Chemistry vs. Toxicity Sufficient Similarity Analyses</a><ul>
<li class="chapter" data-level="7.11.1" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html#visual-representation-of-some-of-the-major-grouping-similarities-vs-differences"><i class="fa fa-check"></i><b>7.11.1</b> Visual representation of some of the major grouping similarities vs differences</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="these-plots-can-help-us-answer-environmental-health-question-7.html"><a href="these-plots-can-help-us-answer-environmental-health-question-7.html"><i class="fa fa-check"></i><b>7.12</b> These plots can help us answer <strong>Environmental Health Question 7</strong>:</a></li>
<li class="chapter" data-level="7.13" data-path="concluding-remarks-6.html"><a href="concluding-remarks-6.html"><i class="fa fa-check"></i><b>7.13</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html"><i class="fa fa-check"></i><b>8</b> -Omics Analyses and Systems Biology</a><ul>
<li class="chapter" data-level="8.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html"><i class="fa fa-check"></i><b>8.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="8.1.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i><b>8.1.1</b> Introduction to the Field of “-Omics”</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="introduction-to-the-omics-endpoint-transcriptomics.html"><a href="introduction-to-the-omics-endpoint-transcriptomics.html"><i class="fa fa-check"></i><b>8.2</b> Introduction to the -Omics Endpoint, Transcriptomics</a></li>
<li class="chapter" data-level="8.3" data-path="introduction-to-training-module-5.html"><a href="introduction-to-training-module-5.html"><i class="fa fa-check"></i><b>8.3</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="8.4" data-path="training-modules-environmental-health-questions-2.html"><a href="training-modules-environmental-health-questions-2.html"><i class="fa fa-check"></i><b>8.4</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="8.5" data-path="script-preparations-4.html"><a href="script-preparations-4.html"><i class="fa fa-check"></i><b>8.5</b> Script Preparations</a></li>
<li class="chapter" data-level="8.6" data-path="loading-the-example-transcriptomic-dataset-and-metadata.html"><a href="loading-the-example-transcriptomic-dataset-and-metadata.html"><i class="fa fa-check"></i><b>8.6</b> Loading the Example Transcriptomic Dataset and Metadata</a></li>
<li class="chapter" data-level="8.7" data-path="data-viewing-2.html"><a href="data-viewing-2.html"><i class="fa fa-check"></i><b>8.7</b> Data Viewing</a><ul>
<li class="chapter" data-level="8.7.1" data-path="data-viewing-2.html"><a href="data-viewing-2.html#with-this-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>8.7.1</b> With this, we can answer <strong>Environmental Health Question 1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="formatting-data-for-downstream-statistics.html"><a href="formatting-data-for-downstream-statistics.html"><i class="fa fa-check"></i><b>8.8</b> Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="8.9" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html"><i class="fa fa-check"></i><b>8.9</b> Transcriptomics Data QA/QC</a><ul>
<li class="chapter" data-level="8.9.1" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#background-filter"><i class="fa fa-check"></i><b>8.9.1</b> Background Filter</a></li>
<li class="chapter" data-level="8.9.2" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#sample-filtering"><i class="fa fa-check"></i><b>8.9.2</b> Sample Filtering</a></li>
<li class="chapter" data-level="8.9.3" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#identifying-removing-sample-outliers"><i class="fa fa-check"></i><b>8.9.3</b> Identifying &amp; Removing Sample Outliers</a></li>
<li class="chapter" data-level="8.9.4" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#with-this-we-can-answer-environmental-health-question-2-1"><i class="fa fa-check"></i><b>8.9.4</b> With this, we can answer <strong>Environmental Health Question 2</strong>:</a></li>
<li class="chapter" data-level="8.9.5" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#we-can-also-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>8.9.5</b> We can also answer <strong>Environmental Health Question 3</strong>:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html"><i class="fa fa-check"></i><b>9</b> Controlling for Sources of Sample Heterogeneity</a><ul>
<li class="chapter" data-level="9.0.1" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html#with-this-we-can-answer-environmental-health-question-4-1"><i class="fa fa-check"></i><b>9.0.1</b> With this, we can answer <strong>Environmental Health Question 4</strong>:</a></li>
<li class="chapter" data-level="9.1" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><i class="fa fa-check"></i><b>9.1</b> Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a><ul>
<li class="chapter" data-level="9.1.1" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-5-2"><i class="fa fa-check"></i><b>9.1.1</b> With this, we can answer <strong>Environmental Health Question 5</strong>:</a></li>
<li class="chapter" data-level="9.1.2" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-6-1"><i class="fa fa-check"></i><b>9.1.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
<li class="chapter" data-level="9.1.3" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-7"><i class="fa fa-check"></i><b>9.1.3</b> With this, we can answer <strong>Environmental Health Question 7</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="visualizing-statistical-results-using-ma-plots.html"><a href="visualizing-statistical-results-using-ma-plots.html"><i class="fa fa-check"></i><b>9.2</b> Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="9.3" data-path="visualizing-statistical-results-using-volcano-plots.html"><a href="visualizing-statistical-results-using-volcano-plots.html"><i class="fa fa-check"></i><b>9.3</b> Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="9.4" data-path="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><i class="fa fa-check"></i><b>9.4</b> Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a><ul>
<li class="chapter" data-level="9.4.1" data-path="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#with-this-we-can-answer-environmental-health-question-8"><i class="fa fa-check"></i><b>9.4.1</b> With this, we can answer <strong>Environmental Health Question 8</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="concluding-remarks-7.html"><a href="concluding-remarks-7.html"><i class="fa fa-check"></i><b>9.5</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i><b>10</b> Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="11" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html"><i class="fa fa-check"></i><b>11</b> Read-Across Toxicity Predictions</a></li>
<li class="chapter" data-level="12" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html"><i class="fa fa-check"></i><b>12</b> Environmental Health Databases: Comparative Toxicogenomics Database</a><ul>
<li class="chapter" data-level="12.1" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html"><i class="fa fa-check"></i><b>12.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="12.1.1" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html#introduction-to-comparative-toxicogenomics-database-ctd"><i class="fa fa-check"></i><b>12.1.1</b> Introduction to Comparative Toxicogenomics Database (CTD)</a></li>
<li class="chapter" data-level="12.1.2" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html#introduction-to-training-module-6"><i class="fa fa-check"></i><b>12.1.2</b> Introduction to Training Module</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="training-modules-environmental-health-questions-3.html"><a href="training-modules-environmental-health-questions-3.html"><i class="fa fa-check"></i><b>12.2</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="12.3" data-path="script-preparations-5.html"><a href="script-preparations-5.html"><i class="fa fa-check"></i><b>12.3</b> Script Preparations</a></li>
<li class="chapter" data-level="12.4" data-path="organizing-example-dataset-from-ctd.html"><a href="organizing-example-dataset-from-ctd.html"><i class="fa fa-check"></i><b>12.4</b> Organizing Example Dataset from CTD</a></li>
<li class="chapter" data-level="12.5" data-path="loading-the-example-ctd-dataset-into-r.html"><a href="loading-the-example-ctd-dataset-into-r.html"><i class="fa fa-check"></i><b>12.5</b> Loading the Example CTD Dataset into R</a></li>
<li class="chapter" data-level="12.6" data-path="data-viewing-3.html"><a href="data-viewing-3.html"><i class="fa fa-check"></i><b>12.6</b> Data Viewing</a></li>
<li class="chapter" data-level="12.7" data-path="filtering-the-data-for-genes-with-altered-expression.html"><a href="filtering-the-data-for-genes-with-altered-expression.html"><i class="fa fa-check"></i><b>12.7</b> Filtering the Data for Genes with Altered Expression</a><ul>
<li class="chapter" data-level="12.7.1" data-path="filtering-the-data-for-genes-with-altered-expression.html"><a href="filtering-the-data-for-genes-with-altered-expression.html#with-this-we-can-answer-environmental-health-question-1-1"><i class="fa fa-check"></i><b>12.7.1</b> With this, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="identifying-genes-under-epigenetic-control.html"><a href="identifying-genes-under-epigenetic-control.html"><i class="fa fa-check"></i><b>12.8</b> Identifying Genes under Epigenetic Control</a></li>
<li class="chapter" data-level="12.9" data-path="method-1-for-list-comparisons-merging.html"><a href="method-1-for-list-comparisons-merging.html"><i class="fa fa-check"></i><b>12.9</b> Method 1 for list comparisons: Merging</a><ul>
<li class="chapter" data-level="12.9.1" data-path="method-1-for-list-comparisons-merging.html"><a href="method-1-for-list-comparisons-merging.html#with-this-we-can-answer-environmental-health-question-2-2"><i class="fa fa-check"></i><b>12.9.1</b> With this, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="12.10" data-path="method-2-for-list-comparisons-intersection.html"><a href="method-2-for-list-comparisons-intersection.html"><i class="fa fa-check"></i><b>12.10</b> Method 2 for list comparisons: Intersection</a></li>
<li class="chapter" data-level="12.11" data-path="concluding-remarks-8.html"><a href="concluding-remarks-8.html"><i class="fa fa-check"></i><b>12.11</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html"><i class="fa fa-check"></i><b>13</b> Environmental Health Databases: Gene Expression Omnibus</a><ul>
<li class="chapter" data-level="13.1" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html"><i class="fa fa-check"></i><b>13.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="13.1.1" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html#introduction-to-the-environmental-health-database-gene-expression-omnibus-geo"><i class="fa fa-check"></i><b>13.1.1</b> Introduction to the Environmental Health Database, Gene Expression Omnibus (GEO)</a></li>
<li class="chapter" data-level="13.1.2" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html#introduction-to-training-module-7"><i class="fa fa-check"></i><b>13.1.2</b> Introduction to Training Module</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="training-modules-environmental-health-questions-4.html"><a href="training-modules-environmental-health-questions-4.html"><i class="fa fa-check"></i><b>13.2</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="13.3" data-path="script-preparations-6.html"><a href="script-preparations-6.html"><i class="fa fa-check"></i><b>13.3</b> Script Preparations</a></li>
<li class="chapter" data-level="13.4" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html"><i class="fa fa-check"></i><b>13.4</b> Loading and Organizing the Example Dataset</a><ul>
<li class="chapter" data-level="13.4.1" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html#manually-downloading-and-uploading-geo-files"><i class="fa fa-check"></i><b>13.4.1</b> 1. Manually Downloading and Uploading GEO Files</a></li>
<li class="chapter" data-level="13.4.2" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html#uploading-and-organizing-geo-files-through-the-geoquery-package"><i class="fa fa-check"></i><b>13.4.2</b> 2. Uploading and Organizing GEO Files through the GEOquery Package</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><i class="fa fa-check"></i><b>13.5</b> Adding Gene Symbol Information to -Omic Data Sets through Platform Annotation Files</a><ul>
<li class="chapter" data-level="13.5.1" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html#with-this-we-can-now-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>13.5.1</b> With this, we can now answer <strong>Environmental Health Question #1</strong>:</a></li>
<li class="chapter" data-level="13.5.2" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html#we-can-also-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>13.5.2</b> We can also answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="visualizing-gene-expression-data-using-boxplots-and-heat-maps.html"><a href="visualizing-gene-expression-data-using-boxplots-and-heat-maps.html"><i class="fa fa-check"></i><b>13.6</b> Visualizing Gene Expression Data using Boxplots and Heat Maps</a></li>
<li class="chapter" data-level="13.7" data-path="boxplot-visualizations-1.html"><a href="boxplot-visualizations-1.html"><i class="fa fa-check"></i><b>13.7</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="13.8" data-path="heatmap-visualizations-1.html"><a href="heatmap-visualizations-1.html"><i class="fa fa-check"></i><b>13.8</b> Heatmap Visualizations</a><ul>
<li class="chapter" data-level="13.8.1" data-path="heatmap-visualizations-1.html"><a href="heatmap-visualizations-1.html#we-can-also-answer-environmental-health-question-3-1"><i class="fa fa-check"></i><b>13.8.1</b> We can also answer <strong>Environmental Health Question #3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="statistical-analyses-to-identify-genes-altered-by-formaldehyde.html"><a href="statistical-analyses-to-identify-genes-altered-by-formaldehyde.html"><i class="fa fa-check"></i><b>13.9</b> Statistical Analyses to Identify Genes altered by Formaldehyde</a></li>
<li class="chapter" data-level="13.10" data-path="interpreting-results.html"><a href="interpreting-results.html"><i class="fa fa-check"></i><b>13.10</b> Interpreting Results</a><ul>
<li class="chapter" data-level="13.10.1" data-path="interpreting-results.html"><a href="interpreting-results.html#with-this-we-can-answer-environmental-health-question-4-2"><i class="fa fa-check"></i><b>13.10.1</b> With this, we can answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="13.10.2" data-path="interpreting-results.html"><a href="interpreting-results.html#with-this-we-can-answer-environmental-health-question-5-3"><i class="fa fa-check"></i><b>13.10.2</b> With this, we can answer <strong>Environmental Health Question #5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="13.11" data-path="concluding-remarks-9.html"><a href="concluding-remarks-9.html"><i class="fa fa-check"></i><b>13.11</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="database-integration-air-quality-study-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-study-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i><b>14</b> Database Integration: Air Quality Study, Mortality, and Environmental Justice Data</a></li>
<li class="chapter" data-level="15" data-path="additional-resources.html"><a href="additional-resources.html"><i class="fa fa-check"></i><b>15</b> Additional Resources</a><ul>
<li class="chapter" data-level="15.1" data-path="r-programming-resources.html"><a href="r-programming-resources.html"><i class="fa fa-check"></i><b>15.1</b> R Programming Resources:</a></li>
<li class="chapter" data-level="15.2" data-path="r-packages-resources.html"><a href="r-packages-resources.html"><i class="fa fa-check"></i><b>15.2</b> R Packages Resources:</a></li>
<li class="chapter" data-level="15.3" data-path="community-discussions-on-r-and-r-packages.html"><a href="community-discussions-on-r-and-r-packages.html"><i class="fa fa-check"></i><b>15.3</b> Community Discussions on R and R Packages:</a></li>
<li class="chapter" data-level="15.4" data-path="data-science-and-statistical-analysis-resources.html"><a href="data-science-and-statistical-analysis-resources.html"><i class="fa fa-check"></i><b>15.4</b> Data Science and Statistical Analysis Resources:</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="controlling-for-sources-of-sample-heterogeneity" class="section level1">
<h1><span class="header-section-number">Chapter 9</span> Controlling for Sources of Sample Heterogeneity</h1>
<p>Because these transcriptomic data were generated from mouse lung tissues, there is potential for these samples to show heterogeneity based on underlying shifts in cell populations (e.g., neutrophil influx) or other aspects of sample heterogeneity (e.g., batch effects from plating, among other sources of heterogeneity that we may want to control for). For these kinds of complex samples, there are data processing methods that can be leveraged to minimize the influence of these sources of heterogeneity. Example methods include Remove Unwanted Variable (RUV), which is discussed here, as well as others (e.g., <a href="https://academic.oup.com/nar/article/42/21/e161/2903156">Surrogate Variable Analysis (SVA)</a>).</p>
<p>Here, we leverage the package called <strong>RUVseq</strong> to employ RUV on this sequencing dataset. Script was developed based off a helpful online training tutorial (LAUREN- DO YOU REMEMBER THIS WEBSITE THAT WE USED HERE?), as well as its parent <a href="https://bioconductor.org/packages/release/bioc/html/RUVSeq.html">Bioconductor website</a>, <a href="http://bioconductor.org/packages/release/bioc/vignettes/RUVSeq/inst/doc/RUVSeq.pdf">vignette</a>, and original <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4404308/">publication</a>.</p>
<div id="steps-in-carrying-out-ruv-using-ruvseq-on-this-example-dataset" class="section level4">
<h4><span class="header-section-number">9.0.0.1</span> Steps in carrying out RUV using RUVseq on this example dataset:</h4>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb44-1"></a><span class="co"># First we store the treatment IDs and exposure conditions as a separate vector</span></span>
<span id="cb44-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb44-2"></a>ID &lt;-<span class="st"> </span>coldata<span class="op">$</span>ID</span>
<span id="cb44-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb44-3"></a></span>
<span id="cb44-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb44-4"></a><span class="co"># And differentiate our treatments and control conditions, first by grabbing the groups associated with each sample</span></span>
<span id="cb44-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb44-5"></a>groups &lt;-<span class="st"> </span><span class="kw">as.factor</span>(coldata<span class="op">$</span>Group)</span>
<span id="cb44-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb44-6"></a></span>
<span id="cb44-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb44-7"></a><span class="co"># let&#39;s view all the groups</span></span>
<span id="cb44-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb44-8"></a>groups</span></code></pre></div>
<pre><code>##  [1] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [3] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [5] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [7] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
##  [9] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
## [11] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
## [13] Saline_4h_Lung             Saline_4h_Lung            
## [15] Saline_4h_Lung             Saline_4h_Lung            
## [17] Saline_4h_Lung             Saline_4h_Lung            
## [19] LPS_4h_Lung                LPS_4h_Lung               
## [21] LPS_4h_Lung                LPS_4h_Lung               
## 4 Levels: LPS_4h_Lung PineNeedlesFlame_4h_Lung ... Saline_4h_Lung</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb46-1"></a><span class="co"># then setting a control label</span></span>
<span id="cb46-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb46-2"></a>ctrl &lt;-<span class="st"> &quot;Saline_4h_Lung&quot;</span></span>
<span id="cb46-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb46-3"></a></span>
<span id="cb46-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb46-4"></a><span class="co"># and extracting a vector of just our treatment groups</span></span>
<span id="cb46-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb46-5"></a>trt_groups &lt;-<span class="st"> </span><span class="kw">setdiff</span>(groups,ctrl)</span>
<span id="cb46-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb46-6"></a></span>
<span id="cb46-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb46-7"></a><span class="co"># let&#39;s view this vector</span></span>
<span id="cb46-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb46-8"></a>trt_groups</span></code></pre></div>
<pre><code>## [1] &quot;PineNeedlesSmolder_4h_Lung&quot; &quot;PineNeedlesFlame_4h_Lung&quot;  
## [3] &quot;LPS_4h_Lung&quot;</code></pre>
<p>RUVseq contains its own set of plotting and normalization functions, though requires input of what’s called an object of S4 class SeqExpressionSet. Let’s go ahead and make this object, using the RUVseq function ‘newSeqExpressionSet’:</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb48-1"></a>exprSet &lt;-<span class="st"> </span><span class="kw">newSeqExpressionSet</span>(<span class="kw">as.matrix</span>(countdata),<span class="dt">phenoData =</span> <span class="kw">data.frame</span>(groups,<span class="dt">row.names=</span><span class="kw">colnames</span>(countdata)))</span></code></pre></div>
<p>And then use this object to generate some exploratory plots using built-in tools within RUVseq.
First starting with some bar charts summarizing overall data distributions per sample:</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb49-1"></a>colors &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set2&quot;</span>)</span>
<span id="cb49-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb49-2"></a><span class="kw">plotRLE</span>(exprSet, <span class="dt">outline=</span><span class="ot">FALSE</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">4</span>, <span class="dv">4</span>), <span class="dt">col=</span>colors[groups])</span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can see from this plot that some of the samples show distributions that may vary from the overall - for instance, one of the flaming pine needles-exposed samples (in orange) is far lower than the rest.</p>
<p>Then viewing a PCA plot of these samples:</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb50-1"></a>colors &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set2&quot;</span>)</span>
<span id="cb50-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb50-2"></a><span class="kw">plotPCA</span>(exprSet, <span class="dt">col=</span>colors[groups], <span class="dt">cex=</span><span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This PCA plot shows pretty good data distributions, with samples mainly showing groupings based upon exposure condition (e.g., LPS), which is to be expected. With this, we can conclude that there may be some sources of unwanted variation, but not a huge amount. Let’s see what the data look like after running RUV.</p>
<p>Now to actually run the RUVseq algorithm, to control for potential sources of sample heterogeneity, we need to first construct a matrix specifying the replicates (samples of the same exposure condition):</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb51-1"></a><span class="co"># Construct a matrix specifying the replicates (samples of the same exposure condition) for running RUV</span></span>
<span id="cb51-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb51-2"></a>differences &lt;-<span class="st"> </span><span class="kw">makeGroups</span>(groups)</span>
<span id="cb51-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb51-3"></a></span>
<span id="cb51-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb51-4"></a><span class="co"># Viewing this new matrix</span></span>
<span id="cb51-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb51-5"></a><span class="kw">head</span>(differences)</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3] [,4] [,5] [,6]
## [1,]   19   20   21   22   -1   -1
## [2,]    7    8    9   10   11   12
## [3,]    1    2    3    4    5    6
## [4,]   13   14   15   16   17   18</code></pre>
<p>LAUREN - CAN WE PLEASE ADD SOME TEXT HERE DESCRIBING WHAT THIS MATRIX IS SHOWING?</p>
<p>LK DRAFTED TEXT- This matrix groups the samples by exposure condition. Here, each of the four rows represents one of the four exposure conditions, and each of the six columns represents a possible sample. Since the LPS exposure condition only had four samples, instead of six like the rest of the exposure conditions, a value of -1 is automatically used as a place holder to fill out the matrix. The samples in the matrix are identified by the index of the sample in the previously defined ‘groups’ factor that was used to generate the matrix. For example, the PineNeedlesSmolder_4h_Lung samples are the the first six samples contained in the ‘groups’ factor, so in the matrix, samples of this exposure condition are identified as ‘1’,‘2’,‘3’,‘4’,‘5’, and ‘6’.</p>
<p>Let’s now implement the RUVseq algorithm and, for this example, capture one factor (k=1) of unwanted variation. Note that the k parameter can be modified to capture additional factors, though in this case, it appears that the majority of unwanted variation is likely captured through just a single factor, as shown below.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb53-1"></a><span class="co"># Now capture 1 factor (k=1) of unwanted variation</span></span>
<span id="cb53-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb53-2"></a>ruv_set &lt;-<span class="st"> </span><span class="kw">RUVs</span>(exprSet, <span class="kw">rownames</span>(countdata), <span class="dt">k=</span><span class="dv">1</span>, differences) </span></code></pre></div>
<p>This results in a list of objects within ‘ruv_set’, which include the following important pieces of information that are used in the final statistical model:</p>
<ol style="list-style-type: decimal">
<li>Estimated factors of unwanted variation are provided in the phenoData object, as viewed using the following:</li>
</ol>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb54-1"></a><span class="co"># viewing the estimated factors of unwanted variation in the column W_1</span></span>
<span id="cb54-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb54-2"></a><span class="kw">pData</span>(ruv_set)</span></code></pre></div>
<pre><code>##                                            groups       W_1
## M13_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.3777900
## M14_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.2392662
## M15_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.0687702
## M16_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.2448747
## M17_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.3156003
## M18_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.4136600
## M49_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.3343115
## M50_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 0.4446084
## M51_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.0172990
## M52_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.0332373
## M53_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 0.7746752
## M54_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.0011086
## M31_Saline                         Saline_4h_Lung 0.6582117
## M32_Saline                         Saline_4h_Lung 1.2518829
## M33_Saline                         Saline_4h_Lung 1.1892456
## M34_Saline                         Saline_4h_Lung 1.2607852
## M35_Saline                         Saline_4h_Lung 1.3141520
## M36_Saline                         Saline_4h_Lung 1.4335200
## M67_LPS                               LPS_4h_Lung 1.0341947
## M68_LPS                               LPS_4h_Lung 1.2823240
## M69_LPS                               LPS_4h_Lung 1.4460347
## M70_LPS                               LPS_4h_Lung 1.4376930</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Normalized counts obtained by regressing the original counts on the unwanted factors (normalizedCounts object within ‘ruv_set’), as viewed using the following:</li>
</ol>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb56-1"></a><span class="co"># viewing the head of the normalized count data, accounting for unwanted variation</span></span>
<span id="cb56-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb56-2"></a><span class="kw">head</span>(<span class="kw">normCounts</span>(ruv_set))</span></code></pre></div>
<pre><code>##                     M13_PineNeedlesSmolder M14_PineNeedlesSmolder
## 0610009B22Rik_56046                      4                      5
## 0610010F05Rik_69119                   2552                   2630
## 0610010F05Rik_74637                     17                     13
## 0610010K14Rik_31619                   1602                   1774
## 0610010K14Rik_31873                    711                   1009
## 0610010K14Rik_68949                     23                     25
##                     M15_PineNeedlesSmolder M16_PineNeedlesSmolder
## 0610009B22Rik_56046                      8                      8
## 0610010F05Rik_69119                   2474                   1634
## 0610010F05Rik_74637                     17                      3
## 0610010K14Rik_31619                   1639                   1722
## 0610010K14Rik_31873                    819                    992
## 0610010K14Rik_68949                     19                     28
##                     M17_PineNeedlesSmolder M18_PineNeedlesSmolder
## 0610009B22Rik_56046                      6                      4
## 0610010F05Rik_69119                   2343                   2352
## 0610010F05Rik_74637                     12                      7
## 0610010K14Rik_31619                   1979                   1726
## 0610010K14Rik_31873                    889                    888
## 0610010K14Rik_68949                     20                     30
##                     M49_PineNeedlesFlame M50_PineNeedlesFlame
## 0610009B22Rik_56046                    4                    5
## 0610010F05Rik_69119                 1767                 1563
## 0610010F05Rik_74637                   12                    6
## 0610010K14Rik_31619                 1480                 1134
## 0610010K14Rik_31873                  586                  599
## 0610010K14Rik_68949                   23                   27
##                     M51_PineNeedlesFlame M52_PineNeedlesFlame
## 0610009B22Rik_56046                    4                    4
## 0610010F05Rik_69119                 1918                 1711
## 0610010F05Rik_74637                    4                    9
## 0610010K14Rik_31619                 1390                 1763
## 0610010K14Rik_31873                  683                  797
## 0610010K14Rik_68949                   26                   27
##                     M53_PineNeedlesFlame M54_PineNeedlesFlame M31_Saline
## 0610009B22Rik_56046                    5                    5          3
## 0610010F05Rik_69119                 1751                 1582       1891
## 0610010F05Rik_74637                    6                    3         10
## 0610010K14Rik_31619                 1370                 1419       1340
## 0610010K14Rik_31873                  593                  647        561
## 0610010K14Rik_68949                   20                   24         19
##                     M32_Saline M33_Saline M34_Saline M35_Saline M36_Saline
## 0610009B22Rik_56046          2          4          3          4          4
## 0610010F05Rik_69119       2507       2518       2524       1831       1878
## 0610010F05Rik_74637          7          7          5          7          8
## 0610010K14Rik_31619       1563       1632       1438       1280        967
## 0610010K14Rik_31873        685        815        703        646        526
## 0610010K14Rik_68949         16         21         19         22         20
##                     M67_LPS M68_LPS M69_LPS M70_LPS
## 0610009B22Rik_56046       3       2       4       5
## 0610010F05Rik_69119    1906    1597    1461    1201
## 0610010F05Rik_74637       6       6       7       5
## 0610010K14Rik_31619     964    1012     776     638
## 0610010K14Rik_31873     622     702     522     496
## 0610010K14Rik_68949      21      21      18      15</code></pre>
<p>Let’s again generate an exploratory plot using this updated dataset, focusing on the bar chart view since that was the most informative pre-RUV. Here are the updated bar charts summarizing overall data distributions per sample:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb58-1"></a>colors &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set2&quot;</span>)</span>
<span id="cb58-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb58-2"></a><span class="kw">plotRLE</span>(ruv_set, <span class="dt">outline=</span><span class="ot">FALSE</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">4</span>, <span class="dv">4</span>), <span class="dt">col=</span>colors[groups])</span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This plot shows overall tighter data that are more similarly distributed across samples. Therefore, it is looking like this RUV addition improved the overall distribution of this dataset. It is important not to over-correct/over-smooth your datasets, so implement these types of pre-processing steps with caution. One strategy that we commonly employ to gage whether data smoothing is needed/applied correctly is to run the statistical models with and without correction of potential sources of heterogeneity, and critically evaluate similarities vs differences produced in the results.</p>
</div>
<div id="with-this-we-can-answer-environmental-health-question-4-1" class="section level3">
<h3><span class="header-section-number">9.0.1</span> With this, we can answer <strong>Environmental Health Question 4</strong>:</h3>
<div id="what-is-an-approach-that-analysts-can-use-when-evaluating-transcriptomic-data-from-tissues-of-mixed-cellular-composition-to-aid-in-controlling-for-sources-of-sample-heterogeneity" class="section level4">
<h4><span class="header-section-number">9.0.1.1</span> What is an approach that analysts can use when evaluating transcriptomic data from tissues of mixed cellular composition to aid in controlling for sources of sample heterogeneity?</h4>
</div>
<div id="answer-remove-unwanted-variation-ruv-among-other-approaches-including-surrogate-variable-analysis-sva." class="section level4">
<h4><span class="header-section-number">9.0.1.2</span> <em>Answer: Remove unwanted variation (RUV), among other approaches, including surrogate variable analysis (SVA).</em></h4>
<p><br></p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="transcriptomics-data-qaqc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
