<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Controlling for Sources of Sample Heterogeneity | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="Controlling for Sources of Sample Heterogeneity | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Controlling for Sources of Sample Heterogeneity | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Controlling for Sources of Sample Heterogeneity | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle Roell, Lauren Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/Favicon_v1.png" type="image/x-icon" />
<link rel="prev" href="omics-analyses-and-systems-biology.html"/>
<link rel="next" href="toxicokinetic-modeling.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/Data Training Icon for Top Left Bookdown_v2.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html"><i class="fa fa-check"></i>1.1 Introduction to Coding in R</a><ul>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#r-packages"><i class="fa fa-check"></i>R Packages</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation using Base R</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#basic-data-anaylsis"><i class="fa fa-check"></i>Basic Data Anaylsis</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#regression-modeling"><i class="fa fa-check"></i>Regression Modeling</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#categorical-data-analysis"><i class="fa fa-check"></i>Categorical Data Analysis</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#high-dimensional-data-visualizations"><i class="fa fa-check"></i>High-Dimensional Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#density-plot-visualizations"><i class="fa fa-check"></i>Density Plot Visualizations</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#ggally-visualizations"><i class="fa fa-check"></i>GGally Visualizations</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#boxplot-visualizations"><i class="fa fa-check"></i>Boxplot Visualizations</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#correlation-plot-visualizations"><i class="fa fa-check"></i>Correlation Plot Visualizations</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#hierarchical-clustering-visualizations"><i class="fa fa-check"></i>Hierarchical Clustering Visualizations</a><ul>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#heatmap-visualizations"><i class="fa fa-check"></i>Heatmap Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#helpful-resources-on-fair"><i class="fa fa-check"></i>Helpful Resources on FAIR</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Chemical-<br>Biological Analyses <br>and Predictive Modeling</b></span></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html"><i class="fa fa-check"></i>2.1 Dose-Response Modeling</a><ul>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#plotting-data-in-dose-response"><i class="fa fa-check"></i>Plotting Data in Dose-Response</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#dose-response-curve-fitting"><i class="fa fa-check"></i>Dose-Response Curve Fitting</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#comparing-curve-fits"><i class="fa fa-check"></i>Comparing Curve Fits</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#curve-fitting-example"><i class="fa fa-check"></i>Curve Fitting Example</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#deriving-benchmark-dose"><i class="fa fa-check"></i>Deriving Benchmark Dose</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>2.2 Machine Learning and Predictive Modeling for Environmental Health</a><ul>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#k-means-analysis"><i class="fa fa-check"></i>K-means Analysis</a></li>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#principal-component-analysis"><i class="fa fa-check"></i>Principal Component Analysis</a></li>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#combining-k-means-with-pca"><i class="fa fa-check"></i>Combining K-Means with PCA</a></li>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html"><i class="fa fa-check"></i>2.3 Mixtures Analysis</a><ul>
<li class="chapter" data-level="" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#chemistry-based-approach"><i class="fa fa-check"></i>Chemistry-based Approach</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#toxicity-based-approach"><i class="fa fa-check"></i>Toxicity-based Approach</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#comparing-results"><i class="fa fa-check"></i>Comparing Results</a><ul>
<li class="chapter" data-level="" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#visual-representation-of-some-of-the-major-grouping-similarities-vs-differences"><i class="fa fa-check"></i>Visual representation of some of the major grouping similarities vs differences</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html"><i class="fa fa-check"></i>2.4 -Omics Analyses and Systems Biology</a><ul>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li><a href="omics-analyses-and-systems-biology.html#training-modules-environmental-health-questions-3">Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#identifying-removing-sample-outliers"><i class="fa fa-check"></i>Identifying &amp; Removing Sample Outliers</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a><ul>
<li class="chapter" data-level="" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html#significantly-differentially-expressed-genes"><i class="fa fa-check"></i>Significantly Differentially Expressed Genes</a></li>
<li class="chapter" data-level="" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html#visualizing-statistical-results"><i class="fa fa-check"></i>Visualizing Statistical Results</a></li>
<li class="chapter" data-level="" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html#ma-plots"><i class="fa fa-check"></i>MA Plots</a></li>
<li class="chapter" data-level="" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html#volcano-plots"><i class="fa fa-check"></i>Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html#pathway-enrichment-analysis"><i class="fa fa-check"></i>Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>2.5 Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html"><i class="fa fa-check"></i>2.6 Read-Across Toxicity Predictions</a></li>
<li class="part"><span><b>Chapter 3 Environmental <br>Health DatabaseMining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>3.1 Comparative Toxicogenomics Database</a><ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a><ul>
<li><a href="comparative-toxicogenomics-database.html#training-modules-environmental-health-questions-5">Training Module’s <strong>Environmental Health Questions</strong></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Additional Resources</b></span><ul>
<li class="chapter" data-level="" data-path=""><a href="#r-programming-resources"><i class="fa fa-check"></i>R Programming Resources:</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#r-packages-resources"><i class="fa fa-check"></i>R Packages Resources:</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#community-discussions-on-r-and-r-packages"><i class="fa fa-check"></i>Community Discussions on R and R Packages:</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#data-science-and-statistical-analysis-resources"><i class="fa fa-check"></i>Data Science and Statistical Analysis Resources:</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="controlling-for-sources-of-sample-heterogeneity" class="section level1">
<h1>Controlling for Sources of Sample Heterogeneity</h1>
<p>Because these transcriptomic data were generated from mouse lung tissues, there is potential for these samples to show heterogeneity based on underlying shifts in cell populations (e.g., neutrophil influx) or other aspects of sample heterogeneity (e.g., batch effects from plating, among other sources of heterogeneity that we may want to control for). For these kinds of complex samples, there are data processing methods that can be leveraged to minimize the influence of these sources of heterogeneity. Example methods include Remove Unwanted Variable (RUV), which is discussed here, as well as others (e.g., <a href="https://academic.oup.com/nar/article/42/21/e161/2903156">Surrogate Variable Analysis (SVA)</a>).</p>
<p>Here, we leverage the package called <strong>RUVseq</strong> to employ RUV on this sequencing dataset. Script was developed based off <a href="https://bioconductor.org/packages/release/bioc/html/RUVSeq.html">Bioconductor website</a>, <a href="http://bioconductor.org/packages/release/bioc/vignettes/RUVSeq/inst/doc/RUVSeq.pdf">vignette</a>, and original <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4404308/">publication</a>.</p>
<div id="steps-in-carrying-out-ruv-using-ruvseq-on-this-example-dataset" class="section level4">
<h4>Steps in carrying out RUV using RUVseq on this example dataset:</h4>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb184-1"></a><span class="co"># First we store the treatment IDs and exposure conditions as a separate vector</span></span>
<span id="cb184-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb184-2"></a>ID &lt;-<span class="st"> </span>coldata<span class="op">$</span>ID</span>
<span id="cb184-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb184-3"></a></span>
<span id="cb184-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb184-4"></a><span class="co"># And differentiate our treatments and control conditions, first by grabbing </span></span>
<span id="cb184-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb184-5"></a><span class="co"># the groups associated with each sample</span></span>
<span id="cb184-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb184-6"></a>groups &lt;-<span class="st"> </span><span class="kw">as.factor</span>(coldata<span class="op">$</span>Group)</span>
<span id="cb184-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb184-7"></a></span>
<span id="cb184-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb184-8"></a><span class="co"># Let&#39;s view all the groups</span></span>
<span id="cb184-9"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb184-9"></a>groups</span></code></pre></div>
<pre><code>##  [1] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [3] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [5] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [7] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
##  [9] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
## [11] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
## [13] Saline_4h_Lung             Saline_4h_Lung            
## [15] Saline_4h_Lung             Saline_4h_Lung            
## [17] Saline_4h_Lung             Saline_4h_Lung            
## [19] LPS_4h_Lung                LPS_4h_Lung               
## [21] LPS_4h_Lung                LPS_4h_Lung               
## 4 Levels: LPS_4h_Lung PineNeedlesFlame_4h_Lung ... Saline_4h_Lung</code></pre>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb186-1"></a><span class="co"># then setting a control label</span></span>
<span id="cb186-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb186-2"></a>ctrl &lt;-<span class="st"> &quot;Saline_4h_Lung&quot;</span></span>
<span id="cb186-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb186-3"></a></span>
<span id="cb186-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb186-4"></a><span class="co"># and extracting a vector of just our treatment groups</span></span>
<span id="cb186-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb186-5"></a>trt_groups &lt;-<span class="st"> </span><span class="kw">setdiff</span>(groups,ctrl)</span>
<span id="cb186-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb186-6"></a></span>
<span id="cb186-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb186-7"></a><span class="co"># let&#39;s view this vector</span></span>
<span id="cb186-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb186-8"></a>trt_groups</span></code></pre></div>
<pre><code>## [1] &quot;PineNeedlesSmolder_4h_Lung&quot; &quot;PineNeedlesFlame_4h_Lung&quot;  
## [3] &quot;LPS_4h_Lung&quot;</code></pre>
<p>RUVseq contains its own set of plotting and normalization functions, though requires input of what’s called an object of S4 class SeqExpressionSet. Let’s go ahead and make this object, using the RUVseq function ‘newSeqExpressionSet’:</p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb188-1"></a>exprSet &lt;-<span class="st"> </span><span class="kw">newSeqExpressionSet</span>(<span class="kw">as.matrix</span>(countdata),<span class="dt">phenoData =</span> </span>
<span id="cb188-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb188-2"></a>                                 <span class="kw">data.frame</span>(groups,<span class="dt">row.names=</span><span class="kw">colnames</span>(countdata)))</span></code></pre></div>
<p>And then use this object to generate some exploratory plots using built-in tools within RUVseq.
First starting with some bar charts summarizing overall data distributions per sample:</p>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb189-1"></a>colors &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set2&quot;</span>)</span>
<span id="cb189-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb189-2"></a><span class="kw">plotRLE</span>(exprSet, <span class="dt">outline=</span><span class="ot">FALSE</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">4</span>, <span class="dv">4</span>), <span class="dt">col=</span>colors[groups])</span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-124-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can see from this plot that some of the samples show distributions that may vary from the overall - for instance, one of the flaming pine needles-exposed samples (in orange) is far lower than the rest.</p>
<p>Then viewing a PCA plot of these samples:</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb190-1"></a>colors &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set2&quot;</span>)</span>
<span id="cb190-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb190-2"></a><span class="kw">plotPCA</span>(exprSet, <span class="dt">col=</span>colors[groups], <span class="dt">cex=</span><span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-125-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This PCA plot shows pretty good data distributions, with samples mainly showing groupings based upon exposure condition (e.g., LPS), which is to be expected. With this, we can conclude that there may be some sources of unwanted variation, but not a huge amount. Let’s see what the data look like after running RUV.</p>
<p>Now to actually run the RUVseq algorithm, to control for potential sources of sample heterogeneity, we need to first construct a matrix specifying the replicates (samples of the same exposure condition):</p>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb191-1"></a><span class="co"># Construct a matrix specifying the replicates (samples of the same exposure condition) </span></span>
<span id="cb191-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb191-2"></a><span class="co"># for running RUV</span></span>
<span id="cb191-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb191-3"></a>differences &lt;-<span class="st"> </span><span class="kw">makeGroups</span>(groups)</span>
<span id="cb191-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb191-4"></a></span>
<span id="cb191-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb191-5"></a><span class="co"># Viewing this new matrix</span></span>
<span id="cb191-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb191-6"></a><span class="kw">head</span>(differences)</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3] [,4] [,5] [,6]
## [1,]   19   20   21   22   -1   -1
## [2,]    7    8    9   10   11   12
## [3,]    1    2    3    4    5    6
## [4,]   13   14   15   16   17   18</code></pre>
<p>This matrix groups the samples by exposure condition. Here, each of the four rows represents one of the four exposure conditions, and each of the six columns represents a possible sample. Since the LPS exposure condition only had four samples, instead of six like the rest of the exposure conditions, a value of -1 is automatically used as a place holder to fill out the matrix. The samples in the matrix are identified by the index of the sample in the previously defined ‘groups’ factor that was used to generate the matrix. For example, the PineNeedlesSmolder_4h_Lung samples are the the first six samples contained in the ‘groups’ factor, so in the matrix, samples of this exposure condition are identified as ‘1’,‘2’,‘3’,‘4’,‘5’, and ‘6’.</p>
<p>Let’s now implement the RUVseq algorithm and, for this example, capture one factor (k=1) of unwanted variation. Note that the k parameter can be modified to capture additional factors, if necessary.</p>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb193-1"></a><span class="co"># Now capture 1 factor (k=1) of unwanted variation</span></span>
<span id="cb193-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb193-2"></a>ruv_set &lt;-<span class="st"> </span><span class="kw">RUVs</span>(exprSet, <span class="kw">rownames</span>(countdata), <span class="dt">k=</span><span class="dv">1</span>, differences) </span></code></pre></div>
<p>This results in a list of objects within ‘ruv_set’, which include the following important pieces of information:</p>
<ol style="list-style-type: decimal">
<li>Estimated factors of unwanted variation are provided in the phenoData object, as viewed using the following:</li>
</ol>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb194-1"></a><span class="co"># viewing the estimated factors of unwanted variation in the column W_1</span></span>
<span id="cb194-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb194-2"></a><span class="kw">pData</span>(ruv_set)</span></code></pre></div>
<pre><code>##                                            groups       W_1
## M13_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.3777900
## M14_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.2392662
## M15_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.0687702
## M16_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.2448747
## M17_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.3156003
## M18_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.4136600
## M49_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.3343115
## M50_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 0.4446084
## M51_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.0172990
## M52_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.0332373
## M53_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 0.7746752
## M54_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.0011086
## M31_Saline                         Saline_4h_Lung 0.6582117
## M32_Saline                         Saline_4h_Lung 1.2518829
## M33_Saline                         Saline_4h_Lung 1.1892456
## M34_Saline                         Saline_4h_Lung 1.2607852
## M35_Saline                         Saline_4h_Lung 1.3141520
## M36_Saline                         Saline_4h_Lung 1.4335200
## M67_LPS                               LPS_4h_Lung 1.0341947
## M68_LPS                               LPS_4h_Lung 1.2823240
## M69_LPS                               LPS_4h_Lung 1.4460347
## M70_LPS                               LPS_4h_Lung 1.4376930</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Normalized counts obtained by regressing the original counts on the unwanted factors (normalizedCounts object within ‘ruv_set’). Note that the normalized counts should only used for exploratory purposes and not subsequent differential expression analyses. For additional information on this topic, please refer official RUVSeq documentation. The normalized counts can be viewed using the following:</li>
</ol>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb196-1"></a><span class="co"># Viewing the head of the normalized count data, accounting for unwanted variation</span></span>
<span id="cb196-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb196-2"></a><span class="kw">head</span>(<span class="kw">normCounts</span>(ruv_set))</span></code></pre></div>
<pre><code>##                     M13_PineNeedlesSmolder M14_PineNeedlesSmolder
## 0610009B22Rik_56046                      4                      5
## 0610010F05Rik_69119                   2552                   2630
## 0610010F05Rik_74637                     17                     13
## 0610010K14Rik_31619                   1602                   1774
## 0610010K14Rik_31873                    711                   1009
## 0610010K14Rik_68949                     23                     25
##                     M15_PineNeedlesSmolder M16_PineNeedlesSmolder
## 0610009B22Rik_56046                      8                      8
## 0610010F05Rik_69119                   2474                   1634
## 0610010F05Rik_74637                     17                      3
## 0610010K14Rik_31619                   1639                   1722
## 0610010K14Rik_31873                    819                    992
## 0610010K14Rik_68949                     19                     28
##                     M17_PineNeedlesSmolder M18_PineNeedlesSmolder
## 0610009B22Rik_56046                      6                      4
## 0610010F05Rik_69119                   2343                   2352
## 0610010F05Rik_74637                     12                      7
## 0610010K14Rik_31619                   1979                   1726
## 0610010K14Rik_31873                    889                    888
## 0610010K14Rik_68949                     20                     30
##                     M49_PineNeedlesFlame M50_PineNeedlesFlame
## 0610009B22Rik_56046                    4                    5
## 0610010F05Rik_69119                 1767                 1563
## 0610010F05Rik_74637                   12                    6
## 0610010K14Rik_31619                 1480                 1134
## 0610010K14Rik_31873                  586                  599
## 0610010K14Rik_68949                   23                   27
##                     M51_PineNeedlesFlame M52_PineNeedlesFlame
## 0610009B22Rik_56046                    4                    4
## 0610010F05Rik_69119                 1918                 1711
## 0610010F05Rik_74637                    4                    9
## 0610010K14Rik_31619                 1390                 1763
## 0610010K14Rik_31873                  683                  797
## 0610010K14Rik_68949                   26                   27
##                     M53_PineNeedlesFlame M54_PineNeedlesFlame M31_Saline
## 0610009B22Rik_56046                    5                    5          3
## 0610010F05Rik_69119                 1751                 1582       1891
## 0610010F05Rik_74637                    6                    3         10
## 0610010K14Rik_31619                 1370                 1419       1340
## 0610010K14Rik_31873                  593                  647        561
## 0610010K14Rik_68949                   20                   24         19
##                     M32_Saline M33_Saline M34_Saline M35_Saline M36_Saline
## 0610009B22Rik_56046          2          4          3          4          4
## 0610010F05Rik_69119       2507       2518       2524       1831       1878
## 0610010F05Rik_74637          7          7          5          7          8
## 0610010K14Rik_31619       1563       1632       1438       1280        967
## 0610010K14Rik_31873        685        815        703        646        526
## 0610010K14Rik_68949         16         21         19         22         20
##                     M67_LPS M68_LPS M69_LPS M70_LPS
## 0610009B22Rik_56046       3       2       4       5
## 0610010F05Rik_69119    1906    1597    1461    1201
## 0610010F05Rik_74637       6       6       7       5
## 0610010K14Rik_31619     964    1012     776     638
## 0610010K14Rik_31873     622     702     522     496
## 0610010K14Rik_68949      21      21      18      15</code></pre>
<p>Let’s again generate an exploratory plot using this updated dataset, focusing on the bar chart view since that was the most informative pre-RUV. Here are the updated bar charts summarizing overall data distributions per sample:</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb198-1"></a>colors &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set2&quot;</span>)</span>
<span id="cb198-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb198-2"></a><span class="kw">plotRLE</span>(ruv_set, <span class="dt">outline=</span><span class="ot">FALSE</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">4</span>, <span class="dv">4</span>), <span class="dt">col=</span>colors[groups])</span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-130-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This plot shows overall tighter data that are more similarly distributed across samples. Therefore, it is looking like this RUV addition improved the overall distribution of this dataset. It is important not to over-correct/over-smooth your datasets, so implement these types of pre-processing steps with caution. One strategy that we commonly employ to gage whether data smoothing is needed/applied correctly is to run the statistical models with and without correction of potential sources of heterogeneity, and critically evaluate similarities vs differences produced in the results.</p>
<!-- ### With this, we can answer **Environmental Health Question 4**: -->
<!-- #### What is an approach that analysts can use when evaluating transcriptomic data from tissues of mixed cellular composition to aid in controlling for sources of sample heterogeneity? -->
<!-- #### *Answer: Remove unwanted variation (RUV), among other approaches, including surrogate variable analysis (SVA).* -->
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 4</strong>:</i>
What is an approach that analysts can use when evaluating transcriptomic data from tissues of mixed cellular composition to aid in controlling for sources of sample heterogeneity?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Remove unwanted variation (RUV), among other approaches, including surrogate variable analysis (SVA).</p>
</div>
</div>
<div id="significantly-differentially-expressed-genes" class="section level2">
<h2>Significantly Differentially Expressed Genes</h2>
<div id="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure" class="section level4">
<h4>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</h4>
<p>At this point, we have completed several data pre-processing, QA/QC, and additional steps to prepare our example transcriptomics data for statistical analysis. And finally, we are ready to run the overall statistical model to identify genes that are altered in expression in association with different biomass burn conditions.</p>
<p>Here we leverage the DESeq2 package to carry out these statistical comparisons. This package is now the most commonly implemented analysis pipeline used for transcriptomic data, including sequencing data as well as transcriptomic data produced via other technologies (e.g., Nanostring, Fluidigm, and other gene expression technologies). This package is extremely well-documented and we encourage trainees to leverage these resources in parallel with the current training module when carrying out their own transcriptomics analyses in R:</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">Bioconductor website</a><br />
</li>
<li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">Vignette</a><br />
</li>
<li><a href="https://bioconductor.org/packages/devel/bioc/manuals/DESeq2/man/DESeq2.pdf">Manual</a><br />
</li>
<li>Primary citation: Love MI, Huber W, Anders S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol. 2014;15(12):550. doi: 10.1186/s13059-014-0550-8. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/25516281/">25516281</a></li>
</ul>
<p>In brief, the basic calculations employed within the DESeq2 underlying algorithms include the following:</p>
<p><strong>1. Estimate size factors.</strong>
In the first step, size factors are estimated to help account for potential differences in the sequencing depth across samples. It is similar to a normalization parameter in the model.</p>
<p><strong>2. Normalize count data.</strong>
DESeq2 employs different normalization algorithms depending on the parameters selected / stage of analysis. The most commonly employed method is called the <strong>median of ratios</strong>, which takes into account sequencing depth and RNA composition, as described <a href="https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html">here</a>. Specifically, these normalized values are calculated as counts divided by sample-specific size factors determined by median ratio of gene counts relative to geometric mean per gene. DESeq2 then transforms these data using variance stabilization within the final statistical model. Because of these two steps, we prefer to export both the median of ratios normalized data as well as the variance stabilization transformed data, to save in our records and use when generating plots of expression levels for specific genes we are interested in. These steps are detailed below.</p>
<p><strong>3. Estimate dispersion.</strong>
The dispersion estimate takes into account the relationship between the variance of an observed count and its mean value. It is similar to a variance parameter. In DESeq2, dispersion is estimated using a maximum likelihood and empirical bayes approach.</p>
<p><strong>4. Fit negative binomial generalized linear model (GLM).</strong>
Finally, a negative binomial model is fit for each gene using the design formula that will be described within the proceeding code. The Wald test is performed to test if log fold changes in expression (typically calculated as log(average exposed / average unexposed)) significantly differ from zero. Statistical p-values are reported from this test and also adjusted for multiple testing using the Benjamini and Hochberg procedure.</p>
<p>Note that these calculations, among others, are embedded within the DESeq2 functions, so we do not need to code for them ourselves. Instead, we just need to make sure that we set-up the DESeq2 functions correctly, such that these calculations are carried out appropriately in our final transcriptomics analyses.</p>
</div>
<div id="setting-up-the-deseq2-experiment" class="section level4">
<h4>Setting up the DESeq2 experiment</h4>
<p>Here we provide example script that is used to identify which genes are significantly differentially expressed in association with the example biomass smoke exposures, smoldering pine needles and flaming pine needles, as well as a positive inflammation control, LPS.</p>
<p>First, we need to set-up the DESeq2 experiment:</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb199-1"></a><span class="co"># Set up our experiment using our RUV adjusted count and phenotype data.</span></span>
<span id="cb199-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb199-2"></a><span class="co"># Our design indicates that our count data is dependent on the exposure condition </span></span>
<span id="cb199-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb199-3"></a><span class="co"># (groups variable) and our factor of unwanted variation, and we have specified </span></span>
<span id="cb199-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb199-4"></a><span class="co"># that there not be an intercept term through the use of &#39;~0&#39;</span></span>
<span id="cb199-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb199-5"></a></span>
<span id="cb199-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb199-6"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSetFromMatrix</span>(<span class="dt">countData =</span> <span class="kw">counts</span>(ruv_set), <span class="co"># Grabbing count data from the &#39;ruv_set&#39; object</span></span>
<span id="cb199-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb199-7"></a>                                 <span class="dt">colData =</span> <span class="kw">pData</span>(ruv_set), <span class="co"># Grabbing the phenotype data and corresponding factor of unwanted variation from the &#39;ruv_set&#39; object</span></span>
<span id="cb199-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb199-8"></a>                                 <span class="dt">design =</span> <span class="op">~</span><span class="dv">0</span><span class="op">+</span>groups<span class="op">+</span>W_<span class="dv">1</span>) <span class="co"># Setting up the statistical formula (see below)</span></span></code></pre></div>
<p>For the forumula design, we use a ‘~0’ at the front to not include an intercept term, and then also account for the exposure condition (groups) and the previously calculated factors of unwanted variation (W_1) of the samples. Formula design is an important step and should be carefully considered for each individual analysis. Other resources, including official DESeq2 documentation, are avialble for consultation regarding formula design, as the specifics of formula design are beyond the scope of this training module.</p>
</div>
<div id="estimating-size-factors" class="section level4">
<h4>Estimating size factors</h4>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb200-1"></a><span class="co"># Estimate size factors from the dds object that was just created as the experiment above</span></span>
<span id="cb200-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb200-2"></a>dds &lt;-<span class="st"> </span><span class="kw">estimateSizeFactors</span>(dds)</span>
<span id="cb200-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb200-3"></a><span class="kw">sizeFactors</span>(dds)  <span class="co"># viewing the size factors</span></span></code></pre></div>
<pre><code>## M13_PineNeedlesSmolder M14_PineNeedlesSmolder M15_PineNeedlesSmolder 
##              1.2668895              1.3040762              1.1782071 
## M16_PineNeedlesSmolder M17_PineNeedlesSmolder M18_PineNeedlesSmolder 
##              1.1723749              1.2904950              1.3405500 
##   M49_PineNeedlesFlame   M50_PineNeedlesFlame   M51_PineNeedlesFlame 
##              1.1313237              0.6313099              0.9328970 
##   M52_PineNeedlesFlame   M53_PineNeedlesFlame   M54_PineNeedlesFlame 
##              1.0050414              0.7649632              0.8747854 
##             M31_Saline             M32_Saline             M33_Saline 
##              0.8050127              1.0480725              1.0389623 
##             M34_Saline             M35_Saline             M36_Saline 
##              1.0282261              0.9519266              1.0409103 
##                M67_LPS                M68_LPS                M69_LPS 
##              0.9308588              1.1002076              0.9793051 
##                M70_LPS 
##              0.8575718</code></pre>
</div>
<div id="calculating-and-exporting-normalized-counts" class="section level4">
<h4>Calculating and exporting normalized counts</h4>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb202-1"></a><span class="co"># Calculating normalized count data</span></span>
<span id="cb202-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb202-2"></a>normcounts &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">counts</span>(dds, <span class="dt">normalized=</span><span class="ot">TRUE</span>))</span>
<span id="cb202-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb202-3"></a><span class="kw">write.csv</span>(normcounts, <span class="st">&quot;Module2_4_Output_NormalizedCounts.csv&quot;</span>)</span>
<span id="cb202-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb202-4"></a></span>
<span id="cb202-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb202-5"></a><span class="co"># Transforming normalized counts through variance stabilization</span></span>
<span id="cb202-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb202-6"></a>vsd &lt;-<span class="st"> </span><span class="kw">varianceStabilizingTransformation</span>(dds, <span class="dt">blind=</span><span class="ot">FALSE</span>)</span>
<span id="cb202-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb202-7"></a>vsd_matrix &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">assay</span>(vsd))</span>
<span id="cb202-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb202-8"></a><span class="kw">write.csv</span>(vsd_matrix, <span class="st">&quot;Module2_4_Output_VSDCounts.csv&quot;</span>, <span class="dt">row.names=</span><span class="ot">TRUE</span>)</span></code></pre></div>
</div>
<div id="running-the-final-deseq2-experiment" class="section level4">
<h4>Running the final DESeq2 experiment</h4>
<p>Here, we are finally ready to run the actual statistical comparisons (exposed vs control samples) to calculate fold changes and p-values that describe the degree to which each gene may or may not be altered at the expression level in association with treatment.</p>
<p>For this example, we would like to run three different comparisons:<br />
(1) Smoldering Pine Needles vs. Control<br />
(2) Flaming Pine Needles vs. Control<br />
(3) LPS vs. Control<br />
which we can easily code for using a loop function, as detailed below.</p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-1"></a><span class="co"># Run experiment</span></span>
<span id="cb203-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-2"></a>dds_run &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds, <span class="dt">betaPrior=</span><span class="ot">FALSE</span>) </span>
<span id="cb203-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-3"></a></span>
<span id="cb203-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-4"></a><span class="co"># Loop through and extract and export results for all contrasts (treatments vs. control)</span></span>
<span id="cb203-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-5"></a><span class="cf">for</span> (trt <span class="cf">in</span> trt_groups){ <span class="co"># Iterate for each of the treatments listed in &#39;trt_groups&#39;</span></span>
<span id="cb203-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-6"></a>  <span class="kw">cat</span>(trt) <span class="co"># Print which treatment group we are on in the loop </span></span>
<span id="cb203-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-7"></a>  res &lt;-<span class="st"> </span><span class="kw">results</span>(dds_run, <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;groups&quot;</span>,trt,ctrl)) <span class="co"># Extract the results of the DESeq2 analysis specifically for the comparison of the treatment group for the current iteration of the loop with the control group</span></span>
<span id="cb203-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-8"></a>  <span class="kw">summary</span>(res) <span class="co"># Print out a high-level summary of the results</span></span>
<span id="cb203-9"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-9"></a>  ordered &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(res[<span class="kw">order</span>(res<span class="op">$</span>padj),]) <span class="co"># Make a dataframe of the results and order them by adjusted p-value from lowest to highest</span></span>
<span id="cb203-10"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-10"></a>  top10 &lt;-<span class="st"> </span><span class="kw">head</span>(ordered, <span class="dt">n=</span><span class="dv">10</span>) <span class="co"># Make dataframe of the first ten rows of the ordered results</span></span>
<span id="cb203-11"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-11"></a>  <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">The 10 most significantly differentially expressed genes by adjusted p-value:</span><span class="ch">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb203-12"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-12"></a>  <span class="kw">print</span>(top10) <span class="co"># View the first ten rows of the ordered results</span></span>
<span id="cb203-13"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-13"></a>  pfilt<span class="fl">.05</span> &lt;-<span class="st"> </span><span class="kw">nrow</span>(ordered <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(padj<span class="op">&lt;</span><span class="fl">0.05</span>)) <span class="co"># Get the number of genes that are significantly differentially expressed where padj &lt; 0.05</span></span>
<span id="cb203-14"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-14"></a>  <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">The number of genes showing significant differential expression where padj &lt; 0.05 is &quot;</span>, pfilt<span class="fl">.05</span>)</span>
<span id="cb203-15"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-15"></a>  pfilt<span class="fl">.10</span> &lt;-<span class="st"> </span><span class="kw">nrow</span>(ordered <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(padj<span class="op">&lt;</span><span class="fl">0.1</span>)) <span class="co"># Get the number of genes that are significantly differentially expressed where padj &lt; 0.10</span></span>
<span id="cb203-16"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-16"></a>  <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">The number of genes showing significant differential expression where padj &lt; 0.10 is &quot;</span>, pfilt<span class="fl">.10</span>,<span class="st">&quot;</span><span class="ch">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb203-17"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-17"></a>  <span class="kw">write.csv</span>(ordered, <span class="kw">paste0</span>(<span class="st">&quot;Module2_4_Output_StatisticalResults_&quot;</span>,trt ,<span class="st">&quot;.csv&quot;</span>)) <span class="co"># Export the full dataframe of ordered results as a csv</span></span>
<span id="cb203-18"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb203-18"></a>}</span></code></pre></div>
<pre><code>## PineNeedlesSmolder_4h_Lung
## out of 16664 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 457, 2.7%
## LFC &lt; 0 (down)     : 697, 4.2%
## outliers [1]       : 0, 0%
## low counts [2]     : 1293, 7.8%
## (mean count &lt; 21)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results
## 
## 
## The 10 most significantly differentially expressed genes by adjusted p-value:
## 
##                baseMean log2FoldChange      lfcSE      stat       pvalue
## Hbs1l_72436   367.64772     -1.1465230 0.13614949 -8.421060 3.730933e-17
## Cetn3_30290  2350.31984     -0.7197604 0.09379276 -7.673944 1.667868e-14
## Grasp_56900   277.43936      0.9759274 0.14177298  6.883734 5.830390e-12
## Pcsk7_57804   180.24040      0.8261941 0.13529825  6.106465 1.018617e-09
## Plscr4_55764  178.95216      1.3193654 0.21751536  6.065620 1.314459e-09
## Socs3_79479   183.28293      1.2535533 0.20740514  6.043983 1.503550e-09
## Kat2b_64780   444.97603     -0.7197105 0.12034633 -5.980327 2.226896e-09
## Ptp4a2_62762 1840.75247     -0.7802857 0.13076704 -5.966990 2.416698e-09
## Rmi1_56327    377.57936     -0.8548903 0.14288443 -5.983089 2.189449e-09
## Abcd2_62271    57.46455     -1.2504459 0.21464501 -5.825646 5.689193e-09
##                      padj
## Hbs1l_72436  5.734817e-13
## Cetn3_30290  1.281840e-10
## Grasp_56900  2.987297e-08
## Pcsk7_57804  3.851843e-06
## Plscr4_55764 3.851843e-06
## Socs3_79479  3.851843e-06
## Kat2b_64780  4.127451e-06
## Ptp4a2_62762 4.127451e-06
## Rmi1_56327   4.127451e-06
## Abcd2_62271  8.744859e-06
## 
## The number of genes showing significant differential expression where padj &lt; 0.05 is  679
## The number of genes showing significant differential expression where padj &lt; 0.10 is  1154 
## 
## PineNeedlesFlame_4h_Lung
## out of 16664 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 411, 2.5%
## LFC &lt; 0 (down)     : 575, 3.5%
## outliers [1]       : 0, 0%
## low counts [2]     : 0, 0%
## (mean count &lt; 10)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results
## 
## 
## The 10 most significantly differentially expressed genes by adjusted p-value:
## 
##                 baseMean log2FoldChange     lfcSE      stat       pvalue
## Tmem109_70229  351.49267      1.2681226 0.1637785  7.742913 9.716419e-15
## Rhof_64396      58.57258      1.4767798 0.2210640  6.680328 2.384081e-11
## Plscr4_55764   178.95216      1.5349460 0.2393976  6.411701 1.439049e-10
## Pip4k2a_67992  796.64597      0.8262520 0.1299839  6.356571 2.063073e-10
## Zranb2_68671   739.91379      0.8131628 0.1330277  6.112735 9.793822e-10
## Cul1_31957     896.52286     -0.7061150 0.1203618 -5.866602 4.448172e-09
## Ppfibp1_75630  655.07197      1.4148222 0.2481843  5.700692 1.193223e-08
## Mbnl1_30670    851.18586     -1.0159639 0.1811247 -5.609197 2.032670e-08
## Atp5k_68198   1096.49479      1.1003470 0.1983469  5.547588 2.896380e-08
## Gclm_29258     138.54250      1.0993904 0.1985396  5.537387 3.070173e-08
##                       padj
## Tmem109_70229 1.619144e-10
## Rhof_64396    1.986416e-07
## Plscr4_55764  7.993439e-07
## Pip4k2a_67992 8.594761e-07
## Zranb2_68671  3.264085e-06
## Cul1_31957    1.235406e-05
## Ppfibp1_75630 2.840553e-05
## Mbnl1_30670   4.234052e-05
## Atp5k_68198   5.116136e-05
## Gclm_29258    5.116136e-05
## 
## The number of genes showing significant differential expression where padj &lt; 0.05 is  515
## The number of genes showing significant differential expression where padj &lt; 0.10 is  986 
## 
## LPS_4h_Lung
## out of 16664 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 2960, 18%
## LFC &lt; 0 (down)     : 2947, 18%
## outliers [1]       : 0, 0%
## low counts [2]     : 0, 0%
## (mean count &lt; 10)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results
## 
## 
## The 10 most significantly differentially expressed genes by adjusted p-value:
## 
##                baseMean log2FoldChange     lfcSE     stat        pvalue
## Usp18_32270   1319.2259       3.829526 0.1459140 26.24508 8.133515e-152
## Usp18_67659    414.5959       3.739864 0.1679851 22.26307 8.428171e-110
## Ifit1_29859    862.3286       5.000365 0.2259553 22.12989 1.629877e-108
## Oasl2_67419   1437.6182       3.710164 0.1904583 19.48019  1.616902e-84
## Gbp5_77887     600.7219       5.272287 0.2769982 19.03365  8.977555e-81
## Oasl2_67043    292.4482       3.898484 0.2096435 18.59578  3.476310e-77
## Rnf213_57840  1052.0454       3.168052 0.1731212 18.29962  8.332621e-75
## Cmpk2_70664    429.4232       4.592488 0.2518916 18.23200  2.875888e-74
## Cmpk2_29189    695.1997       4.264136 0.2386279 17.86939  2.042134e-71
## Eif2ak2_30238 1585.6071       2.671974 0.1495526 17.86645  2.152596e-71
##                        padj
## Usp18_32270   1.355369e-147
## Usp18_67659   7.022352e-106
## Ifit1_29859   9.053424e-105
## Oasl2_67419    6.736013e-81
## Gbp5_77887     2.992039e-77
## Oasl2_67043    9.654872e-74
## Rnf213_57840   1.983640e-71
## Cmpk2_70664    5.990475e-71
## Cmpk2_29189    3.587085e-68
## Eif2ak2_30238  3.587085e-68
## 
## The number of genes showing significant differential expression where padj &lt; 0.05 is  4813
## The number of genes showing significant differential expression where padj &lt; 0.10 is  5907</code></pre>
<!-- ### With this, we can answer **Environmental Health Question 5**: -->
<!-- #### How many genes showed significant differential expression associated with flaming pine needles exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) < 0.05? -->
<!-- #### *Answer: 515 genes.* -->
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 5</strong>:</i>
How many genes showed significant differential expression associated with flaming pine needles exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) &lt; 0.05?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: 515 genes.</p>
</div>
<p><br></p>
<!-- ### With this, we can answer **Environmental Health Question 6**: -->
<!-- #### How many genes showed significant differential expression associated with smoldering pine needles exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) < 0.05? -->
<!-- #### *Answer: 679 genes.* -->
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 6</strong>:</i>
How many genes showed significant differential expression associated with smoldering pine needles exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) &lt; 0.05?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: 679 genes.</p>
</div>
<p><br></p>
<!-- ### With this, we can answer **Environmental Health Question 7**: -->
<!-- #### How many genes showed significant differential expression associated with lipopolysaccharide (LPS) exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) < 0.05? -->
<!-- #### *Answer: 4813 genes.* -->
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 7</strong>:</i>
How many genes showed significant differential expression associated with lipopolysaccharide (LPS) exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) &lt; 0.05?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: 4813 genes.</p>
</div>
<p><em>Together, we find that exposure to both flaming and smoldering of pine needles caused substantial disruptions in gene expression profiles. LPS serves as a positive control for inflammation and produced the greatest transcriptomic response.</em></p>
</div>
</div>
<div id="visualizing-statistical-results" class="section level2">
<h2>Visualizing Statistical Results</h2>
</div>
<div id="ma-plots" class="section level2">
<h2>MA Plots</h2>
<p><a href="https://en.wikipedia.org/wiki/MA_plot">MA plots</a> represent a common method of visualization that illustrates differences between measurements taken in two samples, by transforming the data onto M (log ratio) and A (mean average) scales, then plotting these values.</p>
<p>Here, we leverage MA plots to show how log fold changes relate to expression levels. In these plots, the log fold change is plotted on the y-axis and expression values are plotted along the x-axis, and dots are colored according to statistical significance (using padj&lt;0.05 as the statistical filter). Here we will generate an MA plot for Flaming Pine Needles.</p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-1"></a>res &lt;-<span class="st"> </span><span class="kw">results</span>(dds_run, <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="dt">contrast =</span> </span>
<span id="cb205-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-2"></a>                 <span class="kw">c</span>(<span class="st">&quot;groups&quot;</span>,<span class="st">&quot;PineNeedlesFlame_4h_Lung&quot;</span>,ctrl)) <span class="co"># Re-extract the DESeq2 results for the flamming pine needles </span></span>
<span id="cb205-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-3"></a>MA &lt;-<span class="st"> </span><span class="kw">data.frame</span>(res) <span class="co"># Make a prelimiary dataframe of the flaming pine needle results</span></span>
<span id="cb205-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-4"></a>MA_ns &lt;-<span class="st"> </span>MA[ <span class="kw">which</span>(MA<span class="op">$</span>padj<span class="op">&gt;=</span><span class="fl">0.05</span>),]  <span class="co"># Non-significant genes to plot</span></span>
<span id="cb205-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-5"></a>MA_up &lt;-<span class="st"> </span>MA[ <span class="kw">which</span>(MA<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>MA<span class="op">$</span>log2FoldChange <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>),]   <span class="co"># Significant up-regulated genes to plot</span></span>
<span id="cb205-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-6"></a>MA_down &lt;-<span class="st"> </span>MA[ <span class="kw">which</span>(MA<span class="op">$</span>padj<span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>MA<span class="op">$</span>log2FoldChange <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>),]  <span class="co">#Significant down-regulated genes to plot</span></span>
<span id="cb205-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-7"></a></span>
<span id="cb205-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-8"></a><span class="kw">ggplot</span>(MA_ns, <span class="kw">aes</span>(<span class="dt">x =</span> baseMean, <span class="dt">y =</span> log2FoldChange)) <span class="op">+</span><span class="st"> </span><span class="co"># Plot data with counts on x-axis and log2 fold change on y-axis</span></span>
<span id="cb205-9"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-9"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">color=</span><span class="st">&quot;gray75&quot;</span>, <span class="dt">size =</span> <span class="fl">.5</span>) <span class="op">+</span><span class="st"> </span><span class="co"># Set point size and color</span></span>
<span id="cb205-10"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-10"></a><span class="st">  </span></span>
<span id="cb205-11"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-11"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">data =</span> MA_up, <span class="dt">color=</span><span class="st">&quot;firebrick&quot;</span>, <span class="dt">size=</span><span class="dv">1</span>, <span class="dt">show.legend =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="co"># Plot the up-regulated significant genes</span></span>
<span id="cb205-12"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-12"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">data =</span> MA_down, <span class="dt">color=</span><span class="st">&quot;dodgerblue2&quot;</span>, <span class="dt">size=</span><span class="dv">1</span>, <span class="dt">show.legend =</span> <span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="co"># Plot down-regulated significant genes</span></span>
<span id="cb205-13"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-13"></a><span class="st">  </span></span>
<span id="cb205-14"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-14"></a><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span><span class="co"># Change theme of plot from gray to black and white</span></span>
<span id="cb205-15"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-15"></a><span class="st">  </span></span>
<span id="cb205-16"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-16"></a><span class="st">  </span><span class="co"># We want to log10 transform x-axis for better visualizations</span></span>
<span id="cb205-17"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-17"></a><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">trans =</span> <span class="st">&quot;log10&quot;</span>, <span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">10</span>,<span class="dv">100</span>, <span class="dv">1000</span>, <span class="dv">10000</span>, <span class="dv">100000</span>, <span class="dv">1000000</span>), </span>
<span id="cb205-18"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-18"></a>                     <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;1&quot;</span>,<span class="st">&quot;10&quot;</span>,<span class="st">&quot;100&quot;</span>, <span class="st">&quot;1000&quot;</span>, <span class="st">&quot;10000&quot;</span>, <span class="st">&quot;100000&quot;</span>, <span class="st">&quot;1000000&quot;</span>)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb205-19"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-19"></a><span class="st">  </span><span class="co"># We will bound y axis as well to better fit data while not leaving out too many points</span></span>
<span id="cb205-20"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-20"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">2</span>, <span class="dv">2</span>)) <span class="op">+</span></span>
<span id="cb205-21"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-21"></a><span class="st">  </span></span>
<span id="cb205-22"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-22"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Expression (Normalized Count)&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Fold Change (log2)&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="co"># Add labels for axes</span></span>
<span id="cb205-23"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-23"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;MA Plot Flaming Pine Needles 4h Lung&quot;</span>) <span class="op">+</span><span class="st">   </span><span class="co"># Add title</span></span>
<span id="cb205-24"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb205-24"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept=</span><span class="dv">0</span>) <span class="co"># Add horizontal line at 0</span></span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-135-1.png" width="672" /></p>
</div>
<div id="volcano-plots" class="section level2">
<h2>Volcano Plots</h2>
<p>Similar to MA plots, volcano plots provide visualizations of fold changes in expression from transcriptomic data. However, instead of plotting these values against expression, log fold change is plotted against (adjusted) p-values in volcano plots. Here, we use functions within the <a href="https://www.rdocumentation.org/packages/EnhancedVolcano/versions/1.11.3/topics/EnhancedVolcano">EnhancedVolcano package</a> to generate a volcano plot for Flaming Pine Needles.</p>
<p>Running the ‘EnhancedVolcano’ function to generate an example volcano plot:</p>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-1"></a>Vol &lt;-<span class="st"> </span><span class="kw">data.frame</span>(res) <span class="co"># Dataset to use for plotting</span></span>
<span id="cb206-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-2"></a><span class="kw">EnhancedVolcano</span>(Vol, </span>
<span id="cb206-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-3"></a>                <span class="dt">lab =</span> <span class="kw">rownames</span>(res), <span class="co"># Label information from dataset (can be a column name)</span></span>
<span id="cb206-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-4"></a>                <span class="dt">x =</span> <span class="st">&#39;log2FoldChange&#39;</span>, <span class="co"># Column name in dataset with l2fc information</span></span>
<span id="cb206-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-5"></a>                <span class="dt">y =</span> <span class="st">&#39;padj&#39;</span>, <span class="co"># Column name in dataset with adjusted p-value information</span></span>
<span id="cb206-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-6"></a>                <span class="dt">ylab =</span> <span class="st">&quot;-Log(FDR-adjusted p)&quot;</span>, <span class="co"># Y-axis label</span></span>
<span id="cb206-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-7"></a>                <span class="dt">pCutoff=</span> <span class="fl">0.05</span>, <span class="co"># Set p-value cutoff</span></span>
<span id="cb206-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-8"></a>                <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">5</span>), <span class="co"># Limit y-axis for better plot visuals</span></span>
<span id="cb206-9"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-9"></a>                <span class="dt">xlim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">2</span>,<span class="dv">2</span>), <span class="co"># Limit x-axis (similar to in MA plot y-axis)</span></span>
<span id="cb206-10"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-10"></a>                <span class="dt">title=</span><span class="st">&quot;Volcano Plot&quot;</span>, <span class="co"># Set title</span></span>
<span id="cb206-11"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-11"></a>                <span class="dt">subtitle =</span> <span class="st">&quot;Flaming Pine Needles 4h Lung&quot;</span>, <span class="co"># Set subtitle</span></span>
<span id="cb206-12"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb206-12"></a>                <span class="dt">legendPosition =</span> <span class="st">&#39;bottom&#39;</span>) <span class="co"># Put legend on bottom</span></span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-136-1.png" width="672" /></p>
</div>
<div id="pathway-enrichment-analysis" class="section level2">
<h2>Pathway Enrichment Analysis</h2>
<div id="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis" class="section level4">
<h4>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</h4>
<p>Pathway enrichment analysis is a very helpful tool that can be applied to interpret transcriptomic changes of interest in terms of systems biology. In these types of analyses, gene lists of interest are used to identify biological pathways that include genes present in your dataset more often than expected by chance alone. There are many tools that can be used to carry out pathway enrichment analyses. Here, we are using the R package, PIANO, to carry out the statistical enrichment analysis based on the lists of genes we previously identified with differential expression associated with flaming pine needles exposure.</p>
<p>To detail, the following input data are required to run PIANO:<br />
(1) Your background gene sets, which represent all genes queried from your experiment (aka your ‘gene universe’)
(2) The list of genes you are interested in evaluating pathway enrichment of; here, this represents the genes identified with significant differential expression associated with flaming pine needles<br />
(3) A underlying pathway dataset; here, we’re using the KEGG PATHWAY Database (<a href="https://www.genome.jp/kegg/pathway.html">KEGG</a>), summarized through the Molecular Signature Database (<a href="https://www.gsea-msigdb.org/gsea/msigdb/">MSigDB</a>) into pre-formatted input files (.gmt) ready for PIANO.</p>
<p><em>Let’s organize these three required data inputs.</em></p>
<ol style="list-style-type: decimal">
<li>Background gene set:</li>
</ol>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-1"></a><span class="co"># First grab the rownames of the &#39;res&#39; object, which was redefined as the DESeq2 </span></span>
<span id="cb207-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-2"></a><span class="co"># results for flaming pine needles prior to MA plot generation, and remove the </span></span>
<span id="cb207-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-3"></a><span class="co"># BioSpyder numeric identifier using a sub function, while maintaining the gene </span></span>
<span id="cb207-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-4"></a><span class="co"># symbol and place these IDs into a new list within the &#39;res&#39; object (saved as &#39;id&#39;)</span></span>
<span id="cb207-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-5"></a>res<span class="op">$</span>id &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">rownames</span>(res));</span>
<span id="cb207-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-6"></a></span>
<span id="cb207-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-7"></a><span class="co"># Because these IDs now contain duplicate gene symbols, we need to remove duplicates</span></span>
<span id="cb207-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-8"></a><span class="co"># One way to do this is to preferentially retain rows of data with the largest </span></span>
<span id="cb207-9"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-9"></a><span class="co"># fold change (it doesn&#39;t really matter here, because we&#39;re just identifying </span></span>
<span id="cb207-10"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-10"></a><span class="co"># unique genes within the background set)</span></span>
<span id="cb207-11"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-11"></a>res.ordered &lt;-<span class="st"> </span>res[<span class="kw">order</span>(res<span class="op">$</span>id, <span class="op">-</span><span class="kw">abs</span>(res<span class="op">$</span>log2FoldChange) ), ] <span class="co"># sort by id and reverse of abs(log2foldchange)</span></span>
<span id="cb207-12"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-12"></a>res.ordered &lt;-<span class="st"> </span>res.ordered[ <span class="op">!</span><span class="kw">duplicated</span>(res.ordered<span class="op">$</span>id), ]     <span class="co"># removing gene duplicates</span></span>
<span id="cb207-13"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-13"></a></span>
<span id="cb207-14"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-14"></a><span class="co"># Setting this as the background list</span></span>
<span id="cb207-15"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-15"></a>Background &lt;-<span class="st"> </span><span class="kw">toupper</span>(<span class="kw">as.character</span>(res.ordered<span class="op">$</span>id))</span>
<span id="cb207-16"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb207-16"></a>Background[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] <span class="co"># viewing the first 200 genes in this background list</span></span></code></pre></div>
<pre><code>##   [1] &quot;0610009B22RIK&quot;  &quot;0610010F05RIK&quot;  &quot;0610010K14RIK&quot;  &quot;0610012G03RIK&quot; 
##   [5] &quot;0610030E20RIK&quot;  &quot;0610040J01RIK&quot;  &quot;1110004F10RIK&quot;  &quot;1110008P14RIK&quot; 
##   [9] &quot;1110012L19RIK&quot;  &quot;1110017D15RIK&quot;  &quot;1110032A03RIK&quot;  &quot;1110038F14RIK&quot; 
##  [13] &quot;1110059E24RIK&quot;  &quot;1110059G10RIK&quot;  &quot;1110065P20RIK&quot;  &quot;1190002N15RIK&quot; 
##  [17] &quot;1190007I07RIK&quot;  &quot;1500011B03RIK&quot;  &quot;1500011K16RIK&quot;  &quot;1600002K03RIK&quot; 
##  [21] &quot;1600012H06RIK&quot;  &quot;1600014C10RIK&quot;  &quot;1700001L19RIK&quot;  &quot;1700003E16RIK&quot; 
##  [25] &quot;1700007K13RIK&quot;  &quot;1700012B09RIK&quot;  &quot;1700013F07RIK&quot;  &quot;1700016K19RIK&quot; 
##  [29] &quot;1700017B05RIK&quot;  &quot;1700020D05RIK&quot;  &quot;1700024G13RIK&quot;  &quot;1700025G04RIK&quot; 
##  [33] &quot;1700028P14RIK&quot;  &quot;1700029I15RIK&quot;  &quot;1700029J07RIK&quot;  &quot;1700030K09RIK&quot; 
##  [37] &quot;1700037C18RIK&quot;  &quot;1700037H04RIK&quot;  &quot;1700086D15RIK&quot;  &quot;1700088E04RIK&quot; 
##  [41] &quot;1700102P08RIK&quot;  &quot;1700109H08RIK&quot;  &quot;1700123O20RIK&quot;  &quot;1810009A15RIK&quot; 
##  [45] &quot;1810010H24RIK&quot;  &quot;1810013L24RIK&quot;  &quot;1810022K09RIK&quot;  &quot;1810030O07RIK&quot; 
##  [49] &quot;1810037I17RIK&quot;  &quot;1810043G02RIK&quot;  &quot;1810055G02RIK&quot;  &quot;2010005H15RIK&quot; 
##  [53] &quot;2010111I01RIK&quot;  &quot;2010300C02RIK&quot;  &quot;2010309G21RIK&quot;  &quot;2200002D01RIK&quot; 
##  [57] &quot;2210011C24RIK&quot;  &quot;2210016H18RIK&quot;  &quot;2210016L21RIK&quot;  &quot;2210408I21RIK&quot; 
##  [61] &quot;2300009A05RIK&quot;  &quot;2310007B03RIK&quot;  &quot;2310009B15RIK&quot;  &quot;2310011J03RIK&quot; 
##  [65] &quot;2310022A10RIK&quot;  &quot;2310022B05RIK&quot;  &quot;2310030G06RIK&quot;  &quot;2310033P09RIK&quot; 
##  [69] &quot;2310057M21RIK&quot;  &quot;2310061I04RIK&quot;  &quot;2410002F23RIK&quot;  &quot;2410004B18RIK&quot; 
##  [73] &quot;2410015M20RIK&quot;  &quot;2410131K14RIK&quot;  &quot;2510002D24RIK&quot;  &quot;2510009E07RIK&quot; 
##  [77] &quot;2600001M11RIK&quot;  &quot;2610001J05RIK&quot;  &quot;2610002M06RIK&quot;  &quot;2610008E11RIK&quot; 
##  [81] &quot;2610028H24RIK&quot;  &quot;2610301B20RIK&quot;  &quot;2610507B11RIK&quot;  &quot;2700049A03RIK&quot; 
##  [85] &quot;2700062C07RIK&quot;  &quot;2700081O15RIK&quot;  &quot;2700097O09RIK&quot;  &quot;2810004N23RIK&quot; 
##  [89] &quot;2810021J22RIK&quot;  &quot;2810408A11RIK&quot;  &quot;2900026A02RIK&quot;  &quot;3110001I22RIK&quot; 
##  [93] &quot;3110040N11RIK&quot;  &quot;3110082I17RIK&quot;  &quot;3830406C13RIK&quot;  &quot;4430402I18RIK&quot; 
##  [97] &quot;4732423E21RIK&quot;  &quot;4833414E09RIK&quot;  &quot;4833420G17RIK&quot;  &quot;4833427G06RIK&quot; 
## [101] &quot;4833439L19RIK&quot;  &quot;4921524J17RIK&quot;  &quot;4930402H24RIK&quot;  &quot;4930430F08RIK&quot; 
## [105] &quot;4930451G09RIK&quot;  &quot;4930453N24RIK&quot;  &quot;4930486L24RIK&quot;  &quot;4930503L19RIK&quot; 
## [109] &quot;4930523C07RIK&quot;  &quot;4930550C14RIK&quot;  &quot;4930562C15RIK&quot;  &quot;4931406C07RIK&quot; 
## [113] &quot;4931406P16RIK&quot;  &quot;4931414P19RIK&quot;  &quot;4932438A13RIK&quot;  &quot;4933408B17RIK&quot; 
## [117] &quot;4933415F23RIK&quot;  &quot;4933427D14RIK&quot;  &quot;4933434E20RIK&quot;  &quot;5330417C22RIK&quot; 
## [121] &quot;5430427O19RIK&quot;  &quot;5730480H06RIK&quot;  &quot;5830417I10RIK&quot;  &quot;6030458C11RIK&quot; 
## [125] &quot;6030468B19RIK&quot;  &quot;6330403K07RIK&quot;  &quot;6330417A16RIK&quot;  &quot;6430531B16RIK&quot; 
## [129] &quot;6430548M08RIK&quot;  &quot;6430550D23RIK&quot;  &quot;6720489N17RIK&quot;  &quot;8030462N17RIK&quot; 
## [133] &quot;9030624G23RIK&quot;  &quot;9130008F23RIK&quot;  &quot;9130019O22RIK&quot;  &quot;9130023H24RIK&quot; 
## [137] &quot;9130230L23RIK&quot;  &quot;9230104L09RIK&quot;  &quot;9330182L06RIK&quot;  &quot;9530068E07RIK&quot; 
## [141] &quot;9930012K11RIK&quot;  &quot;9930021J03RIK&quot;  &quot;9930111J21RIK1&quot; &quot;9930111J21RIK2&quot;
## [145] &quot;A130071D04RIK&quot;  &quot;A230050P20RIK&quot;  &quot;A2ML1&quot;          &quot;A430005L14RIK&quot; 
## [149] &quot;A430033K04RIK&quot;  &quot;A430078G23RIK&quot;  &quot;A530032D15RIK&quot;  &quot;A630001G21RIK&quot; 
## [153] &quot;A730034C02&quot;     &quot;A730049H05RIK&quot;  &quot;A830018L16RIK&quot;  &quot;A930002H24RIK&quot; 
## [157] &quot;A930004D18RIK&quot;  &quot;AA415038&quot;       &quot;AA986860&quot;       &quot;AACS&quot;          
## [161] &quot;AAED1&quot;          &quot;AAGAB&quot;          &quot;AAK1&quot;           &quot;AAMDC&quot;         
## [165] &quot;AAMP&quot;           &quot;AAR2&quot;           &quot;AARS&quot;           &quot;AARS2&quot;         
## [169] &quot;AARSD1&quot;         &quot;AASDH&quot;          &quot;AASS&quot;           &quot;AATF&quot;          
## [173] &quot;AATK&quot;           &quot;AB124611&quot;       &quot;ABCA1&quot;          &quot;ABCA17&quot;        
## [177] &quot;ABCA2&quot;          &quot;ABCA3&quot;          &quot;ABCA5&quot;          &quot;ABCA7&quot;         
## [181] &quot;ABCA8A&quot;         &quot;ABCA8B&quot;         &quot;ABCA9&quot;          &quot;ABCB10&quot;        
## [185] &quot;ABCB1A&quot;         &quot;ABCB1B&quot;         &quot;ABCB6&quot;          &quot;ABCB7&quot;         
## [189] &quot;ABCB8&quot;          &quot;ABCC1&quot;          &quot;ABCC3&quot;          &quot;ABCC4&quot;         
## [193] &quot;ABCC5&quot;          &quot;ABCD1&quot;          &quot;ABCD2&quot;          &quot;ABCD3&quot;         
## [197] &quot;ABCD4&quot;          &quot;ABCE1&quot;          &quot;ABCF1&quot;          &quot;ABCF3&quot;</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>The list of genes identified with significant differential expression associated with flaming pine needles:</li>
</ol>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-1"></a><span class="co"># Similar to the above script, but starting with the res$id object</span></span>
<span id="cb209-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-2"></a><span class="co"># and filtering for genes with padj &lt; 0.05</span></span>
<span id="cb209-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-3"></a></span>
<span id="cb209-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-4"></a><span class="co"># Pulling the genes with padj &lt; 0.05</span></span>
<span id="cb209-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-5"></a>res.ordered &lt;-<span class="st"> </span>res[<span class="kw">order</span>(res<span class="op">$</span>id, <span class="op">-</span><span class="kw">abs</span>(res<span class="op">$</span>log2FoldChange) ), ] </span>
<span id="cb209-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-6"></a></span>
<span id="cb209-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-7"></a><span class="co"># Pulling the genes with padj &lt; 0.05</span></span>
<span id="cb209-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-8"></a>SigGenes &lt;-<span class="st"> </span><span class="kw">toupper</span>(<span class="kw">as.character</span>(res.ordered[<span class="kw">which</span>(res.ordered<span class="op">$</span>padj<span class="op">&lt;</span>.<span class="dv">05</span>),<span class="st">&quot;id&quot;</span>])) </span>
<span id="cb209-9"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-9"></a></span>
<span id="cb209-10"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-10"></a><span class="co"># Removing gene duplicates</span></span>
<span id="cb209-11"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-11"></a>SigGenes &lt;-<span class="st"> </span>SigGenes[ <span class="op">!</span><span class="kw">duplicated</span>(SigGenes)] </span>
<span id="cb209-12"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-12"></a></span>
<span id="cb209-13"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-13"></a><span class="co"># Viewing the length of this significant gene list</span></span>
<span id="cb209-14"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb209-14"></a><span class="kw">length</span>(SigGenes) </span></code></pre></div>
<pre><code>## [1] 488</code></pre>
<p>Therefore, this gene set includes 488 <em>unique</em> genes significantly associated with the Flaming Pine Needles condition, based on padj&lt;0.05.</p>
<ol start="3" style="list-style-type: decimal">
<li>The underlying KEGG pathway dataset.
Note that this file was simply downloaded from <a href="https://www.gsea-msigdb.org/gsea/msigdb/">MSigDB</a>, ready for upload as a .gmt file. Here, we use the ‘loadGSC’ function enabled through the PIANO package to upload and organize these pathways.</li>
</ol>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb211-1"></a>KEGG_Pathways &lt;-<span class="st"> </span><span class="kw">loadGSC</span>(<span class="dt">file=</span><span class="st">&quot;Module2_4/Module2_4_kegg_v7_symbols.gmt&quot;</span>, <span class="dt">type=</span><span class="st">&quot;gmt&quot;</span>)</span>
<span id="cb211-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb211-2"></a></span>
<span id="cb211-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb211-3"></a><span class="co"># Viewing the number of biological pathways contained in the database</span></span>
<span id="cb211-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb211-4"></a><span class="kw">length</span>(KEGG_Pathways<span class="op">$</span>gsc) </span></code></pre></div>
<pre><code>## [1] 186</code></pre>
<p>This KEGG pathway database therefore includes 186 biological pathways available to query</p>
<p>With these data inputs ready, we can now run the pathway enrichment analysis. The enrichment statistic that is commonly employed through the PIANO package is based of a hypergeometric test, run through the ‘runGSAhyper’ function. This returns a p-value for each gene set from which you can determine enrichment status.</p>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb213-1"></a><span class="co"># Running the piano function based on the hypergeometric statistic</span></span>
<span id="cb213-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb213-2"></a>Results_GSA &lt;-<span class="st"> </span>piano<span class="op">::</span><span class="kw">runGSAhyper</span>(<span class="dt">genes=</span>SigGenes, <span class="dt">universe=</span>Background,</span>
<span id="cb213-3"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb213-3"></a>                                  <span class="dt">gsc=</span>KEGG_Pathways, <span class="dt">gsSizeLim=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="ot">Inf</span>), <span class="dt">adjMethod =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb213-4"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb213-4"></a></span>
<span id="cb213-5"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb213-5"></a><span class="co"># Pulling the pathway enrichment results into a separate dataframe</span></span>
<span id="cb213-6"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb213-6"></a>PathwayResults &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(Results_GSA<span class="op">$</span>resTab)</span>
<span id="cb213-7"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb213-7"></a></span>
<span id="cb213-8"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb213-8"></a><span class="co"># Viewing the top of these pathway enrichment results (which are not ordered at the moment)</span></span>
<span id="cb213-9"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb213-9"></a><span class="kw">head</span>(PathwayResults)</span></code></pre></div>
<pre><code>##                                                        p-value Adjusted p-value
## KEGG_N_GLYCAN_BIOSYNTHESIS                          0.77021314        1.0000000
## KEGG_OTHER_GLYCAN_DEGRADATION                       1.00000000        1.0000000
## KEGG_O_GLYCAN_BIOSYNTHESIS                          0.03553158        0.6656139
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                  0.42056787        1.0000000
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE 1.00000000        1.0000000
## KEGG_GLYCEROLIPID_METABOLISM                        0.16241736        1.0000000
##                                                     Significant (in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                  1
## KEGG_OTHER_GLYCAN_DEGRADATION                                               0
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                  3
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                          1
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                         0
## KEGG_GLYCEROLIPID_METABOLISM                                                3
##                                                     Non-significant (in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                     34
## KEGG_OTHER_GLYCAN_DEGRADATION                                                  13
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                     15
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                             12
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                            12
## KEGG_GLYCEROLIPID_METABOLISM                                                   31
##                                                     Significant (not in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                    487
## KEGG_OTHER_GLYCAN_DEGRADATION                                                 488
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                    485
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                            487
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                           488
## KEGG_GLYCEROLIPID_METABOLISM                                                  485
##                                                     Non-significant (not in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                      11355
## KEGG_OTHER_GLYCAN_DEGRADATION                                                   11376
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                      11374
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                              11377
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                             11377
## KEGG_GLYCEROLIPID_METABOLISM                                                    11358</code></pre>
<p>This dataframe therefore summarizes the enrichment p-value for each pathway, FDR adjusted p-value, number of significant genes in the gene set that intersect with genes in the pathway, etc</p>
<p>With these results, let’s identify which pathways meet a statistical enrichment p-value filter of 0.05:</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb215-1"></a>SigPathways &lt;-<span class="st"> </span>PathwayResults[<span class="kw">which</span>(PathwayResults<span class="op">$</span><span class="st">`</span><span class="dt">p-value</span><span class="st">`</span> <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>),]</span>
<span id="cb215-2"><a href="controlling-for-sources-of-sample-heterogeneity.html#cb215-2"></a><span class="kw">rownames</span>(SigPathways)</span></code></pre></div>
<pre><code>##  [1] &quot;KEGG_O_GLYCAN_BIOSYNTHESIS&quot;                               
##  [2] &quot;KEGG_HYPERTROPHIC_CARDIOMYOPATHY_HCM&quot;                     
##  [3] &quot;KEGG_ARRHYTHMOGENIC_RIGHT_VENTRICULAR_CARDIOMYOPATHY_ARVC&quot;
##  [4] &quot;KEGG_PROTEASOME&quot;                                          
##  [5] &quot;KEGG_OOCYTE_MEIOSIS&quot;                                      
##  [6] &quot;KEGG_VASCULAR_SMOOTH_MUSCLE_CONTRACTION&quot;                  
##  [7] &quot;KEGG_WNT_SIGNALING_PATHWAY&quot;                               
##  [8] &quot;KEGG_HEDGEHOG_SIGNALING_PATHWAY&quot;                          
##  [9] &quot;KEGG_FOCAL_ADHESION&quot;                                      
## [10] &quot;KEGG_ECM_RECEPTOR_INTERACTION&quot;                            
## [11] &quot;KEGG_COMPLEMENT_AND_COAGULATION_CASCADES&quot;                 
## [12] &quot;KEGG_GNRH_SIGNALING_PATHWAY&quot;</code></pre>
<!-- ### With this, we can answer **Environmental Health Question 8**: -->
<!-- #### What biological pathways are disrupted in association with flaming pine needles exposure in the lung, identified through systems level analyses? -->
<!-- #### *Answer: Biological pathways involved in cardiopulmonary function (e.g., arrhythmogenic right ventricular cardiomyopathy, hypertrophic cardiomyopathy, vascular smooth muscle contraction), carcinogenesis signaling (e.g., Wnt signaling pathway, hedgehog signaling pathway), and hormone signaling (e.g., Gnrh signaling pathway), among others.* -->
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 8</strong>:</i>
What biological pathways are disrupted in association with flaming pine needles exposure in the lung, identified through systems level analyses?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Biological pathways involved in cardiopulmonary function (e.g., arrhythmogenic right ventricular cardiomyopathy, hypertrophic cardiomyopathy, vascular smooth muscle contraction), carcinogenesis signaling (e.g., Wnt signaling pathway, hedgehog signaling pathway), and hormone signaling (e.g., Gnrh signaling pathway), among others.</p>
</div>
</div>
</div>
<div id="concluding-remarks-9" class="section level2">
<h2>Concluding Remarks</h2>
<p>In this module, users are guided through the uploading, organization, QA/QC, statistical analysis, and systems level analysis of an example -omics dataset based on transcriptomic responses to biomass burn scenarios, representing environmental exposure scenarios of growing concern worldwide. It is worth noting that the methods described herein represent a fraction of the approaches and tools that can be leveraged in the analysis of -omics datasets, and methods should be tailored to the purposes of each individual analysis’ goal. For additional example research projects that have leveraged -omics and systems biology to address environmental health questions, see the following select relevant publications:</p>
<p><strong>Genomic publications evaluating gene-environment interactions and relations to disease etiology:</strong></p>
<ul>
<li><p>Balik-Meisner M, Truong L, Scholl EH, La Du JK, Tanguay RL, Reif DM. Elucidating Gene-by-Environment Interactions Associated with Differential Susceptibility to Chemical Exposure. Environ Health Perspect. 2018 Jun 28;126(6):067010. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/29968567/">29968567</a>.</p></li>
<li><p>Ward-Caviness CK, Neas LM, Blach C, Haynes CS, LaRocque-Abramson K, Grass E, Dowdy ZE, Devlin RB, Diaz-Sanchez D, Cascio WE, Miranda ML, Gregory SG, Shah SH, Kraus WE, Hauser ER. A genome-wide trans-ethnic interaction study links the PIGR-FCAMR locus to coronary atherosclerosis via interactions between genetic variants and residential exposure to traffic. PLoS One. 2017 Mar 29;12(3):e0173880. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/28355232/">28355232</a>.</p></li>
</ul>
<p><strong>Transcriptomic publications evaluating gene expression responses to environmental exposures and relations to disease etiology:</strong></p>
<ul>
<li><p>Chang Y, Rager JE, Tilton SC. Linking Coregulated Gene Modules with Polycyclic Aromatic Hydrocarbon-Related Cancer Risk in the 3D Human Bronchial Epithelium. Chem Res Toxicol. 2021 Jun 21;34(6):1445-1455. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/34048650/">34048650</a>.</p></li>
<li><p>Chappell GA, Rager JE, Wolf J, Babic M, LeBlanc KJ, Ring CL, Harris MA, Thompson CM. Comparison of Gene Expression Responses in the Small Intestine of Mice Following Exposure to 3 Carcinogens Using the S1500+ Gene Set Informs a Potential Common Adverse Outcome Pathway. Toxicol Pathol. 2019 Oct;47(7):851-864. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/31558096/">31558096</a>.</p></li>
<li><p>Manuck TA, Eaves LA, Rager JE, Fry RC. Mid-pregnancy maternal blood nitric oxide-related gene and miRNA expression are associated with preterm birth. Epigenomics. 2021 May;13(9):667-682. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/33890487/">33890487</a>.</p></li>
</ul>
<p><strong>Epigenomic publications</strong> evaluating microRNA, CpG methylation, and/or histone methylation responses to environmental exposures and relations to disease etiology:</p>
<ul>
<li><p>Chappell GA, Rager JE. Epigenetics in chemical-induced genotoxic carcinogenesis. Curr Opinion Toxicol. <a href="https://www.sciencedirect.com/science/article/abs/pii/S2468202017300396">2017 Oct; 6:10-17</a>.</p></li>
<li><p>Rager JE, Bailey KA, Smeester L, Miller SK, Parker JS, Laine JE, Drobná Z, Currier J, Douillet C, Olshan AF, Rubio-Andrade M, Stýblo M, García-Vargas G, Fry RC. Prenatal arsenic exposure and the epigenome: altered microRNAs associated with innate and adaptive immune signaling in newborn cord blood. Environ Mol Mutagen. 2014 Apr;55(3):196-208. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/24327377/">24327377</a>.</p></li>
<li><p>Rager JE, Bauer RN, Müller LL, Smeester L, Carson JL, Brighton LE, Fry RC, Jaspers I. DNA methylation in nasal epithelial cells from smokers: identification of ULBP3-related effects. Am J Physiol Lung Cell Mol Physiol. 2013 Sep 15;305(6):L432-8. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/23831618/">23831618</a>.</p></li>
<li><p>Smeester L, Rager JE, Bailey KA, Guan X, Smith N, García-Vargas G, Del Razo LM, Drobná Z, Kelkar H, Stýblo M, Fry RC. Epigenetic changes in individuals with arsenicosis. Chem Res Toxicol. 2011 Feb 18;24(2):165-7. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/21291286/">21291286</a>.</p></li>
</ul>
<p><strong>Metabolomic publications</strong> evaluating changes in the metabolome in response to environmental exposures and involved in disease etiology:</p>
<ul>
<li><p>Lu K, Abo RP, Schlieper KA, Graffam ME, Levine S, Wishnok JS, Swenberg JA, Tannenbaum SR, Fox JG. Arsenic exposure perturbs the gut microbiome and its metabolic profile in mice: an integrated metagenomics and metabolomics analysis. Environ Health Perspect. 2014 Mar;122(3):284-91. PMID: 24413286; PMCID: <a href="https://pubmed.ncbi.nlm.nih.gov/24413286/">PMC3948040</a>.</p></li>
<li><p>Manuck TA, Lai Y, Ru H, Glover AV, Rager JE, Fry RC, Lu K. Metabolites from midtrimester plasma of pregnant patients at high risk for preterm birth. Am J Obstet Gynecol MFM. 2021 Jul;3(4):100393. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/33991707/">33991707</a>.</p></li>
</ul>
<p><strong>Microbiome publications</strong> evaluating changes in microbiome profiles in relation to the environment and human disease:</p>
<ul>
<li><p>Chi L, Bian X, Gao B, Ru H, Tu P, Lu K. Sex-Specific Effects of Arsenic Exposure on the Trajectory and Function of the Gut Microbiome. Chem Res Toxicol. 2016 Jun 20;29(6):949-51.PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/27268458/">27268458</a>.</p></li>
<li><p>Cho I, Blaser MJ. The human microbiome: at the interface of health and disease. Nat Rev Genet. 2012 Mar 13;13(4):260-70. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/22411464/">22411464</a>.</p></li>
<li><p>Lu K, Abo RP, Schlieper KA, Graffam ME, Levine S, Wishnok JS, Swenberg JA, Tannenbaum SR, Fox JG. Arsenic exposure perturbs the gut microbiome and its metabolic profile in mice: an integrated metagenomics and metabolomics analysis. Environ Health Perspect. 2014 Mar;122(3):284-91. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/24413286/">24413286</a>.</p></li>
</ul>
<p><strong>Exposome publications</strong> evaluating changes in chemical signatures in relation to the environment and human disease:</p>
<ul>
<li><p>Rager JE, Strynar MJ, Liang S, McMahen RL, Richard AM, Grulke CM, Wambaugh JF, Isaacs KK, Judson R, Williams AJ, Sobus JR. Linking high resolution mass spectrometry data with exposure and toxicity forecasts to advance high-throughput environmental monitoring. Environ Int. 2016 Mar;88:269-280. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/26812473/">26812473</a>.</p></li>
<li><p>Rappaport SM, Barupal DK, Wishart D, Vineis P, Scalbert A. The blood exposome and its role in discovering causes of disease. Environ Health Perspect. 2014 Aug;122(8):769-74. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/24659601/">24659601</a>.</p></li>
<li><p>Viet SM, Falman JC, Merrill LS, Faustman EM, Savitz DA, Mervish N, Barr DB, Peterson LA, Wright R, Balshaw D, O’Brien B. Human Health Exposure Analysis Resource (HHEAR): A model for incorporating the exposome into health studies. Int J Hyg Environ Health. 2021 Jun;235:113768. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/34034040/">34034040</a>.</p></li>
</ul>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="omics-analyses-and-systems-biology.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="toxicokinetic-modeling.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
