<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1.2 Data Organization Basics | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="1.2 Data Organization Basics | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="1.2 Data Organization Basics | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1.2 Data Organization Basics | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle R. Roell, Lauren E. Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/Favicon_v1.png" type="image/x-icon" />
<link rel="prev" href="introduction-to-coding-in-r.html"/>
<link rel="next" href="finding-and-visualizing-data-trends.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/Data Training Icon for Top Left Bookdown_v2.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html"><i class="fa fa-check"></i>1.1 Introduction to Coding in R</a><ul>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#r-packages"><i class="fa fa-check"></i>R Packages</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-organization-basics.html"><a href="data-organization-basics.html"><i class="fa fa-check"></i>1.2 Data Organization Basics</a><ul>
<li class="chapter" data-level="" data-path="data-organization-basics.html"><a href="data-organization-basics.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-organization-basics.html"><a href="data-organization-basics.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation using Base R</a></li>
<li class="chapter" data-level="" data-path="data-organization-basics.html"><a href="data-organization-basics.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-organization-basics.html"><a href="data-organization-basics.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html"><i class="fa fa-check"></i>1.3 Finding and Visualizing Data Trends</a><ul>
<li class="chapter" data-level="" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#basic-data-analysis"><i class="fa fa-check"></i>Basic Data Analysis</a></li>
<li class="chapter" data-level="" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#regression-modeling"><i class="fa fa-check"></i>Regression Modeling</a></li>
<li class="chapter" data-level="" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#categorical-data-analysis"><i class="fa fa-check"></i>Categorical Data Analysis</a></li>
<li class="chapter" data-level="" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html"><i class="fa fa-check"></i>1.4 High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#high-dimensional-data-visualizations-1"><i class="fa fa-check"></i>High-Dimensional Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#density-plot-visualizations"><i class="fa fa-check"></i>Density Plot Visualizations</a></li>
<li class="chapter" data-level="" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#ggally-visualizations"><i class="fa fa-check"></i>GGally Visualizations</a></li>
<li class="chapter" data-level="" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#boxplot-visualizations"><i class="fa fa-check"></i>Boxplot Visualizations</a></li>
<li class="chapter" data-level="" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#correlation-plot-visualizations"><i class="fa fa-check"></i>Correlation Plot Visualizations</a></li>
<li class="chapter" data-level="" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#hierarchical-clustering-visualizations"><i class="fa fa-check"></i>Hierarchical Clustering Visualizations</a><ul>
<li class="chapter" data-level="" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#heat-map-visualizations"><i class="fa fa-check"></i>Heat map Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>1.5 FAIR Data Management Practices</a><ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#helpful-resources-on-fair"><i class="fa fa-check"></i>Helpful Resources on FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Chemical-<br>Biological Analyses <br>and Predictive Modeling</b></span></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html"><i class="fa fa-check"></i>2.1 Dose-Response Modeling</a><ul>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#plotting-data-in-dose-response"><i class="fa fa-check"></i>Plotting Data in Dose-Response</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#dose-response-curve-fitting"><i class="fa fa-check"></i>Dose-Response Curve Fitting</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#comparing-curve-fits"><i class="fa fa-check"></i>Comparing Curve Fits</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#curve-fitting-example"><i class="fa fa-check"></i>Curve Fitting Example</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#deriving-benchmark-dose"><i class="fa fa-check"></i>Deriving Benchmark Dose</a></li>
<li class="chapter" data-level="" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling.html"><a href="machine-learning-and-predictive-modeling.html"><i class="fa fa-check"></i>2.2 Machine Learning and Predictive Modeling</a><ul>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling.html"><a href="machine-learning-and-predictive-modeling.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling.html"><a href="machine-learning-and-predictive-modeling.html#k-means-analysis"><i class="fa fa-check"></i>K-means Analysis</a></li>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling.html"><a href="machine-learning-and-predictive-modeling.html#principal-component-analysis"><i class="fa fa-check"></i>Principal Component Analysis</a></li>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling.html"><a href="machine-learning-and-predictive-modeling.html#combining-k-means-with-pca"><i class="fa fa-check"></i>Combining K-Means with PCA</a></li>
<li class="chapter" data-level="" data-path="machine-learning-and-predictive-modeling.html"><a href="machine-learning-and-predictive-modeling.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analyses.html"><a href="mixtures-analyses.html"><i class="fa fa-check"></i>2.3 Mixtures Analyses</a><ul>
<li class="chapter" data-level="" data-path="mixtures-analyses.html"><a href="mixtures-analyses.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analyses.html"><a href="mixtures-analyses.html#chemistry-based-approach"><i class="fa fa-check"></i>Chemistry-based Approach</a></li>
<li class="chapter" data-level="" data-path="mixtures-analyses.html"><a href="mixtures-analyses.html#toxicity-based-approach"><i class="fa fa-check"></i>Toxicity-based Approach</a></li>
<li class="chapter" data-level="" data-path="mixtures-analyses.html"><a href="mixtures-analyses.html#comparing-results"><i class="fa fa-check"></i>Comparing Results</a></li>
<li class="chapter" data-level="" data-path="mixtures-analyses.html"><a href="mixtures-analyses.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html"><i class="fa fa-check"></i>2.4 -Omics Analyses and Systems Biology</a><ul>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#the-field-of--omics"><i class="fa fa-check"></i>The Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#transcriptomics"><i class="fa fa-check"></i>Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#statistical-analysis-of-gene-expression"><i class="fa fa-check"></i>Statistical Analysis of Gene Expression</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#ma-plots"><i class="fa fa-check"></i>MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#volcano-plots"><i class="fa fa-check"></i>Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#pathway-enrichment-analysis"><i class="fa fa-check"></i>Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>2.5 Toxicokinetic Modeling</a><ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#calculating-steady-state-concentration"><i class="fa fa-check"></i>Calculating steady-state concentration</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#monte-carlo-approach"><i class="fa fa-check"></i>Monte Carlo Approach</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#chemical-specific-example"><i class="fa fa-check"></i>Chemical-Specific Example</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#calculating-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Calculating Bioactivity-Exposure Ratios (BERs)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html"><i class="fa fa-check"></i>2.6 Read-Across Toxicity Predictions</a><ul>
<li class="chapter" data-level="" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html#introduction-to-chemical-read-across"><i class="fa fa-check"></i>Introduction to Chemical Read-Across</a></li>
<li class="chapter" data-level="" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html#read-across-example-analysis"><i class="fa fa-check"></i>Read Across Example Analysis</a></li>
<li class="chapter" data-level="" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html#calculating-chemical-similarities"><i class="fa fa-check"></i>Calculating Chemical Similarities</a></li>
<li class="chapter" data-level="" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html#chemical-read-across-to-predict-acute-toxicity"><i class="fa fa-check"></i>Chemical Read-Across to Predict Acute Toxicity</a></li>
<li class="chapter" data-level="" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Environmental <br>Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>3.1 Comparative Toxicogenomics Database</a><ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>3.2 Gene Expression Omnibus</a><ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a><ul>
<li><a href="gene-expression-omnibus.html#training-modules-environmental-health-questions-7">Training Module’s <strong>Environmental Health Questions</strong></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i>3.3 Database Integration: Air Quality, Mortality, and Environmental Justice Data</a><ul>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-exposure-and-health-databases-used-in-this-module"><i class="fa fa-check"></i>Introduction to Exposure and Health Databases used in this Module</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#population-weighted-vs-unweighted-exposures"><i class="fa fa-check"></i>Population-weighted vs Unweighted Exposures</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#regression-modeling-1"><i class="fa fa-check"></i>Regression Modeling</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations"><i class="fa fa-check"></i>Environmental Justice Considerations</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Additional Resources</b></span></li>
<li class="chapter" data-level="" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i>Resources</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-organization-basics" class="section level1">
<h1>1.2 Data Organization Basics</h1>
<p>This training module was developed by Dr. Kyle R. Roell and Dr. Julia E. Rager</p>
<p>Fall 2021</p>
<div id="introduction-to-training-module-1" class="section level2">
<h2>Introduction to Training Module</h2>
<p>Data within the fields of exposure science, toxicology, and public health are very rarely prepared and ready for all statistical analyses/visualization code. The beginning of almost any scripted analysis includes important formatting steps. These steps largely encompass data organization, manipulation, and other steps in preparation for actual statistical analyses/visualizations. Data organization and manipulation generally refers to organizing and formatting data in a way that makes it easier to read and work with. This can be done through several approaches, including:</p>
<ul>
<li>Base R operations and functions, or</li>
<li>A collection of packages (and philosophy) known as <a href="https://www.tidyverse.org">Tidyverse</a>.</li>
</ul>
<p>In this training tutorial we will go over some of the most common ways you can organize and manipulate data, including:</p>
<ul>
<li>Merging data</li>
<li>Filtering and subsetting data</li>
<li>Melting and casting data</li>
</ul>
<p>These approaches will first be taught using the basic operations and functions in R. Then, the exact same approaches will be taught using the Tidyverse package and associated functions and syntax.</p>
<p>These data manipulation and organization methods are demonstrated using an example environmentally relevant human cohort dataset. This cohort was generated by creating data distributions randomly pulled from our previously published cohorts, resulting in a bespoke dataset for these training purposes with associated demographic data and variable environmental exposure metrics from metal levels obtained using sources of drinking water and human urine samples.</p>
<div id="set-your-working-directory-1" class="section level4">
<h4>Set your working directory</h4>
<p>In preparation, first let’s set our working directory to the folderpath that contains our input files</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="data-organization-basics.html#cb18-1"></a><span class="kw">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
<p>Note that in macOS, filepaths use “/” as folder separaters; whereas in PCs, filepaths use "".</p>
</div>
<div id="importing-example-datasets" class="section level4">
<h4>Importing example datasets</h4>
<p>Then let’s read in our example datasets</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="data-organization-basics.html#cb19-1"></a>demo.data &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;Module1_2/Module1_2_DemographicData.csv&quot;</span>)</span>
<span id="cb19-2"><a href="data-organization-basics.html#cb19-2"></a>chem.data &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;Module1_2/Module1_2_ChemicalData.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="viewing-example-datasets" class="section level4">
<h4>Viewing example datasets</h4>
<p>Let’s see what these datasets look like, starting with the chemical measures:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="data-organization-basics.html#cb20-1"></a><span class="kw">dim</span>(chem.data)</span></code></pre></div>
<pre><code>## [1] 200   7</code></pre>
<p>The chemical measurement dataset includes 200 rows x 7 columns</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="data-organization-basics.html#cb22-1"></a>chem.data[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">7</span>]</span></code></pre></div>
<pre><code>##    ID     DWAs     DWCd     DWCr       UAs       UCd      UCr
## 1   1 6.426464 1.292941 51.67987 10.192695 0.7537104 42.60187
## 2   2 7.832384 1.798535 50.10409 11.815088 0.9789506 41.30757
## 3   3 7.516569 1.288461 48.74001 10.079057 0.1903262 36.47716
## 4   4 5.906656 2.075259 50.92745  8.719123 0.9364825 42.47987
## 5   5 7.181873 2.762643 55.16882  9.436559 1.4977829 47.78528
## 6   6 9.723429 3.054057 51.14812 11.589403 1.6645837 38.26386
## 7   7 6.268547 1.218410 52.08578  8.887948 0.6347667 39.45535
## 8   8 6.718448 1.414975 54.96740  9.304968 0.6658849 45.09987
## 9   9 9.074928 2.727755 55.72826 10.818153 1.6585757 42.58577
## 10 10 5.771691 2.410993 47.06552  8.747217 1.7354305 34.80661</code></pre>
<p>These data are organized according to subject ID (first column), followed by measures of:</p>
<ul>
<li>DWAs (drinking water arsenic levels in µg/L)</li>
<li>DWCd (drinking water cadmium levels in µg/L)</li>
<li>DWCr (drinking water chromium levels in µg/L)</li>
<li>UAs (urinary arsenic levels in µg/L)</li>
<li>UCd (urinary cadmium levels in µg/L)</li>
<li>UCr (urinary chromium levels in µg/L)</li>
</ul>
<p>Now let’s view the demographic data:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="data-organization-basics.html#cb24-1"></a><span class="kw">dim</span>(demo.data)</span></code></pre></div>
<pre><code>## [1] 200   6</code></pre>
<p>The subject demographic dataset includes 200 rows x 6 columns</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="data-organization-basics.html#cb26-1"></a>demo.data[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>]</span></code></pre></div>
<pre><code>##    ID  BMI     MAge MEdu       BW GA
## 1   1 27.7 22.99928    3 3180.058 34
## 2   2 26.8 30.05142    3 3210.823 43
## 3   3 33.2 28.04660    3 3311.551 40
## 4   4 30.1 34.81796    3 3266.844 32
## 5   5 37.4 42.68440    3 3664.088 35
## 6   6 33.3 24.94960    3 3328.988 40
## 7   7 24.8 29.54798    3 3061.949 30
## 8   8 16.9 24.94954    3 3332.539 38
## 9   9 36.9 33.58589    3 3260.482 39
## 10 10 21.7 39.29018    3 3141.723 35</code></pre>
<p>These data are organized according to subject ID (first column) followed by the following subject information:</p>
<ul>
<li>BMI (body mass index)</li>
<li>MAge (maternal age, years)</li>
<li>MEdu (maternal education, 1= “less than high school”; 2= “high school or some college”; 3= “college or greater”)</li>
<li>BW (body weight, grams)</li>
<li>GA (gestational age, week)</li>
</ul>
</div>
</div>
<div id="data-manipulation-using-base-r" class="section level2">
<h2>Data Manipulation using Base R</h2>
<div id="merging-data-using-base-r-syntax" class="section level4">
<h4>Merging Data using Base R Syntax</h4>
<p>Merging datasets represents the joining together of two or more datasets, while connecting the datasets using a common identifier (generally some sort of ID). This is useful if you have multiple datasets describing different aspects of the study, different variables, or different measures across the same samples. Samples could correspond to the same study participants, animals, cell culture samples, environmental media samples, etc, depending on the study design. In the current example, we will be joining human demographic data and environmental metals exposure data collected from drinking water and human urine samples.</p>
<p>Let’s start by merging the example demographic data with the chemical measurement data using the base R function of “merge”. To learn more about this function, you can type the following:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="data-organization-basics.html#cb28-1"></a>?merge</span></code></pre></div>
<p>which brings up helpful information in the R console</p>
<p>To merge these datasets using the merge function, use the following code:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="data-organization-basics.html#cb29-1"></a><span class="co"># Note that we specify to merge these datasets by their shared ID column</span></span>
<span id="cb29-2"><a href="data-organization-basics.html#cb29-2"></a>full.data &lt;-<span class="st"> </span><span class="kw">merge</span>(demo.data, chem.data, <span class="dt">by=</span><span class="st">&quot;ID&quot;</span>) </span>
<span id="cb29-3"><a href="data-organization-basics.html#cb29-3"></a><span class="kw">dim</span>(full.data) </span></code></pre></div>
<pre><code>## [1] 200  12</code></pre>
<p>This merged dataframe contains 200 rows x 12 columns</p>
<p>Viewing this merged dataframe</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="data-organization-basics.html#cb31-1"></a>full.data[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">12</span>]</span></code></pre></div>
<pre><code>##    ID  BMI     MAge MEdu       BW GA     DWAs     DWCd     DWCr       UAs
## 1   1 27.7 22.99928    3 3180.058 34 6.426464 1.292941 51.67987 10.192695
## 2   2 26.8 30.05142    3 3210.823 43 7.832384 1.798535 50.10409 11.815088
## 3   3 33.2 28.04660    3 3311.551 40 7.516569 1.288461 48.74001 10.079057
## 4   4 30.1 34.81796    3 3266.844 32 5.906656 2.075259 50.92745  8.719123
## 5   5 37.4 42.68440    3 3664.088 35 7.181873 2.762643 55.16882  9.436559
## 6   6 33.3 24.94960    3 3328.988 40 9.723429 3.054057 51.14812 11.589403
## 7   7 24.8 29.54798    3 3061.949 30 6.268547 1.218410 52.08578  8.887948
## 8   8 16.9 24.94954    3 3332.539 38 6.718448 1.414975 54.96740  9.304968
## 9   9 36.9 33.58589    3 3260.482 39 9.074928 2.727755 55.72826 10.818153
## 10 10 21.7 39.29018    3 3141.723 35 5.771691 2.410993 47.06552  8.747217
##          UCd      UCr
## 1  0.7537104 42.60187
## 2  0.9789506 41.30757
## 3  0.1903262 36.47716
## 4  0.9364825 42.47987
## 5  1.4977829 47.78528
## 6  1.6645837 38.26386
## 7  0.6347667 39.45535
## 8  0.6658849 45.09987
## 9  1.6585757 42.58577
## 10 1.7354305 34.80661</code></pre>
<p>We can see that the merge function retained the first column in each original dataframe (ID), though did not replicate it since it was used as the identifier to merge off of. All other columns include their original data, just merged together by the IDs in the first column.</p>
<p>These datasets were actually quite easy to merge, since they had the same exact column identifier and number of rows. You can edit your script to include more specifics in instances when these may differ across datasets that you would like to merge. For example:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="data-organization-basics.html#cb33-1"></a>full.data &lt;-<span class="st"> </span><span class="kw">merge</span>(demo.data, chem.data, <span class="dt">by.x=</span><span class="st">&quot;ID&quot;</span>, <span class="dt">by.y=</span><span class="st">&quot;ID&quot;</span>) </span>
<span id="cb33-2"><a href="data-organization-basics.html#cb33-2"></a><span class="co"># This option allows you to edit the column header text that is used in each </span></span>
<span id="cb33-3"><a href="data-organization-basics.html#cb33-3"></a><span class="co"># dataframe. Here, these are still the same &quot;ID&quot;, but you can see that adding </span></span>
<span id="cb33-4"><a href="data-organization-basics.html#cb33-4"></a><span class="co"># this script allows you to specify instances when differ header text is used.</span></span></code></pre></div>
</div>
<div id="filtering-and-subsetting-data-using-base-r-syntax" class="section level4">
<h4>Filtering and Subsetting Data using Base R Syntax</h4>
<p>Filtering and subsetting data are useful tools when you need to focus your dataset to highlight data you are interested in analyzing downstream. These could represent, for example, specific samples or participants that meet certain criteria that you are interested in evaluating. It is also useful for simply removing particular variables or samples from dataframes as you are working through your script. These methods are illustrated here.</p>
<p>For this example, let’s first define a vector of columns that we want to keep in our analysis</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="data-organization-basics.html#cb34-1"></a>subset.columns &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;BMI&quot;</span>, <span class="st">&quot;MAge&quot;</span>, <span class="st">&quot;MEdu&quot;</span>)</span>
<span id="cb34-2"><a href="data-organization-basics.html#cb34-2"></a>subset.columns</span></code></pre></div>
<pre><code>## [1] &quot;BMI&quot;  &quot;MAge&quot; &quot;MEdu&quot;</code></pre>
<p>Now we can simply subset our data using those columns</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="data-organization-basics.html#cb36-1"></a><span class="co"># Subsetting the data by selecting the columns represented in the defined </span></span>
<span id="cb36-2"><a href="data-organization-basics.html#cb36-2"></a><span class="co"># &#39;subset.columns&#39; vector</span></span>
<span id="cb36-3"><a href="data-organization-basics.html#cb36-3"></a>subset.data1 &lt;-<span class="st"> </span>full.data[,subset.columns] </span>
<span id="cb36-4"><a href="data-organization-basics.html#cb36-4"></a></span>
<span id="cb36-5"><a href="data-organization-basics.html#cb36-5"></a><span class="co"># Viewing the top of this subsetted dataframe</span></span>
<span id="cb36-6"><a href="data-organization-basics.html#cb36-6"></a><span class="kw">head</span>(subset.data1) </span></code></pre></div>
<pre><code>##    BMI     MAge MEdu
## 1 27.7 22.99928    3
## 2 26.8 30.05142    3
## 3 33.2 28.04660    3
## 4 30.1 34.81796    3
## 5 37.4 42.68440    3
## 6 33.3 24.94960    3</code></pre>
<p>Conversely, if we want to remove all columns except those that we are interested in within the ‘subset.columns’ vector, we can write the code as follows (to achieve the same results).</p>
<p>Note that we have to first create a vector of TRUE/FALSE’s here to execute the removal script written below:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="data-organization-basics.html#cb38-1"></a><span class="co"># First specify which columns we would like to remove</span></span>
<span id="cb38-2"><a href="data-organization-basics.html#cb38-2"></a>remove.columns &lt;-<span class="st"> </span><span class="kw">colnames</span>(full.data) <span class="op">%in%</span><span class="st"> </span>subset.columns </span>
<span id="cb38-3"><a href="data-organization-basics.html#cb38-3"></a></span>
<span id="cb38-4"><a href="data-organization-basics.html#cb38-4"></a><span class="co"># Viewing this new vector</span></span>
<span id="cb38-5"><a href="data-organization-basics.html#cb38-5"></a>remove.columns </span></code></pre></div>
<pre><code>##  [1] FALSE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</code></pre>
<p>This creates a vector of TRUE/FALSE’s denoting whether or not each column is included in the ‘subset.columns’ vector</p>
<p>Now we can subset our dataset. Here, we decide to keep those that are labeled ‘FALSE’ in the remove.columns vector. This will remove the columns that are NOT contained in the subset.columns vector</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="data-organization-basics.html#cb40-1"></a>subset.data2 &lt;-<span class="st"> </span>full.data[,<span class="op">!</span>remove.columns]</span>
<span id="cb40-2"><a href="data-organization-basics.html#cb40-2"></a></span>
<span id="cb40-3"><a href="data-organization-basics.html#cb40-3"></a><span class="co"># Viewing the top of this dataframe</span></span>
<span id="cb40-4"><a href="data-organization-basics.html#cb40-4"></a><span class="kw">head</span>(subset.data2)</span></code></pre></div>
<pre><code>##   ID       BW GA     DWAs     DWCd     DWCr       UAs       UCd      UCr
## 1  1 3180.058 34 6.426464 1.292941 51.67987 10.192695 0.7537104 42.60187
## 2  2 3210.823 43 7.832384 1.798535 50.10409 11.815088 0.9789506 41.30757
## 3  3 3311.551 40 7.516569 1.288461 48.74001 10.079057 0.1903262 36.47716
## 4  4 3266.844 32 5.906656 2.075259 50.92745  8.719123 0.9364825 42.47987
## 5  5 3664.088 35 7.181873 2.762643 55.16882  9.436559 1.4977829 47.78528
## 6  6 3328.988 40 9.723429 3.054057 51.14812 11.589403 1.6645837 38.26386</code></pre>
<p>We can also easily subset data based on row numbers. For example, to keep only the first 100 rows:</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="data-organization-basics.html#cb42-1"></a>subset.data3 &lt;-<span class="st"> </span>full.data[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>,]</span>
<span id="cb42-2"><a href="data-organization-basics.html#cb42-2"></a></span>
<span id="cb42-3"><a href="data-organization-basics.html#cb42-3"></a><span class="co"># Viewing the dimensions of this new dataframe</span></span>
<span id="cb42-4"><a href="data-organization-basics.html#cb42-4"></a><span class="kw">dim</span>(subset.data3)</span></code></pre></div>
<pre><code>## [1] 100  12</code></pre>
<p>To remove the first 100 rows:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="data-organization-basics.html#cb44-1"></a>subset.data4 &lt;-<span class="st"> </span>full.data[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>),]</span>
<span id="cb44-2"><a href="data-organization-basics.html#cb44-2"></a></span>
<span id="cb44-3"><a href="data-organization-basics.html#cb44-3"></a><span class="co"># Viewing the dimensions of this new dataframe</span></span>
<span id="cb44-4"><a href="data-organization-basics.html#cb44-4"></a><span class="kw">dim</span>(subset.data4)</span></code></pre></div>
<pre><code>## [1] 100  12</code></pre>
<p>To filter data using <strong>conditional statements</strong>:</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="data-organization-basics.html#cb46-1"></a>subset.data5 &lt;-<span class="st"> </span>full.data[<span class="kw">which</span>(full.data<span class="op">$</span>BMI <span class="op">&gt;</span><span class="st"> </span><span class="dv">25</span> <span class="op">&amp;</span><span class="st"> </span>full.data<span class="op">$</span>MAge <span class="op">&gt;</span><span class="st"> </span><span class="dv">31</span>),]</span>
<span id="cb46-2"><a href="data-organization-basics.html#cb46-2"></a></span>
<span id="cb46-3"><a href="data-organization-basics.html#cb46-3"></a><span class="co"># Viewing the top of this new dataframe</span></span>
<span id="cb46-4"><a href="data-organization-basics.html#cb46-4"></a><span class="kw">head</span>(subset.data5)</span></code></pre></div>
<pre><code>##    ID  BMI     MAge MEdu       BW GA     DWAs      DWCd     DWCr       UAs
## 4   4 30.1 34.81796    3 3266.844 32 5.906656 2.0752589 50.92745  8.719123
## 5   5 37.4 42.68440    3 3664.088 35 7.181873 2.7626433 55.16882  9.436559
## 9   9 36.9 33.58589    3 3260.482 39 9.074928 2.7277549 55.72826 10.818153
## 13 13 33.7 33.82961    3 3481.293 36 7.101634 0.8443918 47.11677  9.967185
## 22 22 25.7 37.08028    3 3387.046 43 7.207447 2.8088453 48.08648  9.446643
## 31 31 28.4 47.85761    3 3173.033 30 6.032807 2.1929549 45.71856  9.917588
##           UCd      UCr
## 4   0.9364825 42.47987
## 5   1.4977829 47.78528
## 9   1.6585757 42.58577
## 13 -0.3466431 36.74220
## 22  1.9891049 34.16921
## 31  1.1194851 37.82297</code></pre>
<p>Filtering data based on conditions can also be done using the subset function:</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="data-organization-basics.html#cb48-1"></a>subset.data6 &lt;-<span class="st"> </span><span class="kw">subset</span>(full.data, BMI <span class="op">&gt;</span><span class="st"> </span><span class="dv">25</span> <span class="op">&amp;</span><span class="st"> </span>MAge <span class="op">&gt;</span><span class="st"> </span><span class="dv">31</span>)</span></code></pre></div>
<p>Additionally, we can subset and select specific columns we would like to keep, using ‘select’ within the subset function:</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="data-organization-basics.html#cb49-1"></a>subset.data7 &lt;-<span class="st"> </span><span class="kw">subset</span>(full.data, BMI <span class="op">&lt;</span><span class="st"> </span><span class="dv">22</span> <span class="op">|</span><span class="st"> </span>BMI <span class="op">&gt;</span><span class="st"> </span><span class="dv">27</span>, </span>
<span id="cb49-2"><a href="data-organization-basics.html#cb49-2"></a>                       <span class="dt">select=</span><span class="kw">c</span>(<span class="st">&quot;BMI&quot;</span>, <span class="st">&quot;MAge&quot;</span>, <span class="st">&quot;MEdu&quot;</span>))</span></code></pre></div>
<p>For more information on the subset function, see its associated <a href="https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/subset">RDocumentation website</a>.</p>
</div>
<div id="melting-and-casting-data-using-base-r-syntax" class="section level4">
<h4>Melting and Casting Data using Base R Syntax</h4>
<p>Melting and casting refers to the conversion of data to “long” or “wide” form. You will often see data within the environmental health field in wide format; though long format is necessary for some procedures, such as plotting with <a href="https://ggplot2.tidyverse.org">ggplot2</a>.</p>
<p>Here, we’ll illustrate some example script to melt and cast data using the <a href="https://www.rdocumentation.org/packages/reshape2/versions/1.4.4">reshape2 package</a>.
Let’s first load the reshape2 library:</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="data-organization-basics.html#cb50-1"></a><span class="kw">library</span>(reshape2)</span></code></pre></div>
<p>Using the fully merged dataframe, let’s remind ourselves what these data look like in the current dataframe format:</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="data-organization-basics.html#cb51-1"></a><span class="kw">head</span>(full.data)</span></code></pre></div>
<pre><code>##   ID  BMI     MAge MEdu       BW GA     DWAs     DWCd     DWCr       UAs
## 1  1 27.7 22.99928    3 3180.058 34 6.426464 1.292941 51.67987 10.192695
## 2  2 26.8 30.05142    3 3210.823 43 7.832384 1.798535 50.10409 11.815088
## 3  3 33.2 28.04660    3 3311.551 40 7.516569 1.288461 48.74001 10.079057
## 4  4 30.1 34.81796    3 3266.844 32 5.906656 2.075259 50.92745  8.719123
## 5  5 37.4 42.68440    3 3664.088 35 7.181873 2.762643 55.16882  9.436559
## 6  6 33.3 24.94960    3 3328.988 40 9.723429 3.054057 51.14812 11.589403
##         UCd      UCr
## 1 0.7537104 42.60187
## 2 0.9789506 41.30757
## 3 0.1903262 36.47716
## 4 0.9364825 42.47987
## 5 1.4977829 47.78528
## 6 1.6645837 38.26386</code></pre>
<p>These data are represented by single subject identifiers listed as unique IDs per row, with associated environmental measures and demographic data organized across the columns. Thus, this dataframe is currently in <strong>wide (also known as casted)</strong> format.</p>
<p>Let’s convert this dataframe to <strong>long (also known as melted)</strong> format:</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="data-organization-basics.html#cb53-1"></a><span class="co"># Here, we are saying that we want a row for each unique </span></span>
<span id="cb53-2"><a href="data-organization-basics.html#cb53-2"></a><span class="co"># sample ID - variable measure pair</span></span>
<span id="cb53-3"><a href="data-organization-basics.html#cb53-3"></a>full.melted &lt;-<span class="st"> </span><span class="kw">melt</span>(full.data, <span class="dt">id=</span><span class="st">&quot;ID&quot;</span>) </span>
<span id="cb53-4"><a href="data-organization-basics.html#cb53-4"></a></span>
<span id="cb53-5"><a href="data-organization-basics.html#cb53-5"></a><span class="co"># Viewing this new dataframe</span></span>
<span id="cb53-6"><a href="data-organization-basics.html#cb53-6"></a><span class="kw">head</span>(full.melted) </span></code></pre></div>
<pre><code>##   ID variable value
## 1  1      BMI  27.7
## 2  2      BMI  26.8
## 3  3      BMI  33.2
## 4  4      BMI  30.1
## 5  5      BMI  37.4
## 6  6      BMI  33.3</code></pre>
<p>You can see here that each measure that was originally contained as a unique column has been reoriented, such that the original column header is now listed throughout the second column labeled “variable”. Then, the third column contains the value of this variable.</p>
<p>Let’s see an example view of the middle of this new dataframe</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="data-organization-basics.html#cb55-1"></a>full.melted[<span class="dv">1100</span><span class="op">:</span><span class="dv">1110</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##       ID variable    value
## 1100 100     DWAs 7.928885
## 1101 101     DWAs 8.677403
## 1102 102     DWAs 8.115183
## 1103 103     DWAs 7.134189
## 1104 104     DWAs 8.816142
## 1105 105     DWAs 7.487227
## 1106 106     DWAs 7.541973
## 1107 107     DWAs 6.313516
## 1108 108     DWAs 6.654474
## 1109 109     DWAs 7.564429
## 1110 110     DWAs 7.357122</code></pre>
<p>Here, we can see a different variable (DWAs) now being listed. This continues throughout the entire dataframe, which has the following dimensions:</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="data-organization-basics.html#cb57-1"></a><span class="kw">dim</span>(full.melted)</span></code></pre></div>
<pre><code>## [1] 2200    3</code></pre>
<p>Thus, this dataframe is clearly melted, in long format.</p>
<p>Let’s now re-cast this dataframe back into wide format using the ‘dcast’ function</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="data-organization-basics.html#cb59-1"></a><span class="co"># Here, we are telling the dcast </span></span>
<span id="cb59-2"><a href="data-organization-basics.html#cb59-2"></a><span class="co"># function to give us a sample (ID) for every variable in the column labeled &#39;variable&#39;. </span></span>
<span id="cb59-3"><a href="data-organization-basics.html#cb59-3"></a><span class="co"># Then it automatically fills the dataframe with values from the &#39;value&#39; column</span></span>
<span id="cb59-4"><a href="data-organization-basics.html#cb59-4"></a>full.cast &lt;-<span class="st"> </span><span class="kw">dcast</span>(full.melted, ID <span class="op">~</span><span class="st"> </span>variable) </span>
<span id="cb59-5"><a href="data-organization-basics.html#cb59-5"></a><span class="kw">head</span>(full.cast)</span></code></pre></div>
<pre><code>##   ID  BMI     MAge MEdu       BW GA     DWAs     DWCd     DWCr       UAs
## 1  1 27.7 22.99928    3 3180.058 34 6.426464 1.292941 51.67987 10.192695
## 2  2 26.8 30.05142    3 3210.823 43 7.832384 1.798535 50.10409 11.815088
## 3  3 33.2 28.04660    3 3311.551 40 7.516569 1.288461 48.74001 10.079057
## 4  4 30.1 34.81796    3 3266.844 32 5.906656 2.075259 50.92745  8.719123
## 5  5 37.4 42.68440    3 3664.088 35 7.181873 2.762643 55.16882  9.436559
## 6  6 33.3 24.94960    3 3328.988 40 9.723429 3.054057 51.14812 11.589403
##         UCd      UCr
## 1 0.7537104 42.60187
## 2 0.9789506 41.30757
## 3 0.1903262 36.47716
## 4 0.9364825 42.47987
## 5 1.4977829 47.78528
## 6 1.6645837 38.26386</code></pre>
<p>Here, we can see that this dataframe is back in its original casted (or wide) format.</p>
</div>
</div>
<div id="introduction-to-tidyverse" class="section level2">
<h2>Introduction to Tidyverse</h2>
<p><a href="https://www.tidyverse.org">Tidyverse</a> is a collection of packages that are commonly used to more efficiently organize and manipulate datasets in R. This collection of packages has its own specific type of syntax, dataset and formatting protocols that slightly differ from the Base R functions. Here, we will carry out all the of the same data organization exercises described above using Tidyverse.</p>
<div id="downloading-and-loading-the-tidyverse-package" class="section level4">
<h4>Downloading and Loading the Tidyverse Package</h4>
<p>If you don’t have tidyverse already installed, you will need to install it using:</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="data-organization-basics.html#cb61-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">require</span>(tidyverse)) <span class="kw">install.packages</span>(<span class="st">&quot;tidyverse&quot;</span>)</span></code></pre></div>
<p>And then load the tidyverse package using:</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="data-organization-basics.html#cb62-1"></a><span class="kw">library</span>(tidyverse)</span></code></pre></div>
</div>
<div id="merging-data-using-tidyverse-syntax" class="section level4">
<h4>Merging Data using Tidyverse Syntax</h4>
<p>To merge the same example dataframes using tidyverse, you can run the following script:</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="data-organization-basics.html#cb63-1"></a>full.data.tidy &lt;-<span class="st"> </span><span class="kw">inner_join</span>(demo.data, chem.data, <span class="dt">by=</span><span class="st">&quot;ID&quot;</span>)</span>
<span id="cb63-2"><a href="data-organization-basics.html#cb63-2"></a><span class="co"># Note, for future scripting purposes, we can still merge with different IDs </span></span>
<span id="cb63-3"><a href="data-organization-basics.html#cb63-3"></a><span class="co"># using: by = c(&quot;ID.Demo&quot;=&quot;ID.Chem&quot;)</span></span>
<span id="cb63-4"><a href="data-organization-basics.html#cb63-4"></a></span>
<span id="cb63-5"><a href="data-organization-basics.html#cb63-5"></a><span class="kw">head</span>(full.data.tidy)</span></code></pre></div>
<pre><code>##   ID  BMI     MAge MEdu       BW GA     DWAs     DWCd     DWCr       UAs
## 1  1 27.7 22.99928    3 3180.058 34 6.426464 1.292941 51.67987 10.192695
## 2  2 26.8 30.05142    3 3210.823 43 7.832384 1.798535 50.10409 11.815088
## 3  3 33.2 28.04660    3 3311.551 40 7.516569 1.288461 48.74001 10.079057
## 4  4 30.1 34.81796    3 3266.844 32 5.906656 2.075259 50.92745  8.719123
## 5  5 37.4 42.68440    3 3664.088 35 7.181873 2.762643 55.16882  9.436559
## 6  6 33.3 24.94960    3 3328.988 40 9.723429 3.054057 51.14812 11.589403
##         UCd      UCr
## 1 0.7537104 42.60187
## 2 0.9789506 41.30757
## 3 0.1903262 36.47716
## 4 0.9364825 42.47987
## 5 1.4977829 47.78528
## 6 1.6645837 38.26386</code></pre>
</div>
<div id="filtering-and-subsetting-data-using-tidyverse-syntax" class="section level4">
<h4>Filtering and Subsetting Data using Tidyverse Syntax</h4>
<p>To subset columns in tidyverse, run the following:</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="data-organization-basics.html#cb65-1"></a>subset.tidy1 =<span class="st"> </span>full.data.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="kw">all_of</span>(subset.columns))</span>
<span id="cb65-2"><a href="data-organization-basics.html#cb65-2"></a><span class="kw">head</span>(subset.tidy1)</span></code></pre></div>
<pre><code>##    BMI     MAge MEdu
## 1 27.7 22.99928    3
## 2 26.8 30.05142    3
## 3 33.2 28.04660    3
## 4 30.1 34.81796    3
## 5 37.4 42.68440    3
## 6 33.3 24.94960    3</code></pre>
<p>Note that you can also include column identifiers that may get dropped in the subsetting vector here:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="data-organization-basics.html#cb67-1"></a><span class="co"># Note that we&#39;re including a &#39;fake&#39; column here &#39;NotAColName&#39; to illustrate </span></span>
<span id="cb67-2"><a href="data-organization-basics.html#cb67-2"></a><span class="co"># how to incorporate additional columns; though this column gets dropped in </span></span>
<span id="cb67-3"><a href="data-organization-basics.html#cb67-3"></a><span class="co"># the next line of code</span></span>
<span id="cb67-4"><a href="data-organization-basics.html#cb67-4"></a>subset.columns2 &lt;-<span class="st"> </span><span class="kw">c</span>(subset.columns, <span class="st">&quot;NotAColName&quot;</span>)</span>
<span id="cb67-5"><a href="data-organization-basics.html#cb67-5"></a></span>
<span id="cb67-6"><a href="data-organization-basics.html#cb67-6"></a><span class="co"># Viewing this new vector</span></span>
<span id="cb67-7"><a href="data-organization-basics.html#cb67-7"></a>subset.columns2</span></code></pre></div>
<pre><code>## [1] &quot;BMI&quot;         &quot;MAge&quot;        &quot;MEdu&quot;        &quot;NotAColName&quot;</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="data-organization-basics.html#cb69-1"></a>subset.tidy2 &lt;-<span class="st"> </span>full.data.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="kw">any_of</span>(subset.columns2))</span>
<span id="cb69-2"><a href="data-organization-basics.html#cb69-2"></a></span>
<span id="cb69-3"><a href="data-organization-basics.html#cb69-3"></a><span class="co"># Viewing the top of this new dataframe</span></span>
<span id="cb69-4"><a href="data-organization-basics.html#cb69-4"></a><span class="kw">head</span>(subset.tidy2) </span></code></pre></div>
<pre><code>##    BMI     MAge MEdu
## 1 27.7 22.99928    3
## 2 26.8 30.05142    3
## 3 33.2 28.04660    3
## 4 30.1 34.81796    3
## 5 37.4 42.68440    3
## 6 33.3 24.94960    3</code></pre>
<p>Note that the ‘fake’ column ‘NotAColName’ gets automatically dropped here</p>
<p>To remove columns using tidyverse, you can run the following:</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="data-organization-basics.html#cb71-1"></a><span class="co"># Removing columns</span></span>
<span id="cb71-2"><a href="data-organization-basics.html#cb71-2"></a>subset.tidy3 &lt;-<span class="st"> </span>full.data.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>subset.columns)</span>
<span id="cb71-3"><a href="data-organization-basics.html#cb71-3"></a></span>
<span id="cb71-4"><a href="data-organization-basics.html#cb71-4"></a><span class="co"># Viewing this new dataframe</span></span>
<span id="cb71-5"><a href="data-organization-basics.html#cb71-5"></a><span class="kw">head</span>(subset.tidy3) </span></code></pre></div>
<pre><code>##   ID       BW GA     DWAs     DWCd     DWCr       UAs       UCd      UCr
## 1  1 3180.058 34 6.426464 1.292941 51.67987 10.192695 0.7537104 42.60187
## 2  2 3210.823 43 7.832384 1.798535 50.10409 11.815088 0.9789506 41.30757
## 3  3 3311.551 40 7.516569 1.288461 48.74001 10.079057 0.1903262 36.47716
## 4  4 3266.844 32 5.906656 2.075259 50.92745  8.719123 0.9364825 42.47987
## 5  5 3664.088 35 7.181873 2.762643 55.16882  9.436559 1.4977829 47.78528
## 6  6 3328.988 40 9.723429 3.054057 51.14812 11.589403 1.6645837 38.26386</code></pre>
<p>Subsetting rows using tidyverse:</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="data-organization-basics.html#cb73-1"></a><span class="co"># Selecting to retain the first 100 rows</span></span>
<span id="cb73-2"><a href="data-organization-basics.html#cb73-2"></a>subset.tidy4 &lt;-<span class="st"> </span>full.data.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>) </span>
<span id="cb73-3"><a href="data-organization-basics.html#cb73-3"></a><span class="kw">dim</span>(subset.tidy4)</span></code></pre></div>
<pre><code>## [1] 100  12</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="data-organization-basics.html#cb75-1"></a><span class="co"># Selecting to remove the first 100 rows</span></span>
<span id="cb75-2"><a href="data-organization-basics.html#cb75-2"></a>subset.tidy5 &lt;-<span class="st"> </span>full.data.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">slice</span>(<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>))</span>
<span id="cb75-3"><a href="data-organization-basics.html#cb75-3"></a><span class="kw">dim</span>(subset.tidy5)</span></code></pre></div>
<pre><code>## [1] 100  12</code></pre>
<p>Filtering data based on conditional statements using tidyverse:</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="data-organization-basics.html#cb77-1"></a>subset.tidy6 &lt;-<span class="st"> </span>full.data.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(BMI <span class="op">&gt;</span><span class="st"> </span><span class="dv">25</span> <span class="op">&amp;</span><span class="st"> </span>MAge <span class="op">&gt;</span><span class="st"> </span><span class="dv">31</span>)</span>
<span id="cb77-2"><a href="data-organization-basics.html#cb77-2"></a><span class="kw">dim</span>(subset.tidy6)</span></code></pre></div>
<pre><code>## [1] 49 12</code></pre>
<p>Another example of a conditional statement that can be used to filter data:</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="data-organization-basics.html#cb79-1"></a>subset.tidy7 &lt;-<span class="st"> </span>full.data.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(BMI <span class="op">&gt;</span><span class="st"> </span><span class="dv">25</span> <span class="op">&amp;</span><span class="st"> </span>MAge <span class="op">&gt;</span><span class="st"> </span><span class="dv">31</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(BMI, MAge, MEdu)</span></code></pre></div>
</div>
<div id="melting-and-casting-data-using-tidyverse-syntax" class="section level4">
<h4>Melting and Casting Data using Tidyverse Syntax</h4>
<p>To melt and cast data in tidyverse, you can use the ‘pivot’ functions (i.e., ‘pivot_longer’ or ‘pivot_wider’). These are exemplified below.</p>
<p>Melting to long format using tidyverse:</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="data-organization-basics.html#cb80-1"></a>full.pivotlong &lt;-<span class="st"> </span>full.data.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="op">-</span>ID, <span class="dt">names_to =</span> <span class="st">&quot;var&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;value&quot;</span>)</span>
<span id="cb80-2"><a href="data-organization-basics.html#cb80-2"></a><span class="kw">head</span>(full.pivotlong, <span class="dv">15</span>)</span></code></pre></div>
<pre><code>## # A tibble: 15 x 3
##       ID var      value
##    &lt;int&gt; &lt;chr&gt;    &lt;dbl&gt;
##  1     1 BMI     27.7  
##  2     1 MAge    23.0  
##  3     1 MEdu     3    
##  4     1 BW    3180.   
##  5     1 GA      34    
##  6     1 DWAs     6.43 
##  7     1 DWCd     1.29 
##  8     1 DWCr    51.7  
##  9     1 UAs     10.2  
## 10     1 UCd      0.754
## 11     1 UCr     42.6  
## 12     2 BMI     26.8  
## 13     2 MAge    30.1  
## 14     2 MEdu     3    
## 15     2 BW    3211.</code></pre>
<p>Casting to wide format using tidyverse:</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="data-organization-basics.html#cb82-1"></a>full.pivotwide &lt;-<span class="st"> </span>full.pivotlong <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;var&quot;</span>, <span class="dt">values_from=</span><span class="st">&quot;value&quot;</span>)</span>
<span id="cb82-2"><a href="data-organization-basics.html#cb82-2"></a><span class="kw">head</span>(full.pivotwide)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 12
##      ID   BMI  MAge  MEdu    BW    GA  DWAs  DWCd  DWCr   UAs   UCd   UCr
##   &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1     1  27.7  23.0     3 3180.    34  6.43  1.29  51.7 10.2  0.754  42.6
## 2     2  26.8  30.1     3 3211.    43  7.83  1.80  50.1 11.8  0.979  41.3
## 3     3  33.2  28.0     3 3312.    40  7.52  1.29  48.7 10.1  0.190  36.5
## 4     4  30.1  34.8     3 3267.    32  5.91  2.08  50.9  8.72 0.936  42.5
## 5     5  37.4  42.7     3 3664.    35  7.18  2.76  55.2  9.44 1.50   47.8
## 6     6  33.3  24.9     3 3329.    40  9.72  3.05  51.1 11.6  1.66   38.3</code></pre>
</div>
</div>
<div id="concluding-remarks-2" class="section level2">
<h2>Concluding Remarks</h2>
<p>Together, this training module provides introductory level information on the basics of data organization in R. The important data organization / manipulation methods of merging, filtering, subsetting, melting, and casted are presented on an environmentally relevant dataset.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction-to-coding-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="finding-and-visualizing-data-trends.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
