<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 The Basics for Data Organization | DMAC Training Modules</title>
  <meta name="description" content="This is a resource for R training using environmental health datasets along standard code pipelines and methods." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 The Basics for Data Organization | DMAC Training Modules" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a resource for R training using environmental health datasets along standard code pipelines and methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 The Basics for Data Organization | DMAC Training Modules" />
  
  <meta name="twitter:description" content="This is a resource for R training using environmental health datasets along standard code pipelines and methods." />
  

<meta name="author" content="Kyle Roell, Lauren Koval, Julia Rager" />


<meta name="date" content="2021-06-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="intro.html"/>
<link rel="next" href="finding-and-visualizing-data-trends.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DMAC Training Modules</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Setting Up Your R Environment</a><ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#r-and-rstudio"><i class="fa fa-check"></i><b>2.1</b> R and RStudio</a><ul>
<li class="chapter" data-level="2.1.1" data-path="intro.html"><a href="intro.html#downloading-r-and-rstudio"><i class="fa fa-check"></i><b>2.1.1</b> Downloading R and RStudio</a></li>
<li class="chapter" data-level="2.1.2" data-path="intro.html"><a href="intro.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>2.1.2</b> Installing R and RStudio</a></li>
<li class="chapter" data-level="2.1.3" data-path="intro.html"><a href="intro.html#installing-and-loading-packages"><i class="fa fa-check"></i><b>2.1.3</b> Installing and Loading Packages</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#scripting-basics"><i class="fa fa-check"></i><b>2.2</b> Scripting Basics</a><ul>
<li class="chapter" data-level="2.2.1" data-path="intro.html"><a href="intro.html#setting-working-directory"><i class="fa fa-check"></i><b>2.2.1</b> Setting Working Directory</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro.html"><a href="intro.html#importing-and-exporting-files"><i class="fa fa-check"></i><b>2.2.2</b> Importing and Exporting Files</a></li>
<li class="chapter" data-level="2.2.3" data-path="intro.html"><a href="intro.html#viewing-data"><i class="fa fa-check"></i><b>2.2.3</b> Viewing Data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dataorg.html"><a href="dataorg.html"><i class="fa fa-check"></i><b>3</b> The Basics for Data Organization</a><ul>
<li class="chapter" data-level="3.1" data-path="dataorg.html"><a href="dataorg.html#basic-data-manipulation"><i class="fa fa-check"></i><b>3.1</b> Basic Data Manipulation</a><ul>
<li class="chapter" data-level="3.1.1" data-path="dataorg.html"><a href="dataorg.html#merging"><i class="fa fa-check"></i><b>3.1.1</b> Merging</a></li>
<li class="chapter" data-level="3.1.2" data-path="dataorg.html"><a href="dataorg.html#filtering-subsetting"><i class="fa fa-check"></i><b>3.1.2</b> Filtering &amp; subsetting</a></li>
<li class="chapter" data-level="3.1.3" data-path="dataorg.html"><a href="dataorg.html#melt-and-cast"><i class="fa fa-check"></i><b>3.1.3</b> Melt and Cast</a></li>
<li class="chapter" data-level="3.1.4" data-path="dataorg.html"><a href="dataorg.html#tidyverse"><i class="fa fa-check"></i><b>3.1.4</b> Tidyverse</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html"><i class="fa fa-check"></i><b>4</b> Finding and Visualizing Data Trends</a><ul>
<li class="chapter" data-level="4.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#basic-statistical-tests-and-visualizations-of-data"><i class="fa fa-check"></i><b>4.1</b> Basic Statistical Tests and Visualizations of Data</a><ul>
<li class="chapter" data-level="4.1.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#normality"><i class="fa fa-check"></i><b>4.1.1</b> Normality</a></li>
<li class="chapter" data-level="4.1.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#t-tests"><i class="fa fa-check"></i><b>4.1.2</b> T-tests</a></li>
<li class="chapter" data-level="4.1.3" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#anova"><i class="fa fa-check"></i><b>4.1.3</b> ANOVA</a></li>
<li class="chapter" data-level="4.1.4" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#regression-linear-regression-and-logistic-regression"><i class="fa fa-check"></i><b>4.1.4</b> Regression: linear regression and logistic regression</a></li>
<li class="chapter" data-level="4.1.5" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#chi-squared-test-box-plots"><i class="fa fa-check"></i><b>4.1.5</b> Chi-squared test – box plots</a></li>
<li class="chapter" data-level="4.1.6" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#fishers-exact-test"><i class="fa fa-check"></i><b>4.1.6</b> Fisher’s exact test</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#heat-maps"><i class="fa fa-check"></i><b>4.2</b> Heat maps</a><ul>
<li class="chapter" data-level="4.2.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#pheatmap"><i class="fa fa-check"></i><b>4.2.1</b> pheatmap</a></li>
<li class="chapter" data-level="4.2.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#heatmap2"><i class="fa fa-check"></i><b>4.2.2</b> heatmap2</a></li>
<li class="chapter" data-level="4.2.3" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#superheat"><i class="fa fa-check"></i><b>4.2.3</b> superheat</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#clustering"><i class="fa fa-check"></i><b>4.3</b> Clustering</a><ul>
<li class="chapter" data-level="4.3.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#hierarchical"><i class="fa fa-check"></i><b>4.3.1</b> Hierarchical</a></li>
<li class="chapter" data-level="4.3.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#k-means"><i class="fa fa-check"></i><b>4.3.2</b> K-means</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#data-reduction-pca"><i class="fa fa-check"></i><b>4.4</b> Data reduction (PCA)</a><ul>
<li class="chapter" data-level="4.4.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#visualize-pca-plot"><i class="fa fa-check"></i><b>4.4.1</b> Visualize PCA Plot</a></li>
<li class="chapter" data-level="4.4.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#identify-of-variance-captured"><i class="fa fa-check"></i><b>4.4.2</b> Identify % of variance captured</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html"><i class="fa fa-check"></i><b>5</b> Multi-Omics Analyses for Environmental Health</a><ul>
<li class="chapter" data-level="5.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#exposomics"><i class="fa fa-check"></i><b>5.1</b> Exposomics</a><ul>
<li class="chapter" data-level="5.1.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#placenta-exposome"><i class="fa fa-check"></i><b>5.1.1</b> Placenta Exposome</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#transcriptomics"><i class="fa fa-check"></i><b>5.2</b> Transcriptomics</a><ul>
<li class="chapter" data-level="5.2.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#deseq2-rnaseq"><i class="fa fa-check"></i><b>5.2.1</b> DESeq2 / RNAseq</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#genome-wide-microrna"><i class="fa fa-check"></i><b>5.3</b> Genome-wide MicroRNA</a></li>
<li class="chapter" data-level="5.4" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#genome-wide-dna-methylation"><i class="fa fa-check"></i><b>5.4</b> Genome-wide DNA Methylation</a><ul>
<li class="chapter" data-level="5.4.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#illumina-array-data"><i class="fa fa-check"></i><b>5.4.1</b> Illumina array data</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#proteomics"><i class="fa fa-check"></i><b>5.5</b> Proteomics</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mixtures-analyses-for-environmental-health.html"><a href="mixtures-analyses-for-environmental-health.html"><i class="fa fa-check"></i><b>6</b> Mixtures Analyses for Environmental Health</a><ul>
<li class="chapter" data-level="6.1" data-path="mixtures-analyses-for-environmental-health.html"><a href="mixtures-analyses-for-environmental-health.html#sufficient-similarity"><i class="fa fa-check"></i><b>6.1</b> Sufficient Similarity</a></li>
<li class="chapter" data-level="6.2" data-path="mixtures-analyses-for-environmental-health.html"><a href="mixtures-analyses-for-environmental-health.html#mixtures-modeling-through-qgcomp"><i class="fa fa-check"></i><b>6.2</b> Mixtures Modeling through qgcomp</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html"><i class="fa fa-check"></i><b>7</b> Environmental Health Databases</a><ul>
<li class="chapter" data-level="7.1" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html#comparative-toxicogenomics-database-ctd"><i class="fa fa-check"></i><b>7.1</b> Comparative Toxicogenomics Database (CTD)</a></li>
<li class="chapter" data-level="7.2" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html#gene-expression-omnibus-geo"><i class="fa fa-check"></i><b>7.2</b> Gene Expression Omnibus (GEO)</a></li>
<li class="chapter" data-level="7.3" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html#nhanes"><i class="fa fa-check"></i><b>7.3</b> NHANES</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">DMAC Training Modules</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dataorg" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> The Basics for Data Organization</h1>
<p>One of the first things that is useful to learn about is data organization and manipulation. While this will not be a complete guide, as there are many things you can do in R to manipulate your data, hopefully this will get you started and will provide you with the fundamentals for working with data in R.</p>
<div id="basic-data-manipulation" class="section level2">
<h2><span class="header-section-number">3.1</span> Basic Data Manipulation</h2>
<p>Data manipulation generally refers to organizing and formatting data in a way that makes it easier to read and work with. This can be done through base R operations and functions or through a collection of packages (and philosophy) known as <a href="https://www.tidyverse.org">Tidyverse</a>.</p>
<p>In this brief tutorial we will first go over the basic operations and functions you can use to manipulate data, and then show the exact same steps using the Tidyverse packages, functions, and syntax.</p>
<div id="merging" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Merging</h3>
<p>Here, we will be looking at merging as the joining together of two or more datasets, connected using a common identifier, generally some sort of ID. This is useful if you various datasets describing different variables across the same participants. In our example, we will be joining demographic data and environmental exposure data.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="dataorg.html#cb13-1"></a><span class="co">#First, let&#39;s read in our datasets</span></span>
<span id="cb13-2"><a href="dataorg.html#cb13-2"></a>demo.data =<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;~/Downloads/demo_data.csv&quot;</span>);</span>
<span id="cb13-3"><a href="dataorg.html#cb13-3"></a>chem.data =<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;~/Downloads/chem_data.csv&quot;</span>);</span>
<span id="cb13-4"><a href="dataorg.html#cb13-4"></a></span>
<span id="cb13-5"><a href="dataorg.html#cb13-5"></a><span class="co">#We can merge these by their shared ID column</span></span>
<span id="cb13-6"><a href="dataorg.html#cb13-6"></a>full.data =<span class="st"> </span><span class="kw">merge</span>(demo.data, chem.data, <span class="dt">by=</span><span class="st">&quot;ID&quot;</span>)</span>
<span id="cb13-7"><a href="dataorg.html#cb13-7"></a></span>
<span id="cb13-8"><a href="dataorg.html#cb13-8"></a><span class="co">#We could also merge by different ID columns using: by.x = &quot;ID.Demo&quot;, by.y=&quot;ID.Chem&quot;</span></span></code></pre></div>
</div>
<div id="filtering-subsetting" class="section level3">
<h3><span class="header-section-number">3.1.2</span> Filtering &amp; subsetting</h3>
<p>Filtering and subsetting data is useful when you need to restrict your dataset to samples or participants with specific conditions. It is also useful for simply removing particular variables or samples.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="dataorg.html#cb14-1"></a><span class="co">#Let&#39;s define a vector of columns we want to keep</span></span>
<span id="cb14-2"><a href="dataorg.html#cb14-2"></a>subset.columns =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;BMI&quot;</span>, <span class="st">&quot;MAge&quot;</span>, <span class="st">&quot;MEdu&quot;</span>);</span>
<span id="cb14-3"><a href="dataorg.html#cb14-3"></a></span>
<span id="cb14-4"><a href="dataorg.html#cb14-4"></a><span class="co">#Now we can simply subset our data using those columns</span></span>
<span id="cb14-5"><a href="dataorg.html#cb14-5"></a>subset.data1 =<span class="st"> </span>full.data[,subset.columns];</span>
<span id="cb14-6"><a href="dataorg.html#cb14-6"></a></span>
<span id="cb14-7"><a href="dataorg.html#cb14-7"></a><span class="co">#If we want to remove those columns, need to identify which columns need removing</span></span>
<span id="cb14-8"><a href="dataorg.html#cb14-8"></a>remove.columns =<span class="st"> </span><span class="kw">colnames</span>(full.data) <span class="op">%in%</span><span class="st"> </span>subset.columns;</span>
<span id="cb14-9"><a href="dataorg.html#cb14-9"></a><span class="kw">cat</span>(remove.columns);</span></code></pre></div>
<pre><code>## FALSE TRUE TRUE TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="dataorg.html#cb16-1"></a><span class="co">#Now we can subset our dataset by only keeping those that we didn&#39;t specify</span></span>
<span id="cb16-2"><a href="dataorg.html#cb16-2"></a>subset.data1a =<span class="st"> </span>full.data[,<span class="op">!</span>remove.columns];</span>
<span id="cb16-3"><a href="dataorg.html#cb16-3"></a></span>
<span id="cb16-4"><a href="dataorg.html#cb16-4"></a><span class="co">#It is easy to subset data based on rows</span></span>
<span id="cb16-5"><a href="dataorg.html#cb16-5"></a><span class="co">#Keeping only first 100 rows</span></span>
<span id="cb16-6"><a href="dataorg.html#cb16-6"></a>subset.data2 =<span class="st"> </span>full.data[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>,];</span>
<span id="cb16-7"><a href="dataorg.html#cb16-7"></a><span class="co">#Removing the first 100 rows</span></span>
<span id="cb16-8"><a href="dataorg.html#cb16-8"></a>subset.data2a =<span class="st"> </span>full.data[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>),];</span>
<span id="cb16-9"><a href="dataorg.html#cb16-9"></a></span>
<span id="cb16-10"><a href="dataorg.html#cb16-10"></a><span class="co">#We can also filter data using conditional statements</span></span>
<span id="cb16-11"><a href="dataorg.html#cb16-11"></a>subset.data3 =<span class="st"> </span>full.data[<span class="kw">which</span>(full.data<span class="op">$</span>BMI <span class="op">&gt;</span><span class="st"> </span><span class="dv">25</span> <span class="op">&amp;</span><span class="st"> </span>full.data<span class="op">$</span>MAge <span class="op">&gt;</span><span class="st"> </span><span class="dv">31</span>),];</span>
<span id="cb16-12"><a href="dataorg.html#cb16-12"></a><span class="kw">head</span>(subset.data3);</span></code></pre></div>
<pre><code>##    ID      BMI     MAge     MEdu       BW       GA     DWAs      DWCd     DWCr
## 4   4 30.09710 34.81796 3.760030 3266.844 18.60876 5.906656 2.0752589 50.92745
## 5   5 37.41737 42.68440 4.484686 3664.088 32.57351 7.181873 2.7626433 55.16882
## 9   9 36.94794 33.58589 3.242318 3260.482 37.37241 9.074928 2.7277549 55.72826
## 13 13 33.65870 33.82961 2.924433 3481.293 33.68744 7.101634 0.8443918 47.11677
## 22 22 25.68414 37.08028 2.704409 3387.046 52.48203 7.207447 2.8088453 48.08648
## 31 31 28.39896 47.85761 6.453362 3173.033 20.10080 6.032807 2.1929549 45.71856
##          UAs        UCd      UCr
## 4   8.719123  0.9364825 42.47987
## 5   9.436559  1.4977829 47.78528
## 9  10.818153  1.6585757 42.58577
## 13  9.967185 -0.3466431 36.74220
## 22  9.446643  1.9891049 34.16921
## 31  9.917588  1.1194851 37.82297</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="dataorg.html#cb18-1"></a><span class="co">#This can also be done using the subset function</span></span>
<span id="cb18-2"><a href="dataorg.html#cb18-2"></a>subset.data4 =<span class="st"> </span><span class="kw">subset</span>(full.data, BMI <span class="op">&gt;</span><span class="st"> </span><span class="dv">25</span> <span class="op">&amp;</span><span class="st"> </span>MAge <span class="op">&gt;</span><span class="st"> </span><span class="dv">31</span>);</span>
<span id="cb18-3"><a href="dataorg.html#cb18-3"></a></span>
<span id="cb18-4"><a href="dataorg.html#cb18-4"></a><span class="co">#Additionally, we can subset and select only specific columns to keep</span></span>
<span id="cb18-5"><a href="dataorg.html#cb18-5"></a>subset.data4 =<span class="st"> </span><span class="kw">subset</span>(full.data, BMI <span class="op">&lt;</span><span class="st"> </span><span class="dv">22</span> <span class="op">|</span><span class="st"> </span>BMI <span class="op">&gt;</span><span class="st"> </span><span class="dv">27</span>, <span class="dt">select=</span><span class="kw">c</span>(<span class="st">&quot;BMI&quot;</span>, <span class="st">&quot;MAge&quot;</span>, <span class="st">&quot;MEdu&quot;</span>));</span></code></pre></div>
</div>
<div id="melt-and-cast" class="section level3">
<h3><span class="header-section-number">3.1.3</span> Melt and Cast</h3>
<p>Melting and casting refers to coverting data to “long” or “wide” form. Generally, our data is in wide format, but long format is necessary for some procedures, such as plotting with <a href="https://ggplot2.tidyverse.org">ggplot2</a>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="dataorg.html#cb19-1"></a><span class="co">#Melt and cast functions from reshape2 package</span></span>
<span id="cb19-2"><a href="dataorg.html#cb19-2"></a><span class="kw">library</span>(reshape2);</span>
<span id="cb19-3"><a href="dataorg.html#cb19-3"></a></span>
<span id="cb19-4"><a href="dataorg.html#cb19-4"></a><span class="co">#Melting data converts it to long format</span></span>
<span id="cb19-5"><a href="dataorg.html#cb19-5"></a><span class="co">#Here, we are saying we want a row for each sample (ID) and each column </span></span>
<span id="cb19-6"><a href="dataorg.html#cb19-6"></a>full.melted =<span class="st"> </span><span class="kw">melt</span>(full.data, <span class="dt">id=</span><span class="st">&quot;ID&quot;</span>)</span>
<span id="cb19-7"><a href="dataorg.html#cb19-7"></a><span class="co">#Let&#39;s look at the new data</span></span>
<span id="cb19-8"><a href="dataorg.html#cb19-8"></a><span class="kw">head</span>(full.melted);</span></code></pre></div>
<pre><code>##   ID variable    value
## 1  1      BMI 27.69249
## 2  2      BMI 26.79125
## 3  3      BMI 33.23209
## 4  4      BMI 30.09710
## 5  5      BMI 37.41737
## 6  6      BMI 33.28761</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="dataorg.html#cb21-1"></a><span class="co">#Casting puts data into wide format</span></span>
<span id="cb21-2"><a href="dataorg.html#cb21-2"></a><span class="co">#We are telling the cast function to give us a sample (ID) </span></span>
<span id="cb21-3"><a href="dataorg.html#cb21-3"></a><span class="co">#for every variable in the variable column</span></span>
<span id="cb21-4"><a href="dataorg.html#cb21-4"></a>full.cast =<span class="st"> </span><span class="kw">dcast</span>(full.melted, ID <span class="op">~</span><span class="st"> </span>variable);</span>
<span id="cb21-5"><a href="dataorg.html#cb21-5"></a><span class="kw">head</span>(full.cast);</span></code></pre></div>
<pre><code>##   ID      BMI     MAge     MEdu       BW       GA     DWAs     DWCd     DWCr
## 1  1 27.69249 22.99928 3.136759 3180.058 31.79569 6.426464 1.292941 51.67987
## 2  2 26.79125 30.05142 2.656019 3210.823 53.34991 7.832384 1.798535 50.10409
## 3  3 33.23209 28.04660 3.259116 3311.551 53.60599 7.516569 1.288461 48.74001
## 4  4 30.09710 34.81796 3.760030 3266.844 18.60876 5.906656 2.075259 50.92745
## 5  5 37.41737 42.68440 4.484686 3664.088 32.57351 7.181873 2.762643 55.16882
## 6  6 33.28761 24.94960 3.596075 3328.988 38.12531 9.723429 3.054057 51.14812
##         UAs       UCd      UCr
## 1 10.192695 0.7537104 42.60187
## 2 11.815088 0.9789506 41.30757
## 3 10.079057 0.1903262 36.47716
## 4  8.719123 0.9364825 42.47987
## 5  9.436559 1.4977829 47.78528
## 6 11.589403 1.6645837 38.26386</code></pre>
</div>
<div id="tidyverse" class="section level3">
<h3><span class="header-section-number">3.1.4</span> Tidyverse</h3>
<p><a href="https://www.tidyverse.org">Tidyverse</a> is a collection of packages that are used together along with a specific type of syntax, dataset and formatting protocols, etc.</p>
<p>Here, we will peform all the of the previous exercises using Tidyverse!</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="dataorg.html#cb23-1"></a><span class="kw">library</span>(tidyverse);</span>
<span id="cb23-2"><a href="dataorg.html#cb23-2"></a></span>
<span id="cb23-3"><a href="dataorg.html#cb23-3"></a><span class="co">#Merging</span></span>
<span id="cb23-4"><a href="dataorg.html#cb23-4"></a>full.tidy =<span class="st"> </span><span class="kw">inner_join</span>(demo.data, chem.data, <span class="dt">by=</span><span class="st">&quot;ID&quot;</span>); </span>
<span id="cb23-5"><a href="dataorg.html#cb23-5"></a></span>
<span id="cb23-6"><a href="dataorg.html#cb23-6"></a><span class="co">#To merge with different IDs use: by = c(&quot;ID.Demo&quot;=&quot;ID.Chem&quot;)</span></span>
<span id="cb23-7"><a href="dataorg.html#cb23-7"></a></span>
<span id="cb23-8"><a href="dataorg.html#cb23-8"></a><span class="co">#Susbetting columns</span></span>
<span id="cb23-9"><a href="dataorg.html#cb23-9"></a>subset.tidy1 =<span class="st"> </span>full.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="kw">all_of</span>(subset.columns));</span>
<span id="cb23-10"><a href="dataorg.html#cb23-10"></a>subset.columns1 =<span class="st"> </span><span class="kw">c</span>(subset.columns, <span class="st">&quot;NotAColName&quot;</span>);</span>
<span id="cb23-11"><a href="dataorg.html#cb23-11"></a>subset.tidy1a =<span class="st"> </span>full.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="kw">any_of</span>(subset.columns1));</span>
<span id="cb23-12"><a href="dataorg.html#cb23-12"></a></span>
<span id="cb23-13"><a href="dataorg.html#cb23-13"></a><span class="co">#Removing columns</span></span>
<span id="cb23-14"><a href="dataorg.html#cb23-14"></a>subset.tidy1b =<span class="st"> </span>full.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>subset.columns);</span>
<span id="cb23-15"><a href="dataorg.html#cb23-15"></a></span>
<span id="cb23-16"><a href="dataorg.html#cb23-16"></a><span class="co">#Susbetting Rows</span></span>
<span id="cb23-17"><a href="dataorg.html#cb23-17"></a>subset.tidy2a =<span class="st"> </span>full.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>);</span>
<span id="cb23-18"><a href="dataorg.html#cb23-18"></a>subset.tidy2b =<span class="st"> </span>full.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">slice</span>(<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>));</span>
<span id="cb23-19"><a href="dataorg.html#cb23-19"></a></span>
<span id="cb23-20"><a href="dataorg.html#cb23-20"></a><span class="co">#Filtering data</span></span>
<span id="cb23-21"><a href="dataorg.html#cb23-21"></a>subset.tidy3 =<span class="st"> </span>full.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(BMI <span class="op">&gt;</span><span class="st"> </span><span class="dv">25</span> <span class="op">&amp;</span><span class="st"> </span>MAge <span class="op">&gt;</span><span class="st"> </span><span class="dv">31</span>);</span>
<span id="cb23-22"><a href="dataorg.html#cb23-22"></a><span class="kw">head</span>(subset.tidy3);</span></code></pre></div>
<pre><code>##   ID      BMI     MAge     MEdu       BW       GA     DWAs      DWCd     DWCr
## 1  4 30.09710 34.81796 3.760030 3266.844 18.60876 5.906656 2.0752589 50.92745
## 2  5 37.41737 42.68440 4.484686 3664.088 32.57351 7.181873 2.7626433 55.16882
## 3  9 36.94794 33.58589 3.242318 3260.482 37.37241 9.074928 2.7277549 55.72826
## 4 13 33.65870 33.82961 2.924433 3481.293 33.68744 7.101634 0.8443918 47.11677
## 5 22 25.68414 37.08028 2.704409 3387.046 52.48203 7.207447 2.8088453 48.08648
## 6 31 28.39896 47.85761 6.453362 3173.033 20.10080 6.032807 2.1929549 45.71856
##         UAs        UCd      UCr
## 1  8.719123  0.9364825 42.47987
## 2  9.436559  1.4977829 47.78528
## 3 10.818153  1.6585757 42.58577
## 4  9.967185 -0.3466431 36.74220
## 5  9.446643  1.9891049 34.16921
## 6  9.917588  1.1194851 37.82297</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="dataorg.html#cb25-1"></a>subset.tidy4 =<span class="st"> </span>full.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(BMI <span class="op">&gt;</span><span class="st"> </span><span class="dv">25</span> <span class="op">&amp;</span><span class="st"> </span>MAge <span class="op">&gt;</span><span class="st"> </span><span class="dv">31</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(BMI, MAge, MEdu);</span>
<span id="cb25-2"><a href="dataorg.html#cb25-2"></a></span>
<span id="cb25-3"><a href="dataorg.html#cb25-3"></a><span class="co">#Melting and Casting (pivoting in Tidyverse)</span></span>
<span id="cb25-4"><a href="dataorg.html#cb25-4"></a>full.pivotlong =<span class="st"> </span>full.tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="op">-</span>ID, <span class="dt">names_to =</span> <span class="st">&quot;var&quot;</span>, <span class="dt">values_to =</span> <span class="st">&quot;value&quot;</span>);</span>
<span id="cb25-5"><a href="dataorg.html#cb25-5"></a><span class="kw">head</span>(full.pivotlong, <span class="dv">15</span>);</span></code></pre></div>
<pre><code>## # A tibble: 15 x 3
##       ID var      value
##    &lt;int&gt; &lt;chr&gt;    &lt;dbl&gt;
##  1     1 BMI     27.7  
##  2     1 MAge    23.0  
##  3     1 MEdu     3.14 
##  4     1 BW    3180.   
##  5     1 GA      31.8  
##  6     1 DWAs     6.43 
##  7     1 DWCd     1.29 
##  8     1 DWCr    51.7  
##  9     1 UAs     10.2  
## 10     1 UCd      0.754
## 11     1 UCr     42.6  
## 12     2 BMI     26.8  
## 13     2 MAge    30.1  
## 14     2 MEdu     2.66 
## 15     2 BW    3211.</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="dataorg.html#cb27-1"></a>full.pivotwide =<span class="st"> </span>full.pivotlong <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;var&quot;</span>, <span class="dt">values_from=</span><span class="st">&quot;value&quot;</span>);</span>
<span id="cb27-2"><a href="dataorg.html#cb27-2"></a><span class="kw">head</span>(full.pivotwide);</span></code></pre></div>
<pre><code>## # A tibble: 6 x 12
##      ID   BMI  MAge  MEdu    BW    GA  DWAs  DWCd  DWCr   UAs   UCd   UCr
##   &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1     1  27.7  23.0  3.14 3180.  31.8  6.43  1.29  51.7 10.2  0.754  42.6
## 2     2  26.8  30.1  2.66 3211.  53.3  7.83  1.80  50.1 11.8  0.979  41.3
## 3     3  33.2  28.0  3.26 3312.  53.6  7.52  1.29  48.7 10.1  0.190  36.5
## 4     4  30.1  34.8  3.76 3267.  18.6  5.91  2.08  50.9  8.72 0.936  42.5
## 5     5  37.4  42.7  4.48 3664.  32.6  7.18  2.76  55.2  9.44 1.50   47.8
## 6     6  33.3  24.9  3.60 3329.  38.1  9.72  3.05  51.1 11.6  1.66   38.3</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="finding-and-visualizing-data-trends.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["test1.pdf", "test1.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
