<!DOCTYPE html>
<html lang="" xml:lang="">
<head>
  
  <!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-M60LEM76PT"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-M60LEM76PT');
</script>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1.3 Downloading and Installing R Studio | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="1.3 Downloading and Installing R Studio | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="1.3 Downloading and Installing R Studio | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1.3 Downloading and Installing R Studio | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle Roell, Lauren Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="downloading-and-installing-r.html"/>
<link rel="next" href="concluding-remarks-1.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TAME Toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#background"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#objectives"><i class="fa fa-check"></i>Objectives</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#module-development-overview"><i class="fa fa-check"></i>Module Development Overview</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#concluding-remarks"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#funding"><i class="fa fa-check"></i>Funding</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#author-affiliations"><i class="fa fa-check"></i>Author Affiliations</a></li>
</ul></li>
<li class="part"><span><b>I Introductory Data Science</b></span></li>
<li class="chapter" data-level="1" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html"><i class="fa fa-check"></i><b>1</b> Introduction to Coding in R</a><ul>
<li class="chapter" data-level="1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html"><i class="fa fa-check"></i><b>1.1</b> General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="1.2" data-path="downloading-and-installing-r.html"><a href="downloading-and-installing-r.html"><i class="fa fa-check"></i><b>1.2</b> Downloading and Installing R</a></li>
<li class="chapter" data-level="1.3" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html"><i class="fa fa-check"></i><b>1.3</b> Downloading and Installing R Studio</a><ul>
<li class="chapter" data-level="1.3.1" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html#importing-files"><i class="fa fa-check"></i><b>1.3.1</b> Importing Files</a></li>
<li class="chapter" data-level="1.3.2" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html#viewing-data"><i class="fa fa-check"></i><b>1.3.2</b> Viewing Data</a></li>
<li class="chapter" data-level="1.3.3" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html#exporting-data"><i class="fa fa-check"></i><b>1.3.3</b> Exporting Data</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="concluding-remarks-1.html"><a href="concluding-remarks-1.html"><i class="fa fa-check"></i><b>1.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-organization-basics.html"><a href="data-organization-basics.html"><i class="fa fa-check"></i><b>2</b> Data Organization Basics</a><ul>
<li class="chapter" data-level="2.0.1" data-path="data-organization-basics.html"><a href="data-organization-basics.html#introduction-to-training-module"><i class="fa fa-check"></i><b>2.0.1</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="2.0.2" data-path="data-organization-basics.html"><a href="data-organization-basics.html#merging-data-using-base-r-syntax"><i class="fa fa-check"></i><b>2.0.2</b> Merging Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.0.3" data-path="data-organization-basics.html"><a href="data-organization-basics.html#filtering-and-subsetting-data-using-base-r-syntax"><i class="fa fa-check"></i><b>2.0.3</b> Filtering and Subsetting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.0.4" data-path="data-organization-basics.html"><a href="data-organization-basics.html#melting-and-casting-data-using-base-r-syntax"><i class="fa fa-check"></i><b>2.0.4</b> Melting and Casting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.0.5" data-path="data-organization-basics.html"><a href="data-organization-basics.html#introduction-to-tidyverse"><i class="fa fa-check"></i><b>2.0.5</b> Introduction to Tidyverse</a></li>
<li class="chapter" data-level="2.0.6" data-path="data-organization-basics.html"><a href="data-organization-basics.html#concluding-remarks-2"><i class="fa fa-check"></i><b>2.0.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html"><i class="fa fa-check"></i><b>3</b> Finding and Visualizing Data Trend</a><ul>
<li class="chapter" data-level="3.0.1" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html#introduction-to-training-module-1"><i class="fa fa-check"></i><b>3.0.1</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="3.0.2" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html#basic-statistical-tests-and-visualizations-of-data-trends"><i class="fa fa-check"></i><b>3.0.2</b> Basic Statistical Tests and Visualizations of Data Trends</a></li>
<li class="chapter" data-level="3.0.3" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i><b>3.0.3</b> Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="3.0.4" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i><b>3.0.4</b> Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="3.0.5" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html#concluding-remarks-3"><i class="fa fa-check"></i><b>3.0.5</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html"><i class="fa fa-check"></i><b>4</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.0.1" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#introduction-to-training-module-2"><i class="fa fa-check"></i><b>4.0.1</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="4.0.2" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#script-preparations-1"><i class="fa fa-check"></i><b>4.0.2</b> Script Preparations</a></li>
<li class="chapter" data-level="4.0.3" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#high-dimensional-data-visualizations-1"><i class="fa fa-check"></i><b>4.0.3</b> High-Dimensional Data Visualizations</a></li>
<li class="chapter" data-level="4.0.4" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html#concluding-remarks-4"><i class="fa fa-check"></i><b>4.0.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i><b>5</b> FAIR Data Management Practices</a><ul>
<li class="chapter" data-level="5.0.1" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#background-on-training-module"><i class="fa fa-check"></i><b>5.0.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="5.0.2" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#training-modules-questions"><i class="fa fa-check"></i><b>5.0.2</b> Training Module’s <strong>Questions</strong></a></li>
<li class="chapter" data-level="5.0.3" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i><b>5.0.3</b> Introduction to FAIR</a></li>
<li class="chapter" data-level="5.0.4" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i><b>5.0.4</b> Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="5.0.5" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#f-findable-in-fair"><i class="fa fa-check"></i><b>5.0.5</b> F (Findable) in FAIR</a></li>
<li class="chapter" data-level="5.0.6" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#a-accessible-in-fair"><i class="fa fa-check"></i><b>5.0.6</b> A (Accessible) in FAIR</a></li>
<li class="chapter" data-level="5.0.7" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#i-interoperable-in-fair"><i class="fa fa-check"></i><b>5.0.7</b> I (Interoperable) in FAIR</a></li>
<li class="chapter" data-level="5.0.8" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#r-reusable-in-fair"><i class="fa fa-check"></i><b>5.0.8</b> R (Reusable) in FAIR</a></li>
<li class="chapter" data-level="5.0.9" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-to-you"><i class="fa fa-check"></i><b>5.0.9</b> What does this mean to you?</a></li>
<li class="chapter" data-level="5.0.10" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i><b>5.0.10</b> Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="5.0.11" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#helpful-resources-on-fair"><i class="fa fa-check"></i><b>5.0.11</b> Helpful Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chemical-biological-analyses-and-predictive-modeling.html"><a href="chemical-biological-analyses-and-predictive-modeling.html"><i class="fa fa-check"></i><b>6</b> Chemical-Biological Analyses and Predictive Modeling</a><ul>
<li class="chapter" data-level="6.1" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html"><i class="fa fa-check"></i><b>6.1</b> Dose-Response Modeling</a><ul>
<li class="chapter" data-level="6.1.1" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#background-on-training-module-1"><i class="fa fa-check"></i><b>6.1.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="6.1.2" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#introduction-to-training-module-3"><i class="fa fa-check"></i><b>6.1.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="6.1.3" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#training-modules-environmental-health-questions"><i class="fa fa-check"></i><b>6.1.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="6.1.4" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#script-preparations-2"><i class="fa fa-check"></i><b>6.1.4</b> Script Preparations</a></li>
<li class="chapter" data-level="6.1.5" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#loading-the-example-dataset"><i class="fa fa-check"></i><b>6.1.5</b> Loading the Example Dataset</a></li>
<li class="chapter" data-level="6.1.6" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#data-viewing"><i class="fa fa-check"></i><b>6.1.6</b> Data Viewing</a></li>
<li class="chapter" data-level="6.1.7" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#plotting-data-in-dose-response"><i class="fa fa-check"></i><b>6.1.7</b> Plotting Data in Dose-Response</a></li>
<li class="chapter" data-level="6.1.8" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#fitting-curves-to-dose-response-plots"><i class="fa fa-check"></i><b>6.1.8</b> Fitting Curves to Dose-Response Plots</a></li>
<li class="chapter" data-level="6.1.9" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#comparing-curve-fits"><i class="fa fa-check"></i><b>6.1.9</b> Comparing Curve Fits</a></li>
<li class="chapter" data-level="6.1.10" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#curve-fitting-for-liver-outcomes"><i class="fa fa-check"></i><b>6.1.10</b> Curve Fitting for Liver Outcomes</a></li>
<li class="chapter" data-level="6.1.11" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#deriving-the-final-benchmark-dose-bmd-estimates"><i class="fa fa-check"></i><b>6.1.11</b> Deriving the Final Benchmark Dose (BMD) Estimates</a></li>
<li class="chapter" data-level="6.1.12" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html#concluding-remarks-5"><i class="fa fa-check"></i><b>6.1.12</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i><b>6.2</b> Machine Learning and Predictive Modeling for Environmental Health</a><ul>
<li class="chapter" data-level="6.2.1" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#background-on-training-module-2"><i class="fa fa-check"></i><b>6.2.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="6.2.2" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-4"><i class="fa fa-check"></i><b>6.2.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="6.2.3" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#training-modules-environmental-health-questions-1"><i class="fa fa-check"></i><b>6.2.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="6.2.4" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#script-preparations-3"><i class="fa fa-check"></i><b>6.2.4</b> Script Preparations</a></li>
<li class="chapter" data-level="6.2.5" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#loading-the-example-dataset-1"><i class="fa fa-check"></i><b>6.2.5</b> Loading the Example Dataset</a></li>
<li class="chapter" data-level="6.2.6" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#data-viewing-1"><i class="fa fa-check"></i><b>6.2.6</b> Data Viewing</a></li>
<li class="chapter" data-level="6.2.7" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#evaluating-the-original-physicochemical-property-data-across-substances"><i class="fa fa-check"></i><b>6.2.7</b> Evaluating the Original Physicochemical Property Data across Substances</a></li>
<li class="chapter" data-level="6.2.8" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i><b>6.2.8</b> Identifying Clusters of Chemicals through k-means</a></li>
<li class="chapter" data-level="6.2.9" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>6.2.9</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="6.2.10" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-6"><i class="fa fa-check"></i><b>6.2.10</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html"><i class="fa fa-check"></i><b>6.3</b> Mixtures Analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#background-on-training-module-3"><i class="fa fa-check"></i><b>6.3.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="6.3.2" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#introduction-to-training-module-5"><i class="fa fa-check"></i><b>6.3.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="6.3.3" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#training-modules-environmental-health-questions-2"><i class="fa fa-check"></i><b>6.3.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="6.3.4" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#script-preparations-4"><i class="fa fa-check"></i><b>6.3.4</b> Script Preparations</a></li>
<li class="chapter" data-level="6.3.5" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#loading-the-example-ginkgo-biloba-datasets"><i class="fa fa-check"></i><b>6.3.5</b> Loading the Example <em>Ginkgo biloba</em> Datasets</a></li>
<li class="chapter" data-level="6.3.6" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#data-viewing-2"><i class="fa fa-check"></i><b>6.3.6</b> Data Viewing</a></li>
<li class="chapter" data-level="6.3.7" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i><b>6.3.7</b> Chemistry-based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="6.3.8" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#creating-a-heat-map-of-the-mixtures-chemistry-data"><i class="fa fa-check"></i><b>6.3.8</b> Creating a Heat Map of the Mixtures Chemistry Data</a></li>
<li class="chapter" data-level="6.3.9" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i><b>6.3.9</b> Toxicity-based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="6.3.10" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#creating-a-heat-map-of-the-mixtures-toxicity-data"><i class="fa fa-check"></i><b>6.3.10</b> Creating a Heat Map of the Mixtures Toxicity Data</a></li>
<li class="chapter" data-level="6.3.11" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#comparing-results-from-the-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i><b>6.3.11</b> Comparing Results from the Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="6.3.12" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#these-plots-can-help-us-answer-environmental-health-question-7"><i class="fa fa-check"></i><b>6.3.12</b> These plots can help us answer <strong>Environmental Health Question 7</strong>:</a></li>
<li class="chapter" data-level="6.3.13" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html#concluding-remarks-7"><i class="fa fa-check"></i><b>6.3.13</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html"><i class="fa fa-check"></i><b>6.4</b> -Omics Analyses and Systems Biology</a></li>
<li class="chapter" data-level="6.5" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i><b>6.5</b> Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="6.6" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html"><i class="fa fa-check"></i><b>6.6</b> Read-Across Toxicity Predictions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="environmental-health-database-mining.html"><a href="environmental-health-database-mining.html"><i class="fa fa-check"></i><b>7</b> Environmental Health Database Mining</a><ul>
<li class="chapter" data-level="7.1" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html"><i class="fa fa-check"></i><b>7.1</b> Environmental Health Databases: Comparative Toxicogenomics Database</a><ul>
<li class="chapter" data-level="7.1.1" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#background-on-training-module-4"><i class="fa fa-check"></i><b>7.1.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="7.1.2" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#training-modules-environmental-health-questions-3"><i class="fa fa-check"></i><b>7.1.2</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="7.1.3" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#script-preparations-5"><i class="fa fa-check"></i><b>7.1.3</b> Script Preparations</a></li>
<li class="chapter" data-level="7.1.4" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#organizing-example-dataset-from-ctd"><i class="fa fa-check"></i><b>7.1.4</b> Organizing Example Dataset from CTD</a></li>
<li class="chapter" data-level="7.1.5" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#loading-the-example-ctd-dataset-into-r"><i class="fa fa-check"></i><b>7.1.5</b> Loading the Example CTD Dataset into R</a></li>
<li class="chapter" data-level="7.1.6" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#data-viewing-3"><i class="fa fa-check"></i><b>7.1.6</b> Data Viewing</a></li>
<li class="chapter" data-level="7.1.7" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#filtering-the-data-for-genes-with-altered-expression"><i class="fa fa-check"></i><b>7.1.7</b> Filtering the Data for Genes with Altered Expression</a></li>
<li class="chapter" data-level="7.1.8" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i><b>7.1.8</b> Identifying Genes under Epigenetic Control</a></li>
<li class="chapter" data-level="7.1.9" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#method-1-for-list-comparisons-merging"><i class="fa fa-check"></i><b>7.1.9</b> Method 1 for list comparisons: Merging</a></li>
<li class="chapter" data-level="7.1.10" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#method-2-for-list-comparisons-intersection"><i class="fa fa-check"></i><b>7.1.10</b> Method 2 for list comparisons: Intersection</a></li>
<li class="chapter" data-level="7.1.11" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html#concluding-remarks-8"><i class="fa fa-check"></i><b>7.1.11</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html"><i class="fa fa-check"></i><b>7.2</b> Environmental Health Databases: Gene Expression Omnibus</a><ul>
<li class="chapter" data-level="7.2.1" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#background-on-training-module-5"><i class="fa fa-check"></i><b>7.2.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="7.2.2" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#training-modules-environmental-health-questions-4"><i class="fa fa-check"></i><b>7.2.2</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="7.2.3" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#script-preparations-6"><i class="fa fa-check"></i><b>7.2.3</b> Script Preparations</a></li>
<li class="chapter" data-level="7.2.4" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#loading-and-organizing-the-example-dataset"><i class="fa fa-check"></i><b>7.2.4</b> Loading and Organizing the Example Dataset</a></li>
<li class="chapter" data-level="7.2.5" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#adding-gene-symbol-information-to--omic-data-sets-through-platform-annotation-files"><i class="fa fa-check"></i><b>7.2.5</b> Adding Gene Symbol Information to -Omic Data Sets through Platform Annotation Files</a></li>
<li class="chapter" data-level="7.2.6" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#visualizing-gene-expression-data-using-boxplots-and-heat-maps"><i class="fa fa-check"></i><b>7.2.6</b> Visualizing Gene Expression Data using Boxplots and Heat Maps</a></li>
<li class="chapter" data-level="7.2.7" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#boxplot-visualizations-1"><i class="fa fa-check"></i><b>7.2.7</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="7.2.8" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#heatmap-visualizations-1"><i class="fa fa-check"></i><b>7.2.8</b> Heatmap Visualizations</a></li>
<li class="chapter" data-level="7.2.9" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#statistical-analyses-to-identify-genes-altered-by-formaldehyde"><i class="fa fa-check"></i><b>7.2.9</b> Statistical Analyses to Identify Genes altered by Formaldehyde</a></li>
<li class="chapter" data-level="7.2.10" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#interpreting-results"><i class="fa fa-check"></i><b>7.2.10</b> Interpreting Results</a></li>
<li class="chapter" data-level="7.2.11" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html#concluding-remarks-9"><i class="fa fa-check"></i><b>7.2.11</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="database-integration-air-quality-study-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-study-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i><b>7.3</b> Database Integration: Air Quality Study, Mortality, and Environmental Justice Data</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="additional-resources.html"><a href="additional-resources.html"><i class="fa fa-check"></i><b>8</b> Additional Resources</a><ul>
<li class="chapter" data-level="8.1" data-path="r-programming-resources.html"><a href="r-programming-resources.html"><i class="fa fa-check"></i><b>8.1</b> R Programming Resources:</a></li>
<li class="chapter" data-level="8.2" data-path="r-packages-resources.html"><a href="r-packages-resources.html"><i class="fa fa-check"></i><b>8.2</b> R Packages Resources:</a></li>
<li class="chapter" data-level="8.3" data-path="community-discussions-on-r-and-r-packages.html"><a href="community-discussions-on-r-and-r-packages.html"><i class="fa fa-check"></i><b>8.3</b> Community Discussions on R and R Packages:</a></li>
<li class="chapter" data-level="8.4" data-path="data-science-and-statistical-analysis-resources.html"><a href="data-science-and-statistical-analysis-resources.html"><i class="fa fa-check"></i><b>8.4</b> Data Science and Statistical Analysis Resources:</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="downloading-and-installing-r-studio" class="section level2">
<h2><span class="header-section-number">1.3</span> Downloading and Installing R Studio</h2>
  <h4>Important update: we released an updated version of TAME, called TAME 2.0, now publicly available at: <a href="https://uncsrp.github.io/TAME2/">Tame 2.0</a></h4>

<p><strong>What is R Studio?</strong>
RStudio is an Integrated Development Environment (IDE) for R, which makes it more ‘user friendly’ when developing and using R script.</p>
<p>RStudio Desktop is a desktop application that can be downloaded for free, online. To download RStudio:</p>
<ul>
<li>Navigate to: <a href="https://rstudio.com/products/rstudio/download/" class="uri">https://rstudio.com/products/rstudio/download/</a></li>
<li>Select the free RStudio Desktop option, and click “DOWNLOAD”</li>
<li>Then download the top (most recent) RStudio Desktop option for your operating system</li>
</ul>
<p>Here is a screenshot of what R script looks like within RStudio:</p>
<p><img src="_book/TAME_Toolkit_files/figure-html/Module1_1_RStudio_Image.png" width="684" /></p>
<p><br>
## Introduction to R Packages
One of the major benefits to coding in the R language is access to the continually expanding resource of thousands of user-developed packages that aid in improved data analyses and methods sharing. Packages have varying utilities, spanning basic organization and manipulation of data, visualizing data, and more advanced approaches to parse and analyze data, with examples included in all of the proceeding training modules.</p>
<p>In brief, packages represent compilations of code fitted for a specialized focus or purpose. These are
often written by R users and submitted to the <a href="https://cran.r-project.org/web/packages/">CRAN</a>, or another host such as <a href="https://www.bioconductor.org/">BioConductor</a> or <a href="https://github.com/">Github</a>.</p>
<p>Examples of some common packages that we’ll be using throughout these training modules include the following:
+ tidyverse: A collection of open source R packages that share an underlying design philosophy, grammar, and data structures of tidy data. For more information on the tidyverse package, see its associated <a href="https://cran.r-project.org/web/packages/tidyverse/index.html">CRAN webpage</a>, primary <a href="https://www.tidyverse.org/packages/">webpage</a>, and peer-reviewed <a href="https://onlinelibrary.wiley.com/doi/10.1002/sdr.1600">article released in 2018</a>.
+ ggplot2: A system for declaratively creating graphics. Users provide the data, tell ggplot2 how to map variables to aesthetics, what graphical primitives to use, and ggplot2 drafts corresponding visualizations. For more information on the ggplot2 package, see its associated <a href="https://cran.r-project.org/web/packages/ggplot2/index.html">CRAN webpage</a> and <a href="https://www.rdocumentation.org/packages/ggplot2/versions/3.3.5">R Documentation</a>.</p>
<p>More information on these packages, as well as many others, is included throughout the training modules.</p>
<p><br>
### Downloading/Installing R Packages
R packages often do not need to be downloaded from a website. Instead, you can just load packages through running script in R, like:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="downloading-and-installing-r-studio.html#cb1-1"></a><span class="kw">install.packages</span>(“tidyverse”)</span>
<span id="cb1-2"><a href="downloading-and-installing-r-studio.html#cb1-2"></a><span class="kw">library</span>(tidyverse)</span></code></pre></div>
<p>It is worth noting that a function can be queried in RStudio by typing a question mark before the name of the function. For example:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="downloading-and-installing-r-studio.html#cb2-1"></a>?install.packages</span></code></pre></div>
<p>This will bring up documentation in the viewer window.</p>
<p><br><br>
## Scripting Basics
<br>
### Comments
R allows for scripts to contain non-code elements, called comments, that will not be run or interpreted. Comments are commonly included when annotating code, or describing what your code does, where your data came from, and just general textual reminders throughout the script.</p>
<p>To make a comment, simply use a # followed by the comment.
A # only comments out a single line of code. In other words, only that line will be commented and therefore not be run, but lines directly above/below it still will be</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="downloading-and-installing-r-studio.html#cb3-1"></a><span class="co"># This is an R comment!</span></span></code></pre></div>
<p>Comments are useful to help make code more interpretable for others or to add reminders of what and why parts of code may have been written.</p>
<p><br>
### Starting Code
RStudio will autofill function names, variable names, etc. by pressing tab while typing.If multiple matches are found, RStudio will provide you with a drop down list to select from, which may be useful when searching through newly installed packages or trying to quickly type variable names in an R script.</p>
<p>One of the first lines of code in any script will likely include the loading of packages needed to run the script. Here is an example line of code to load a package:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="downloading-and-installing-r-studio.html#cb4-1"></a><span class="co"># Loading ggplot2 package (should already be installed as a base package)</span></span>
<span id="cb4-2"><a href="downloading-and-installing-r-studio.html#cb4-2"></a><span class="kw">library</span>(ggplot2)</span></code></pre></div>
<p>Many packages also exist as part of the baseline configuration of an R working environment, and do not require manual loading each time you launch R. These include the following packages:
+ datasets
+ graphics
+ methods
+ stats
+ utils</p>
<p><br>
### Setting Your Working Directory
Another step that is commonly done at the very beginning of your code is setting your working direction. This points you to where you have files that you want to upload / where the default is to deposit output files produced during your scripted activities.</p>
<p>You must set the working directory to a local directory where data are located or output files will be saved.</p>
<p>To view where your current working directory is (by default), run the following:</p>
<div id="show-your-working-directory" class="section level4">
<h4><span class="header-section-number">1.3.0.1</span> Show your working directory</h4>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="downloading-and-installing-r-studio.html#cb5-1"></a><span class="kw">getwd</span>()</span></code></pre></div>
<p>To set the location of your working directory, run the following:</p>
</div>
<div id="set-your-working-directory" class="section level4">
<h4><span class="header-section-number">1.3.0.2</span> Set your working directory</h4>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="downloading-and-installing-r-studio.html#cb6-1"></a><span class="kw">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
<p>Note that in macOS, filepaths use “/” as folder separaters; whereas in PCs, filepaths use "".
<br></p>
</div>
<div id="importing-files" class="section level3">
<h3><span class="header-section-number">1.3.1</span> Importing Files</h3>
<p>After setting the working directory, importing and exporting files can be done using various functions based on the type of file being read or written. Often, it is easiest to import data into R that are in a comma separated values / comma delimited file (.csv) or tab / text delimited file (.txt).</p>
<p>Other datatypes such as SAS data files, large .csv files, etc. may require different functions to be more efficiently read in and some of these file formats will be discussed in future modules.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="downloading-and-installing-r-studio.html#cb7-1"></a><span class="co"># Read in the .csv data that&#39;s located in our working directory</span></span>
<span id="cb7-2"><a href="downloading-and-installing-r-studio.html#cb7-2"></a>csv.dataset &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;Module1_1/Module1_1_ExampleData.csv&quot;</span>)</span>
<span id="cb7-3"><a href="downloading-and-installing-r-studio.html#cb7-3"></a></span>
<span id="cb7-4"><a href="downloading-and-installing-r-studio.html#cb7-4"></a><span class="co"># Read in the .txt data</span></span>
<span id="cb7-5"><a href="downloading-and-installing-r-studio.html#cb7-5"></a>txt.dataset &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;Module1_1/Module1_1_ExampleData.txt&quot;</span>)</span></code></pre></div>
<p>These datasets now appear as saved dataframes (“csv.dataset” and “txt.dataset”) in our working environment in R.
<br></p>
</div>
<div id="viewing-data" class="section level3">
<h3><span class="header-section-number">1.3.2</span> Viewing Data</h3>
<p>After data have been loaded into R, or created within R, you will likely want to view what these datasets look like.
Datasets can be viewed in their entirety, or datasets can be subsetted to quickly look at part of the data.</p>
<p>Here’s some example script to view just the beginnings of a dataframe, using the “head” function in R (a part of the baseline packages)</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="downloading-and-installing-r-studio.html#cb8-1"></a><span class="kw">head</span>(csv.dataset)</span></code></pre></div>
<pre><code>##    Sample Var1 Var2 Var3
## 1 sample1    1    2    1
## 2 sample2    2    4    4
## 3 sample3    3    6    9
## 4 sample4    4    8   16
## 5 sample5    5   10   25</code></pre>
<p>Here, you can see that this automatically brings up a view of the first five rows of the dataframe.</p>
<p>Another way to view the first five rows of a dataframe is to run the following:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="downloading-and-installing-r-studio.html#cb10-1"></a>csv.dataset[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,]</span></code></pre></div>
<pre><code>##    Sample Var1 Var2 Var3
## 1 sample1    1    2    1
## 2 sample2    2    4    4
## 3 sample3    3    6    9
## 4 sample4    4    8   16
## 5 sample5    5   10   25</code></pre>
<p>Expanding on this, to view the first 5 rows x 2 columns, run the following:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="downloading-and-installing-r-studio.html#cb12-1"></a>csv.dataset[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</span></code></pre></div>
<pre><code>##    Sample Var1
## 1 sample1    1
## 2 sample2    2
## 3 sample3    3
## 4 sample4    4
## 5 sample5    5</code></pre>
<p>To view the entire dataset in RStudio, use the “View” function:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="downloading-and-installing-r-studio.html#cb14-1"></a><span class="kw">View</span>(csv.dataset)</span></code></pre></div>
<p><br></p>
</div>
<div id="exporting-data" class="section level3">
<h3><span class="header-section-number">1.3.3</span> Exporting Data</h3>
<p>Now that we have these datasets saved as dataframes, we can use these as examples to export data files from the R environment back into our local directory.</p>
<p>There are many ways to export data in R. Data can be written out into a .csv file, tab delimited .txt file, RData file, etc. There are also many functions within packages that write out specific datasets generated by that package.
To write out to a .csv file:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="downloading-and-installing-r-studio.html#cb15-1"></a><span class="kw">write.csv</span>(csv.dataset, <span class="st">&quot;Module1_1_SameCSVFileNowOut.csv&quot;</span>)</span></code></pre></div>
<p>To write out a .txt tab delimited file:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="downloading-and-installing-r-studio.html#cb16-1"></a><span class="kw">write.table</span>(txt.dataset, <span class="st">&quot;Module1_1_SameTXTFileNowOut.txt&quot;</span>)</span></code></pre></div>
<p>R also allows objects to be saved in RData files. These files can be read into R, as well, and will load the object into the current workspace. Entire workspaces are also able to be saved.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="downloading-and-installing-r-studio.html#cb17-1"></a><span class="co"># Read in saved single R data object (note that this file is not provided, just example code is for future reference)</span></span>
<span id="cb17-2"><a href="downloading-and-installing-r-studio.html#cb17-2"></a>r.obj =<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;data.rds&quot;</span>)</span>
<span id="cb17-3"><a href="downloading-and-installing-r-studio.html#cb17-3"></a></span>
<span id="cb17-4"><a href="downloading-and-installing-r-studio.html#cb17-4"></a><span class="co"># Write single R object to file (note that this file is not provided, just example code is for future reference)</span></span>
<span id="cb17-5"><a href="downloading-and-installing-r-studio.html#cb17-5"></a><span class="kw">saveRDS</span>(object, <span class="st">&quot;single_object.rds&quot;</span>)</span>
<span id="cb17-6"><a href="downloading-and-installing-r-studio.html#cb17-6"></a></span>
<span id="cb17-7"><a href="downloading-and-installing-r-studio.html#cb17-7"></a><span class="co"># Read in multiple saved R objects (note that this file is not provided, just example code is for future reference)</span></span>
<span id="cb17-8"><a href="downloading-and-installing-r-studio.html#cb17-8"></a><span class="kw">load</span>(<span class="st">&quot;multiple_data.RData&quot;</span>)</span>
<span id="cb17-9"><a href="downloading-and-installing-r-studio.html#cb17-9"></a></span>
<span id="cb17-10"><a href="downloading-and-installing-r-studio.html#cb17-10"></a><span class="co"># Save multiple R objects (note that this file is not provided, just example code is for future reference)</span></span>
<span id="cb17-11"><a href="downloading-and-installing-r-studio.html#cb17-11"></a><span class="kw">save</span>(object1, object2, <span class="st">&quot;multiple_objects.RData&quot;</span>)</span>
<span id="cb17-12"><a href="downloading-and-installing-r-studio.html#cb17-12"></a></span>
<span id="cb17-13"><a href="downloading-and-installing-r-studio.html#cb17-13"></a><span class="co"># Save entire workspace</span></span>
<span id="cb17-14"><a href="downloading-and-installing-r-studio.html#cb17-14"></a><span class="kw">save.image</span>(<span class="st">&quot;entire_workspace.RData&quot;</span>)</span>
<span id="cb17-15"><a href="downloading-and-installing-r-studio.html#cb17-15"></a></span>
<span id="cb17-16"><a href="downloading-and-installing-r-studio.html#cb17-16"></a><span class="co"># Load entire workspace</span></span>
<span id="cb17-17"><a href="downloading-and-installing-r-studio.html#cb17-17"></a><span class="kw">load</span>(<span class="st">&quot;entire_workspace.RData&quot;</span>)</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="downloading-and-installing-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="concluding-remarks-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
