<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1.5 FAIR Data Management Practices | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="1.5 FAIR Data Management Practices | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="1.5 FAIR Data Management Practices | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1.5 FAIR Data Management Practices | The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle R. Roell, Lauren E. Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/Favicon_v1.png" type="image/x-icon" />
<link rel="prev" href="high-dimensional-data-visualizations.html"/>
<link rel="next" href="dose-response-modeling.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/Data Training Icon for Top Left Bookdown_v2.png"><br></a></li>

<li class="divider"></li>
<li><a href="index.html#preface">Preface<span></span></a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science<span></span></b></span></li>
<li><a href="introduction-to-coding-in-r.html#introduction-to-coding-in-r">1.1 Introduction to Coding in R<span></span></a>
<ul>
<li><a href="introduction-to-coding-in-r.html#introduction-to-training-module">Introduction to Training Module<span></span></a></li>
<li><a href="introduction-to-coding-in-r.html#r-packages">R Packages<span></span></a></li>
<li><a href="introduction-to-coding-in-r.html#scripting-basics">Scripting Basics<span></span></a></li>
<li><a href="introduction-to-coding-in-r.html#concluding-remarks-1">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="data-organization-basics.html#data-organization-basics">1.2 Data Organization Basics<span></span></a>
<ul>
<li><a href="data-organization-basics.html#introduction-to-training-module-1">Introduction to Training Module<span></span></a></li>
<li><a href="data-organization-basics.html#data-manipulation-using-base-r">Data Manipulation using Base R<span></span></a></li>
<li><a href="data-organization-basics.html#introduction-to-tidyverse">Introduction to Tidyverse<span></span></a></li>
<li><a href="data-organization-basics.html#concluding-remarks-2">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="finding-and-visualizing-data-trends.html#finding-and-visualizing-data-trends">1.3 Finding and Visualizing Data Trends<span></span></a>
<ul>
<li><a href="finding-and-visualizing-data-trends.html#introduction-to-training-module-2">Introduction to Training Module<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#basic-data-analysis">Basic Data Analysis<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#regression-modeling">Regression Modeling<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#categorical-data-analysis">Categorical Data Analysis<span></span></a></li>
<li><a href="finding-and-visualizing-data-trends.html#concluding-remarks-3">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="high-dimensional-data-visualizations.html#high-dimensional-data-visualizations">1.4 High-Dimensional Data Visualizations<span></span></a>
<ul>
<li><a href="high-dimensional-data-visualizations.html#introduction-to-training-module-3">Introduction to Training Module<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#high-dimensional-data-visualizations-1">High-Dimensional Data Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#density-plot-visualizations">Density Plot Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#ggally-visualizations">GGally Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#boxplot-visualizations">Boxplot Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#correlation-plot-visualizations">Correlation Plot Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#hierarchical-clustering-visualizations">Hierarchical Clustering Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#heat-map-visualizations">Heat Map Visualizations<span></span></a></li>
<li><a href="high-dimensional-data-visualizations.html#concluding-remarks-4">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="fair-data-management-practices.html#fair-data-management-practices">1.5 FAIR Data Management Practices<span></span></a>
<ul>
<li><a href="fair-data-management-practices.html#introduction-to-training-module-4">Introduction to Training Module<span></span></a></li>
<li><a href="fair-data-management-practices.html#introduction-to-fair">Introduction to FAIR<span></span></a></li>
<li><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter">Breaking Down FAIR, Letter-by-Letter<span></span></a></li>
<li><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data">Data Repositories for Sharing of Data<span></span></a></li>
<li><a href="fair-data-management-practices.html#helpful-resources-on-fair">Helpful Resources on FAIR<span></span></a></li>
<li><a href="fair-data-management-practices.html#concluding-remarks-5">Concluding Remarks<span></span></a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Chemical-<br>Biological Analyses <br>and Predictive Modeling<span></span></b></span></li>
<li><a href="dose-response-modeling.html#dose-response-modeling">2.1 Dose-Response Modeling<span></span></a>
<ul>
<li><a href="dose-response-modeling.html#introduction-to-training-module-5">Introduction to Training Module<span></span></a></li>
<li><a href="dose-response-modeling.html#plotting-data-in-dose-response">Plotting Data in Dose-Response<span></span></a></li>
<li><a href="dose-response-modeling.html#dose-response-curve-fitting">Dose-Response Curve Fitting<span></span></a></li>
<li><a href="dose-response-modeling.html#comparing-curve-fits">Comparing Curve Fits<span></span></a></li>
<li><a href="dose-response-modeling.html#curve-fitting-example">Curve Fitting Example<span></span></a></li>
<li><a href="dose-response-modeling.html#deriving-benchmark-dose">Deriving Benchmark Dose<span></span></a></li>
<li><a href="dose-response-modeling.html#concluding-remarks-6">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="machine-learning-and-predictive-modeling.html#machine-learning-and-predictive-modeling">2.2 Machine Learning and Predictive Modeling<span></span></a>
<ul>
<li><a href="machine-learning-and-predictive-modeling.html#introduction-to-training-module-6">Introduction to Training Module<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#k-means-analysis">K-means Analysis<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#principal-component-analysis">Principal Component Analysis<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#combining-k-means-with-pca">Combining K-Means with PCA<span></span></a></li>
<li><a href="machine-learning-and-predictive-modeling.html#concluding-remarks-7">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="mixtures-analyses.html#mixtures-analyses">2.3 Mixtures Analyses<span></span></a>
<ul>
<li><a href="mixtures-analyses.html#introduction-to-training-module-7">Introduction to Training Module<span></span></a></li>
<li><a href="mixtures-analyses.html#chemistry-based-approach">Chemistry-based Approach<span></span></a></li>
<li><a href="mixtures-analyses.html#toxicity-based-approach">Toxicity-based Approach<span></span></a></li>
<li><a href="mixtures-analyses.html#comparing-results">Comparing Results<span></span></a></li>
<li><a href="mixtures-analyses.html#concluding-remarks-8">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="omics-analyses-and-systems-biology.html#omics-analyses-and-systems-biology">2.4 -Omics Analyses and Systems Biology<span></span></a>
<ul>
<li><a href="omics-analyses-and-systems-biology.html#introduction-to-training-module-8">Introduction to Training Module<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#transcriptomics-data-qaqc">Transcriptomics Data QA/QC<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#statistical-analysis-of-gene-expression">Statistical Analysis of Gene Expression<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#ma-plots">MA Plots<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#volcano-plots">Volcano Plots<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#pathway-enrichment-analysis">Pathway Enrichment Analysis<span></span></a></li>
<li><a href="omics-analyses-and-systems-biology.html#concluding-remarks-9">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="toxicokinetic-modeling.html#toxicokinetic-modeling">2.5 Toxicokinetic Modeling<span></span></a>
<ul>
<li><a href="toxicokinetic-modeling.html#introduction-to-training-module-9">Introduction to Training Module<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk">Data and Models used in Toxicokinetic Modeling (TK)<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#calculating-steady-state-concentration">Calculating steady-state concentration<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#reverse-toxicokinetics">Reverse Toxicokinetics<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#monte-carlo-approach">Monte Carlo Approach<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#chemical-specific-example">Chemical-Specific Example<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#calculating-bioactivity-exposure-ratios-bers">Calculating Bioactivity-Exposure Ratios (BERs)<span></span></a></li>
<li><a href="toxicokinetic-modeling.html#concluding-remarks-10">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="read-across-toxicity-predictions.html#read-across-toxicity-predictions">2.6 Read-Across Toxicity Predictions<span></span></a>
<ul>
<li><a href="read-across-toxicity-predictions.html#introduction-to-training-module-10">Introduction to Training Module<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#read-across-example-analysis">Read-Across Example Analysis<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#calculating-chemical-similarities">Calculating Chemical Similarities<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#chemical-read-across-to-predict-acute-toxicity">Chemical Read-Across to Predict Acute Toxicity<span></span></a></li>
<li><a href="read-across-toxicity-predictions.html#concluding-remarks-11">Concluding Remarks<span></span></a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Environmental <br>Health Database Mining<span></span></b></span></li>
<li><a href="comparative-toxicogenomics-database.html#comparative-toxicogenomics-database">3.1 Comparative Toxicogenomics Database<span></span></a>
<ul>
<li><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-11">Introduction to Training Module<span></span></a></li>
<li><a href="comparative-toxicogenomics-database.html#ctd-data-in-r">CTD Data in R<span></span></a></li>
<li><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control">Identifying Genes under Epigenetic Control<span></span></a></li>
<li><a href="comparative-toxicogenomics-database.html#concluding-remarks-12">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="gene-expression-omnibus.html#gene-expression-omnibus">3.2 Gene Expression Omnibus<span></span></a>
<ul>
<li><a href="gene-expression-omnibus.html#introduction-to-training-module-12">Introduction to Training Module<span></span></a>
<ul>
<li><a href="gene-expression-omnibus.html#training-modules-environmental-health-questions-7">Training Module’s <strong>Environmental Health Questions</strong><span></span></a></li>
</ul></li>
<li><a href="gene-expression-omnibus.html#geo-data-in-r">GEO Data in R<span></span></a></li>
<li><a href="gene-expression-omnibus.html#visualizing-data">Visualizing Data<span></span></a></li>
<li><a href="gene-expression-omnibus.html#statistical-analyses">Statistical Analyses<span></span></a></li>
<li><a href="gene-expression-omnibus.html#concluding-remarks-13">Concluding Remarks<span></span></a></li>
</ul></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#database-integration-air-quality-mortality-and-environmental-justice-data">3.3 Database Integration: Air Quality, Mortality, and Environmental Justice Data<span></span></a>
<ul>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-13">Introduction to Training Module<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#population-weighted-vs-unweighted-exposures">Population-weighted vs Unweighted Exposures<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#regression-modeling-1">Regression Modeling<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations">Environmental Justice Considerations<span></span></a></li>
<li><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-14">Concluding Remarks<span></span></a></li>
</ul></li>
<li class="part"><span><b>Additional Resources<span></span></b></span></li>
<li><a href="resources.html#resources">Resources<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The inTelligence And Machine lEarning (TAME) Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="fair-data-management-practices" class="section level1 hasAnchor">
<h1>1.5 FAIR Data Management Practices<a href="fair-data-management-practices.html#fair-data-management-practices" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Ms. Rebecca Boyles, MSPH, with contributions from Dr. Julia E. Rager</p>
<p>Fall 2021</p>
<!-- ## Background on Training Module -->
<div id="introduction-to-training-module-4" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="fair-data-management-practices.html#introduction-to-training-module-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This training module provides a description of FAIR data management practices, and points participants to important resources to help ensure generated data meet current FAIR guidelines. This training module is descriptive-based (as opposed to coding-based), in order to present information clearly and serve as an important resource alongside the other scripted training activities.</p>
<div id="training-modules-questions" class="section level4 hasAnchor">
<h4>Training Module’s <strong>Questions</strong><a href="fair-data-management-practices.html#training-modules-questions" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This training module was specifically developed to answer the following questions:</p>
<ol style="list-style-type: decimal">
<li>What is FAIR?</li>
<li>When was FAIR first developed?</li>
<li>When making data ‘Findable’, who and what should be able to find your data?</li>
<li>When saving/formatting your data, which of the following formats is preferred to meet FAIR principles: .pdf, .csv, or a proprietary output file from your lab instrument?</li>
<li>How can I find a suitable data repository for my data?</li>
</ol>
</div>
</div>
<div id="introduction-to-fair" class="section level2 hasAnchor">
<h2>Introduction to FAIR<a href="fair-data-management-practices.html#introduction-to-fair" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Proper data management is of utmost importance while leading data analyses within the field of environmental health science. A method to ensure proper data management is the implementation of Findability, Accessibility, Interoperability, and Reusability (FAIR) practices. A landmark paper that describes FAIR practices in environmental health research is the following:</p>
<ul>
<li>Wilkinson MD, Dumontier M, Aalbersberg IJ, et al. The FAIR Guiding Principles for scientific data management and stewardship. Sci Data. 2016 Mar 15. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/26978244/">26978244</a>.</li>
</ul>
<p>The FAIR principles describe a framework for data management and stewardship aimed at increasing the value of data by enabling sharing and reuse. These principles were originally developed from discussions during the <a href="https://www.lorentzcenter.nl/jointly-designing-a-data-fairport.html">Jointly Designing a Data FAIRport</a> meeting at the Lorentz Center in Leiden, The Netherlands in 2014, which brought together stakeholders to discuss the creation of an environment for virtual computational science. The resulting principles are technology agnostic, discipline independent, community driven, and internationally adopted.</p>
<p>Below is a schematic providing an overview of this guiding principle:</p>
<p><img src="_book/TAME_Toolkit_files/figure-html/Module1_5_FAIR_Fig.png" width="524" style="display: block; margin: auto;" /></p>
<div class="question">
<p><i>With this background, we can now answer <strong>Question 1</strong>:</i></p>
<p>What is FAIR?</p>
</div>
<!-- #### With this background, we can now answer **Question 1**: -->
<!-- ##### (1) What is FAIR? -->
<!-- ##### *Answer: A guiding framework that was recently established to promote best data management practices, to ensure that data are Findable, Accessibility, Interoperable, and Reusable.* -->
<div class="answer">
<p><strong>Answer</strong>: A guiding framework that was recently established to promote best data management practices, to ensure that data are Findable, Accessibility, Interoperable, and Reusable.</p>
</div>
<p><br></p>
<!-- #### We can also answer **Question 2**: -->
<!-- ##### (2) When was FAIR first developed? -->
<!-- ##### *Answer: 2014- which means that these principles are very new and continuing to evolve!* -->
<div class="question">
<p><i>We can also answer <strong>Question 2</strong>:</i></p>
<p>When was FAIR first developed?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: 2014- which means that these principles are very new and continuing to evolve!</p>
</div>
<p><br></p>
</div>
<div id="breaking-down-fair-letter-by-letter" class="section level2 hasAnchor">
<h2>Breaking Down FAIR, Letter-by-Letter<a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The aspects of the FAIR principles apply to data and metadata with the aim of making the information available to people and computers as described in the seminal paper by <a href="https://pubmed.ncbi.nlm.nih.gov/26978244/">Wilkinson et al., 2016</a>.</p>
<div id="f-findable-in-fair" class="section level4 hasAnchor">
<h4>F (Findable) in FAIR<a href="fair-data-management-practices.html#f-findable-in-fair" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The F in FAIR identifies components of the principles needed to make the meta(data) findable through the application of unique persistent identifiers, thoroughly described, reference the unique identifiers, and that the descriptive information (i.e., metadata) could be searched by both <em>humans and computer systems</em>.</p>
<p><strong>F1. (Meta)data are assigned a globally unique and persistent identifier</strong></p>
<ul>
<li>Each data set is assigned a globally unique and persistent identifier (PID), for example a DOI. These identifiers allow to find, cite and track (meta)data.<br />
</li>
<li>A DOI looks like: <a href="https://doi.org/10.1109/5.771073" class="uri">https://doi.org/10.1109/5.771073</a><br />
</li>
<li>Action: Ensure that each data set is assigned a globally unique and persistent identifier. Certain repositories automatically assign identifiers to data sets as a service. If not, obtain a PID via a <a href="https://pidservices.org/">PID registration service</a>.</li>
</ul>
<p><strong>F2. Data are described with rich metadata</strong></p>
<ul>
<li>Each data set is thoroughly (see R1) described: these metadata document how the data was generated, under what term (license) and how it can be (re)used and provide the necessary context for proper interpretation. This information needs to be machine-readable.<br />
</li>
<li>Action: Fully document each data set in the metadata, which may include descriptive information about the context, quality and condition, or characteristics of the data. Another researcher in any field, or their computer, should be able to properly understand the nature of your dataset. Be as generous as possible with your metadata (see R1).</li>
</ul>
<p><strong>F3. Metadata clearly and explicitly include the identifier of the data it describes</strong></p>
<ul>
<li>Explanation: The metadata and the data set they describe are separate files. The association between a metadata file and the data set is obvious thanks to the mention of the data set’s PID in the metadata.<br />
</li>
<li>Action: Make sure that the metadata contains the data set’s PID.</li>
</ul>
<p><strong>F4. (Meta)data are registered or indexed in a searchable resource</strong></p>
<ul>
<li>Explanation: Metadata are used to build easily searchable indexes of data sets. These resources will allow to search for existing data sets similarly to searching for a book in a library.<br />
</li>
<li>Action: Provide detailed and complete metadata for each data set (see F2).</li>
</ul>
<!-- #### With this information, can can now answer **Question 3**: -->
<!-- ##### (3) When making data ‘Findable’, who and what should be able to find your data? -->
<!-- ##### *Answer: Both humans and computer systems should be able to find your data.* -->
<p><br></p>
<div class="question">
<p><i>With this information, can can now answer <strong>Question 3</strong>:</i></p>
<p>When making data ‘Findable’, who and what should be able to find your data?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Both humans and computer systems should be able to find your data.</p>
</div>
<p><br></p>
</div>
<div id="a-accessible-in-fair" class="section level4 hasAnchor">
<h4>A (Accessible) in FAIR<a href="fair-data-management-practices.html#a-accessible-in-fair" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The A components are designed to enable meta(data) be available long-term, accessed by humans and machines using standard communication protocols with clearly described limitations on reuse.</p>
<p><strong>A1. (Meta)data are retrievable by their identifier using a standardized communications protocol</strong></p>
<ul>
<li>Explanation: If one knows a data set’s identifier and the location where it is archived, one can access at least the metadata. Furthermore, the user knows how to proceed to get access to the data.<br />
</li>
<li>Action: Clearly define who can access the actual data and specify how. It is possible that data will not be downloaded, but rather reused <em>in situ</em>. If so, the metadata must specify the conditions under which this is allowed (sometimes versus the conditions needed to fulfill for external usage/“download”).</li>
</ul>
<p><strong>A1.1 The protocol is open, free, and universally implementable</strong></p>
<ul>
<li>Explanation: Anyone with a computer and an internet connection can access at least the metadata.</li>
</ul>
<p><strong>A1.2 The protocol allows for an authentication and authorization procedure, where necessary</strong></p>
<ul>
<li>Explanation: It often makes sense to request users to create a user account on a repository. This allows to authenticate the owner (or contributor) of each data set, and to potentially set user specific rights.</li>
</ul>
<p><strong>A2. Metadata are accessible, even when the data are no longer available</strong></p>
<ul>
<li>Explanation: Maintaining all data sets in a readily usable state eternally would require an enormous amount of curation work (adapting to new standards for formats, converting to different format if specifically needed software is discontinued, etc). Keeping the metadata describing each data set accessible, however, can be done with much less resources. This allows to build comprehensive data indexes including all current, past, and potentially arising data sets.</li>
<li>Action: Provide detailed and complete metadata for each data set (see R1).</li>
</ul>
</div>
<div id="i-interoperable-in-fair" class="section level4 hasAnchor">
<h4>I (Interoperable) in FAIR<a href="fair-data-management-practices.html#i-interoperable-in-fair" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The I components of the principles address needs for data exchange and interpretation by humans and machines which includes the use of controlled vocabularies or ontologies to describe meta(data) and to describe provenance relationships through appropriate data citation.</p>
<p><strong>I1. (Meta)data use a formal, accessible, shared, and broadly applicable language</strong></p>
<ul>
<li>Explanation: Interoperability typically means that each computer system has at least knowledge of the other system’s formats in which data is exchanged. If (meta)data are to be searchable and if compatible data sources should be combinable in a (semi)automatic way, computer systems need to be able to decide if the content of data sets are comparable.</li>
<li>Action: Provide machine readable data and metadata in an accessible language, using a well-established formalism. Data and metadata are annotated with resolvable vocabularies/ontologies/thesauri that are commonly used in the field (see I2).</li>
</ul>
<p><strong>I2. (Meta)data use vocabularies that follow FAIR principles</strong></p>
<ul>
<li>Explanation: The controlled vocabulary (e.g., <a href="https://www.ncbi.nlm.nih.gov/mesh/">MESH</a>) used to describe data sets needs to be documented. This documentation needs to be easily findable and accessible by anyone who uses the data set.</li>
<li>Action: The vocabularies/ontologies/thesauri are themselves findable, accessible, interoperable and thoroughly documented, hence FAIR. Lists of these standards can be found at: <a href="https://bioportal.bioontology.org/">NCBO BioPortal</a>, <a href="https://fairsharing.org/">FAIRSharing</a>, <a href="http://www.obofoundry.org/">OBO Foundry</a>.</li>
</ul>
<p><strong>I3. (Meta)data include qualified references to other (meta)data</strong></p>
<ul>
<li>Explanation: If the data set builds on another data set, if additional data sets are needed to complete the data, or if complementary information is stored in a different data set, this needs to be specified. In particular, the scientific link between the data sets needs to be described. Furthermore, all data sets need to be properly cited (i.e. including their persistent identifiers).</li>
<li>Action: Properly cite relevant/associated data sets, by providing their persistent identifiers, in the metadata, and describe the scientific link/relation to your data set.</li>
</ul>
</div>
<div id="r-reusable-in-fair" class="section level4 hasAnchor">
<h4>R (Reusable) in FAIR<a href="fair-data-management-practices.html#r-reusable-in-fair" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The R components highlight needs for the meta(data) to be reused and support integration such as sufficient description of the data and data use limitations.</p>
<p><strong>R1. Meta(data) are richly described with a plurality of accurate and relevant attributes</strong></p>
<p>Explanation: Description of a data set is required at two different levels:</p>
<ul>
<li>Metadata describing the data set: what does the data set contain, how was the data generated, how has it been processed, how can it be reused.<br />
</li>
<li>Metadata describing the data: any needed information to properly use the data, such as definitions of the variable names</li>
</ul>
<p>Action: Provide complete metadata for each data file.</p>
<ul>
<li>Scope of your data: for what purpose was it generated/collected?<br />
</li>
<li>Particularities or limitations about the data that other users should be aware of.<br />
</li>
<li>Date of the data set generation, lab conditions, who prepared the data, parameter settings, name and version of the software used.<br />
</li>
<li>Variable names are explained or self-explanatory.</li>
<li>Version of the archived and/or reused data is clearly specified and documented.</li>
</ul>
</div>
<div id="what-does-this-mean-to-you" class="section level4 hasAnchor">
<h4>What does this mean to you?<a href="fair-data-management-practices.html#what-does-this-mean-to-you" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We advise the following as ‘starting-points’ for participants to start meeting FAIR guidances:</p>
<ul>
<li>Learn how to create a <a href="https://dmptool.org">Data Management Plan</a></li>
<li>Keep good documentation (project &amp; data-level) while working</li>
<li>Do not use proprietary file formats (.csv is a great go-to formats for your data!)</li>
<li>When able, use a domain appropriate metadata standard or ontology</li>
<li>Ruthlessly document any steps in a project</li>
<li>Most of FAIR can be handled by selecting a good data or software repository</li>
<li>Don’t forget to include a <a href="https://resources.data.gov/open-licenses/">license</a>!</li>
</ul>
<!-- #### With this information, can can now answer **Question 4**: -->
<!-- ##### (4) When saving/formatting your data, which of the following formats is preferred to meet FAIR principles: .pdf, .csv, or a proprietary output file from your lab instrument? -->
<!-- ##### *Answer: A .csv file is preferred to enhance data sharing.* -->
<p><br></p>
<div class="question">
<p><i>With this information, can can now answer <strong>Question 4</strong>:</i></p>
<p>When saving/formatting your data, which of the following formats is preferred to meet FAIR principles: .pdf, .csv, or a proprietary output file from your lab instrument?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Answer: A .csv file is preferred to enhance data sharing.</p>
</div>
<p><br></p>
</div>
</div>
<div id="data-repositories-for-sharing-of-data" class="section level2 hasAnchor">
<h2>Data Repositories for Sharing of Data<a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When you are organizing your data to deposit online, it is important to identify an appropriate repository to publish your dataset it. A good starting place is a repository registry such as re3data.org or the NIH list of repositories. Many funding agencies have a list of recommended data repositories.</p>
<p>Below are some examples of two main categories of data repositories:</p>
<p><strong>1. Domain Agnostic Data Repositories</strong>
Domain agnostic repositories allow the deposition of any data type. Some examples include the following:</p>
<ul>
<li>Data in Brief Articles (e.g., <a href="https://www.journals.elsevier.com/data-in-brief">Elsevier’s Data in Brief Journal</a>)</li>
<li><a href="https://www.datadryad.org">Dryad</a></li>
<li><a href="https://figshare.com/">Figshare</a></li>
<li><a href="https://dataverse.org/">The Dataverse Project</a></li>
<li><a href="https://zenodo.org/">Zenodo</a></li>
</ul>
<p><strong>2. Domain Specific Data Repositories</strong>
Domain specific repositories allow the deposition of specific types of data, produced from specific types of technologies or within specific domains. Some examples include the following:</p>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/">Gene Expression Omnibus</a></li>
<li><a href="https://www.immport.org/home">The Immunology Database and Analysis Portal</a></li>
<li><a href="https://upddi.pitt.edu/microphysiology-systems-database/">Microphysiology Systems Database</a></li>
<li><a href="http://www.informatics.jax.org/">Mouse Genome Informatics</a></li>
<li><a href="https://phenome.jax.org/">Mouse Phenome Database</a></li>
<li><a href="https://openneuro.org/">OpenNeuro</a></li>
<li><a href="https://www.rcsb.org/">Protein Data Bank</a></li>
<li><a href="http://www.proteomexchange.org/">ProteomeXchange</a></li>
<li><a href="https://rgd.mcw.edu/">Rat Genome Database</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/gap/">The Database of Genotypes and Phenotypes</a></li>
<li><a href="http://zfin.org/">Zebrafish Model Organism Database</a></li>
<li>and many, many others…</li>
</ul>
<!-- #### With this information, can can now answer **Question 5**: -->
<!-- ##### (5) How can I find a suitable data repository for my data? -->
<!-- ##### *Answer: I can search through a data repository registry service or look for recommendations from NIH or other funding agencies.* -->
<p><br></p>
<div class="question">
<p><i>With this information, can can now answer <strong>Question 5</strong>: </i></p>
<p>How can I find a suitable data repository for my data?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: I can search through a data repository registry service or look for recommendations from NIH or other funding agencies.</p>
</div>
<p><br></p>
</div>
<div id="helpful-resources-on-fair" class="section level2 hasAnchor">
<h2>Helpful Resources on FAIR<a href="fair-data-management-practices.html#helpful-resources-on-fair" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="additional-training-resources-on-fair" class="section level4 hasAnchor">
<h4>Additional Training Resources on FAIR<a href="fair-data-management-practices.html#additional-training-resources-on-fair" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Many organizations, from specific programs to broad organizations, provide training and resources for scientists in FAIR principles. Some of the notable global organizations organizing and providing training that offer opportunities for community involvement are:</p>
<ul>
<li><a href="https://www.codata.org/uploads/CODATA@45years.pdf">Committee on Data for Science and Technology (CODATA)</a></li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/27149219/">Global Alliance for Genomics &amp; Health</a></li>
<li><a href="https://www.go-fair.org/">GoFAIR</a></li>
<li><a href="https://www.force11.org/">Force11</a></li>
<li><a href="http://www.dlib.org/dlib/january14/01guest_editorial.html">Research Data Alliance</a></li>
</ul>
</div>
<div id="additional-examples-and-documents-on-fair" class="section level4 hasAnchor">
<h4>Additional Examples and Documents on FAIR<a href="fair-data-management-practices.html#additional-examples-and-documents-on-fair" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This topic is receiving much attention in recent years, including the following workshops, government reports, and publications.</p>
<p><strong>Example Workshops discussing FAIR</strong>:</p>
<ul>
<li><p>NAS Implementing FAIR Data for People and Machines: Impacts and Implications (2019). Available at: <a href="https://www.nationalacademies.org/our-work/implementing-fair-data-for-people-and-machines-impacts-and-implications" class="uri">https://www.nationalacademies.org/our-work/implementing-fair-data-for-people-and-machines-impacts-and-implications</a></p></li>
<li><p>NIH Catalyzing Knowledge-driven Discovery in Environmental Health Sciences Through a Harmonized Language, Virtual Workshop (2021). Available at: <a href="https://www.niehs.nih.gov/news/events/pastmtg/2021/ehslanguage/index.cfm" class="uri">https://www.niehs.nih.gov/news/events/pastmtg/2021/ehslanguage/index.cfm</a></p></li>
<li><p>NIH Trustworthy Data Repositories Workshop (2019). Available at: <a href="https://datascience.nih.gov/data-ecosystem/trustworthy-data-repositories-workshop" class="uri">https://datascience.nih.gov/data-ecosystem/trustworthy-data-repositories-workshop</a></p></li>
<li><p>NIH Virtual Workshop on Data Metrics (2020). Available at: <a href="https://datascience.nih.gov/data-ecosystem/nih-virtual-workshop-on-data-metrics" class="uri">https://datascience.nih.gov/data-ecosystem/nih-virtual-workshop-on-data-metrics</a></p></li>
<li><p>NIH Workshop on the Role of Generalist Repositories to Enhance Data Discoverability and Reuse: Workshop Summary (2020). Available at: <a href="https://datascience.nih.gov/data-ecosystem/nih-data-repository-workshop-summary" class="uri">https://datascience.nih.gov/data-ecosystem/nih-data-repository-workshop-summary</a></p></li>
</ul>
<p><strong>Example Government Report Documents on FAIR:</strong></p>
<ul>
<li><p>Collins S, Genova F, Harrower N, Hodson S, Jones S, Laaksonen L, Mietchen D, Petrauskaite R, Wittenburg P. Turning FAIR into reality: Final report and action plan from the European Commission expert group on FAIR data: European Union; 2018. Available at: <a href="https://www.vdu.lt/cris/handle/20.500.12259/103794" class="uri">https://www.vdu.lt/cris/handle/20.500.12259/103794</a>.</p></li>
<li><p>EU. FAIR Data Advanced Use Cases: From Principles to Practice in the Netherlands. 2018. European Union. Available at: <a href="doi:10.5281/zenodo.1250535" class="uri">doi:10.5281/zenodo.1250535</a>.</p></li>
<li><p>NIH. Final NIH Policy for Data Management and Sharing and Supplemental Information. National Institutes of Health. Federal Register, vol. 85, 2020-23674, 30 Oct. 2020, pp. 68890–900. Available at: <a href="https://www.federalregister.gov/d/2020-23674" class="uri">https://www.federalregister.gov/d/2020-23674</a>.</p></li>
<li><p>NIH. NIH Strategic Plan for Data Science 2018. National Institutes of Health. Available at: <a href="https://datascience.nih.gov/strategicplan" class="uri">https://datascience.nih.gov/strategicplan</a>.</p></li>
<li><p>NLM. NLM Strategic Plan 2017 to 2027. U.S. National Library of Medicine, Feb. 2018. Available at: <a href="https://www.nlm.nih.gov/about/strategic-plan.html" class="uri">https://www.nlm.nih.gov/about/strategic-plan.html</a>.</p></li>
</ul>
<p><strong>Example Related Publications on FAIR:</strong></p>
<ul>
<li><p>Comess S, Akbay A, Vasiliou M, Hines RN, Joppa L, Vasiliou V, Kleinstreuer N. Bringing Big Data to Bear in Environmental Public Health: Challenges and Recommendations. Front Artif Intell. 2020 May;3:31. doi: 10.3389/frai.2020.00031. Epub 2020 May 15. PMID: 33184612; PMCID: <a href="https://pubmed.ncbi.nlm.nih.gov/33184612/">PMC7654840</a>.</p></li>
<li><p>Koers H, Bangert D, Hermans E, van Horik R, de Jong M, Mokrane M. Recommendations for Services in a FAIR Data Ecosystem. Patterns (N Y). 2020 Jul 7;1(5):100058. doi: 10.1016/j.patter.2020.100058. Erratum in: Patterns (N Y). 2020 Sep 11;1(6):100104. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/33205119/">33205119</a>.</p></li>
<li><p>Kush RD, Warzel D, Kush MA, Sherman A, Navarro EA, Fitzmartin R, Pétavy F, Galvez J, Becnel LB, Zhou FL, Harmon N, Jauregui B, Jackson T, Hudson L. FAIR data sharing: The roles of common data elements and harmonization. J Biomed Inform. 2020 Jul;107:103421. doi: 10.1016/j.jbi.2020.103421. Epub 2020 May 12. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/32407878/">32407878</a>.</p></li>
<li><p>Lin D, Crabtree J, Dillo I, Downs RR, Edmunds R, Giaretta D, De Giusti M, L’Hours H, Hugo W, Jenkyns R, Khodiyar V, Martone ME, Mokrane M, Navale V, Petters J, Sierman B, Sokolova DV, Stockhause M, Westbrook J. The TRUST Principles for digital repositories. Sci Data. 2020 May 14;7(1):144. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/32409645/">32409645</a>.</p></li>
<li><p>Thessen AE, Grondin CJ, Kulkarni RD, Brander S, Truong L, Vasilevsky NA, Callahan TJ, Chan LE, Westra B, Willis M, Rothenberg SE, Jarabek AM, Burgoon L, Korrick SA, Haendel MA. Community Approaches for Integrating Environmental Exposures into Human Models of Disease. Environ Health Perspect. 2020 Dec;128(12):125002. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/33369481/">33369481</a>.</p></li>
<li><p>Roundtable on Environmental Health Sciences, Research, and Medicine; Board on Population Health and Public Health Practice; Health and Medicine Division; National Academies of Sciences, Engineering, and Medicine. Principles and Obstacles for Sharing Data from Environmental Health Research: Workshop Summary. Washington (DC): National Academies Press (US); 2016 Apr 29. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/27227195/">27227195</a>.</p></li>
</ul>
</div>
</div>
<div id="concluding-remarks-5" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="fair-data-management-practices.html#concluding-remarks-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In conclusion, this training module introduces participants to best data management practices in the field of exposure science, toxicology, and environmental health research through the implementation of FAIR practices. This module first provides an introduction to FAIR, including a history of how this term was first developed and implemented. Participants are then guided through each component of FAIR, organized by letter (i.e., Findable, Accessible, Interoperable, and Reusable). The training module then reviews different types of data repositories that can be used to publish datasets in exposure science, toxicology, and environmental health research. Lastly, this module summarizes additional training resources, workshops, government reports, and example publications surrounding the use of FAIR data management practices.</p>

</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="high-dimensional-data-visualizations.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dose-response-modeling.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
