<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Finding and Visualizing Data Trends | DMAC Training Modules</title>
  <meta name="description" content="This is a resource for R training using environmental health datasets along standard code pipelines and methods." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Finding and Visualizing Data Trends | DMAC Training Modules" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a resource for R training using environmental health datasets along standard code pipelines and methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Finding and Visualizing Data Trends | DMAC Training Modules" />
  
  <meta name="twitter:description" content="This is a resource for R training using environmental health datasets along standard code pipelines and methods." />
  

<meta name="author" content="Kyle Roell, Lauren Koval, Julia Rager" />


<meta name="date" content="2021-06-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dataorg.html"/>
<link rel="next" href="multi-omics-analyses-for-environmental-health.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DMAC Training Modules</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Setting Up Your R Environment</a><ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#r-and-rstudio"><i class="fa fa-check"></i><b>2.1</b> R and RStudio</a><ul>
<li class="chapter" data-level="2.1.1" data-path="intro.html"><a href="intro.html#downloading-r-and-rstudio"><i class="fa fa-check"></i><b>2.1.1</b> Downloading R and RStudio</a></li>
<li class="chapter" data-level="2.1.2" data-path="intro.html"><a href="intro.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>2.1.2</b> Installing R and RStudio</a></li>
<li class="chapter" data-level="2.1.3" data-path="intro.html"><a href="intro.html#installing-and-loading-packages"><i class="fa fa-check"></i><b>2.1.3</b> Installing and Loading Packages</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#scripting-basics"><i class="fa fa-check"></i><b>2.2</b> Scripting Basics</a><ul>
<li class="chapter" data-level="2.2.1" data-path="intro.html"><a href="intro.html#setting-working-directory"><i class="fa fa-check"></i><b>2.2.1</b> Setting Working Directory</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro.html"><a href="intro.html#importing-and-exporting-files"><i class="fa fa-check"></i><b>2.2.2</b> Importing and Exporting Files</a></li>
<li class="chapter" data-level="2.2.3" data-path="intro.html"><a href="intro.html#viewing-data"><i class="fa fa-check"></i><b>2.2.3</b> Viewing Data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dataorg.html"><a href="dataorg.html"><i class="fa fa-check"></i><b>3</b> The Basics for Data Organization</a><ul>
<li class="chapter" data-level="3.1" data-path="dataorg.html"><a href="dataorg.html#basic-data-manipulation"><i class="fa fa-check"></i><b>3.1</b> Basic Data Manipulation</a><ul>
<li class="chapter" data-level="3.1.1" data-path="dataorg.html"><a href="dataorg.html#merging"><i class="fa fa-check"></i><b>3.1.1</b> Merging</a></li>
<li class="chapter" data-level="3.1.2" data-path="dataorg.html"><a href="dataorg.html#filtering-subsetting"><i class="fa fa-check"></i><b>3.1.2</b> Filtering &amp; subsetting</a></li>
<li class="chapter" data-level="3.1.3" data-path="dataorg.html"><a href="dataorg.html#melt-and-cast"><i class="fa fa-check"></i><b>3.1.3</b> Melt and Cast</a></li>
<li class="chapter" data-level="3.1.4" data-path="dataorg.html"><a href="dataorg.html#tidyverse"><i class="fa fa-check"></i><b>3.1.4</b> Tidyverse</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html"><i class="fa fa-check"></i><b>4</b> Finding and Visualizing Data Trends</a><ul>
<li class="chapter" data-level="4.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#basic-statistical-tests-and-visualizations-of-data"><i class="fa fa-check"></i><b>4.1</b> Basic Statistical Tests and Visualizations of Data</a><ul>
<li class="chapter" data-level="4.1.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#normality"><i class="fa fa-check"></i><b>4.1.1</b> Normality</a></li>
<li class="chapter" data-level="4.1.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#t-tests"><i class="fa fa-check"></i><b>4.1.2</b> T-tests</a></li>
<li class="chapter" data-level="4.1.3" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#anova"><i class="fa fa-check"></i><b>4.1.3</b> ANOVA</a></li>
<li class="chapter" data-level="4.1.4" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#regression-linear-regression-and-logistic-regression"><i class="fa fa-check"></i><b>4.1.4</b> Regression: linear regression and logistic regression</a></li>
<li class="chapter" data-level="4.1.5" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#chi-squared-test-box-plots"><i class="fa fa-check"></i><b>4.1.5</b> Chi-squared test – box plots</a></li>
<li class="chapter" data-level="4.1.6" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#fishers-exact-test"><i class="fa fa-check"></i><b>4.1.6</b> Fisher’s exact test</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#heat-maps"><i class="fa fa-check"></i><b>4.2</b> Heat maps</a><ul>
<li class="chapter" data-level="4.2.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#pheatmap"><i class="fa fa-check"></i><b>4.2.1</b> pheatmap</a></li>
<li class="chapter" data-level="4.2.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#heatmap2"><i class="fa fa-check"></i><b>4.2.2</b> heatmap2</a></li>
<li class="chapter" data-level="4.2.3" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#superheat"><i class="fa fa-check"></i><b>4.2.3</b> superheat</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#clustering"><i class="fa fa-check"></i><b>4.3</b> Clustering</a><ul>
<li class="chapter" data-level="4.3.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#hierarchical"><i class="fa fa-check"></i><b>4.3.1</b> Hierarchical</a></li>
<li class="chapter" data-level="4.3.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#k-means"><i class="fa fa-check"></i><b>4.3.2</b> K-means</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#data-reduction-pca"><i class="fa fa-check"></i><b>4.4</b> Data reduction (PCA)</a><ul>
<li class="chapter" data-level="4.4.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#visualize-pca-plot"><i class="fa fa-check"></i><b>4.4.1</b> Visualize PCA Plot</a></li>
<li class="chapter" data-level="4.4.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#identify-of-variance-captured"><i class="fa fa-check"></i><b>4.4.2</b> Identify % of variance captured</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html"><i class="fa fa-check"></i><b>5</b> Multi-Omics Analyses for Environmental Health</a><ul>
<li class="chapter" data-level="5.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#exposomics"><i class="fa fa-check"></i><b>5.1</b> Exposomics</a><ul>
<li class="chapter" data-level="5.1.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#placenta-exposome"><i class="fa fa-check"></i><b>5.1.1</b> Placenta Exposome</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#transcriptomics"><i class="fa fa-check"></i><b>5.2</b> Transcriptomics</a><ul>
<li class="chapter" data-level="5.2.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#deseq2-rnaseq"><i class="fa fa-check"></i><b>5.2.1</b> DESeq2 / RNAseq</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#genome-wide-microrna"><i class="fa fa-check"></i><b>5.3</b> Genome-wide MicroRNA</a></li>
<li class="chapter" data-level="5.4" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#genome-wide-dna-methylation"><i class="fa fa-check"></i><b>5.4</b> Genome-wide DNA Methylation</a><ul>
<li class="chapter" data-level="5.4.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#illumina-array-data"><i class="fa fa-check"></i><b>5.4.1</b> Illumina array data</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#proteomics"><i class="fa fa-check"></i><b>5.5</b> Proteomics</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mixtures-analyses-for-environmental-health.html"><a href="mixtures-analyses-for-environmental-health.html"><i class="fa fa-check"></i><b>6</b> Mixtures Analyses for Environmental Health</a><ul>
<li class="chapter" data-level="6.1" data-path="mixtures-analyses-for-environmental-health.html"><a href="mixtures-analyses-for-environmental-health.html#sufficient-similarity"><i class="fa fa-check"></i><b>6.1</b> Sufficient Similarity</a></li>
<li class="chapter" data-level="6.2" data-path="mixtures-analyses-for-environmental-health.html"><a href="mixtures-analyses-for-environmental-health.html#mixtures-modeling-through-qgcomp"><i class="fa fa-check"></i><b>6.2</b> Mixtures Modeling through qgcomp</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html"><i class="fa fa-check"></i><b>7</b> Environmental Health Databases</a><ul>
<li class="chapter" data-level="7.1" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html#comparative-toxicogenomics-database-ctd"><i class="fa fa-check"></i><b>7.1</b> Comparative Toxicogenomics Database (CTD)</a></li>
<li class="chapter" data-level="7.2" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html#gene-expression-omnibus-geo"><i class="fa fa-check"></i><b>7.2</b> Gene Expression Omnibus (GEO)</a></li>
<li class="chapter" data-level="7.3" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html#nhanes"><i class="fa fa-check"></i><b>7.3</b> NHANES</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">DMAC Training Modules</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="finding-and-visualizing-data-trends" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Finding and Visualizing Data Trends</h1>
<div id="basic-statistical-tests-and-visualizations-of-data" class="section level2">
<h2><span class="header-section-number">4.1</span> Basic Statistical Tests and Visualizations of Data</h2>
<p>Need an example dataset – maybe ELGAN shuffled/deidentified, with made-up environmental exposure column?</p>
<div id="normality" class="section level3">
<h3><span class="header-section-number">4.1.1</span> Normality</h3>
<p>Many statistical tests and methods rely on assumptions of normality. There are a few ways to look at the normality of a dataset, both formally and informally. Plotting data using historgrams, densities, or qqplots, can graphically help inform if a variable is normally distributed. There also exist statistical tests, such as the Kolmogorov-Smirnov (K-S) normality test and Shapiro-Wilk’s test, that will formally test if data come from a normal distribution. When using these tests, it is important to remember that the null hypothesis is that the sample distribution is normal, and a significant p-value means the distribution is non-normal.</p>
<p>Even when using more formal statistical tests to test for normality, it is important to visualize the data, as well.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="finding-and-visualizing-data-trends.html#cb29-1"></a><span class="co">#Histograms to visualize data</span></span>
<span id="cb29-2"><a href="finding-and-visualizing-data-trends.html#cb29-2"></a><span class="kw">hist</span>(full.data<span class="op">$</span>BMI);</span></code></pre></div>
<p><img src="test1_files/figure-html/normality-1.png" width="672" /></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="finding-and-visualizing-data-trends.html#cb30-1"></a><span class="co">#Can decrease bin size (using breaks parameter) to better visualize</span></span>
<span id="cb30-2"><a href="finding-and-visualizing-data-trends.html#cb30-2"></a><span class="kw">hist</span>(full.data<span class="op">$</span>BMI, <span class="dt">breaks=</span><span class="dv">20</span>);</span></code></pre></div>
<p><img src="test1_files/figure-html/normality-2.png" width="672" /></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="finding-and-visualizing-data-trends.html#cb31-1"></a><span class="co">#Also look at normal qqplot using qqnorm function </span></span>
<span id="cb31-2"><a href="finding-and-visualizing-data-trends.html#cb31-2"></a><span class="kw">qqnorm</span>(full.data<span class="op">$</span>BMI);</span>
<span id="cb31-3"><a href="finding-and-visualizing-data-trends.html#cb31-3"></a><span class="co">#Add reference line</span></span>
<span id="cb31-4"><a href="finding-and-visualizing-data-trends.html#cb31-4"></a><span class="kw">qqline</span>(full.data<span class="op">$</span>BMI);</span></code></pre></div>
<p><img src="test1_files/figure-html/normality-3.png" width="672" /></p>
<p>From visual inspection, BMI seems to be pretty normally distributed. But, we can also check using a more formal statistical test, the Shapiro-Wilk’s test.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="finding-and-visualizing-data-trends.html#cb32-1"></a><span class="co">#Perform Shapiro-Wilk normality test (from base stats package)</span></span>
<span id="cb32-2"><a href="finding-and-visualizing-data-trends.html#cb32-2"></a><span class="kw">shapiro.test</span>(full.data<span class="op">$</span>BMI);</span></code></pre></div>
<pre><code>## 
##  Shapiro-Wilk normality test
## 
## data:  full.data$BMI
## W = 0.99617, p-value = 0.9014</code></pre>
<p>We get a p-value of .9014, so cannot reject the null hypothesis. This means that we can assume normality of this data.</p>
</div>
<div id="t-tests" class="section level3">
<h3><span class="header-section-number">4.1.2</span> T-tests</h3>
<p>T-tests are used to test for a significant difference between the means of two groups. There are a few types of t-tests, but here, we will be comparing BMI between two groups, smokers and non-smokers, and will be using a two sample t-test (or independent samples t-test).</p>
<p>TALK ABOUT ASSUMPTIONS…</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="finding-and-visualizing-data-trends.html#cb34-1"></a><span class="co">#It is nice to visualize the data across groups using things like boxplots</span></span>
<span id="cb34-2"><a href="finding-and-visualizing-data-trends.html#cb34-2"></a><span class="kw">boxplot</span>(<span class="dt">data=</span>full.data, BMI <span class="op">~</span><span class="st"> </span>Smoker);</span></code></pre></div>
<p><img src="test1_files/figure-html/ttest-1.png" width="672" /></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="finding-and-visualizing-data-trends.html#cb35-1"></a><span class="co">#It is easy to peform a t-test on these data using t.test() from base stats package</span></span>
<span id="cb35-2"><a href="finding-and-visualizing-data-trends.html#cb35-2"></a><span class="kw">t.test</span>(<span class="dt">data=</span>full.data, BMI <span class="op">~</span><span class="st"> </span>Smoker);</span></code></pre></div>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  BMI by Smoker
## t = 1.6189, df = 77.478, p-value = 0.1095
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.4115608  3.9922067
## sample estimates:
## mean in group 0 mean in group 1 
##        25.82896        24.03864</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="finding-and-visualizing-data-trends.html#cb37-1"></a><span class="co">#We can also save the results into a variable and access various output values</span></span>
<span id="cb37-2"><a href="finding-and-visualizing-data-trends.html#cb37-2"></a>ttest.res =<span class="st"> </span><span class="kw">t.test</span>(<span class="dt">data=</span>full.data, BMI <span class="op">~</span><span class="st"> </span>Smoker);</span>
<span id="cb37-3"><a href="finding-and-visualizing-data-trends.html#cb37-3"></a></span>
<span id="cb37-4"><a href="finding-and-visualizing-data-trends.html#cb37-4"></a><span class="co">#For example, we can access the p-value</span></span>
<span id="cb37-5"><a href="finding-and-visualizing-data-trends.html#cb37-5"></a>ttest.res<span class="op">$</span>p.value;</span></code></pre></div>
<pre><code>## [1] 0.1095319</code></pre>
<p>From the boxplots, it’s clear that there are differences in the BMIs between smokers and non-smokers. And, in fact, when running the t-test, we see that means across groups are not equal.</p>
</div>
<div id="anova" class="section level3">
<h3><span class="header-section-number">4.1.3</span> ANOVA</h3>
<p>Analysis of Variance (ANOVA) is a statistical method that is used to compare means of more than two groups.</p>
<p>TALK ABOUT ASSUMPTIONS and other stuff.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="finding-and-visualizing-data-trends.html#cb39-1"></a><span class="co">#Boxplots to look at data across groups</span></span>
<span id="cb39-2"><a href="finding-and-visualizing-data-trends.html#cb39-2"></a><span class="kw">boxplot</span>(<span class="dt">data=</span>full.data, BMI <span class="op">~</span><span class="st"> </span>Smoker3);</span></code></pre></div>
<p><img src="test1_files/figure-html/anova-1.png" width="672" /></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="finding-and-visualizing-data-trends.html#cb40-1"></a><span class="co">#Can also get group means</span></span>
<span id="cb40-2"><a href="finding-and-visualizing-data-trends.html#cb40-2"></a>full.data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(Smoker3) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="kw">mean</span>(BMI));</span></code></pre></div>
<pre><code>## # A tibble: 3 x 2
##   Smoker3 `mean(BMI)`
##   &lt;chr&gt;         &lt;dbl&gt;
## 1 Current        19.0
## 2 Former         27.4
## 3 Never          26.9</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="finding-and-visualizing-data-trends.html#cb42-1"></a><span class="co">#Use ANOVA to compare groups (use aov to fit ANOVA model)</span></span>
<span id="cb42-2"><a href="finding-and-visualizing-data-trends.html#cb42-2"></a><span class="kw">aov</span>(<span class="dt">data=</span>full.data, BMI <span class="op">~</span><span class="st"> </span>Smoker3);</span></code></pre></div>
<pre><code>## Call:
##    aov(formula = BMI ~ Smoker3, data = full.data)
## 
## Terms:
##                  Smoker3 Residuals
## Sum of Squares  2087.198  7130.377
## Deg. of Freedom        2       197
## 
## Residual standard error: 6.016212
## Estimated effects may be unbalanced</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="finding-and-visualizing-data-trends.html#cb44-1"></a><span class="co">#We can get the typical ANOVA table using either summary or anova on fitted object</span></span>
<span id="cb44-2"><a href="finding-and-visualizing-data-trends.html#cb44-2"></a><span class="kw">anova</span>(<span class="kw">aov</span>(<span class="dt">data=</span>full.data, BMI <span class="op">~</span><span class="st"> </span>Smoker3));</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: BMI
##            Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## Smoker3     2 2087.2 1043.60  28.833 1.04e-11 ***
## Residuals 197 7130.4   36.19                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>From the ANOVA table, we can conclude that the group means are not all equal.</p>
</div>
<div id="regression-linear-regression-and-logistic-regression" class="section level3">
<h3><span class="header-section-number">4.1.4</span> Regression: linear regression and logistic regression</h3>
</div>
<div id="chi-squared-test-box-plots" class="section level3">
<h3><span class="header-section-number">4.1.5</span> Chi-squared test – box plots</h3>
</div>
<div id="fishers-exact-test" class="section level3">
<h3><span class="header-section-number">4.1.6</span> Fisher’s exact test</h3>
</div>
</div>
<div id="heat-maps" class="section level2">
<h2><span class="header-section-number">4.2</span> Heat maps</h2>
<div id="pheatmap" class="section level3">
<h3><span class="header-section-number">4.2.1</span> pheatmap</h3>
</div>
<div id="heatmap2" class="section level3">
<h3><span class="header-section-number">4.2.2</span> heatmap2</h3>
</div>
<div id="superheat" class="section level3">
<h3><span class="header-section-number">4.2.3</span> superheat</h3>
</div>
</div>
<div id="clustering" class="section level2">
<h2><span class="header-section-number">4.3</span> Clustering</h2>
<p>Examples with genomics: Rager et al. 2014</p>
<div id="hierarchical" class="section level3">
<h3><span class="header-section-number">4.3.1</span> Hierarchical</h3>
</div>
<div id="k-means" class="section level3">
<h3><span class="header-section-number">4.3.2</span> K-means</h3>
</div>
</div>
<div id="data-reduction-pca" class="section level2">
<h2><span class="header-section-number">4.4</span> Data reduction (PCA)</h2>
<div id="visualize-pca-plot" class="section level3">
<h3><span class="header-section-number">4.4.1</span> Visualize PCA Plot</h3>
</div>
<div id="identify-of-variance-captured" class="section level3">
<h3><span class="header-section-number">4.4.2</span> Identify % of variance captured</h3>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dataorg.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multi-omics-analyses-for-environmental-health.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["test1.pdf", "test1.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
