<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9.1 Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure) | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="9.1 Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure) | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="9.1 Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure) | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9.1 Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure) | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle Roell, Lauren Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="controlling-for-sources-of-sample-heterogeneity.html"/>
<link rel="next" href="visualizing-statistical-results-using-ma-plots.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TAME Toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#background"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#objectives"><i class="fa fa-check"></i>Objectives</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#module-development-overview"><i class="fa fa-check"></i>Module Development Overview</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#concluding-remarks"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#funding"><i class="fa fa-check"></i>Funding</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#author-affiliations"><i class="fa fa-check"></i>Author Affiliations</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html"><i class="fa fa-check"></i><b>1</b> Introduction to Coding in R</a><ul>
<li class="chapter" data-level="1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html"><i class="fa fa-check"></i><b>1.1</b> General Introduction and Installation of R and RStudio</a><ul>
<li class="chapter" data-level="1.1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html#what-is-r"><i class="fa fa-check"></i><b>1.1.1</b> What is R?</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="downloading-and-installing-r.html"><a href="downloading-and-installing-r.html"><i class="fa fa-check"></i><b>1.2</b> Downloading and Installing R</a></li>
<li class="chapter" data-level="1.3" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html"><i class="fa fa-check"></i><b>1.3</b> Downloading and Installing R Studio</a></li>
<li class="chapter" data-level="1.4" data-path="introduction-to-r-packages.html"><a href="introduction-to-r-packages.html"><i class="fa fa-check"></i><b>1.4</b> Introduction to R Packages</a><ul>
<li class="chapter" data-level="1.4.1" data-path="introduction-to-r-packages.html"><a href="introduction-to-r-packages.html#downloadinginstalling-r-packages"><i class="fa fa-check"></i><b>1.4.1</b> Downloading/Installing R Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="scripting-basics.html"><a href="scripting-basics.html"><i class="fa fa-check"></i><b>1.5</b> Scripting Basics</a><ul>
<li class="chapter" data-level="1.5.1" data-path="scripting-basics.html"><a href="scripting-basics.html#comments"><i class="fa fa-check"></i><b>1.5.1</b> Comments</a></li>
<li class="chapter" data-level="1.5.2" data-path="scripting-basics.html"><a href="scripting-basics.html#starting-code"><i class="fa fa-check"></i><b>1.5.2</b> Starting Code</a></li>
<li class="chapter" data-level="1.5.3" data-path="scripting-basics.html"><a href="scripting-basics.html#setting-your-working-directory"><i class="fa fa-check"></i><b>1.5.3</b> Setting Your Working Directory</a></li>
<li class="chapter" data-level="1.5.4" data-path="scripting-basics.html"><a href="scripting-basics.html#importing-files"><i class="fa fa-check"></i><b>1.5.4</b> Importing Files</a></li>
<li class="chapter" data-level="1.5.5" data-path="scripting-basics.html"><a href="scripting-basics.html#viewing-data"><i class="fa fa-check"></i><b>1.5.5</b> Viewing Data</a></li>
<li class="chapter" data-level="1.5.6" data-path="scripting-basics.html"><a href="scripting-basics.html#exporting-data"><i class="fa fa-check"></i><b>1.5.6</b> Exporting Data</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="concluding-remarks-1.html"><a href="concluding-remarks-1.html"><i class="fa fa-check"></i><b>1.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-organization-basics.html"><a href="data-organization-basics.html"><i class="fa fa-check"></i><b>2</b> Data Organization Basics</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html"><i class="fa fa-check"></i><b>2.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="2.1.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#set-your-working-directory-1"><i class="fa fa-check"></i><b>2.1.1</b> Set your working directory</a></li>
<li class="chapter" data-level="2.1.2" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#importing-example-datasets"><i class="fa fa-check"></i><b>2.1.2</b> Importing example datasets</a></li>
<li class="chapter" data-level="2.1.3" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#viewing-example-datasets"><i class="fa fa-check"></i><b>2.1.3</b> Viewing example datasets</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="merging-data-using-base-r-syntax.html"><a href="merging-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.2</b> Merging Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.3" data-path="filtering-and-subsetting-data-using-base-r-syntax.html"><a href="filtering-and-subsetting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.3</b> Filtering and Subsetting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.4" data-path="melting-and-casting-data-using-base-r-syntax.html"><a href="melting-and-casting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.4</b> Melting and Casting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.5" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html"><i class="fa fa-check"></i><b>2.5</b> Introduction to Tidyverse</a><ul>
<li class="chapter" data-level="2.5.1" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#downloading-and-loading-the-tidyverse-package"><i class="fa fa-check"></i><b>2.5.1</b> Downloading and Loading the Tidyverse Package</a></li>
<li class="chapter" data-level="2.5.2" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#merging-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.2</b> Merging Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.3" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#filtering-and-subsetting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.3</b> Filtering and Subsetting Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.4" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#melting-and-casting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.4</b> Melting and Casting Data using Tidyverse Syntax</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="concluding-remarks-2.html"><a href="concluding-remarks-2.html"><i class="fa fa-check"></i><b>2.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html"><i class="fa fa-check"></i><b>3</b> Finding and Visualizing Data Trend</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html"><i class="fa fa-check"></i><b>3.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="3.1.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html#script-preparations"><i class="fa fa-check"></i><b>3.1.1</b> Script Preparations</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html"><i class="fa fa-check"></i><b>3.2</b> Basic Statistical Tests and Visualizations of Data Trends</a><ul>
<li class="chapter" data-level="3.2.1" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#visualize-and-test-data-for-normality"><i class="fa fa-check"></i><b>3.2.1</b> Visualize and Test Data for Normality</a></li>
<li class="chapter" data-level="3.2.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i><b>3.2.2</b> Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="3.2.3" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i><b>3.2.3</b> Two-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><i class="fa fa-check"></i><b>3.3</b> Regression Modeling and Visualization: Linear and Logistic Regressions</a><ul>
<li class="chapter" data-level="3.3.1" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#linear-regression"><i class="fa fa-check"></i><b>3.3.1</b> Linear Regression</a></li>
<li class="chapter" data-level="3.3.2" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#logistic-regression"><i class="fa fa-check"></i><b>3.3.2</b> Logistic Regression</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="concluding-remarks-3.html"><a href="concluding-remarks-3.html"><i class="fa fa-check"></i><b>3.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html"><i class="fa fa-check"></i><b>4</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-training-module-2.html"><a href="introduction-to-training-module-2.html"><i class="fa fa-check"></i><b>4.1</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="4.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html"><i class="fa fa-check"></i><b>4.2</b> Script Preparations</a><ul>
<li class="chapter" data-level="4.2.1" data-path="script-preparations-1.html"><a href="script-preparations-1.html#formatting-dataframes-for-downstream-visualization-code"><i class="fa fa-check"></i><b>4.2.1</b> Formatting dataframes for downstream visualization code</a></li>
<li class="chapter" data-level="4.2.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html#scaling-dataframes-for-downstream-data-visualizations"><i class="fa fa-check"></i><b>4.2.2</b> Scaling dataframes for downstream data visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html"><i class="fa fa-check"></i><b>4.3</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.3.1" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#density-plot-visualizations"><i class="fa fa-check"></i><b>4.3.1</b> Density Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.2" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#ggally-visualizations"><i class="fa fa-check"></i><b>4.3.2</b> GGally Visualizations</a></li>
<li class="chapter" data-level="4.3.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#boxplot-visualizations"><i class="fa fa-check"></i><b>4.3.3</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="4.3.4" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#correlation-plot-visualizations"><i class="fa fa-check"></i><b>4.3.4</b> Correlation Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.5" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#hierarchical-clustering-visualizations"><i class="fa fa-check"></i><b>4.3.5</b> Hierarchical Clustering Visualizations</a></li>
<li class="chapter" data-level="4.3.6" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#heatmap-visualizations"><i class="fa fa-check"></i><b>4.3.6</b> Heatmap Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="concluding-remarks-4.html"><a href="concluding-remarks-4.html"><i class="fa fa-check"></i><b>4.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i><b>5</b> FAIR Data Management Practices</a><ul>
<li class="chapter" data-level="5.1" data-path="background-on-training-module.html"><a href="background-on-training-module.html"><i class="fa fa-check"></i><b>5.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="5.2" data-path="training-modules-questions.html"><a href="training-modules-questions.html"><i class="fa fa-check"></i><b>5.2</b> Training Module’s <strong>Questions</strong></a></li>
<li class="chapter" data-level="5.3" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html"><i class="fa fa-check"></i><b>5.3</b> Introduction to FAIR</a><ul>
<li class="chapter" data-level="5.3.1" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#with-this-background-we-can-now-answer-question-1"><i class="fa fa-check"></i><b>5.3.1</b> With this background, we can now answer <strong>Question 1</strong>:</a></li>
<li class="chapter" data-level="5.3.2" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#we-can-also-answer-question-2"><i class="fa fa-check"></i><b>5.3.2</b> We can also answer <strong>Question 2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="breaking-down-fair-letter-by-letter.html"><a href="breaking-down-fair-letter-by-letter.html"><i class="fa fa-check"></i><b>5.4</b> Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="5.5" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html"><i class="fa fa-check"></i><b>5.5</b> F (Findable) in FAIR</a><ul>
<li class="chapter" data-level="5.5.1" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html#with-this-information-can-can-now-answer-question-3"><i class="fa fa-check"></i><b>5.5.1</b> With this information, can can now answer <strong>Question 3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="a-accessible-in-fair.html"><a href="a-accessible-in-fair.html"><i class="fa fa-check"></i><b>5.6</b> A (Accessible) in FAIR</a></li>
<li class="chapter" data-level="5.7" data-path="i-interoperable-in-fair.html"><a href="i-interoperable-in-fair.html"><i class="fa fa-check"></i><b>5.7</b> I (Interoperable) in FAIR</a></li>
<li class="chapter" data-level="5.8" data-path="r-reusable-in-fair.html"><a href="r-reusable-in-fair.html"><i class="fa fa-check"></i><b>5.8</b> R (Reusable) in FAIR</a></li>
<li class="chapter" data-level="5.9" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html"><i class="fa fa-check"></i><b>5.9</b> What does this mean to you?</a><ul>
<li class="chapter" data-level="5.9.1" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html#with-this-information-can-can-now-answer-question-4"><i class="fa fa-check"></i><b>5.9.1</b> With this information, can can now answer <strong>Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.10" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html"><i class="fa fa-check"></i><b>5.10</b> Data Repositories for Sharing of Data</a><ul>
<li class="chapter" data-level="5.10.1" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html#with-this-information-can-can-now-answer-question-5"><i class="fa fa-check"></i><b>5.10.1</b> With this information, can can now answer <strong>Question 5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html"><i class="fa fa-check"></i><b>5.11</b> Helpful Resources on FAIR</a><ul>
<li class="chapter" data-level="5.11.1" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-training-resources-on-fair"><i class="fa fa-check"></i><b>5.11.1</b> Additional Training Resources on FAIR</a></li>
<li class="chapter" data-level="5.11.2" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-examples-and-documents-on-fair"><i class="fa fa-check"></i><b>5.11.2</b> Additional Examples and Documents on FAIR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="6" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i><b>6</b> Machine Learning and Predictive Modeling for Environmental Health</a><ul>
<li class="chapter" data-level="6.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html"><i class="fa fa-check"></i><b>6.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="6.1.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#introduction-to-machine-learning-ml-and-predictive-modeling"><i class="fa fa-check"></i><b>6.1.1</b> Introduction to Machine Learning (ML) and Predictive Modeling</a></li>
<li class="chapter" data-level="6.1.2" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#defining-predictive-modeling-in-the-context-of-toxicology-and-environmental-health"><i class="fa fa-check"></i><b>6.1.2</b> Defining predictive modeling in the context of toxicology and environmental health</a></li>
<li class="chapter" data-level="6.1.3" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#distinguish-between-machine-learning-ml-and-traditional-statistical-methods"><i class="fa fa-check"></i><b>6.1.3</b> Distinguish between machine learning (ML) and traditional statistical methods</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="introduction-to-training-module-3.html"><a href="introduction-to-training-module-3.html"><i class="fa fa-check"></i><b>6.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="6.3" data-path="training-modules-environmental-health-questions.html"><a href="training-modules-environmental-health-questions.html"><i class="fa fa-check"></i><b>6.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="6.4" data-path="script-preparations-2.html"><a href="script-preparations-2.html"><i class="fa fa-check"></i><b>6.4</b> Script Preparations</a></li>
<li class="chapter" data-level="6.5" data-path="loading-the-example-dataset.html"><a href="loading-the-example-dataset.html"><i class="fa fa-check"></i><b>6.5</b> Loading the Example Dataset</a></li>
<li class="chapter" data-level="6.6" data-path="data-viewing.html"><a href="data-viewing.html"><i class="fa fa-check"></i><b>6.6</b> Data Viewing</a></li>
<li class="chapter" data-level="6.7" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html"><i class="fa fa-check"></i><b>6.7</b> Evaluating the Original Physicochemical Property Data across Substances</a><ul>
<li class="chapter" data-level="6.7.1" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html#with-these-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>6.7.1</b> With these, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html"><i class="fa fa-check"></i><b>6.8</b> Identifying Clusters of Chemicals through k-means</a><ul>
<li class="chapter" data-level="6.8.1" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html#with-this-we-can-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>6.8.1</b> With this, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html"><i class="fa fa-check"></i><b>6.9</b> Principal Component Analysis (PCA)</a><ul>
<li class="chapter" data-level="6.9.1" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-these-results-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>6.9.1</b> With these results, we can answer <strong>Environmental Health Question #3</strong>:</a></li>
<li class="chapter" data-level="6.9.2" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>6.9.2</b> With this, we can answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="6.9.3" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-5"><i class="fa fa-check"></i><b>6.9.3</b> With this, we can answer <strong>Environmental Health Question #5</strong>:</a></li>
<li class="chapter" data-level="6.9.4" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#we-can-also-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>6.9.4</b> We can also answer <strong>Environmental Health Question #6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="concluding-remarks-5.html"><a href="concluding-remarks-5.html"><i class="fa fa-check"></i><b>6.10</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html"><i class="fa fa-check"></i><b>7</b> Mixtures Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html"><i class="fa fa-check"></i><b>7.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="7.1.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html#introduction-to-mixtures-toxicology-and-in-silico-modeling-to-address-mixtures"><i class="fa fa-check"></i><b>7.1.1</b> Introduction to Mixtures Toxicology and <em>In Silico</em> Modeling to Address Mixtures</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="introduction-to-training-module-4.html"><a href="introduction-to-training-module-4.html"><i class="fa fa-check"></i><b>7.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="7.3" data-path="training-modules-environmental-health-questions-1.html"><a href="training-modules-environmental-health-questions-1.html"><i class="fa fa-check"></i><b>7.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="7.4" data-path="script-preparations-3.html"><a href="script-preparations-3.html"><i class="fa fa-check"></i><b>7.4</b> Script Preparations</a></li>
<li class="chapter" data-level="7.5" data-path="loading-the-example-ginkgo-biloba-datasets.html"><a href="loading-the-example-ginkgo-biloba-datasets.html"><i class="fa fa-check"></i><b>7.5</b> Loading the Example <em>Ginkgo biloba</em> Datasets</a></li>
<li class="chapter" data-level="7.6" data-path="data-viewing-1.html"><a href="data-viewing-1.html"><i class="fa fa-check"></i><b>7.6</b> Data Viewing</a><ul>
<li class="chapter" data-level="7.6.1" data-path="data-viewing-1.html"><a href="data-viewing-1.html#lets-also-see-what-kind-of-data-are-organized-within-the-datasets"><i class="fa fa-check"></i><b>7.6.1</b> Let’s also see what kind of data are organized within the datasets</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>7.7</b> Chemistry-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="7.7.1" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-chemical-profiles"><i class="fa fa-check"></i><b>7.7.1</b> This plot tells us a lot about sample groupings based on chemical profiles!</a></li>
<li class="chapter" data-level="7.7.2" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-questions-1-2"><i class="fa fa-check"></i><b>7.7.2</b> With this, we can answer <strong>Environmental Health Questions 1-2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><i class="fa fa-check"></i><b>7.8</b> Creating a Heat Map of the Mixtures Chemistry Data</a><ul>
<li class="chapter" data-level="7.8.1" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#this-plot-tells-us-a-lot-about-the-individual-chemicals-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>7.8.1</b> This plot tells us a lot about the individual chemicals that differentiate the sample groupings</a></li>
<li class="chapter" data-level="7.8.2" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-this-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>7.8.2</b> With this, we can answer <strong>Environmental Health Question 3</strong>:</a></li>
<li class="chapter" data-level="7.8.3" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#lets-now-revisit-the-pca-plot"><i class="fa fa-check"></i><b>7.8.3</b> Let’s now revisit the PCA plot</a></li>
<li class="chapter" data-level="7.8.4" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-these-plots-side-by-side-we-can-now-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>7.8.4</b> With these plots, side-by-side, we can now answer <strong>Environmental Health Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>7.9</b> Toxicity-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="7.9.1" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-toxicity-profiles"><i class="fa fa-check"></i><b>7.9.1</b> This plot tells us a lot about sample groupings based on toxicity profiles!</a></li>
<li class="chapter" data-level="7.9.2" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-question-5-1"><i class="fa fa-check"></i><b>7.9.2</b> With this, we can answer <em>Environmental Health Question 5</em>:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><i class="fa fa-check"></i><b>7.10</b> Creating a Heat Map of the Mixtures Toxicity Data</a><ul>
<li class="chapter" data-level="7.10.1" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#this-plot-tells-us-a-lot-about-the-individual-genes-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>7.10.1</b> This plot tells us a lot about the individual genes that differentiate the sample groupings</a></li>
<li class="chapter" data-level="7.10.2" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#with-this-we-can-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>7.10.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><i class="fa fa-check"></i><b>7.11</b> Comparing Results from the Chemistry vs. Toxicity Sufficient Similarity Analyses</a><ul>
<li class="chapter" data-level="7.11.1" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html#visual-representation-of-some-of-the-major-grouping-similarities-vs-differences"><i class="fa fa-check"></i><b>7.11.1</b> Visual representation of some of the major grouping similarities vs differences</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="these-plots-can-help-us-answer-environmental-health-question-7.html"><a href="these-plots-can-help-us-answer-environmental-health-question-7.html"><i class="fa fa-check"></i><b>7.12</b> These plots can help us answer <strong>Environmental Health Question 7</strong>:</a></li>
<li class="chapter" data-level="7.13" data-path="concluding-remarks-6.html"><a href="concluding-remarks-6.html"><i class="fa fa-check"></i><b>7.13</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html"><i class="fa fa-check"></i><b>8</b> -Omics Analyses and Systems Biology</a><ul>
<li class="chapter" data-level="8.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html"><i class="fa fa-check"></i><b>8.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="8.1.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i><b>8.1.1</b> Introduction to the Field of “-Omics”</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="introduction-to-the-omics-endpoint-transcriptomics.html"><a href="introduction-to-the-omics-endpoint-transcriptomics.html"><i class="fa fa-check"></i><b>8.2</b> Introduction to the -Omics Endpoint, Transcriptomics</a></li>
<li class="chapter" data-level="8.3" data-path="introduction-to-training-module-5.html"><a href="introduction-to-training-module-5.html"><i class="fa fa-check"></i><b>8.3</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="8.4" data-path="training-modules-environmental-health-questions-2.html"><a href="training-modules-environmental-health-questions-2.html"><i class="fa fa-check"></i><b>8.4</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="8.5" data-path="script-preparations-4.html"><a href="script-preparations-4.html"><i class="fa fa-check"></i><b>8.5</b> Script Preparations</a></li>
<li class="chapter" data-level="8.6" data-path="loading-the-example-transcriptomic-dataset-and-metadata.html"><a href="loading-the-example-transcriptomic-dataset-and-metadata.html"><i class="fa fa-check"></i><b>8.6</b> Loading the Example Transcriptomic Dataset and Metadata</a></li>
<li class="chapter" data-level="8.7" data-path="data-viewing-2.html"><a href="data-viewing-2.html"><i class="fa fa-check"></i><b>8.7</b> Data Viewing</a><ul>
<li class="chapter" data-level="8.7.1" data-path="data-viewing-2.html"><a href="data-viewing-2.html#with-this-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>8.7.1</b> With this, we can answer <strong>Environmental Health Question 1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="formatting-data-for-downstream-statistics.html"><a href="formatting-data-for-downstream-statistics.html"><i class="fa fa-check"></i><b>8.8</b> Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="8.9" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html"><i class="fa fa-check"></i><b>8.9</b> Transcriptomics Data QA/QC</a><ul>
<li class="chapter" data-level="8.9.1" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#background-filter"><i class="fa fa-check"></i><b>8.9.1</b> Background Filter</a></li>
<li class="chapter" data-level="8.9.2" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#sample-filtering"><i class="fa fa-check"></i><b>8.9.2</b> Sample Filtering</a></li>
<li class="chapter" data-level="8.9.3" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#identifying-removing-sample-outliers"><i class="fa fa-check"></i><b>8.9.3</b> Identifying &amp; Removing Sample Outliers</a></li>
<li class="chapter" data-level="8.9.4" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#with-this-we-can-answer-environmental-health-question-2-1"><i class="fa fa-check"></i><b>8.9.4</b> With this, we can answer <strong>Environmental Health Question 2</strong>:</a></li>
<li class="chapter" data-level="8.9.5" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#we-can-also-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>8.9.5</b> We can also answer <strong>Environmental Health Question 3</strong>:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html"><i class="fa fa-check"></i><b>9</b> Controlling for Sources of Sample Heterogeneity</a><ul>
<li class="chapter" data-level="9.0.1" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html#with-this-we-can-answer-environmental-health-question-4-1"><i class="fa fa-check"></i><b>9.0.1</b> With this, we can answer <strong>Environmental Health Question 4</strong>:</a></li>
<li class="chapter" data-level="9.1" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><i class="fa fa-check"></i><b>9.1</b> Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a><ul>
<li class="chapter" data-level="9.1.1" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-5-2"><i class="fa fa-check"></i><b>9.1.1</b> With this, we can answer <strong>Environmental Health Question 5</strong>:</a></li>
<li class="chapter" data-level="9.1.2" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-6-1"><i class="fa fa-check"></i><b>9.1.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
<li class="chapter" data-level="9.1.3" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-7"><i class="fa fa-check"></i><b>9.1.3</b> With this, we can answer <strong>Environmental Health Question 7</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="visualizing-statistical-results-using-ma-plots.html"><a href="visualizing-statistical-results-using-ma-plots.html"><i class="fa fa-check"></i><b>9.2</b> Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="9.3" data-path="visualizing-statistical-results-using-volcano-plots.html"><a href="visualizing-statistical-results-using-volcano-plots.html"><i class="fa fa-check"></i><b>9.3</b> Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="9.4" data-path="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><i class="fa fa-check"></i><b>9.4</b> Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a><ul>
<li class="chapter" data-level="9.4.1" data-path="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#with-this-we-can-answer-environmental-health-question-8"><i class="fa fa-check"></i><b>9.4.1</b> With this, we can answer <strong>Environmental Health Question 8</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="concluding-remarks-7.html"><a href="concluding-remarks-7.html"><i class="fa fa-check"></i><b>9.5</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i><b>10</b> Toxicokinetic Modeling</a><ul>
<li class="chapter" data-level="10.0.1" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#background-1"><i class="fa fa-check"></i><b>10.0.1</b> Background</a></li>
<li class="chapter" data-level="10.0.2" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#objectives-1"><i class="fa fa-check"></i><b>10.0.2</b> Objectives</a></li>
<li class="chapter" data-level="10.0.3" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#module-development-overview-1"><i class="fa fa-check"></i><b>10.0.3</b> Module Development Overview</a></li>
<li class="chapter" data-level="10.0.4" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#concluding-remarks-8"><i class="fa fa-check"></i><b>10.0.4</b> Concluding Remarks</a></li>
<li class="chapter" data-level="10.0.5" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#funding-1"><i class="fa fa-check"></i><b>10.0.5</b> Funding</a></li>
<li class="chapter" data-level="10.0.6" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#author-affiliations-1"><i class="fa fa-check"></i><b>10.0.6</b> Author Affiliations</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure" class="section level2">
<h2><span class="header-section-number">9.1</span> Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</h2>
<p>At this point, we have completed several data pre-processing, QA/QC, and additional steps to prepare our example transcriptomics data for statistical analysis. And finally, we are ready to run the overall statistical model to identify genes that are altered in expression in association with different biomass burn conditions.</p>
<p>Here we leverage the DESeq2 package to carry out these statistical comparisons. This package is now the most commonly implemented analysis pipeline used for transcriptomic data, including sequencing data as well as transcriptomic data produced via other technologies (e.g., Nanostring, Fluidigm, and other gene expression technologies). This package is extremely well-documented and we encourage trainees to leverage these resources in parallel with the current training module when carrying out their own transcriptomics analyses in R:</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">Bioconductor website</a><br />
</li>
<li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">Vignette</a><br />
</li>
<li><a href="https://bioconductor.org/packages/devel/bioc/manuals/DESeq2/man/DESeq2.pdf">Manual</a><br />
</li>
<li>Primary citation: Love MI, Huber W, Anders S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol. 2014;15(12):550. doi: 10.1186/s13059-014-0550-8. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/25516281/">25516281</a></li>
</ul>
<p>In brief, the basic calculations employed within the DESeq2 underlying algorithms include the following:</p>
<p><strong>1. Estimate size factors.</strong>
In the first step, size factors are estimated to help account for potential differences in the sequencing depth across samples. It is similar to a normalization parameter in the model.</p>
<p><strong>2. Normalize count data.</strong>
DESeq2 employs different normalization algorithms depending on the parameters selected / stage of analysis. The most commonly employed method is called the <strong>median of ratios</strong>, which takes into account sequencing depth and RNA composition, as described <a href="https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html">here</a>. Specifically, these normalized values are calculated as counts divided by sample-specific size factors determined by median ratio of gene counts relative to geometric mean per gene. DESeq2 then transforms these data using variance stabilization within the final statistical model. Because of these two steps, we prefer to export both the median of ratios normalized data as well as the variance stabilization transformed data, to save in our records and use when generating plots of expression levels for specific genes we are interested in. These steps are detailed below.</p>
<p><strong>3. Estimate dispersion.</strong>
The dispersion estimate takes into account the relationship between the variance of an observed count and its mean value. It is similar to a variance parameter. In DESeq2, dispersion is estimated using a maximum likelihood and empirical bayes approach.</p>
<p><strong>4. Fit negative binomial generalized linear model (GLM).</strong>
Finally, a negative binomial model is fit for each gene using the design formula that will be described within the proceeding code. The Wald test is performed to test if log fold changes in expression (typically calculated as log(average exposed / average unexposed)) significantly differ from zero. Statistical p-values are reported from this test and also adjusted for multiple testing using the Benjamini and Hochberg procedure.</p>
<p>Note that these calculations, among others, are embedded within the DESeq2 functions, so we do not need to code for them ourselves. Instead, we just need to make sure that we set-up the DESeq2 functions correctly, such that these calculations are carried out appropriately in our final transcriptomics analyses.</p>
<div id="setting-up-the-deseq2-experiment" class="section level4">
<h4><span class="header-section-number">9.1.0.1</span> Setting up the DESeq2 experiment</h4>
<p>Here we provide example script that is used to identify which genes are significantly differentially expressed in association with the example biomass smoke exposures, smoldering pine needles and flaming pine needles.</p>
<p>First, we need to set-up the DESeq2 experiment:</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb59-1"></a><span class="co"># Set up our experiment using our RUV adjusted count and phenotype data.</span></span>
<span id="cb59-2"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb59-2"></a><span class="co"># Our design indicates that our count data is dependent on the exposure condition (groups variable) and our factor of unwanted variation.</span></span>
<span id="cb59-3"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb59-3"></a>dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSetFromMatrix</span>(<span class="dt">countData =</span> <span class="kw">counts</span>(ruv_set), <span class="co"># grabbing the RUV adjusted count data from the &#39;ruv_set&#39; object</span></span>
<span id="cb59-4"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb59-4"></a>                                 <span class="dt">colData =</span> <span class="kw">pData</span>(ruv_set), <span class="co"># grabbing the phenotype data (metadata) from the &#39;ruv_set&#39; object</span></span>
<span id="cb59-5"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb59-5"></a>                                 <span class="dt">design =</span> <span class="op">~</span><span class="dv">0</span><span class="op">+</span>groups<span class="op">+</span>W_<span class="dv">1</span>) <span class="co"># setting up the statistical formula (see below)</span></span></code></pre></div>
<p>For the design formula, we use a ‘~0’ at the front to indicate that we will not include covariates (e.g., certain human demographic data as potential confounders) or other factors to adjust for (e.g., batch). Therefore, this analysis is a pretty simple one simply comparing differences between sample groups without adjusting for covariates/other variables. The groups are set within the ‘groups’ factor, shown here as a reminder:</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb60-1"></a>groups</span></code></pre></div>
<pre><code>##  [1] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [3] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [5] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [7] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
##  [9] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
## [11] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
## [13] Saline_4h_Lung             Saline_4h_Lung            
## [15] Saline_4h_Lung             Saline_4h_Lung            
## [17] Saline_4h_Lung             Saline_4h_Lung            
## [19] LPS_4h_Lung                LPS_4h_Lung               
## [21] LPS_4h_Lung                LPS_4h_Lung               
## 4 Levels: LPS_4h_Lung PineNeedlesFlame_4h_Lung ... Saline_4h_Lung</code></pre>
<p>Lastly, the W_1
LAUREN I CAN’T FIND THE W_1 OBJECT/WHAT THIS REFERS TO? CAN YOU PLEASE EXPLAIN HERE?</p>
</div>
<div id="estimating-size-factors" class="section level4">
<h4><span class="header-section-number">9.1.0.2</span> Estimating size factors</h4>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb62-1"></a><span class="co"># Estimate size factors from the dds object that was just created as the experiment above</span></span>
<span id="cb62-2"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb62-2"></a>dds &lt;-<span class="st"> </span><span class="kw">estimateSizeFactors</span>(dds)</span>
<span id="cb62-3"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb62-3"></a><span class="kw">sizeFactors</span>(dds)  <span class="co"># viewing the size factors</span></span></code></pre></div>
<pre><code>## M13_PineNeedlesSmolder M14_PineNeedlesSmolder M15_PineNeedlesSmolder 
##              1.2668895              1.3040762              1.1782071 
## M16_PineNeedlesSmolder M17_PineNeedlesSmolder M18_PineNeedlesSmolder 
##              1.1723749              1.2904950              1.3405500 
##   M49_PineNeedlesFlame   M50_PineNeedlesFlame   M51_PineNeedlesFlame 
##              1.1313237              0.6313099              0.9328970 
##   M52_PineNeedlesFlame   M53_PineNeedlesFlame   M54_PineNeedlesFlame 
##              1.0050414              0.7649632              0.8747854 
##             M31_Saline             M32_Saline             M33_Saline 
##              0.8050127              1.0480725              1.0389623 
##             M34_Saline             M35_Saline             M36_Saline 
##              1.0282261              0.9519266              1.0409103 
##                M67_LPS                M68_LPS                M69_LPS 
##              0.9308588              1.1002076              0.9793051 
##                M70_LPS 
##              0.8575718</code></pre>
</div>
<div id="calculating-and-exporting-normalized-counts" class="section level4">
<h4><span class="header-section-number">9.1.0.3</span> Calculating and exporting normalized counts</h4>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb64-1"></a><span class="co"># Calculating normalized count data</span></span>
<span id="cb64-2"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb64-2"></a>normcounts &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">counts</span>(dds, <span class="dt">normalized=</span><span class="ot">TRUE</span>))</span>
<span id="cb64-3"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb64-3"></a><span class="kw">write.csv</span>(normcounts, <span class="st">&quot;Module2_4_Output_NormalizedCounts.csv&quot;</span>)</span>
<span id="cb64-4"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb64-4"></a></span>
<span id="cb64-5"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb64-5"></a><span class="co"># Transforming normalized counts through variance stabilization</span></span>
<span id="cb64-6"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb64-6"></a>vsd &lt;-<span class="st"> </span><span class="kw">varianceStabilizingTransformation</span>(dds, <span class="dt">blind=</span><span class="ot">FALSE</span>)</span>
<span id="cb64-7"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb64-7"></a>vsd_matrix &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">assay</span>(vsd))</span>
<span id="cb64-8"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb64-8"></a><span class="kw">write.csv</span>(vsd_matrix, <span class="st">&quot;Module2_4_Output_VSDCounts.csv&quot;</span>, <span class="dt">row.names=</span><span class="ot">TRUE</span>)</span></code></pre></div>
</div>
<div id="running-the-final-deseq2-experiment" class="section level4">
<h4><span class="header-section-number">9.1.0.4</span> Running the final DESeq2 experiment</h4>
<p>Here, we are finally ready to run the actual statistical comparisons (exposed vs control samples) to calculate fold changes and p-values that describe the degree to which each gene may or may not be altered at the expression level in association with treatment.</p>
<p>For this example, we would like to run three different comparisons:<br />
(1) Smoldering Pine Needles vs. Control<br />
(2) Flaming Pine Needles vs. Control<br />
(3) LPS vs. Control<br />
which we can easily code for using a loop function, as detailed below.</p>
<p>LAUREN CAN WE PLEASE ADD SOME MORE DETAIL BESIDE EACH LINE OF CODE HERE AS WELL, PARTICULALRY IN THE LOOP?</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-1"></a><span class="co"># Run experiment</span></span>
<span id="cb65-2"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-2"></a>dds_run &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds, <span class="dt">betaPrior=</span><span class="ot">FALSE</span>) </span>
<span id="cb65-3"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-3"></a></span>
<span id="cb65-4"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-4"></a><span class="co"># loop through and extract and export results for all contrasts (treatments vs. control)</span></span>
<span id="cb65-5"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-5"></a><span class="cf">for</span> (trt <span class="cf">in</span> trt_groups){</span>
<span id="cb65-6"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-6"></a>  <span class="kw">cat</span>(trt)</span>
<span id="cb65-7"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-7"></a>  res &lt;-<span class="st"> </span><span class="kw">results</span>(dds_run, <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="dt">contrast =</span> <span class="kw">c</span>(<span class="st">&quot;groups&quot;</span>,trt,ctrl))</span>
<span id="cb65-8"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-8"></a>  <span class="kw">summary</span>(res)</span>
<span id="cb65-9"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-9"></a>  ordered &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(res[<span class="kw">order</span>(res<span class="op">$</span>padj),])</span>
<span id="cb65-10"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-10"></a>  <span class="kw">write.csv</span>(ordered, <span class="kw">paste0</span>(<span class="st">&quot;Module2_4_Output_StatisticalResults_&quot;</span>,trt ,<span class="st">&quot;.csv&quot;</span>))</span>
<span id="cb65-11"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#cb65-11"></a>}</span></code></pre></div>
<pre><code>## PineNeedlesSmolder_4h_Lung
## out of 16664 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 457, 2.7%
## LFC &lt; 0 (down)     : 697, 4.2%
## outliers [1]       : 0, 0%
## low counts [2]     : 1293, 7.8%
## (mean count &lt; 21)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results
## 
## PineNeedlesFlame_4h_Lung
## out of 16664 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 411, 2.5%
## LFC &lt; 0 (down)     : 575, 3.5%
## outliers [1]       : 0, 0%
## low counts [2]     : 0, 0%
## (mean count &lt; 10)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results
## 
## LPS_4h_Lung
## out of 16664 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 2960, 18%
## LFC &lt; 0 (down)     : 2947, 18%
## outliers [1]       : 0, 0%
## low counts [2]     : 0, 0%
## (mean count &lt; 10)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p>LAUREN-
WE NEED TO ADD SOME CODE VIEWING THE TOP OF THE STATISTICAL RESULTS FILES HERE. THIS MAY MEAN ADDING SCRIPT WITHIN THE ABOVE LOOP FUNCTION TO PULL EACH SET OF RESULTS INTO A DATAFRAME, SUCH THAT WE CAN THEN VIEW THE STATISTICAL OUTPUTS WITHIN THE SCRIPT (IN ADDITION TO EXPORTING THEM)</p>
<p>WE ALSO NEED TO GRAB THE NUMBER OF GENES THAT PASS PADJ&lt;0.05 &amp; PADJ &lt; 0.10 IN EACH OF THESE RESULTING OUTPUTS, SO WE CAN ANSWER:</p>
</div>
<div id="with-this-we-can-answer-environmental-health-question-5-2" class="section level3">
<h3><span class="header-section-number">9.1.1</span> With this, we can answer <strong>Environmental Health Question 5</strong>:</h3>
<div id="how-many-genes-showed-significant-differential-expression-associated-with-flaming-pine-needles-exposure-in-the-mouse-lung-based-on-two-filters-multiple-test-corrected-p-value-padj-0.05-padj-0.10" class="section level4">
<h4><span class="header-section-number">9.1.1.1</span> How many genes showed significant differential expression associated with flaming pine needles exposure in the mouse lung, based on two filters: multiple test corrected p-value (padj) &lt; 0.05; &amp; padj &lt; 0.10?</h4>
</div>
<div id="answer-x-genes-meet-the-padj-0.05-filter-and-x-genes-meet-the-padj-0.10-filter." class="section level4">
<h4><span class="header-section-number">9.1.1.2</span> <em>Answer: X genes meet the padj &lt;0.05 filter and X genes meet the padj &lt; 0.10 filter.</em></h4>
</div>
</div>
<div id="with-this-we-can-answer-environmental-health-question-6-1" class="section level3">
<h3><span class="header-section-number">9.1.2</span> With this, we can answer <strong>Environmental Health Question 6</strong>:</h3>
<div id="how-many-genes-showed-significant-differential-expression-associated-with-smoldering-pine-needles-exposure-in-the-mouse-lung-based-on-two-filters-multiple-test-corrected-p-value-padj-0.05-padj-0.10" class="section level4">
<h4><span class="header-section-number">9.1.2.1</span> How many genes showed significant differential expression associated with smoldering pine needles exposure in the mouse lung, based on two filters: multiple test corrected p-value (padj) &lt; 0.05; &amp; padj &lt; 0.10?</h4>
</div>
<div id="answer-x-genes." class="section level4">
<h4><span class="header-section-number">9.1.2.2</span> <em>Answer: X genes.</em></h4>
</div>
</div>
<div id="with-this-we-can-answer-environmental-health-question-7" class="section level3">
<h3><span class="header-section-number">9.1.3</span> With this, we can answer <strong>Environmental Health Question 7</strong>:</h3>
<div id="how-many-genes-showed-significant-differential-expression-associated-with-lipopolysaccharide-lps-exposure-in-the-mouse-lung-based-on-two-filters-multiple-test-corrected-p-value-padj-0.05-padj-0.10" class="section level4">
<h4><span class="header-section-number">9.1.3.1</span> How many genes showed significant differential expression associated with lipopolysaccharide (LPS) exposure in the mouse lung, based on two filters: multiple test corrected p-value (padj) &lt; 0.05; &amp; padj &lt; 0.10?</h4>
</div>
<div id="answer-x-genes.-1" class="section level4">
<h4><span class="header-section-number">9.1.3.2</span> <em>Answer: X genes.</em></h4>
<ol start="8" style="list-style-type: decimal">
<li>What biological pathways are disrupted in association with flaming/smoldering pine needles exposure in the lung, identified through systems level analyses?</li>
</ol>
<p><br></p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="controlling-for-sources-of-sample-heterogeneity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualizing-statistical-results-using-ma-plots.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
