<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9.4 Interpretting Findings at the Systems Level through Pathway Enrichment Analysis | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="9.4 Interpretting Findings at the Systems Level through Pathway Enrichment Analysis | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="9.4 Interpretting Findings at the Systems Level through Pathway Enrichment Analysis | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9.4 Interpretting Findings at the Systems Level through Pathway Enrichment Analysis | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle Roell, Lauren Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visualizing-statistical-results-using-volcano-plots.html"/>
<link rel="next" href="concluding-remarks-7.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TAME Toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#background"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#objectives"><i class="fa fa-check"></i>Objectives</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#module-development-overview"><i class="fa fa-check"></i>Module Development Overview</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#concluding-remarks"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#funding"><i class="fa fa-check"></i>Funding</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#author-affiliations"><i class="fa fa-check"></i>Author Affiliations</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html"><i class="fa fa-check"></i><b>1</b> Introduction to Coding in R</a><ul>
<li class="chapter" data-level="1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html"><i class="fa fa-check"></i><b>1.1</b> General Introduction and Installation of R and RStudio</a><ul>
<li class="chapter" data-level="1.1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html#what-is-r"><i class="fa fa-check"></i><b>1.1.1</b> What is R?</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="downloading-and-installing-r.html"><a href="downloading-and-installing-r.html"><i class="fa fa-check"></i><b>1.2</b> Downloading and Installing R</a></li>
<li class="chapter" data-level="1.3" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html"><i class="fa fa-check"></i><b>1.3</b> Downloading and Installing R Studio</a></li>
<li class="chapter" data-level="1.4" data-path="introduction-to-r-packages.html"><a href="introduction-to-r-packages.html"><i class="fa fa-check"></i><b>1.4</b> Introduction to R Packages</a><ul>
<li class="chapter" data-level="1.4.1" data-path="introduction-to-r-packages.html"><a href="introduction-to-r-packages.html#downloadinginstalling-r-packages"><i class="fa fa-check"></i><b>1.4.1</b> Downloading/Installing R Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="scripting-basics.html"><a href="scripting-basics.html"><i class="fa fa-check"></i><b>1.5</b> Scripting Basics</a><ul>
<li class="chapter" data-level="1.5.1" data-path="scripting-basics.html"><a href="scripting-basics.html#comments"><i class="fa fa-check"></i><b>1.5.1</b> Comments</a></li>
<li class="chapter" data-level="1.5.2" data-path="scripting-basics.html"><a href="scripting-basics.html#starting-code"><i class="fa fa-check"></i><b>1.5.2</b> Starting Code</a></li>
<li class="chapter" data-level="1.5.3" data-path="scripting-basics.html"><a href="scripting-basics.html#setting-your-working-directory"><i class="fa fa-check"></i><b>1.5.3</b> Setting Your Working Directory</a></li>
<li class="chapter" data-level="1.5.4" data-path="scripting-basics.html"><a href="scripting-basics.html#importing-files"><i class="fa fa-check"></i><b>1.5.4</b> Importing Files</a></li>
<li class="chapter" data-level="1.5.5" data-path="scripting-basics.html"><a href="scripting-basics.html#viewing-data"><i class="fa fa-check"></i><b>1.5.5</b> Viewing Data</a></li>
<li class="chapter" data-level="1.5.6" data-path="scripting-basics.html"><a href="scripting-basics.html#exporting-data"><i class="fa fa-check"></i><b>1.5.6</b> Exporting Data</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="concluding-remarks-1.html"><a href="concluding-remarks-1.html"><i class="fa fa-check"></i><b>1.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-organization-basics.html"><a href="data-organization-basics.html"><i class="fa fa-check"></i><b>2</b> Data Organization Basics</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html"><i class="fa fa-check"></i><b>2.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="2.1.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#set-your-working-directory-1"><i class="fa fa-check"></i><b>2.1.1</b> Set your working directory</a></li>
<li class="chapter" data-level="2.1.2" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#importing-example-datasets"><i class="fa fa-check"></i><b>2.1.2</b> Importing example datasets</a></li>
<li class="chapter" data-level="2.1.3" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#viewing-example-datasets"><i class="fa fa-check"></i><b>2.1.3</b> Viewing example datasets</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="merging-data-using-base-r-syntax.html"><a href="merging-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.2</b> Merging Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.3" data-path="filtering-and-subsetting-data-using-base-r-syntax.html"><a href="filtering-and-subsetting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.3</b> Filtering and Subsetting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.4" data-path="melting-and-casting-data-using-base-r-syntax.html"><a href="melting-and-casting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.4</b> Melting and Casting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.5" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html"><i class="fa fa-check"></i><b>2.5</b> Introduction to Tidyverse</a><ul>
<li class="chapter" data-level="2.5.1" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#downloading-and-loading-the-tidyverse-package"><i class="fa fa-check"></i><b>2.5.1</b> Downloading and Loading the Tidyverse Package</a></li>
<li class="chapter" data-level="2.5.2" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#merging-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.2</b> Merging Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.3" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#filtering-and-subsetting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.3</b> Filtering and Subsetting Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.4" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#melting-and-casting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.4</b> Melting and Casting Data using Tidyverse Syntax</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="concluding-remarks-2.html"><a href="concluding-remarks-2.html"><i class="fa fa-check"></i><b>2.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html"><i class="fa fa-check"></i><b>3</b> Finding and Visualizing Data Trend</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html"><i class="fa fa-check"></i><b>3.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="3.1.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html#script-preparations"><i class="fa fa-check"></i><b>3.1.1</b> Script Preparations</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html"><i class="fa fa-check"></i><b>3.2</b> Basic Statistical Tests and Visualizations of Data Trends</a><ul>
<li class="chapter" data-level="3.2.1" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#visualize-and-test-data-for-normality"><i class="fa fa-check"></i><b>3.2.1</b> Visualize and Test Data for Normality</a></li>
<li class="chapter" data-level="3.2.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i><b>3.2.2</b> Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="3.2.3" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i><b>3.2.3</b> Two-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><i class="fa fa-check"></i><b>3.3</b> Regression Modeling and Visualization: Linear and Logistic Regressions</a><ul>
<li class="chapter" data-level="3.3.1" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#linear-regression"><i class="fa fa-check"></i><b>3.3.1</b> Linear Regression</a></li>
<li class="chapter" data-level="3.3.2" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#logistic-regression"><i class="fa fa-check"></i><b>3.3.2</b> Logistic Regression</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="concluding-remarks-3.html"><a href="concluding-remarks-3.html"><i class="fa fa-check"></i><b>3.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html"><i class="fa fa-check"></i><b>4</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-training-module-2.html"><a href="introduction-to-training-module-2.html"><i class="fa fa-check"></i><b>4.1</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="4.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html"><i class="fa fa-check"></i><b>4.2</b> Script Preparations</a><ul>
<li class="chapter" data-level="4.2.1" data-path="script-preparations-1.html"><a href="script-preparations-1.html#formatting-dataframes-for-downstream-visualization-code"><i class="fa fa-check"></i><b>4.2.1</b> Formatting dataframes for downstream visualization code</a></li>
<li class="chapter" data-level="4.2.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html#scaling-dataframes-for-downstream-data-visualizations"><i class="fa fa-check"></i><b>4.2.2</b> Scaling dataframes for downstream data visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html"><i class="fa fa-check"></i><b>4.3</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.3.1" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#density-plot-visualizations"><i class="fa fa-check"></i><b>4.3.1</b> Density Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.2" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#ggally-visualizations"><i class="fa fa-check"></i><b>4.3.2</b> GGally Visualizations</a></li>
<li class="chapter" data-level="4.3.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#boxplot-visualizations"><i class="fa fa-check"></i><b>4.3.3</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="4.3.4" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#correlation-plot-visualizations"><i class="fa fa-check"></i><b>4.3.4</b> Correlation Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.5" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#hierarchical-clustering-visualizations"><i class="fa fa-check"></i><b>4.3.5</b> Hierarchical Clustering Visualizations</a></li>
<li class="chapter" data-level="4.3.6" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#heatmap-visualizations"><i class="fa fa-check"></i><b>4.3.6</b> Heatmap Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="concluding-remarks-4.html"><a href="concluding-remarks-4.html"><i class="fa fa-check"></i><b>4.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i><b>5</b> FAIR Data Management Practices</a><ul>
<li class="chapter" data-level="5.1" data-path="background-on-training-module.html"><a href="background-on-training-module.html"><i class="fa fa-check"></i><b>5.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="5.2" data-path="training-modules-questions.html"><a href="training-modules-questions.html"><i class="fa fa-check"></i><b>5.2</b> Training Module’s <strong>Questions</strong></a></li>
<li class="chapter" data-level="5.3" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html"><i class="fa fa-check"></i><b>5.3</b> Introduction to FAIR</a><ul>
<li class="chapter" data-level="5.3.1" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#with-this-background-we-can-now-answer-question-1"><i class="fa fa-check"></i><b>5.3.1</b> With this background, we can now answer <strong>Question 1</strong>:</a></li>
<li class="chapter" data-level="5.3.2" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#we-can-also-answer-question-2"><i class="fa fa-check"></i><b>5.3.2</b> We can also answer <strong>Question 2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="breaking-down-fair-letter-by-letter.html"><a href="breaking-down-fair-letter-by-letter.html"><i class="fa fa-check"></i><b>5.4</b> Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="5.5" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html"><i class="fa fa-check"></i><b>5.5</b> F (Findable) in FAIR</a><ul>
<li class="chapter" data-level="5.5.1" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html#with-this-information-can-can-now-answer-question-3"><i class="fa fa-check"></i><b>5.5.1</b> With this information, can can now answer <strong>Question 3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="a-accessible-in-fair.html"><a href="a-accessible-in-fair.html"><i class="fa fa-check"></i><b>5.6</b> A (Accessible) in FAIR</a></li>
<li class="chapter" data-level="5.7" data-path="i-interoperable-in-fair.html"><a href="i-interoperable-in-fair.html"><i class="fa fa-check"></i><b>5.7</b> I (Interoperable) in FAIR</a></li>
<li class="chapter" data-level="5.8" data-path="r-reusable-in-fair.html"><a href="r-reusable-in-fair.html"><i class="fa fa-check"></i><b>5.8</b> R (Reusable) in FAIR</a></li>
<li class="chapter" data-level="5.9" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html"><i class="fa fa-check"></i><b>5.9</b> What does this mean to you?</a><ul>
<li class="chapter" data-level="5.9.1" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html#with-this-information-can-can-now-answer-question-4"><i class="fa fa-check"></i><b>5.9.1</b> With this information, can can now answer <strong>Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.10" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html"><i class="fa fa-check"></i><b>5.10</b> Data Repositories for Sharing of Data</a><ul>
<li class="chapter" data-level="5.10.1" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html#with-this-information-can-can-now-answer-question-5"><i class="fa fa-check"></i><b>5.10.1</b> With this information, can can now answer <strong>Question 5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html"><i class="fa fa-check"></i><b>5.11</b> Helpful Resources on FAIR</a><ul>
<li class="chapter" data-level="5.11.1" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-training-resources-on-fair"><i class="fa fa-check"></i><b>5.11.1</b> Additional Training Resources on FAIR</a></li>
<li class="chapter" data-level="5.11.2" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-examples-and-documents-on-fair"><i class="fa fa-check"></i><b>5.11.2</b> Additional Examples and Documents on FAIR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="6" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i><b>6</b> Machine Learning and Predictive Modeling for Environmental Health</a><ul>
<li class="chapter" data-level="6.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html"><i class="fa fa-check"></i><b>6.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="6.1.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#introduction-to-machine-learning-ml-and-predictive-modeling"><i class="fa fa-check"></i><b>6.1.1</b> Introduction to Machine Learning (ML) and Predictive Modeling</a></li>
<li class="chapter" data-level="6.1.2" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#defining-predictive-modeling-in-the-context-of-toxicology-and-environmental-health"><i class="fa fa-check"></i><b>6.1.2</b> Defining predictive modeling in the context of toxicology and environmental health</a></li>
<li class="chapter" data-level="6.1.3" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#distinguish-between-machine-learning-ml-and-traditional-statistical-methods"><i class="fa fa-check"></i><b>6.1.3</b> Distinguish between machine learning (ML) and traditional statistical methods</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="introduction-to-training-module-3.html"><a href="introduction-to-training-module-3.html"><i class="fa fa-check"></i><b>6.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="6.3" data-path="training-modules-environmental-health-questions.html"><a href="training-modules-environmental-health-questions.html"><i class="fa fa-check"></i><b>6.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="6.4" data-path="script-preparations-2.html"><a href="script-preparations-2.html"><i class="fa fa-check"></i><b>6.4</b> Script Preparations</a></li>
<li class="chapter" data-level="6.5" data-path="loading-the-example-dataset.html"><a href="loading-the-example-dataset.html"><i class="fa fa-check"></i><b>6.5</b> Loading the Example Dataset</a></li>
<li class="chapter" data-level="6.6" data-path="data-viewing.html"><a href="data-viewing.html"><i class="fa fa-check"></i><b>6.6</b> Data Viewing</a></li>
<li class="chapter" data-level="6.7" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html"><i class="fa fa-check"></i><b>6.7</b> Evaluating the Original Physicochemical Property Data across Substances</a><ul>
<li class="chapter" data-level="6.7.1" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html#with-these-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>6.7.1</b> With these, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html"><i class="fa fa-check"></i><b>6.8</b> Identifying Clusters of Chemicals through k-means</a><ul>
<li class="chapter" data-level="6.8.1" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html#with-this-we-can-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>6.8.1</b> With this, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html"><i class="fa fa-check"></i><b>6.9</b> Principal Component Analysis (PCA)</a><ul>
<li class="chapter" data-level="6.9.1" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-these-results-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>6.9.1</b> With these results, we can answer <strong>Environmental Health Question #3</strong>:</a></li>
<li class="chapter" data-level="6.9.2" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>6.9.2</b> With this, we can answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="6.9.3" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-5"><i class="fa fa-check"></i><b>6.9.3</b> With this, we can answer <strong>Environmental Health Question #5</strong>:</a></li>
<li class="chapter" data-level="6.9.4" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#we-can-also-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>6.9.4</b> We can also answer <strong>Environmental Health Question #6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="concluding-remarks-5.html"><a href="concluding-remarks-5.html"><i class="fa fa-check"></i><b>6.10</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html"><i class="fa fa-check"></i><b>7</b> Mixtures Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html"><i class="fa fa-check"></i><b>7.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="7.1.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html#introduction-to-mixtures-toxicology-and-in-silico-modeling-to-address-mixtures"><i class="fa fa-check"></i><b>7.1.1</b> Introduction to Mixtures Toxicology and <em>In Silico</em> Modeling to Address Mixtures</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="introduction-to-training-module-4.html"><a href="introduction-to-training-module-4.html"><i class="fa fa-check"></i><b>7.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="7.3" data-path="training-modules-environmental-health-questions-1.html"><a href="training-modules-environmental-health-questions-1.html"><i class="fa fa-check"></i><b>7.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="7.4" data-path="script-preparations-3.html"><a href="script-preparations-3.html"><i class="fa fa-check"></i><b>7.4</b> Script Preparations</a></li>
<li class="chapter" data-level="7.5" data-path="loading-the-example-ginkgo-biloba-datasets.html"><a href="loading-the-example-ginkgo-biloba-datasets.html"><i class="fa fa-check"></i><b>7.5</b> Loading the Example <em>Ginkgo biloba</em> Datasets</a></li>
<li class="chapter" data-level="7.6" data-path="data-viewing-1.html"><a href="data-viewing-1.html"><i class="fa fa-check"></i><b>7.6</b> Data Viewing</a><ul>
<li class="chapter" data-level="7.6.1" data-path="data-viewing-1.html"><a href="data-viewing-1.html#lets-also-see-what-kind-of-data-are-organized-within-the-datasets"><i class="fa fa-check"></i><b>7.6.1</b> Let’s also see what kind of data are organized within the datasets</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>7.7</b> Chemistry-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="7.7.1" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-chemical-profiles"><i class="fa fa-check"></i><b>7.7.1</b> This plot tells us a lot about sample groupings based on chemical profiles!</a></li>
<li class="chapter" data-level="7.7.2" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-questions-1-2"><i class="fa fa-check"></i><b>7.7.2</b> With this, we can answer <strong>Environmental Health Questions 1-2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><i class="fa fa-check"></i><b>7.8</b> Creating a Heat Map of the Mixtures Chemistry Data</a><ul>
<li class="chapter" data-level="7.8.1" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#this-plot-tells-us-a-lot-about-the-individual-chemicals-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>7.8.1</b> This plot tells us a lot about the individual chemicals that differentiate the sample groupings</a></li>
<li class="chapter" data-level="7.8.2" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-this-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>7.8.2</b> With this, we can answer <strong>Environmental Health Question 3</strong>:</a></li>
<li class="chapter" data-level="7.8.3" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#lets-now-revisit-the-pca-plot"><i class="fa fa-check"></i><b>7.8.3</b> Let’s now revisit the PCA plot</a></li>
<li class="chapter" data-level="7.8.4" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-these-plots-side-by-side-we-can-now-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>7.8.4</b> With these plots, side-by-side, we can now answer <strong>Environmental Health Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>7.9</b> Toxicity-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="7.9.1" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-toxicity-profiles"><i class="fa fa-check"></i><b>7.9.1</b> This plot tells us a lot about sample groupings based on toxicity profiles!</a></li>
<li class="chapter" data-level="7.9.2" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-question-5-1"><i class="fa fa-check"></i><b>7.9.2</b> With this, we can answer <em>Environmental Health Question 5</em>:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><i class="fa fa-check"></i><b>7.10</b> Creating a Heat Map of the Mixtures Toxicity Data</a><ul>
<li class="chapter" data-level="7.10.1" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#this-plot-tells-us-a-lot-about-the-individual-genes-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>7.10.1</b> This plot tells us a lot about the individual genes that differentiate the sample groupings</a></li>
<li class="chapter" data-level="7.10.2" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#with-this-we-can-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>7.10.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><i class="fa fa-check"></i><b>7.11</b> Comparing Results from the Chemistry vs. Toxicity Sufficient Similarity Analyses</a><ul>
<li class="chapter" data-level="7.11.1" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html#visual-representation-of-some-of-the-major-grouping-similarities-vs-differences"><i class="fa fa-check"></i><b>7.11.1</b> Visual representation of some of the major grouping similarities vs differences</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="these-plots-can-help-us-answer-environmental-health-question-7.html"><a href="these-plots-can-help-us-answer-environmental-health-question-7.html"><i class="fa fa-check"></i><b>7.12</b> These plots can help us answer <strong>Environmental Health Question 7</strong>:</a></li>
<li class="chapter" data-level="7.13" data-path="concluding-remarks-6.html"><a href="concluding-remarks-6.html"><i class="fa fa-check"></i><b>7.13</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html"><i class="fa fa-check"></i><b>8</b> -Omics Analyses and Systems Biology</a><ul>
<li class="chapter" data-level="8.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html"><i class="fa fa-check"></i><b>8.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="8.1.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i><b>8.1.1</b> Introduction to the Field of “-Omics”</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="introduction-to-the-omics-endpoint-transcriptomics.html"><a href="introduction-to-the-omics-endpoint-transcriptomics.html"><i class="fa fa-check"></i><b>8.2</b> Introduction to the -Omics Endpoint, Transcriptomics</a></li>
<li class="chapter" data-level="8.3" data-path="introduction-to-training-module-5.html"><a href="introduction-to-training-module-5.html"><i class="fa fa-check"></i><b>8.3</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="8.4" data-path="training-modules-environmental-health-questions-2.html"><a href="training-modules-environmental-health-questions-2.html"><i class="fa fa-check"></i><b>8.4</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="8.5" data-path="script-preparations-4.html"><a href="script-preparations-4.html"><i class="fa fa-check"></i><b>8.5</b> Script Preparations</a></li>
<li class="chapter" data-level="8.6" data-path="loading-the-example-transcriptomic-dataset-and-metadata.html"><a href="loading-the-example-transcriptomic-dataset-and-metadata.html"><i class="fa fa-check"></i><b>8.6</b> Loading the Example Transcriptomic Dataset and Metadata</a></li>
<li class="chapter" data-level="8.7" data-path="data-viewing-2.html"><a href="data-viewing-2.html"><i class="fa fa-check"></i><b>8.7</b> Data Viewing</a><ul>
<li class="chapter" data-level="8.7.1" data-path="data-viewing-2.html"><a href="data-viewing-2.html#with-this-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>8.7.1</b> With this, we can answer <strong>Environmental Health Question 1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="formatting-data-for-downstream-statistics.html"><a href="formatting-data-for-downstream-statistics.html"><i class="fa fa-check"></i><b>8.8</b> Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="8.9" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html"><i class="fa fa-check"></i><b>8.9</b> Transcriptomics Data QA/QC</a><ul>
<li class="chapter" data-level="8.9.1" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#background-filter"><i class="fa fa-check"></i><b>8.9.1</b> Background Filter</a></li>
<li class="chapter" data-level="8.9.2" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#sample-filtering"><i class="fa fa-check"></i><b>8.9.2</b> Sample Filtering</a></li>
<li class="chapter" data-level="8.9.3" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#identifying-removing-sample-outliers"><i class="fa fa-check"></i><b>8.9.3</b> Identifying &amp; Removing Sample Outliers</a></li>
<li class="chapter" data-level="8.9.4" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#with-this-we-can-answer-environmental-health-question-2-1"><i class="fa fa-check"></i><b>8.9.4</b> With this, we can answer <strong>Environmental Health Question 2</strong>:</a></li>
<li class="chapter" data-level="8.9.5" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#we-can-also-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>8.9.5</b> We can also answer <strong>Environmental Health Question 3</strong>:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html"><i class="fa fa-check"></i><b>9</b> Controlling for Sources of Sample Heterogeneity</a><ul>
<li class="chapter" data-level="9.0.1" data-path="controlling-for-sources-of-sample-heterogeneity.html"><a href="controlling-for-sources-of-sample-heterogeneity.html#with-this-we-can-answer-environmental-health-question-4-1"><i class="fa fa-check"></i><b>9.0.1</b> With this, we can answer <strong>Environmental Health Question 4</strong>:</a></li>
<li class="chapter" data-level="9.1" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><i class="fa fa-check"></i><b>9.1</b> Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a><ul>
<li class="chapter" data-level="9.1.1" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-5-2"><i class="fa fa-check"></i><b>9.1.1</b> With this, we can answer <strong>Environmental Health Question 5</strong>:</a></li>
<li class="chapter" data-level="9.1.2" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-6-1"><i class="fa fa-check"></i><b>9.1.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
<li class="chapter" data-level="9.1.3" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-7"><i class="fa fa-check"></i><b>9.1.3</b> With this, we can answer <strong>Environmental Health Question 7</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="visualizing-statistical-results-using-ma-plots.html"><a href="visualizing-statistical-results-using-ma-plots.html"><i class="fa fa-check"></i><b>9.2</b> Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="9.3" data-path="visualizing-statistical-results-using-volcano-plots.html"><a href="visualizing-statistical-results-using-volcano-plots.html"><i class="fa fa-check"></i><b>9.3</b> Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="9.4" data-path="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><i class="fa fa-check"></i><b>9.4</b> Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a><ul>
<li class="chapter" data-level="9.4.1" data-path="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#with-this-we-can-answer-environmental-health-question-8"><i class="fa fa-check"></i><b>9.4.1</b> With this, we can answer <strong>Environmental Health Question 8</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="concluding-remarks-7.html"><a href="concluding-remarks-7.html"><i class="fa fa-check"></i><b>9.5</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i><b>10</b> Toxicokinetic Modeling</a><ul>
<li class="chapter" data-level="10.0.1" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#background-1"><i class="fa fa-check"></i><b>10.0.1</b> Background</a></li>
<li class="chapter" data-level="10.0.2" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#objectives-1"><i class="fa fa-check"></i><b>10.0.2</b> Objectives</a></li>
<li class="chapter" data-level="10.0.3" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#module-development-overview-1"><i class="fa fa-check"></i><b>10.0.3</b> Module Development Overview</a></li>
<li class="chapter" data-level="10.0.4" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#concluding-remarks-8"><i class="fa fa-check"></i><b>10.0.4</b> Concluding Remarks</a></li>
<li class="chapter" data-level="10.0.5" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#funding-1"><i class="fa fa-check"></i><b>10.0.5</b> Funding</a></li>
<li class="chapter" data-level="10.0.6" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#author-affiliations-1"><i class="fa fa-check"></i><b>10.0.6</b> Author Affiliations</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis" class="section level2">
<h2><span class="header-section-number">9.4</span> Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</h2>
<p>Pathway enrichment analysis is a very helpful tool that can be applied to interpret transcriptomic changes of interest in terms of systems biology. In these types of analyses, gene lists of interest are used to identify biological pathways that include genes present in your dataset more often than expected by chance alone. There are many tools that can be used to carry out pathway enrichment analyses. Here, we are using the R package, PIANO, to carry out the statistical enrichment analysis based on the lists of genes we previously identified with differential expression associated with flaming pine needles exposure.</p>
<p>To detail, the following input data are required to run PIANO:<br />
(1) Your background gene sets, which represent all genes queried from your experiment (aka your ‘gene universe’)
(2) The list of genes you are interested in evaluating pathway enrichment of; here, this represents the genes identified with significant differential expression associated with flaming pine needles<br />
(3) A underlying pathway dataset; here, we’re using the KEGG PATHWAY Database (<a href="https://www.genome.jp/kegg/pathway.html">KEGG</a>), summarized through the Molecular Signature Database (<a href="https://www.gsea-msigdb.org/gsea/msigdb/">MSigDB</a>) into pre-formatted input files (.gmt) ready for PIANO.</p>
<p><em>Let’s organize these three required data inputs.</em></p>
<p>I THINK WE’RE HAVING THE SAME ISSUE HERE - WHERE WE MIGHT JUST BE PULLING THE LPS RESULTS? IT’S HARD FOR ME TO TELL. CAN WE PLEASE PULL THE FLAMING PINE NEEDLES RESULTS? WE ALSO MIGHT WANT TO IMPLEMENT THE PADJ &lt; 0.10 FILTER HERE TO INCLUDE MORE GENES.</p>
<ol style="list-style-type: decimal">
<li>Background gene set:</li>
</ol>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-1"></a><span class="co"># First grab the rownames of the &#39;res&#39; object, and remove the BioSpyder numeric identifier using a sub function, while maintaining the gene symbol</span></span>
<span id="cb69-2"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-2"></a><span class="co"># and place these IDs into a new list within the &#39;res&#39; object (saved as &#39;id&#39;)</span></span>
<span id="cb69-3"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-3"></a>res<span class="op">$</span>id &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">rownames</span>(res));</span>
<span id="cb69-4"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-4"></a></span>
<span id="cb69-5"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-5"></a><span class="co"># Because these IDs now contain duplicate gene symbols, we need to remove duplicates</span></span>
<span id="cb69-6"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-6"></a><span class="co"># One way to do this is to preferentially retain rows of data with the largest fold change (it doesn&#39;t really matter here, because we&#39;re just identifying unique genes within the background set)</span></span>
<span id="cb69-7"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-7"></a>res.ordered &lt;-<span class="st"> </span>res[<span class="kw">order</span>(res<span class="op">$</span>id, <span class="op">-</span><span class="kw">abs</span>(res<span class="op">$</span>log2FoldChange) ), ] <span class="co"># sort by id and reverse of abs(log2foldchange)</span></span>
<span id="cb69-8"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-8"></a>res.ordered &lt;-<span class="st"> </span>res.ordered[ <span class="op">!</span><span class="kw">duplicated</span>(res.ordered<span class="op">$</span>id), ]     <span class="co"># removing gene duplicates</span></span>
<span id="cb69-9"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-9"></a></span>
<span id="cb69-10"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-10"></a><span class="co"># Setting this as the background list</span></span>
<span id="cb69-11"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-11"></a>Background &lt;-<span class="st"> </span><span class="kw">toupper</span>(<span class="kw">as.character</span>(res.ordered<span class="op">$</span>id))</span>
<span id="cb69-12"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb69-12"></a>Background[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>] <span class="co"># viewing the first 200 genes in this background list</span></span></code></pre></div>
<pre><code>##   [1] &quot;0610009B22RIK&quot;  &quot;0610010F05RIK&quot;  &quot;0610010K14RIK&quot;  &quot;0610012G03RIK&quot; 
##   [5] &quot;0610030E20RIK&quot;  &quot;0610040J01RIK&quot;  &quot;1110004F10RIK&quot;  &quot;1110008P14RIK&quot; 
##   [9] &quot;1110012L19RIK&quot;  &quot;1110017D15RIK&quot;  &quot;1110032A03RIK&quot;  &quot;1110038F14RIK&quot; 
##  [13] &quot;1110059E24RIK&quot;  &quot;1110059G10RIK&quot;  &quot;1110065P20RIK&quot;  &quot;1190002N15RIK&quot; 
##  [17] &quot;1190007I07RIK&quot;  &quot;1500011B03RIK&quot;  &quot;1500011K16RIK&quot;  &quot;1600002K03RIK&quot; 
##  [21] &quot;1600012H06RIK&quot;  &quot;1600014C10RIK&quot;  &quot;1700001L19RIK&quot;  &quot;1700003E16RIK&quot; 
##  [25] &quot;1700007K13RIK&quot;  &quot;1700012B09RIK&quot;  &quot;1700013F07RIK&quot;  &quot;1700016K19RIK&quot; 
##  [29] &quot;1700017B05RIK&quot;  &quot;1700020D05RIK&quot;  &quot;1700024G13RIK&quot;  &quot;1700025G04RIK&quot; 
##  [33] &quot;1700028P14RIK&quot;  &quot;1700029I15RIK&quot;  &quot;1700029J07RIK&quot;  &quot;1700030K09RIK&quot; 
##  [37] &quot;1700037C18RIK&quot;  &quot;1700037H04RIK&quot;  &quot;1700086D15RIK&quot;  &quot;1700088E04RIK&quot; 
##  [41] &quot;1700102P08RIK&quot;  &quot;1700109H08RIK&quot;  &quot;1700123O20RIK&quot;  &quot;1810009A15RIK&quot; 
##  [45] &quot;1810010H24RIK&quot;  &quot;1810013L24RIK&quot;  &quot;1810022K09RIK&quot;  &quot;1810030O07RIK&quot; 
##  [49] &quot;1810037I17RIK&quot;  &quot;1810043G02RIK&quot;  &quot;1810055G02RIK&quot;  &quot;2010005H15RIK&quot; 
##  [53] &quot;2010111I01RIK&quot;  &quot;2010300C02RIK&quot;  &quot;2010309G21RIK&quot;  &quot;2200002D01RIK&quot; 
##  [57] &quot;2210011C24RIK&quot;  &quot;2210016H18RIK&quot;  &quot;2210016L21RIK&quot;  &quot;2210408I21RIK&quot; 
##  [61] &quot;2300009A05RIK&quot;  &quot;2310007B03RIK&quot;  &quot;2310009B15RIK&quot;  &quot;2310011J03RIK&quot; 
##  [65] &quot;2310022A10RIK&quot;  &quot;2310022B05RIK&quot;  &quot;2310030G06RIK&quot;  &quot;2310033P09RIK&quot; 
##  [69] &quot;2310057M21RIK&quot;  &quot;2310061I04RIK&quot;  &quot;2410002F23RIK&quot;  &quot;2410004B18RIK&quot; 
##  [73] &quot;2410015M20RIK&quot;  &quot;2410131K14RIK&quot;  &quot;2510002D24RIK&quot;  &quot;2510009E07RIK&quot; 
##  [77] &quot;2600001M11RIK&quot;  &quot;2610001J05RIK&quot;  &quot;2610002M06RIK&quot;  &quot;2610008E11RIK&quot; 
##  [81] &quot;2610028H24RIK&quot;  &quot;2610301B20RIK&quot;  &quot;2610507B11RIK&quot;  &quot;2700049A03RIK&quot; 
##  [85] &quot;2700062C07RIK&quot;  &quot;2700081O15RIK&quot;  &quot;2700097O09RIK&quot;  &quot;2810004N23RIK&quot; 
##  [89] &quot;2810021J22RIK&quot;  &quot;2810408A11RIK&quot;  &quot;2900026A02RIK&quot;  &quot;3110001I22RIK&quot; 
##  [93] &quot;3110040N11RIK&quot;  &quot;3110082I17RIK&quot;  &quot;3830406C13RIK&quot;  &quot;4430402I18RIK&quot; 
##  [97] &quot;4732423E21RIK&quot;  &quot;4833414E09RIK&quot;  &quot;4833420G17RIK&quot;  &quot;4833427G06RIK&quot; 
## [101] &quot;4833439L19RIK&quot;  &quot;4921524J17RIK&quot;  &quot;4930402H24RIK&quot;  &quot;4930430F08RIK&quot; 
## [105] &quot;4930451G09RIK&quot;  &quot;4930453N24RIK&quot;  &quot;4930486L24RIK&quot;  &quot;4930503L19RIK&quot; 
## [109] &quot;4930523C07RIK&quot;  &quot;4930550C14RIK&quot;  &quot;4930562C15RIK&quot;  &quot;4931406C07RIK&quot; 
## [113] &quot;4931406P16RIK&quot;  &quot;4931414P19RIK&quot;  &quot;4932438A13RIK&quot;  &quot;4933408B17RIK&quot; 
## [117] &quot;4933415F23RIK&quot;  &quot;4933427D14RIK&quot;  &quot;4933434E20RIK&quot;  &quot;5330417C22RIK&quot; 
## [121] &quot;5430427O19RIK&quot;  &quot;5730480H06RIK&quot;  &quot;5830417I10RIK&quot;  &quot;6030458C11RIK&quot; 
## [125] &quot;6030468B19RIK&quot;  &quot;6330403K07RIK&quot;  &quot;6330417A16RIK&quot;  &quot;6430531B16RIK&quot; 
## [129] &quot;6430548M08RIK&quot;  &quot;6430550D23RIK&quot;  &quot;6720489N17RIK&quot;  &quot;8030462N17RIK&quot; 
## [133] &quot;9030624G23RIK&quot;  &quot;9130008F23RIK&quot;  &quot;9130019O22RIK&quot;  &quot;9130023H24RIK&quot; 
## [137] &quot;9130230L23RIK&quot;  &quot;9230104L09RIK&quot;  &quot;9330182L06RIK&quot;  &quot;9530068E07RIK&quot; 
## [141] &quot;9930012K11RIK&quot;  &quot;9930021J03RIK&quot;  &quot;9930111J21RIK1&quot; &quot;9930111J21RIK2&quot;
## [145] &quot;A130071D04RIK&quot;  &quot;A230050P20RIK&quot;  &quot;A2ML1&quot;          &quot;A430005L14RIK&quot; 
## [149] &quot;A430033K04RIK&quot;  &quot;A430078G23RIK&quot;  &quot;A530032D15RIK&quot;  &quot;A630001G21RIK&quot; 
## [153] &quot;A730034C02&quot;     &quot;A730049H05RIK&quot;  &quot;A830018L16RIK&quot;  &quot;A930002H24RIK&quot; 
## [157] &quot;A930004D18RIK&quot;  &quot;AA415038&quot;       &quot;AA986860&quot;       &quot;AACS&quot;          
## [161] &quot;AAED1&quot;          &quot;AAGAB&quot;          &quot;AAK1&quot;           &quot;AAMDC&quot;         
## [165] &quot;AAMP&quot;           &quot;AAR2&quot;           &quot;AARS&quot;           &quot;AARS2&quot;         
## [169] &quot;AARSD1&quot;         &quot;AASDH&quot;          &quot;AASS&quot;           &quot;AATF&quot;          
## [173] &quot;AATK&quot;           &quot;AB124611&quot;       &quot;ABCA1&quot;          &quot;ABCA17&quot;        
## [177] &quot;ABCA2&quot;          &quot;ABCA3&quot;          &quot;ABCA5&quot;          &quot;ABCA7&quot;         
## [181] &quot;ABCA8A&quot;         &quot;ABCA8B&quot;         &quot;ABCA9&quot;          &quot;ABCB10&quot;        
## [185] &quot;ABCB1A&quot;         &quot;ABCB1B&quot;         &quot;ABCB6&quot;          &quot;ABCB7&quot;         
## [189] &quot;ABCB8&quot;          &quot;ABCC1&quot;          &quot;ABCC3&quot;          &quot;ABCC4&quot;         
## [193] &quot;ABCC5&quot;          &quot;ABCD1&quot;          &quot;ABCD2&quot;          &quot;ABCD3&quot;         
## [197] &quot;ABCD4&quot;          &quot;ABCE1&quot;          &quot;ABCF1&quot;          &quot;ABCF3&quot;</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>The list of genes identified with significant differential expression associated with flaming pine needles:</li>
</ol>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb71-1"></a><span class="co"># Similar to the above script, but starting with the res$id object</span></span>
<span id="cb71-2"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb71-2"></a><span class="co"># and filtering for genes with padj &lt; 0.05</span></span>
<span id="cb71-3"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb71-3"></a></span>
<span id="cb71-4"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb71-4"></a>res.ordered &lt;-<span class="st"> </span>res[<span class="kw">order</span>(res<span class="op">$</span>id, <span class="op">-</span><span class="kw">abs</span>(res<span class="op">$</span>log2FoldChange) ), ] <span class="co">#sort by id and reverse of abs(log2FC)</span></span>
<span id="cb71-5"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb71-5"></a>SigGenes &lt;-<span class="st"> </span><span class="kw">toupper</span>(<span class="kw">as.character</span>(res.ordered[<span class="kw">which</span>(res.ordered<span class="op">$</span>padj<span class="op">&lt;</span>.<span class="dv">05</span>),<span class="st">&quot;id&quot;</span>])) <span class="co"># pulling the genes with padj &lt; 0.05</span></span>
<span id="cb71-6"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb71-6"></a>SigGenes &lt;-<span class="st"> </span>SigGenes[ <span class="op">!</span><span class="kw">duplicated</span>(SigGenes)] <span class="co"># removing gene duplicates</span></span>
<span id="cb71-7"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb71-7"></a></span>
<span id="cb71-8"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb71-8"></a><span class="kw">length</span>(SigGenes) <span class="co"># viewing the length of this significant gene list</span></span></code></pre></div>
<pre><code>## [1] 4017</code></pre>
<p>Therefore, this gene set includes X genes significant associated with X condition???</p>
<ol start="3" style="list-style-type: decimal">
<li>The underlying KEGG pathway dataset.
Note that this file was simply downloaded from <a href="https://www.gsea-msigdb.org/gsea/msigdb/">MSigDB</a>, ready for upload as a .gmt file. Here, we use the ‘loadGSC’ function enabled through the PIANO package to upload and organize these pathways.</li>
</ol>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb73-1"></a>KEGG_Pathways &lt;-<span class="st"> </span><span class="kw">loadGSC</span>(<span class="dt">file=</span><span class="st">&quot;Module2_4/Module2_4_kegg_v7_symbols.gmt&quot;</span>, <span class="dt">type=</span><span class="st">&quot;gmt&quot;</span>)</span>
<span id="cb73-2"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb73-2"></a></span>
<span id="cb73-3"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb73-3"></a><span class="kw">length</span>(KEGG_Pathways<span class="op">$</span>gsc) <span class="co"># viewing the number of biological pathways contained in the database</span></span></code></pre></div>
<pre><code>## [1] 186</code></pre>
<p>This KEGG pathway database therefore includes 186 biological pathways available to query</p>
<p>With these data inputs ready, we can now run the pathway enrichment analysis. The enrichment statistic that is commonly employed through the PIANO package is based of a hypergeometric test, run through the ‘runGSAhyper’ function. This returns a p-value for each gene set from which you can determine enrichment status.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb75-1"></a><span class="co"># Running the piano function based on the hypergeometric statistic</span></span>
<span id="cb75-2"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb75-2"></a>Results_GSA &lt;-<span class="st"> </span>piano<span class="op">::</span><span class="kw">runGSAhyper</span>(<span class="dt">genes=</span>SigGenes, <span class="dt">universe=</span>Background,<span class="dt">gsc=</span>KEGG_Pathways, <span class="dt">gsSizeLim=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="ot">Inf</span>), <span class="dt">adjMethod =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb75-3"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb75-3"></a></span>
<span id="cb75-4"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb75-4"></a><span class="co"># Pulling the pathway enrichment results into a separate dataframe</span></span>
<span id="cb75-5"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb75-5"></a>PathwayResults &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(Results_GSA<span class="op">$</span>resTab)</span>
<span id="cb75-6"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb75-6"></a></span>
<span id="cb75-7"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb75-7"></a><span class="co"># Viewing the top of these pathway enrichment results</span></span>
<span id="cb75-8"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb75-8"></a><span class="kw">head</span>(PathwayResults)</span></code></pre></div>
<pre><code>##                                                        p-value Adjusted p-value
## KEGG_N_GLYCAN_BIOSYNTHESIS                          0.67813509        0.9412920
## KEGG_OTHER_GLYCAN_DEGRADATION                       0.46310322        0.8034029
## KEGG_O_GLYCAN_BIOSYNTHESIS                          0.40876219        0.7527700
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                  0.11063551        0.4477247
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE 0.07136335        0.3493048
## KEGG_GLYCEROLIPID_METABOLISM                        0.76282992        0.9823039
##                                                     Significant (in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                 11
## KEGG_OTHER_GLYCAN_DEGRADATION                                               5
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                  7
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                          7
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                         7
## KEGG_GLYCEROLIPID_METABOLISM                                               10
##                                                     Non-significant (in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                     24
## KEGG_OTHER_GLYCAN_DEGRADATION                                                   8
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                     11
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                              6
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                             5
## KEGG_GLYCEROLIPID_METABOLISM                                                   24
##                                                     Significant (not in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                   4006
## KEGG_OTHER_GLYCAN_DEGRADATION                                                4012
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                   4010
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                           4010
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                          4010
## KEGG_GLYCEROLIPID_METABOLISM                                                 4007
##                                                     Non-significant (not in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                       7836
## KEGG_OTHER_GLYCAN_DEGRADATION                                                    7852
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                       7849
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                               7854
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                              7855
## KEGG_GLYCEROLIPID_METABOLISM                                                     7836</code></pre>
<p>This dataframe therefore summarizes the enrichment p-value for each pathway, FDR adjusted p-value, number of significant genes in the gene set that intersect with genes in the pathway, etc</p>
<p>With these results, let’s identify which pathways meet a statistical enrichment p-value filter of 0.05:</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb77-1"></a>SigPathways &lt;-<span class="st"> </span>PathwayResults[<span class="kw">which</span>(PathwayResults<span class="op">$</span><span class="st">`</span><span class="dt">p-value</span><span class="st">`</span> <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>),]</span>
<span id="cb77-2"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#cb77-2"></a><span class="kw">rownames</span>(SigPathways)</span></code></pre></div>
<pre><code>##  [1] &quot;KEGG_MAPK_SIGNALING_PATHWAY&quot;                 
##  [2] &quot;KEGG_CYTOKINE_CYTOKINE_RECEPTOR_INTERACTION&quot; 
##  [3] &quot;KEGG_CHEMOKINE_SIGNALING_PATHWAY&quot;            
##  [4] &quot;KEGG_NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION&quot;
##  [5] &quot;KEGG_P53_SIGNALING_PATHWAY&quot;                  
##  [6] &quot;KEGG_APOPTOSIS&quot;                              
##  [7] &quot;KEGG_AXON_GUIDANCE&quot;                          
##  [8] &quot;KEGG_ECM_RECEPTOR_INTERACTION&quot;               
##  [9] &quot;KEGG_CELL_ADHESION_MOLECULES_CAMS&quot;           
## [10] &quot;KEGG_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY&quot;   
## [11] &quot;KEGG_NOD_LIKE_RECEPTOR_SIGNALING_PATHWAY&quot;    
## [12] &quot;KEGG_RIG_I_LIKE_RECEPTOR_SIGNALING_PATHWAY&quot;  
## [13] &quot;KEGG_CYTOSOLIC_DNA_SENSING_PATHWAY&quot;          
## [14] &quot;KEGG_JAK_STAT_SIGNALING_PATHWAY&quot;             
## [15] &quot;KEGG_HEMATOPOIETIC_CELL_LINEAGE&quot;             
## [16] &quot;KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY&quot;      
## [17] &quot;KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY&quot;      
## [18] &quot;KEGG_NEUROTROPHIN_SIGNALING_PATHWAY&quot;         
## [19] &quot;KEGG_LONG_TERM_DEPRESSION&quot;                   
## [20] &quot;KEGG_ADIPOCYTOKINE_SIGNALING_PATHWAY&quot;        
## [21] &quot;KEGG_TYPE_II_DIABETES_MELLITUS&quot;              
## [22] &quot;KEGG_AMYOTROPHIC_LATERAL_SCLEROSIS_ALS&quot;      
## [23] &quot;KEGG_LEISHMANIA_INFECTION&quot;                   
## [24] &quot;KEGG_PATHWAYS_IN_CANCER&quot;                     
## [25] &quot;KEGG_ENDOMETRIAL_CANCER&quot;                     
## [26] &quot;KEGG_PROSTATE_CANCER&quot;                        
## [27] &quot;KEGG_BASAL_CELL_CARCINOMA&quot;                   
## [28] &quot;KEGG_BLADDER_CANCER&quot;                         
## [29] &quot;KEGG_ACUTE_MYELOID_LEUKEMIA&quot;                 
## [30] &quot;KEGG_SMALL_CELL_LUNG_CANCER&quot;                 
## [31] &quot;KEGG_GRAFT_VERSUS_HOST_DISEASE&quot;              
## [32] &quot;KEGG_VIRAL_MYOCARDITIS&quot;</code></pre>
<div id="with-this-we-can-answer-environmental-health-question-8" class="section level3">
<h3><span class="header-section-number">9.4.1</span> With this, we can answer <strong>Environmental Health Question 8</strong>:</h3>
<div id="what-biological-pathways-are-disrupted-in-association-with-flamingsmoldering-pine-needles-exposure-in-the-lung-identified-through-systems-level-analyses" class="section level4">
<h4><span class="header-section-number">9.4.1.1</span> What biological pathways are disrupted in association with flaming/smoldering pine needles exposure in the lung, identified through systems level analyses?</h4>
</div>
<div id="answer-to-insert-after-code-updates." class="section level4">
<h4><span class="header-section-number">9.4.1.2</span> <em>Answer: TO INSERT AFTER CODE UPDATES.</em></h4>
<p><br></p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visualizing-statistical-results-using-volcano-plots.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="concluding-remarks-7.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
