<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Setting Up Your R Environment | DMAC Training Modules</title>
  <meta name="description" content="This is a resource for R training using environmental health datasets along standard code pipelines and methods." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Setting Up Your R Environment | DMAC Training Modules" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a resource for R training using environmental health datasets along standard code pipelines and methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Setting Up Your R Environment | DMAC Training Modules" />
  
  <meta name="twitter:description" content="This is a resource for R training using environmental health datasets along standard code pipelines and methods." />
  

<meta name="author" content="Kyle Roell, Lauren Koval, Julia Rager" />


<meta name="date" content="2021-06-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="dataorg.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">DMAC Training Modules</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Setting Up Your R Environment</a><ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#r-and-rstudio"><i class="fa fa-check"></i><b>2.1</b> R and RStudio</a><ul>
<li class="chapter" data-level="2.1.1" data-path="intro.html"><a href="intro.html#downloading-r-and-rstudio"><i class="fa fa-check"></i><b>2.1.1</b> Downloading R and RStudio</a></li>
<li class="chapter" data-level="2.1.2" data-path="intro.html"><a href="intro.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>2.1.2</b> Installing R and RStudio</a></li>
<li class="chapter" data-level="2.1.3" data-path="intro.html"><a href="intro.html#installing-and-loading-packages"><i class="fa fa-check"></i><b>2.1.3</b> Installing and Loading Packages</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#scripting-basics"><i class="fa fa-check"></i><b>2.2</b> Scripting Basics</a><ul>
<li class="chapter" data-level="2.2.1" data-path="intro.html"><a href="intro.html#setting-working-directory"><i class="fa fa-check"></i><b>2.2.1</b> Setting Working Directory</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro.html"><a href="intro.html#importing-and-exporting-files"><i class="fa fa-check"></i><b>2.2.2</b> Importing and Exporting Files</a></li>
<li class="chapter" data-level="2.2.3" data-path="intro.html"><a href="intro.html#viewing-data"><i class="fa fa-check"></i><b>2.2.3</b> Viewing Data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dataorg.html"><a href="dataorg.html"><i class="fa fa-check"></i><b>3</b> The Basics for Data Organization</a><ul>
<li class="chapter" data-level="3.1" data-path="dataorg.html"><a href="dataorg.html#basic-data-manipulation"><i class="fa fa-check"></i><b>3.1</b> Basic Data Manipulation</a><ul>
<li class="chapter" data-level="3.1.1" data-path="dataorg.html"><a href="dataorg.html#merging"><i class="fa fa-check"></i><b>3.1.1</b> Merging</a></li>
<li class="chapter" data-level="3.1.2" data-path="dataorg.html"><a href="dataorg.html#merging-processed-data-with-metadata-file"><i class="fa fa-check"></i><b>3.1.2</b> Merging processed data with metadata file?</a></li>
<li class="chapter" data-level="3.1.3" data-path="dataorg.html"><a href="dataorg.html#cast"><i class="fa fa-check"></i><b>3.1.3</b> Cast</a></li>
<li class="chapter" data-level="3.1.4" data-path="dataorg.html"><a href="dataorg.html#melt"><i class="fa fa-check"></i><b>3.1.4</b> Melt</a></li>
<li class="chapter" data-level="3.1.5" data-path="dataorg.html"><a href="dataorg.html#filtering-subsetting"><i class="fa fa-check"></i><b>3.1.5</b> Filtering &amp; subsetting</a></li>
<li class="chapter" data-level="3.1.6" data-path="dataorg.html"><a href="dataorg.html#tidyverse-stuff-pivots"><i class="fa fa-check"></i><b>3.1.6</b> Tidyverse stuff (pivots)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html"><i class="fa fa-check"></i><b>4</b> Finding and Visualizing Data Trends</a><ul>
<li class="chapter" data-level="4.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#heat-maps"><i class="fa fa-check"></i><b>4.1</b> Heat maps</a><ul>
<li class="chapter" data-level="4.1.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#pheatmap"><i class="fa fa-check"></i><b>4.1.1</b> pheatmap</a></li>
<li class="chapter" data-level="4.1.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#heatmap2"><i class="fa fa-check"></i><b>4.1.2</b> heatmap2</a></li>
<li class="chapter" data-level="4.1.3" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#superheat"><i class="fa fa-check"></i><b>4.1.3</b> superheat</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#clustering"><i class="fa fa-check"></i><b>4.2</b> Clustering</a><ul>
<li class="chapter" data-level="4.2.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#hierarchical"><i class="fa fa-check"></i><b>4.2.1</b> Hierarchical</a></li>
<li class="chapter" data-level="4.2.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#k-means"><i class="fa fa-check"></i><b>4.2.2</b> K-means</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#data-reduction-pca"><i class="fa fa-check"></i><b>4.3</b> Data reduction (PCA)</a><ul>
<li class="chapter" data-level="4.3.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#visualize-pca-plot"><i class="fa fa-check"></i><b>4.3.1</b> Visualize PCA Plot</a></li>
<li class="chapter" data-level="4.3.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#identify-of-variance-captured"><i class="fa fa-check"></i><b>4.3.2</b> Identify % of variance captured</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#basic-statistical-tests-and-visualizations-of-data"><i class="fa fa-check"></i><b>4.4</b> Basic Statistical Tests and Visualizations of Data</a><ul>
<li class="chapter" data-level="4.4.1" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#normality"><i class="fa fa-check"></i><b>4.4.1</b> Normality</a></li>
<li class="chapter" data-level="4.4.2" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#t-tests-column-charts"><i class="fa fa-check"></i><b>4.4.2</b> T-tests – column charts</a></li>
<li class="chapter" data-level="4.4.3" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#regression-linear-regression-and-logistic-regression"><i class="fa fa-check"></i><b>4.4.3</b> Regression: linear regression and logistic regression</a></li>
<li class="chapter" data-level="4.4.4" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#anova-delicate-commentary"><i class="fa fa-check"></i><b>4.4.4</b> ANOVA + delicate commentary</a></li>
<li class="chapter" data-level="4.4.5" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#chi-squared-test-box-plots"><i class="fa fa-check"></i><b>4.4.5</b> Chi-squared test – box plots</a></li>
<li class="chapter" data-level="4.4.6" data-path="finding-and-visualizing-data-trends.html"><a href="finding-and-visualizing-data-trends.html#fishers-exact-test"><i class="fa fa-check"></i><b>4.4.6</b> Fisher’s exact test</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html"><i class="fa fa-check"></i><b>5</b> Multi-Omics Analyses for Environmental Health</a><ul>
<li class="chapter" data-level="5.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#exposomics"><i class="fa fa-check"></i><b>5.1</b> Exposomics</a><ul>
<li class="chapter" data-level="5.1.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#placenta-exposome"><i class="fa fa-check"></i><b>5.1.1</b> Placenta Exposome</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#transcriptomics"><i class="fa fa-check"></i><b>5.2</b> Transcriptomics</a><ul>
<li class="chapter" data-level="5.2.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#deseq2-rnaseq"><i class="fa fa-check"></i><b>5.2.1</b> DESeq2 / RNAseq</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#genome-wide-microrna"><i class="fa fa-check"></i><b>5.3</b> Genome-wide MicroRNA</a></li>
<li class="chapter" data-level="5.4" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#genome-wide-dna-methylation"><i class="fa fa-check"></i><b>5.4</b> Genome-wide DNA Methylation</a><ul>
<li class="chapter" data-level="5.4.1" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#illumina-array-data"><i class="fa fa-check"></i><b>5.4.1</b> Illumina array data</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="multi-omics-analyses-for-environmental-health.html"><a href="multi-omics-analyses-for-environmental-health.html#proteomics"><i class="fa fa-check"></i><b>5.5</b> Proteomics</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mixtures-analyses-for-environmental-health.html"><a href="mixtures-analyses-for-environmental-health.html"><i class="fa fa-check"></i><b>6</b> Mixtures Analyses for Environmental Health</a><ul>
<li class="chapter" data-level="6.1" data-path="mixtures-analyses-for-environmental-health.html"><a href="mixtures-analyses-for-environmental-health.html#sufficient-similarity"><i class="fa fa-check"></i><b>6.1</b> Sufficient Similarity</a></li>
<li class="chapter" data-level="6.2" data-path="mixtures-analyses-for-environmental-health.html"><a href="mixtures-analyses-for-environmental-health.html#mixtures-modeling-through-qgcomp"><i class="fa fa-check"></i><b>6.2</b> Mixtures Modeling through qgcomp</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html"><i class="fa fa-check"></i><b>7</b> Environmental Health Databases</a><ul>
<li class="chapter" data-level="7.1" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html#comparative-toxicogenomics-database-ctd"><i class="fa fa-check"></i><b>7.1</b> Comparative Toxicogenomics Database (CTD)</a></li>
<li class="chapter" data-level="7.2" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html#gene-expression-omnibus-geo"><i class="fa fa-check"></i><b>7.2</b> Gene Expression Omnibus (GEO)</a></li>
<li class="chapter" data-level="7.3" data-path="environmental-health-databases.html"><a href="environmental-health-databases.html#nhanes"><i class="fa fa-check"></i><b>7.3</b> NHANES</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">DMAC Training Modules</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Setting Up Your R Environment</h1>
<p>Before learning about data manipulation and statistical methods for analyzing environmental health datasets, we will provide a brief introduction to R, RStudio, and setting up an R environment and simple scripts.</p>
<div id="r-and-rstudio" class="section level2">
<h2><span class="header-section-number">2.1</span> R and RStudio</h2>
<p><a href="https://www.r-project.org">R</a> is a free, open source programming language for statistical computing and graphics that anyone can download and use. It doesn’t require a license and is good for reproducible analyses. There exists a large, diverse collection of packages and very comprehensive documentation.</p>
<p>It is easy to download, install, and setup R. Additionally, <a href="https://www.rstudio.com">RStudio</a> is an open source integrated development environment for R. RStudio makes programming in R and using R scripts and features more user friendly. R should be downloaded prior to downloading RStudio.</p>
<div id="downloading-r-and-rstudio" class="section level3">
<h3><span class="header-section-number">2.1.1</span> Downloading R and RStudio</h3>
<p>The following is a walkthrough on how to download R and RStudio.</p>
<p><br></br>
<strong>1. Navigate to R Website</strong></p>
<div class="figure"><span id="fig:website"></span>
<img src="_book/test1_files/figure-html/r_website.png" alt="R Website, https://www.r-project.org" width="638" />
<p class="caption">
Figure 2.1: R Website, <a href="https://www.r-project.org" class="uri">https://www.r-project.org</a>
</p>
</div>
<p><br></br></p>
<p><strong>2. Select the appropriate CRAN mirror (Duke is fastest if at UNC)</strong></p>
<div class="figure"><span id="fig:mirror"></span>
<img src="_book/test1_files/figure-html/cran_mirrors.png" alt="CRAN Mirror, https://cran.r-project.org/mirrors.html" width="642" />
<p class="caption">
Figure 2.2: CRAN Mirror, <a href="https://cran.r-project.org/mirrors.html" class="uri">https://cran.r-project.org/mirrors.html</a>
</p>
</div>
<p><br></br></p>
<p><strong>3. Select the appropriate R distribution</strong></p>
<div class="figure"><span id="fig:archive"></span>
<img src="_book/test1_files/figure-html/duke_archive_r.png" alt="R Download Link, http://archive.linux.duke.edu/cran/" width="610" />
<p class="caption">
Figure 2.3: R Download Link, <a href="http://archive.linux.duke.edu/cran/" class="uri">http://archive.linux.duke.edu/cran/</a>
</p>
</div>
<p><br></br></p>
<p><strong>4. Download R</strong></p>
<div class="figure"><span id="fig:download"></span>
<img src="_book/test1_files/figure-html/duke_download_r.png" alt="R Download, http://archive.linux.duke.edu/cran/bin/macosx/" width="646" />
<p class="caption">
Figure 2.4: R Download, <a href="http://archive.linux.duke.edu/cran/bin/macosx/" class="uri">http://archive.linux.duke.edu/cran/bin/macosx/</a>
</p>
</div>
<p><br></br></p>
<p><strong>5. Navigate to RStudio website and download RStudio (free edition)</strong></p>
<div class="figure"><span id="fig:downloadstudio"></span>
<img src="_book/test1_files/figure-html/rstudio.png" alt="RStudio Download, https://rstudio.com/products/rstudio/download/" width="448" />
<p class="caption">
Figure 2.5: RStudio Download, <a href="https://rstudio.com/products/rstudio/download/" class="uri">https://rstudio.com/products/rstudio/download/</a>
</p>
</div>
</div>
<div id="installing-r-and-rstudio" class="section level3">
<h3><span class="header-section-number">2.1.2</span> Installing R and RStudio</h3>
<p>Once R and RStudio have been downloaded, install R first and then RStudio, following the instructions of the installer.</p>
</div>
<div id="installing-and-loading-packages" class="section level3">
<h3><span class="header-section-number">2.1.3</span> Installing and Loading Packages</h3>
<p>Packages in R are units of shareable code that contain functions, data, and documentation on how to use all of these resources. Because R is an open source programming language, packages are constantly being developed and updated. There are many R packages that exist spanning many topics such as graphics and plotting, machine learning, and data manipulation. R packages are often written by R users and submitted to the <a href="https://cran.r-project.org">Comprehensive R Archive Network</a> (CRAN), or another host such as <a href="https://www.bioconductor.org">BioConductor</a> or <a href="https://github.com">GitHub</a>.</p>
<p>Packages can be installed from the host, but need to be loaded into the workspace. Most of the time, you do not need to download anything from a website. Instead, you can install packages through running code in R or RStudio.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="intro.html#cb1-1"></a><span class="kw">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>, <span class="dt">repos =</span> <span class="st">&quot;https://cran.rstudio.com&quot;</span>)</span></code></pre></div>
<p>Once a package is installed, it needs to be loaded using the <em>library</em> function or explicitly referenced to use functions or datasets from that package.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="intro.html#cb2-1"></a><span class="kw">library</span>(ggplot2)</span></code></pre></div>
</div>
</div>
<div id="scripting-basics" class="section level2">
<h2><span class="header-section-number">2.2</span> Scripting Basics</h2>
<p>Before demonstrating the basics of writing R code and scripts, it is worth noting that a function can be queried in RStudio by typing a question mark before the name of the function (e.g. <em>?install.packages</em>). This will bring up documentation in the viewer window. Additionally, R will autofill function names, variable names, etc. by pressing tab while typing. If multiple matches are found, R will provide you with a drop down list to select from, which may be useful when searching through newly installed packages or trying to quickly type variable names in an R script.</p>
<p>R also allows for scripts to contain non-code elements, called comments, that will not be run or interpreted. To make a comment, simply use a # followed by the comment. A # only comments out a single line of code, i.e. only that line will not be run. Comments are useful to help make code more interpretable for others or to add reminders of what and why parts of code may have been written.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="intro.html#cb3-1"></a><span class="co"># This is an R comment!</span></span>
<span id="cb3-2"><a href="intro.html#cb3-2"></a></span>
<span id="cb3-3"><a href="intro.html#cb3-3"></a><span class="co"># Loading ggplot2 package</span></span>
<span id="cb3-4"><a href="intro.html#cb3-4"></a><span class="kw">library</span>(ggplot2)</span></code></pre></div>
<div id="setting-working-directory" class="section level3">
<h3><span class="header-section-number">2.2.1</span> Setting Working Directory</h3>
<p>When working in R, it can be helpful to set the working directory to a local directory where data are located or output files will be saved. The current working directory can also be displayed.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="intro.html#cb4-1"></a><span class="co"># Show current working directory</span></span>
<span id="cb4-2"><a href="intro.html#cb4-2"></a><span class="kw">getwd</span>()</span></code></pre></div>
<pre><code>## [1] &quot;/Users/kroell/Documents/IEHS/UNC-SRP/test1&quot;</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="intro.html#cb6-1"></a><span class="co"># Set working directory</span></span>
<span id="cb6-2"><a href="intro.html#cb6-2"></a><span class="kw">setwd</span>(<span class="st">&quot;~/Documents/UNCSRP/Data/&quot;</span>)</span></code></pre></div>
</div>
<div id="importing-and-exporting-files" class="section level3">
<h3><span class="header-section-number">2.2.2</span> Importing and Exporting Files</h3>
<p>After setting the working directory, importing and exporting files can be done using various functions based on the type of file being read or written. Often, it is easiest to import data into R that are in a comma separated values, comma delimited, (CSV) file or tab delimited file. Other datatypes such as SAS data files, large csv files, etc. may require different functions to be more efficienlty read in and some of these file formats will be discussed in future modules.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="intro.html#cb7-1"></a><span class="co"># Read in CSV data</span></span>
<span id="cb7-2"><a href="intro.html#cb7-2"></a>csv.dataset =<span class="st"> </span><span class="kw">read.csv</span>(<span class="dt">file=</span><span class="st">&quot;~/Documents/UNCSRP/Data/example_data.csv&quot;</span>)</span>
<span id="cb7-3"><a href="intro.html#cb7-3"></a></span>
<span id="cb7-4"><a href="intro.html#cb7-4"></a><span class="co"># Read in tab delimited data</span></span>
<span id="cb7-5"><a href="intro.html#cb7-5"></a>tab.dataset =<span class="st"> </span><span class="kw">read.table</span>(<span class="dt">file=</span><span class="st">&quot;~/Documents/UNCSRP/Data/example_data.txt&quot;</span>)</span></code></pre></div>
<p>There are many ways to export data in R. Data can be written out into a CSV file, tab delimited file, RData file, etc. There are also many functions within packages that write out specific datasets generated by that package.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="intro.html#cb8-1"></a><span class="co"># Write out to a CSV file</span></span>
<span id="cb8-2"><a href="intro.html#cb8-2"></a><span class="kw">write.csv</span>(csv.output, <span class="dt">file=</span><span class="st">&quot;~/Documents/UNCSRP/Output/csv_output.csv&quot;</span>)</span>
<span id="cb8-3"><a href="intro.html#cb8-3"></a></span>
<span id="cb8-4"><a href="intro.html#cb8-4"></a><span class="co"># Write out to a tab delimited file</span></span>
<span id="cb8-5"><a href="intro.html#cb8-5"></a><span class="kw">write.table</span>(tab.output, <span class="dt">file=</span><span class="st">&quot;~/Documents/UNCSRP/Output/tsv_output.txt&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<p>R also allows objects to be saved in RData files. These files can be read into R, as well, and will load the object into the current workspace. Entire workspaces are also able to be saved.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="intro.html#cb9-1"></a><span class="co"># Read in saved single R data object</span></span>
<span id="cb9-2"><a href="intro.html#cb9-2"></a>r.obj =<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file=</span><span class="st">&quot;~/Documents/UNCSRP/Data/data.rds&quot;</span>)</span>
<span id="cb9-3"><a href="intro.html#cb9-3"></a></span>
<span id="cb9-4"><a href="intro.html#cb9-4"></a><span class="co"># Write single R object to file</span></span>
<span id="cb9-5"><a href="intro.html#cb9-5"></a><span class="kw">saveRDS</span>(object, <span class="dt">file=</span><span class="st">&quot;~/Documents/UNCSRP/Output/single_object.rds&quot;</span>)</span>
<span id="cb9-6"><a href="intro.html#cb9-6"></a></span>
<span id="cb9-7"><a href="intro.html#cb9-7"></a><span class="co"># Read in multiple saved R objects</span></span>
<span id="cb9-8"><a href="intro.html#cb9-8"></a><span class="kw">load</span>(<span class="dt">file=</span><span class="st">&quot;~/Documents/UNCSRP/Data/multiple_data.RData&quot;</span>)</span>
<span id="cb9-9"><a href="intro.html#cb9-9"></a></span>
<span id="cb9-10"><a href="intro.html#cb9-10"></a><span class="co"># Save multiple R objects</span></span>
<span id="cb9-11"><a href="intro.html#cb9-11"></a><span class="kw">save</span>(object1, object2, <span class="dt">file=</span><span class="st">&quot;~/Documents/UNCSRP/Output/multiple_objects.RData&quot;</span>)</span>
<span id="cb9-12"><a href="intro.html#cb9-12"></a></span>
<span id="cb9-13"><a href="intro.html#cb9-13"></a><span class="co"># Save entire workspace</span></span>
<span id="cb9-14"><a href="intro.html#cb9-14"></a><span class="kw">save.image</span>(<span class="dt">file=</span><span class="st">&quot;~/Documents/UNCSRP/Output/entire_workspace.RData&quot;</span>)</span>
<span id="cb9-15"><a href="intro.html#cb9-15"></a></span>
<span id="cb9-16"><a href="intro.html#cb9-16"></a><span class="co"># Load entire workspace</span></span>
<span id="cb9-17"><a href="intro.html#cb9-17"></a><span class="kw">load</span>(<span class="dt">file=</span><span class="st">&quot;~/Documents/UNCSRP/Data/entire_workspace.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="viewing-data" class="section level3">
<h3><span class="header-section-number">2.2.3</span> Viewing Data</h3>
<p>After data has been loaded into R, or created within R, it is good to inspect it. Datasets can be viewed in their entirety or subset to quickly look at part of the data.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="intro.html#cb10-1"></a><span class="co"># View first 5 rows of the previously loaded dataset</span></span>
<span id="cb10-2"><a href="intro.html#cb10-2"></a>csv.dataset[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,]</span></code></pre></div>
<pre><code>##    Sample Var1 Var2 Var3
## 1 sample1    1    2    1
## 2 sample2    2    4    4
## 3 sample3    3    6    9
## 4 sample4    4    8   16
## 5 sample5    5   10   25</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="intro.html#cb12-1"></a><span class="co"># View the entire dataset in RStudio</span></span>
<span id="cb12-2"><a href="intro.html#cb12-2"></a><span class="kw">View</span>(csv.dataset)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dataorg.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["test1.pdf", "test1.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
