<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>13.4 Loading and Organizing the Example Dataset | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="13.4 Loading and Organizing the Example Dataset | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="13.4 Loading and Organizing the Example Dataset | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="13.4 Loading and Organizing the Example Dataset | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle Roell, Lauren Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="script-preparations-6.html"/>
<link rel="next" href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TAME Toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#background"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#objectives"><i class="fa fa-check"></i>Objectives</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#module-development-overview"><i class="fa fa-check"></i>Module Development Overview</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#concluding-remarks"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#funding"><i class="fa fa-check"></i>Funding</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#author-affiliations"><i class="fa fa-check"></i>Author Affiliations</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html"><i class="fa fa-check"></i><b>1</b> Introduction to Coding in R</a><ul>
<li class="chapter" data-level="1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html"><i class="fa fa-check"></i><b>1.1</b> General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="1.2" data-path="downloading-and-installing-r.html"><a href="downloading-and-installing-r.html"><i class="fa fa-check"></i><b>1.2</b> Downloading and Installing R</a></li>
<li class="chapter" data-level="1.3" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html"><i class="fa fa-check"></i><b>1.3</b> Downloading and Installing R Studio</a><ul>
<li class="chapter" data-level="1.3.1" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html#importing-files"><i class="fa fa-check"></i><b>1.3.1</b> Importing Files</a></li>
<li class="chapter" data-level="1.3.2" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html#viewing-data"><i class="fa fa-check"></i><b>1.3.2</b> Viewing Data</a></li>
<li class="chapter" data-level="1.3.3" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html#exporting-data"><i class="fa fa-check"></i><b>1.3.3</b> Exporting Data</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="concluding-remarks-1.html"><a href="concluding-remarks-1.html"><i class="fa fa-check"></i><b>1.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-organization-basics.html"><a href="data-organization-basics.html"><i class="fa fa-check"></i><b>2</b> Data Organization Basics</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html"><i class="fa fa-check"></i><b>2.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="2.1.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#set-your-working-directory-1"><i class="fa fa-check"></i><b>2.1.1</b> Set your working directory</a></li>
<li class="chapter" data-level="2.1.2" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#importing-example-datasets"><i class="fa fa-check"></i><b>2.1.2</b> Importing example datasets</a></li>
<li class="chapter" data-level="2.1.3" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#viewing-example-datasets"><i class="fa fa-check"></i><b>2.1.3</b> Viewing example datasets</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="merging-data-using-base-r-syntax.html"><a href="merging-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.2</b> Merging Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.3" data-path="filtering-and-subsetting-data-using-base-r-syntax.html"><a href="filtering-and-subsetting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.3</b> Filtering and Subsetting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.4" data-path="melting-and-casting-data-using-base-r-syntax.html"><a href="melting-and-casting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.4</b> Melting and Casting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.5" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html"><i class="fa fa-check"></i><b>2.5</b> Introduction to Tidyverse</a><ul>
<li class="chapter" data-level="2.5.1" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#downloading-and-loading-the-tidyverse-package"><i class="fa fa-check"></i><b>2.5.1</b> Downloading and Loading the Tidyverse Package</a></li>
<li class="chapter" data-level="2.5.2" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#merging-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.2</b> Merging Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.3" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#filtering-and-subsetting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.3</b> Filtering and Subsetting Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.4" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#melting-and-casting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.4</b> Melting and Casting Data using Tidyverse Syntax</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="concluding-remarks-2.html"><a href="concluding-remarks-2.html"><i class="fa fa-check"></i><b>2.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html"><i class="fa fa-check"></i><b>3</b> Finding and Visualizing Data Trend</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html"><i class="fa fa-check"></i><b>3.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="3.1.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html#script-preparations"><i class="fa fa-check"></i><b>3.1.1</b> Script Preparations</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html"><i class="fa fa-check"></i><b>3.2</b> Basic Statistical Tests and Visualizations of Data Trends</a><ul>
<li class="chapter" data-level="3.2.1" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#visualize-and-test-data-for-normality"><i class="fa fa-check"></i><b>3.2.1</b> Visualize and Test Data for Normality</a></li>
<li class="chapter" data-level="3.2.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i><b>3.2.2</b> Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="3.2.3" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i><b>3.2.3</b> Two-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><i class="fa fa-check"></i><b>3.3</b> Regression Modeling and Visualization: Linear and Logistic Regressions</a><ul>
<li class="chapter" data-level="3.3.1" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#linear-regression"><i class="fa fa-check"></i><b>3.3.1</b> Linear Regression</a></li>
<li class="chapter" data-level="3.3.2" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#logistic-regression"><i class="fa fa-check"></i><b>3.3.2</b> Logistic Regression</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test.html"><a href="statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test.html"><i class="fa fa-check"></i><b>3.4</b> Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="3.5" data-path="concluding-remarks-3.html"><a href="concluding-remarks-3.html"><i class="fa fa-check"></i><b>3.5</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html"><i class="fa fa-check"></i><b>4</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-training-module-2.html"><a href="introduction-to-training-module-2.html"><i class="fa fa-check"></i><b>4.1</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="4.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html"><i class="fa fa-check"></i><b>4.2</b> Script Preparations</a><ul>
<li class="chapter" data-level="4.2.1" data-path="script-preparations-1.html"><a href="script-preparations-1.html#formatting-dataframes-for-downstream-visualization-code"><i class="fa fa-check"></i><b>4.2.1</b> Formatting dataframes for downstream visualization code</a></li>
<li class="chapter" data-level="4.2.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html#scaling-dataframes-for-downstream-data-visualizations"><i class="fa fa-check"></i><b>4.2.2</b> Scaling dataframes for downstream data visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html"><i class="fa fa-check"></i><b>4.3</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.3.1" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#density-plot-visualizations"><i class="fa fa-check"></i><b>4.3.1</b> Density Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.2" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#ggally-visualizations"><i class="fa fa-check"></i><b>4.3.2</b> GGally Visualizations</a></li>
<li class="chapter" data-level="4.3.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#boxplot-visualizations"><i class="fa fa-check"></i><b>4.3.3</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="4.3.4" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#correlation-plot-visualizations"><i class="fa fa-check"></i><b>4.3.4</b> Correlation Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.5" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#hierarchical-clustering-visualizations"><i class="fa fa-check"></i><b>4.3.5</b> Hierarchical Clustering Visualizations</a></li>
<li class="chapter" data-level="4.3.6" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#heatmap-visualizations"><i class="fa fa-check"></i><b>4.3.6</b> Heatmap Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="concluding-remarks-4.html"><a href="concluding-remarks-4.html"><i class="fa fa-check"></i><b>4.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i><b>5</b> FAIR Data Management Practices</a><ul>
<li class="chapter" data-level="5.1" data-path="background-on-training-module.html"><a href="background-on-training-module.html"><i class="fa fa-check"></i><b>5.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="5.2" data-path="training-modules-questions.html"><a href="training-modules-questions.html"><i class="fa fa-check"></i><b>5.2</b> Training Module’s <strong>Questions</strong></a></li>
<li class="chapter" data-level="5.3" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html"><i class="fa fa-check"></i><b>5.3</b> Introduction to FAIR</a><ul>
<li class="chapter" data-level="5.3.1" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#with-this-background-we-can-now-answer-question-1"><i class="fa fa-check"></i><b>5.3.1</b> With this background, we can now answer <strong>Question 1</strong>:</a></li>
<li class="chapter" data-level="5.3.2" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#we-can-also-answer-question-2"><i class="fa fa-check"></i><b>5.3.2</b> We can also answer <strong>Question 2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="breaking-down-fair-letter-by-letter.html"><a href="breaking-down-fair-letter-by-letter.html"><i class="fa fa-check"></i><b>5.4</b> Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="5.5" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html"><i class="fa fa-check"></i><b>5.5</b> F (Findable) in FAIR</a><ul>
<li class="chapter" data-level="5.5.1" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html#with-this-information-can-can-now-answer-question-3"><i class="fa fa-check"></i><b>5.5.1</b> With this information, can can now answer <strong>Question 3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="a-accessible-in-fair.html"><a href="a-accessible-in-fair.html"><i class="fa fa-check"></i><b>5.6</b> A (Accessible) in FAIR</a></li>
<li class="chapter" data-level="5.7" data-path="i-interoperable-in-fair.html"><a href="i-interoperable-in-fair.html"><i class="fa fa-check"></i><b>5.7</b> I (Interoperable) in FAIR</a></li>
<li class="chapter" data-level="5.8" data-path="r-reusable-in-fair.html"><a href="r-reusable-in-fair.html"><i class="fa fa-check"></i><b>5.8</b> R (Reusable) in FAIR</a></li>
<li class="chapter" data-level="5.9" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html"><i class="fa fa-check"></i><b>5.9</b> What does this mean to you?</a><ul>
<li class="chapter" data-level="5.9.1" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html#with-this-information-can-can-now-answer-question-4"><i class="fa fa-check"></i><b>5.9.1</b> With this information, can can now answer <strong>Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.10" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html"><i class="fa fa-check"></i><b>5.10</b> Data Repositories for Sharing of Data</a><ul>
<li class="chapter" data-level="5.10.1" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html#with-this-information-can-can-now-answer-question-5"><i class="fa fa-check"></i><b>5.10.1</b> With this information, can can now answer <strong>Question 5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html"><i class="fa fa-check"></i><b>5.11</b> Helpful Resources on FAIR</a><ul>
<li class="chapter" data-level="5.11.1" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-training-resources-on-fair"><i class="fa fa-check"></i><b>5.11.1</b> Additional Training Resources on FAIR</a></li>
<li class="chapter" data-level="5.11.2" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-examples-and-documents-on-fair"><i class="fa fa-check"></i><b>5.11.2</b> Additional Examples and Documents on FAIR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="dose-response-modeling.html"><a href="dose-response-modeling.html"><i class="fa fa-check"></i><b>6</b> Dose-Response Modeling</a><ul>
<li class="chapter" data-level="6.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html"><i class="fa fa-check"></i><b>6.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="6.1.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#introduction-to-dose-response-modeling"><i class="fa fa-check"></i><b>6.1.1</b> Introduction to Dose-Response Modeling</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="introduction-to-training-module-3.html"><a href="introduction-to-training-module-3.html"><i class="fa fa-check"></i><b>6.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="6.3" data-path="training-modules-environmental-health-questions.html"><a href="training-modules-environmental-health-questions.html"><i class="fa fa-check"></i><b>6.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="6.4" data-path="script-preparations-2.html"><a href="script-preparations-2.html"><i class="fa fa-check"></i><b>6.4</b> Script Preparations</a></li>
<li class="chapter" data-level="6.5" data-path="loading-the-example-dataset.html"><a href="loading-the-example-dataset.html"><i class="fa fa-check"></i><b>6.5</b> Loading the Example Dataset</a></li>
<li class="chapter" data-level="6.6" data-path="data-viewing.html"><a href="data-viewing.html"><i class="fa fa-check"></i><b>6.6</b> Data Viewing</a><ul>
<li class="chapter" data-level="6.6.1" data-path="data-viewing.html"><a href="data-viewing.html#with-this-data-summary-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>6.6.1</b> With this data summary, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="plotting-data-in-dose-response.html"><a href="plotting-data-in-dose-response.html"><i class="fa fa-check"></i><b>6.7</b> Plotting Data in Dose-Response</a><ul>
<li class="chapter" data-level="6.7.1" data-path="plotting-data-in-dose-response.html"><a href="plotting-data-in-dose-response.html#basic-plotting-of-data-in-dose-response"><i class="fa fa-check"></i><b>6.7.1</b> Basic Plotting of Data in Dose-Response</a></li>
<li class="chapter" data-level="6.7.2" data-path="plotting-data-in-dose-response.html"><a href="plotting-data-in-dose-response.html#with-these-plots-we-can-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>6.7.2</b> With these plots, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
<li class="chapter" data-level="6.7.3" data-path="plotting-data-in-dose-response.html"><a href="plotting-data-in-dose-response.html#we-can-also-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>6.7.3</b> We can also answer <strong>Environmental Health Question #3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="fitting-curves-to-dose-response-plots.html"><a href="fitting-curves-to-dose-response-plots.html"><i class="fa fa-check"></i><b>6.8</b> Fitting Curves to Dose-Response Plots</a><ul>
<li class="chapter" data-level="6.8.1" data-path="fitting-curves-to-dose-response-plots.html"><a href="fitting-curves-to-dose-response-plots.html#first-try-fitting-a-log-logistic-ll-model"><i class="fa fa-check"></i><b>6.8.1</b> First try fitting a log-logistic (LL) model</a></li>
<li class="chapter" data-level="6.8.2" data-path="fitting-curves-to-dose-response-plots.html"><a href="fitting-curves-to-dose-response-plots.html#lets-next-try-fitting-a-weibull-model"><i class="fa fa-check"></i><b>6.8.2</b> Let’s next try fitting a Weibull model</a></li>
<li class="chapter" data-level="6.8.3" data-path="fitting-curves-to-dose-response-plots.html"><a href="fitting-curves-to-dose-response-plots.html#with-this-we-can-now-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>6.8.3</b> With this, we can now answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="6.8.4" data-path="fitting-curves-to-dose-response-plots.html"><a href="fitting-curves-to-dose-response-plots.html#lets-try-fitting-another-model-fit-based-on-asymptotic-regression-modeling"><i class="fa fa-check"></i><b>6.8.4</b> Let’s try fitting another model fit based on asymptotic regression modeling</a></li>
<li class="chapter" data-level="6.8.5" data-path="fitting-curves-to-dose-response-plots.html"><a href="fitting-curves-to-dose-response-plots.html#important-note-on-the-variety-of-curve-fit-models-to-consider"><i class="fa fa-check"></i><b>6.8.5</b> Important note on the variety of curve fit models to consider</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="comparing-curve-fits.html"><a href="comparing-curve-fits.html"><i class="fa fa-check"></i><b>6.9</b> Comparing Curve Fits</a></li>
<li class="chapter" data-level="6.10" data-path="curve-fitting-for-liver-outcomes.html"><a href="curve-fitting-for-liver-outcomes.html"><i class="fa fa-check"></i><b>6.10</b> Curve Fitting for Liver Outcomes</a><ul>
<li class="chapter" data-level="6.10.1" data-path="curve-fitting-for-liver-outcomes.html"><a href="curve-fitting-for-liver-outcomes.html#with-this-we-can-now-answer-environmental-health-question-5"><i class="fa fa-check"></i><b>6.10.1</b> With this, we can now answer <strong>Environmental Health Question #5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="deriving-the-final-benchmark-dose-bmd-estimates.html"><a href="deriving-the-final-benchmark-dose-bmd-estimates.html"><i class="fa fa-check"></i><b>6.11</b> Deriving the Final Benchmark Dose (BMD) Estimates</a><ul>
<li class="chapter" data-level="6.11.1" data-path="deriving-the-final-benchmark-dose-bmd-estimates.html"><a href="deriving-the-final-benchmark-dose-bmd-estimates.html#with-this-we-can-now-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>6.11.1</b> With this, we can now answer <strong>Environmental Health Question #6</strong>:</a></li>
<li class="chapter" data-level="6.11.2" data-path="deriving-the-final-benchmark-dose-bmd-estimates.html"><a href="deriving-the-final-benchmark-dose-bmd-estimates.html#with-this-we-can-now-answer-environmental-health-question-7"><i class="fa fa-check"></i><b>6.11.2</b> With this, we can now answer <strong>Environmental Health Question #7</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.12" data-path="concluding-remarks-5.html"><a href="concluding-remarks-5.html"><i class="fa fa-check"></i><b>6.12</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i><b>7</b> Machine Learning and Predictive Modeling for Environmental Health</a><ul>
<li class="chapter" data-level="7.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html"><i class="fa fa-check"></i><b>7.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="7.1.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html#introduction-to-machine-learning-ml-and-predictive-modeling"><i class="fa fa-check"></i><b>7.1.1</b> Introduction to Machine Learning (ML) and Predictive Modeling</a></li>
<li class="chapter" data-level="7.1.2" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html#defining-predictive-modeling-in-the-context-of-toxicology-and-environmental-health"><i class="fa fa-check"></i><b>7.1.2</b> Defining predictive modeling in the context of toxicology and environmental health</a></li>
<li class="chapter" data-level="7.1.3" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html#distinguish-between-machine-learning-ml-and-traditional-statistical-methods"><i class="fa fa-check"></i><b>7.1.3</b> Distinguish between machine learning (ML) and traditional statistical methods</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="introduction-to-training-module-4.html"><a href="introduction-to-training-module-4.html"><i class="fa fa-check"></i><b>7.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="7.3" data-path="training-modules-environmental-health-questions-1.html"><a href="training-modules-environmental-health-questions-1.html"><i class="fa fa-check"></i><b>7.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="7.4" data-path="script-preparations-3.html"><a href="script-preparations-3.html"><i class="fa fa-check"></i><b>7.4</b> Script Preparations</a></li>
<li class="chapter" data-level="7.5" data-path="loading-the-example-dataset-1.html"><a href="loading-the-example-dataset-1.html"><i class="fa fa-check"></i><b>7.5</b> Loading the Example Dataset</a></li>
<li class="chapter" data-level="7.6" data-path="data-viewing-1.html"><a href="data-viewing-1.html"><i class="fa fa-check"></i><b>7.6</b> Data Viewing</a></li>
<li class="chapter" data-level="7.7" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html"><i class="fa fa-check"></i><b>7.7</b> Evaluating the Original Physicochemical Property Data across Substances</a><ul>
<li class="chapter" data-level="7.7.1" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html#with-these-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>7.7.1</b> With these, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html"><i class="fa fa-check"></i><b>7.8</b> Identifying Clusters of Chemicals through k-means</a><ul>
<li class="chapter" data-level="7.8.1" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html#with-this-we-can-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>7.8.1</b> With this, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html"><i class="fa fa-check"></i><b>7.9</b> Principal Component Analysis (PCA)</a><ul>
<li class="chapter" data-level="7.9.1" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-these-results-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>7.9.1</b> With these results, we can answer <strong>Environmental Health Question #3</strong>:</a></li>
<li class="chapter" data-level="7.9.2" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>7.9.2</b> With this, we can answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="7.9.3" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-5"><i class="fa fa-check"></i><b>7.9.3</b> With this, we can answer <strong>Environmental Health Question #5</strong>:</a></li>
<li class="chapter" data-level="7.9.4" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#we-can-also-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>7.9.4</b> We can also answer <strong>Environmental Health Question #6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="concluding-remarks-6.html"><a href="concluding-remarks-6.html"><i class="fa fa-check"></i><b>7.10</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html"><i class="fa fa-check"></i><b>8</b> Mixtures Analysis</a><ul>
<li class="chapter" data-level="8.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html"><i class="fa fa-check"></i><b>8.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="8.1.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html#introduction-to-mixtures-toxicology-and-in-silico-modeling-to-address-mixtures"><i class="fa fa-check"></i><b>8.1.1</b> Introduction to Mixtures Toxicology and <em>In Silico</em> Modeling to Address Mixtures</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="introduction-to-training-module-5.html"><a href="introduction-to-training-module-5.html"><i class="fa fa-check"></i><b>8.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="8.3" data-path="training-modules-environmental-health-questions-2.html"><a href="training-modules-environmental-health-questions-2.html"><i class="fa fa-check"></i><b>8.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="8.4" data-path="script-preparations-4.html"><a href="script-preparations-4.html"><i class="fa fa-check"></i><b>8.4</b> Script Preparations</a></li>
<li class="chapter" data-level="8.5" data-path="loading-the-example-ginkgo-biloba-datasets.html"><a href="loading-the-example-ginkgo-biloba-datasets.html"><i class="fa fa-check"></i><b>8.5</b> Loading the Example <em>Ginkgo biloba</em> Datasets</a></li>
<li class="chapter" data-level="8.6" data-path="data-viewing-2.html"><a href="data-viewing-2.html"><i class="fa fa-check"></i><b>8.6</b> Data Viewing</a><ul>
<li class="chapter" data-level="8.6.1" data-path="data-viewing-2.html"><a href="data-viewing-2.html#lets-also-see-what-kind-of-data-are-organized-within-the-datasets"><i class="fa fa-check"></i><b>8.6.1</b> Let’s also see what kind of data are organized within the datasets</a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>8.7</b> Chemistry-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="8.7.1" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-chemical-profiles"><i class="fa fa-check"></i><b>8.7.1</b> This plot tells us a lot about sample groupings based on chemical profiles!</a></li>
<li class="chapter" data-level="8.7.2" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-questions-1-2"><i class="fa fa-check"></i><b>8.7.2</b> With this, we can answer <strong>Environmental Health Questions 1-2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><i class="fa fa-check"></i><b>8.8</b> Creating a Heat Map of the Mixtures Chemistry Data</a><ul>
<li class="chapter" data-level="8.8.1" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#this-plot-tells-us-a-lot-about-the-individual-chemicals-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>8.8.1</b> This plot tells us a lot about the individual chemicals that differentiate the sample groupings</a></li>
<li class="chapter" data-level="8.8.2" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-this-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>8.8.2</b> With this, we can answer <strong>Environmental Health Question 3</strong>:</a></li>
<li class="chapter" data-level="8.8.3" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#lets-now-revisit-the-pca-plot"><i class="fa fa-check"></i><b>8.8.3</b> Let’s now revisit the PCA plot</a></li>
<li class="chapter" data-level="8.8.4" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-these-plots-side-by-side-we-can-now-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>8.8.4</b> With these plots, side-by-side, we can now answer <strong>Environmental Health Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.9" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>8.9</b> Toxicity-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="8.9.1" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-toxicity-profiles"><i class="fa fa-check"></i><b>8.9.1</b> This plot tells us a lot about sample groupings based on toxicity profiles!</a></li>
<li class="chapter" data-level="8.9.2" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-question-5-1"><i class="fa fa-check"></i><b>8.9.2</b> With this, we can answer <em>Environmental Health Question 5</em>:</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><i class="fa fa-check"></i><b>8.10</b> Creating a Heat Map of the Mixtures Toxicity Data</a><ul>
<li class="chapter" data-level="8.10.1" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#this-plot-tells-us-a-lot-about-the-individual-genes-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>8.10.1</b> This plot tells us a lot about the individual genes that differentiate the sample groupings</a></li>
<li class="chapter" data-level="8.10.2" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#with-this-we-can-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>8.10.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.11" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><i class="fa fa-check"></i><b>8.11</b> Comparing Results from the Chemistry vs. Toxicity Sufficient Similarity Analyses</a><ul>
<li class="chapter" data-level="8.11.1" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html#visual-representation-of-some-of-the-major-grouping-similarities-vs-differences"><i class="fa fa-check"></i><b>8.11.1</b> Visual representation of some of the major grouping similarities vs differences</a></li>
</ul></li>
<li class="chapter" data-level="8.12" data-path="these-plots-can-help-us-answer-environmental-health-question-7.html"><a href="these-plots-can-help-us-answer-environmental-health-question-7.html"><i class="fa fa-check"></i><b>8.12</b> These plots can help us answer <strong>Environmental Health Question 7</strong>:</a></li>
<li class="chapter" data-level="8.13" data-path="concluding-remarks-7.html"><a href="concluding-remarks-7.html"><i class="fa fa-check"></i><b>8.13</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html"><i class="fa fa-check"></i><b>9</b> -Omics Analyses and Systems Biology</a></li>
<li class="chapter" data-level="10" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i><b>10</b> Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="11" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html"><i class="fa fa-check"></i><b>11</b> Read-Across Toxicity Predictions</a></li>
<li class="chapter" data-level="12" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html"><i class="fa fa-check"></i><b>12</b> Environmental Health Databases: Comparative Toxicogenomics Database</a><ul>
<li class="chapter" data-level="12.1" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html"><i class="fa fa-check"></i><b>12.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="12.1.1" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html#introduction-to-comparative-toxicogenomics-database-ctd"><i class="fa fa-check"></i><b>12.1.1</b> Introduction to Comparative Toxicogenomics Database (CTD)</a></li>
<li class="chapter" data-level="12.1.2" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html#introduction-to-training-module-6"><i class="fa fa-check"></i><b>12.1.2</b> Introduction to Training Module</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="training-modules-environmental-health-questions-3.html"><a href="training-modules-environmental-health-questions-3.html"><i class="fa fa-check"></i><b>12.2</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="12.3" data-path="script-preparations-5.html"><a href="script-preparations-5.html"><i class="fa fa-check"></i><b>12.3</b> Script Preparations</a></li>
<li class="chapter" data-level="12.4" data-path="organizing-example-dataset-from-ctd.html"><a href="organizing-example-dataset-from-ctd.html"><i class="fa fa-check"></i><b>12.4</b> Organizing Example Dataset from CTD</a></li>
<li class="chapter" data-level="12.5" data-path="loading-the-example-ctd-dataset-into-r.html"><a href="loading-the-example-ctd-dataset-into-r.html"><i class="fa fa-check"></i><b>12.5</b> Loading the Example CTD Dataset into R</a></li>
<li class="chapter" data-level="12.6" data-path="data-viewing-3.html"><a href="data-viewing-3.html"><i class="fa fa-check"></i><b>12.6</b> Data Viewing</a></li>
<li class="chapter" data-level="12.7" data-path="filtering-the-data-for-genes-with-altered-expression.html"><a href="filtering-the-data-for-genes-with-altered-expression.html"><i class="fa fa-check"></i><b>12.7</b> Filtering the Data for Genes with Altered Expression</a><ul>
<li class="chapter" data-level="12.7.1" data-path="filtering-the-data-for-genes-with-altered-expression.html"><a href="filtering-the-data-for-genes-with-altered-expression.html#with-this-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>12.7.1</b> With this, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="identifying-genes-under-epigenetic-control.html"><a href="identifying-genes-under-epigenetic-control.html"><i class="fa fa-check"></i><b>12.8</b> Identifying Genes under Epigenetic Control</a></li>
<li class="chapter" data-level="12.9" data-path="method-1-for-list-comparisons-merging.html"><a href="method-1-for-list-comparisons-merging.html"><i class="fa fa-check"></i><b>12.9</b> Method 1 for list comparisons: Merging</a><ul>
<li class="chapter" data-level="12.9.1" data-path="method-1-for-list-comparisons-merging.html"><a href="method-1-for-list-comparisons-merging.html#with-this-we-can-answer-environmental-health-question-2-1"><i class="fa fa-check"></i><b>12.9.1</b> With this, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="12.10" data-path="method-2-for-list-comparisons-intersection.html"><a href="method-2-for-list-comparisons-intersection.html"><i class="fa fa-check"></i><b>12.10</b> Method 2 for list comparisons: Intersection</a></li>
<li class="chapter" data-level="12.11" data-path="concluding-remarks-8.html"><a href="concluding-remarks-8.html"><i class="fa fa-check"></i><b>12.11</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html"><i class="fa fa-check"></i><b>13</b> Environmental Health Databases: Gene Expression Omnibus</a><ul>
<li class="chapter" data-level="13.1" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html"><i class="fa fa-check"></i><b>13.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="13.1.1" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html#introduction-to-the-environmental-health-database-gene-expression-omnibus-geo"><i class="fa fa-check"></i><b>13.1.1</b> Introduction to the Environmental Health Database, Gene Expression Omnibus (GEO)</a></li>
<li class="chapter" data-level="13.1.2" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html#introduction-to-training-module-7"><i class="fa fa-check"></i><b>13.1.2</b> Introduction to Training Module</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="training-modules-environmental-health-questions-4.html"><a href="training-modules-environmental-health-questions-4.html"><i class="fa fa-check"></i><b>13.2</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="13.3" data-path="script-preparations-6.html"><a href="script-preparations-6.html"><i class="fa fa-check"></i><b>13.3</b> Script Preparations</a></li>
<li class="chapter" data-level="13.4" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html"><i class="fa fa-check"></i><b>13.4</b> Loading and Organizing the Example Dataset</a><ul>
<li class="chapter" data-level="13.4.1" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html#manually-downloading-and-uploading-geo-files"><i class="fa fa-check"></i><b>13.4.1</b> 1. Manually Downloading and Uploading GEO Files</a></li>
<li class="chapter" data-level="13.4.2" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html#uploading-and-organizing-geo-files-through-the-geoquery-package"><i class="fa fa-check"></i><b>13.4.2</b> 2. Uploading and Organizing GEO Files through the GEOquery Package</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><i class="fa fa-check"></i><b>13.5</b> Adding Gene Symbol Information to -Omic Data Sets through Platform Annotation Files</a><ul>
<li class="chapter" data-level="13.5.1" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html#with-this-we-can-now-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>13.5.1</b> With this, we can now answer <strong>Environmental Health Question #1</strong>:</a></li>
<li class="chapter" data-level="13.5.2" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html#we-can-also-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>13.5.2</b> We can also answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="visualizing-gene-expression-data-using-boxplots-and-heat-maps.html"><a href="visualizing-gene-expression-data-using-boxplots-and-heat-maps.html"><i class="fa fa-check"></i><b>13.6</b> Visualizing Gene Expression Data using Boxplots and Heat Maps</a></li>
<li class="chapter" data-level="13.7" data-path="boxplot-visualizations-1.html"><a href="boxplot-visualizations-1.html"><i class="fa fa-check"></i><b>13.7</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="13.8" data-path="heatmap-visualizations-1.html"><a href="heatmap-visualizations-1.html"><i class="fa fa-check"></i><b>13.8</b> Heatmap Visualizations</a><ul>
<li class="chapter" data-level="13.8.1" data-path="heatmap-visualizations-1.html"><a href="heatmap-visualizations-1.html#we-can-also-answer-environmental-health-question-3-1"><i class="fa fa-check"></i><b>13.8.1</b> We can also answer <strong>Environmental Health Question #3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="statistical-analyses-to-identify-genes-altered-by-formaldehyde.html"><a href="statistical-analyses-to-identify-genes-altered-by-formaldehyde.html"><i class="fa fa-check"></i><b>13.9</b> Statistical Analyses to Identify Genes altered by Formaldehyde</a></li>
<li class="chapter" data-level="13.10" data-path="interpreting-results.html"><a href="interpreting-results.html"><i class="fa fa-check"></i><b>13.10</b> Interpreting Results</a><ul>
<li class="chapter" data-level="13.10.1" data-path="interpreting-results.html"><a href="interpreting-results.html#with-this-we-can-answer-environmental-health-question-4-1"><i class="fa fa-check"></i><b>13.10.1</b> With this, we can answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="13.10.2" data-path="interpreting-results.html"><a href="interpreting-results.html#with-this-we-can-answer-environmental-health-question-5-2"><i class="fa fa-check"></i><b>13.10.2</b> With this, we can answer <strong>Environmental Health Question #5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="13.11" data-path="concluding-remarks-9.html"><a href="concluding-remarks-9.html"><i class="fa fa-check"></i><b>13.11</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="database-integration-air-quality-study-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-study-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i><b>14</b> Database Integration: Air Quality Study, Mortality, and Environmental Justice Data</a></li>
<li class="chapter" data-level="15" data-path="additional-resources.html"><a href="additional-resources.html"><i class="fa fa-check"></i><b>15</b> Additional Resources</a><ul>
<li class="chapter" data-level="15.1" data-path="r-programming-resources.html"><a href="r-programming-resources.html"><i class="fa fa-check"></i><b>15.1</b> R Programming Resources:</a></li>
<li class="chapter" data-level="15.2" data-path="r-packages-resources.html"><a href="r-packages-resources.html"><i class="fa fa-check"></i><b>15.2</b> R Packages Resources:</a></li>
<li class="chapter" data-level="15.3" data-path="community-discussions-on-r-and-r-packages.html"><a href="community-discussions-on-r-and-r-packages.html"><i class="fa fa-check"></i><b>15.3</b> Community Discussions on R and R Packages:</a></li>
<li class="chapter" data-level="15.4" data-path="data-science-and-statistical-analysis-resources.html"><a href="data-science-and-statistical-analysis-resources.html"><i class="fa fa-check"></i><b>15.4</b> Data Science and Statistical Analysis Resources:</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="loading-and-organizing-the-example-dataset" class="section level2">
<h2><span class="header-section-number">13.4</span> Loading and Organizing the Example Dataset</h2>
<p>Let’s start by loading the GEO dataset needed for this training module. As explained in the introduction, this module walks through two methods of uploading GEO data: manual option vs automatic option using the GEOquery package. These two methods are detailed below.
<br></p>
<div id="manually-downloading-and-uploading-geo-files" class="section level3">
<h3><span class="header-section-number">13.4.1</span> 1. Manually Downloading and Uploading GEO Files</h3>
<p>In this first method, we will navigate to the datasets within the GEO website, manually download its associated text data file, save it in our working directory, and then upload it into our global environment in R.</p>
<p>For the purposes of this training exercise, we manually downloaded the GEO series matrix file from the GEO series webpage, located at: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE42394" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE42394</a>. The specific file that was downloaded was noted as <em>GSE42394_series_matrix.txt</em>, pulled by clicking on the link indicated by the red arrow from the GEO series webpage:</p>
<p><img src="_book/TAME_Toolkit_files/figure-html/Module3_2_GEO_Screenshot1.png" width="506" /></p>
<p>For simplicity, we also have already pre-filtered this file for the samples we are interested in, focusing on the rat nasal gene expression data after 7 days of exposure to gaseous formaldehyde. This filtered file was saved as: <em>GSE42394_series_matrix_filtered.txt</em>
<br></p>
<p>At this point, we can simply read in this pre-filtered text file for the purposes of this training module</p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="loading-and-organizing-the-example-dataset.html#cb347-1"></a>geodata_manual &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="dt">file=</span><span class="st">&quot;Module3_2/Module3_2_GSE42394_series_matrix_filtered.txt&quot;</span>, <span class="dt">header=</span>T)</span></code></pre></div>
<p>Because this is a manual approach, we have to also manually define the treated and untreated samples (based on manually opening the surrounding metadata from the GEO webpage)</p>
<p>Manually defining treated and untreated for these samples of interest:</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="loading-and-organizing-the-example-dataset.html#cb348-1"></a>exposed_manual &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GSM1150940&quot;</span>, <span class="st">&quot;GSM1150941&quot;</span>, <span class="st">&quot;GSM1150942&quot;</span>)</span>
<span id="cb348-2"><a href="loading-and-organizing-the-example-dataset.html#cb348-2"></a>unexposed_manual &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GSM1150937&quot;</span>, <span class="st">&quot;GSM1150938&quot;</span>, <span class="st">&quot;GSM1150939&quot;</span>)</span></code></pre></div>
<p><br><br></p>
</div>
<div id="uploading-and-organizing-geo-files-through-the-geoquery-package" class="section level3">
<h3><span class="header-section-number">13.4.2</span> 2. Uploading and Organizing GEO Files through the GEOquery Package</h3>
<p>In this second method, we will leverage the GEOquery package, which allows for easier downloading and reading in of data from GEO without having to manually download raw text files, and manually assign sample attributes (e.g., exposed vs unexposed). This package is set-up to automatically merge sample information from GEO metadata files with raw genome-wide datasets.</p>
<p>Let’s first use the getGEO function (from the GEOquery package) to load data from our series matrix
<em>Note that this line of code may take a couple of minutes</em></p>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="loading-and-organizing-the-example-dataset.html#cb349-1"></a>geo.getGEO.data &lt;-<span class="st"> </span><span class="kw">getGEO</span>(<span class="dt">filename=</span><span class="st">&#39;Module3_2/Module3_2_GSE42394_series_matrix.txt&#39;</span>)</span></code></pre></div>
<p>One of the reasons the getGEO package so helpful is that we can automatically link a dataset with nicely organized sample information using the pData() function</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="loading-and-organizing-the-example-dataset.html#cb350-1"></a>sampleInfo &lt;-<span class="st"> </span><span class="kw">pData</span>(geo.getGEO.data)</span></code></pre></div>
<p>Let’s view this sample information / metadata file, first by viewing what the column headers are</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="loading-and-organizing-the-example-dataset.html#cb351-1"></a><span class="kw">colnames</span>(sampleInfo)</span></code></pre></div>
<pre><code>##  [1] &quot;title&quot;                   &quot;geo_accession&quot;          
##  [3] &quot;status&quot;                  &quot;submission_date&quot;        
##  [5] &quot;last_update_date&quot;        &quot;type&quot;                   
##  [7] &quot;channel_count&quot;           &quot;source_name_ch1&quot;        
##  [9] &quot;organism_ch1&quot;            &quot;characteristics_ch1&quot;    
## [11] &quot;characteristics_ch1.1&quot;   &quot;characteristics_ch1.2&quot;  
## [13] &quot;characteristics_ch1.3&quot;   &quot;characteristics_ch1.4&quot;  
## [15] &quot;characteristics_ch1.5&quot;   &quot;treatment_protocol_ch1&quot; 
## [17] &quot;growth_protocol_ch1&quot;     &quot;molecule_ch1&quot;           
## [19] &quot;extract_protocol_ch1&quot;    &quot;label_ch1&quot;              
## [21] &quot;label_protocol_ch1&quot;      &quot;taxid_ch1&quot;              
## [23] &quot;hyb_protocol&quot;            &quot;scan_protocol&quot;          
## [25] &quot;description&quot;             &quot;data_processing&quot;        
## [27] &quot;platform_id&quot;             &quot;contact_name&quot;           
## [29] &quot;contact_email&quot;           &quot;contact_department&quot;     
## [31] &quot;contact_institute&quot;       &quot;contact_address&quot;        
## [33] &quot;contact_city&quot;            &quot;contact_zip/postal_code&quot;
## [35] &quot;contact_country&quot;         &quot;supplementary_file&quot;     
## [37] &quot;data_row_count&quot;          &quot;age:ch1&quot;                
## [39] &quot;cell type:ch1&quot;           &quot;gender:ch1&quot;             
## [41] &quot;strain:ch1&quot;              &quot;time:ch1&quot;               
## [43] &quot;treatment:ch1&quot;</code></pre>
<p>Then viewing the first five columns</p>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="loading-and-organizing-the-example-dataset.html#cb353-1"></a>sampleInfo[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##                                                    title geo_accession
## GSM1150937            Nose_7DayControl_Rep1 [Affymetrix]    GSM1150937
## GSM1150938            Nose_7DayControl_Rep2 [Affymetrix]    GSM1150938
## GSM1150939            Nose_7DayControl_Rep3 [Affymetrix]    GSM1150939
## GSM1150940            Nose_7DayExposed_Rep1 [Affymetrix]    GSM1150940
## GSM1150941            Nose_7DayExposed_Rep2 [Affymetrix]    GSM1150941
## GSM1150942            Nose_7DayExposed_Rep3 [Affymetrix]    GSM1150942
## GSM1150943 WhiteBloodCells_7DayControl_Rep1 [Affymetrix]    GSM1150943
## GSM1150944 WhiteBloodCells_7DayControl_Rep2 [Affymetrix]    GSM1150944
## GSM1150945 WhiteBloodCells_7DayControl_Rep3 [Affymetrix]    GSM1150945
## GSM1150946 WhiteBloodCells_7DayExposed_Rep1 [Affymetrix]    GSM1150946
##                           status submission_date last_update_date
## GSM1150937 Public on Jan 07 2014     May 29 2013      Jan 07 2014
## GSM1150938 Public on Jan 07 2014     May 29 2013      Jan 07 2014
## GSM1150939 Public on Jan 07 2014     May 29 2013      Jan 07 2014
## GSM1150940 Public on Jan 07 2014     May 29 2013      Jan 07 2014
## GSM1150941 Public on Jan 07 2014     May 29 2013      Jan 07 2014
## GSM1150942 Public on Jan 07 2014     May 29 2013      Jan 07 2014
## GSM1150943 Public on Jan 07 2014     May 29 2013      Jan 07 2014
## GSM1150944 Public on Jan 07 2014     May 29 2013      Jan 07 2014
## GSM1150945 Public on Jan 07 2014     May 29 2013      Jan 07 2014
## GSM1150946 Public on Jan 07 2014     May 29 2013      Jan 07 2014</code></pre>
<p>This shows that each sample is provided with a unique number starting with “GSM”, and these are described by information summarized in the “title” column. We can also see that these data were made public on Jan 7, 2014.</p>
<p>Let’s view the next five columns</p>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="loading-and-organizing-the-example-dataset.html#cb355-1"></a>sampleInfo[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,<span class="dv">6</span><span class="op">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##            type channel_count                                 source_name_ch1
## GSM1150937  RNA             1        Nasal epithelial cells, 7 day, unexposed
## GSM1150938  RNA             1        Nasal epithelial cells, 7 day, unexposed
## GSM1150939  RNA             1        Nasal epithelial cells, 7 day, unexposed
## GSM1150940  RNA             1          Nasal epithelial cells, 7 day, exposed
## GSM1150941  RNA             1          Nasal epithelial cells, 7 day, exposed
## GSM1150942  RNA             1          Nasal epithelial cells, 7 day, exposed
## GSM1150943  RNA             1 Circulating white blood cells, 7 day, unexposed
## GSM1150944  RNA             1 Circulating white blood cells, 7 day, unexposed
## GSM1150945  RNA             1 Circulating white blood cells, 7 day, unexposed
## GSM1150946  RNA             1   Circulating white blood cells, 7 day, exposed
##                 organism_ch1 characteristics_ch1
## GSM1150937 Rattus norvegicus        gender: male
## GSM1150938 Rattus norvegicus        gender: male
## GSM1150939 Rattus norvegicus        gender: male
## GSM1150940 Rattus norvegicus        gender: male
## GSM1150941 Rattus norvegicus        gender: male
## GSM1150942 Rattus norvegicus        gender: male
## GSM1150943 Rattus norvegicus        gender: male
## GSM1150944 Rattus norvegicus        gender: male
## GSM1150945 Rattus norvegicus        gender: male
## GSM1150946 Rattus norvegicus        gender: male</code></pre>
<p>We can see that information is provided here surrounding the type of sample that was analyzed (i.e., RNA), more information on the collected samples within the column ‘source_name_ch1’, and the organism (rat) is provided in the ‘organism_ch1’ column.</p>
<p>More detailed metadata information is provided throughout this file, as seen when viewing the column headers above</p>
<div id="now-we-can-use-this-information-to-define-the-samples-we-want-to-analyze." class="section level4">
<h4><span class="header-section-number">13.4.2.1</span> Now, we can use this information to define the samples we want to analyze.</h4>
<p>Note that this is the same step we did manually above.</p>
<p>In this training exercise, we are focusing on responses in the nose, so we can easily filter for cell type = Nasal epithelial cells (specifically in the “cell type:ch1” variable)</p>
<p>We are also focusing on responses collected after 7 days of exposure, which we can filter for using time = 7 day (specifically in the “time:ch1” variable).</p>
<p>We will also define exposed and unexposed samples using the variable “treatment:ch1”.</p>
<p>First, let’s subset the sampleInfo dataframe to just keep the samples we’re interested in</p>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="loading-and-organizing-the-example-dataset.html#cb357-1"></a><span class="co"># Define a vector variable (here we call it &#39;keep&#39;) that will store rows we want to keep</span></span>
<span id="cb357-2"><a href="loading-and-organizing-the-example-dataset.html#cb357-2"></a>keep &lt;-<span class="st"> </span><span class="kw">rownames</span>(sampleInfo[<span class="kw">which</span>(sampleInfo<span class="op">$</span><span class="st">`</span><span class="dt">cell type:ch1</span><span class="st">`</span><span class="op">==</span><span class="st">&quot;Nasal epithelial cells&quot;</span> </span>
<span id="cb357-3"><a href="loading-and-organizing-the-example-dataset.html#cb357-3"></a>                                  <span class="op">&amp;</span><span class="st"> </span>sampleInfo<span class="op">$</span><span class="st">`</span><span class="dt">time:ch1</span><span class="st">`</span><span class="op">==</span><span class="st">&quot;7 day&quot;</span>),])</span>
<span id="cb357-4"><a href="loading-and-organizing-the-example-dataset.html#cb357-4"></a></span>
<span id="cb357-5"><a href="loading-and-organizing-the-example-dataset.html#cb357-5"></a><span class="co"># Then subset the sample info for just those samples we defined in keep variable</span></span>
<span id="cb357-6"><a href="loading-and-organizing-the-example-dataset.html#cb357-6"></a>sampleInfo &lt;-<span class="st"> </span>sampleInfo[keep,]</span></code></pre></div>
<p><br></p>
<p>Next, we can pull the exposed and unexposed animal IDs. Let’s first see how these are labeled within the “treatment:ch1” variable</p>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="loading-and-organizing-the-example-dataset.html#cb358-1"></a><span class="kw">unique</span>(sampleInfo<span class="op">$</span><span class="st">`</span><span class="dt">treatment:ch1</span><span class="st">`</span>)</span></code></pre></div>
<pre><code>## [1] &quot;unexposed&quot;          &quot;2 ppm formaldehyde&quot;</code></pre>
<p><br></p>
<p>And then search for the rows of data, pulling the sample animal IDs (which are in the variable ‘geo_accession’)</p>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="loading-and-organizing-the-example-dataset.html#cb360-1"></a>exposedIDs  &lt;-<span class="st"> </span>sampleInfo[<span class="kw">which</span>(sampleInfo<span class="op">$</span><span class="st">`</span><span class="dt">treatment:ch1</span><span class="st">`</span><span class="op">==</span><span class="st">&quot;2 ppm formaldehyde&quot;</span>), <span class="st">&quot;geo_accession&quot;</span>]</span>
<span id="cb360-2"><a href="loading-and-organizing-the-example-dataset.html#cb360-2"></a>unexposedIDs  &lt;-<span class="st"> </span>sampleInfo[<span class="kw">which</span>(sampleInfo<span class="op">$</span><span class="st">`</span><span class="dt">treatment:ch1</span><span class="st">`</span><span class="op">==</span><span class="st">&quot;unexposed&quot;</span>), <span class="st">&quot;geo_accession&quot;</span>]</span></code></pre></div>
</div>
<div id="the-next-step-is-to-pull-the-expression-data-we-want-to-use-in-our-analyses." class="section level4">
<h4><span class="header-section-number">13.4.2.2</span> The next step is to pull the expression data we want to use in our analyses.</h4>
<p>The GEOquery function, exprs(), allows us to easily pull these data. Here, we can pull the data we’re interested in using the exprs() function, while defining the data we want to pull based off our previously generated ‘keep’ vector.</p>
<div class="sourceCode" id="cb361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb361-1"><a href="loading-and-organizing-the-example-dataset.html#cb361-1"></a><span class="co">#As a reminder, this is what the &#39;keep&#39; vector includes (i.e., animal IDs that we&#39;re interested in)</span></span>
<span id="cb361-2"><a href="loading-and-organizing-the-example-dataset.html#cb361-2"></a>keep</span></code></pre></div>
<pre><code>## [1] &quot;GSM1150937&quot; &quot;GSM1150938&quot; &quot;GSM1150939&quot; &quot;GSM1150940&quot; &quot;GSM1150941&quot;
## [6] &quot;GSM1150942&quot;</code></pre>
<div class="sourceCode" id="cb363"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb363-1"><a href="loading-and-organizing-the-example-dataset.html#cb363-1"></a><span class="co">#Using the exprs() function</span></span>
<span id="cb363-2"><a href="loading-and-organizing-the-example-dataset.html#cb363-2"></a>geodata &lt;-<span class="st"> </span><span class="kw">exprs</span>(geo.getGEO.data[,keep])</span></code></pre></div>
<p>Let’s view the full dataset as is now</p>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb364-1"><a href="loading-and-organizing-the-example-dataset.html#cb364-1"></a><span class="kw">head</span>(geodata)</span></code></pre></div>
<pre><code>##          GSM1150937 GSM1150938 GSM1150939 GSM1150940 GSM1150941 GSM1150942
## 10700001    5786.60    5830.08    5637.34    5313.33    5557.04    5469.90
## 10700002     192.92     206.86     220.83     183.12     177.16     198.64
## 10700003    1820.98    1795.79    1735.70    1578.02    1681.58    1632.20
## 10700004      66.95      65.61      64.41      60.19      60.41      60.67
## 10700005     770.07     753.41     731.20     684.53     657.25     667.66
## 10700006       5.80       5.35       5.48       5.58       5.39       5.35</code></pre>
<p>This now represents a matrix of data, with animal IDs as column headers and expression levels within the matrix.
<br></p>
</div>
<div id="simplifying-column-names" class="section level4">
<h4><span class="header-section-number">13.4.2.3</span> Simplifying column names</h4>
<p>These column names are not the easiest to interpret, so let’s rename these columns to indicate which animals were from the exposed vs. unexposed groups.</p>
<p>We need to first convert our expression dataset to a dataframe so we can edit columns names, and continue with downstream data manipulations that require dataframe formats</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="loading-and-organizing-the-example-dataset.html#cb366-1"></a>geodata &lt;-<span class="st"> </span><span class="kw">data.frame</span>(geodata)</span></code></pre></div>
<p>Let’s remind ourselves what the column names are</p>
<div class="sourceCode" id="cb367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb367-1"><a href="loading-and-organizing-the-example-dataset.html#cb367-1"></a><span class="kw">colnames</span>(geodata)</span></code></pre></div>
<pre><code>## [1] &quot;GSM1150937&quot; &quot;GSM1150938&quot; &quot;GSM1150939&quot; &quot;GSM1150940&quot; &quot;GSM1150941&quot;
## [6] &quot;GSM1150942&quot;</code></pre>
<p>Which ones of these are exposed vs unexposed animals can be determined by viewing our previously defined vectors</p>
<div class="sourceCode" id="cb369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb369-1"><a href="loading-and-organizing-the-example-dataset.html#cb369-1"></a>exposedIDs</span></code></pre></div>
<pre><code>## [1] &quot;GSM1150940&quot; &quot;GSM1150941&quot; &quot;GSM1150942&quot;</code></pre>
<div class="sourceCode" id="cb371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb371-1"><a href="loading-and-organizing-the-example-dataset.html#cb371-1"></a>unexposedIDs</span></code></pre></div>
<pre><code>## [1] &quot;GSM1150937&quot; &quot;GSM1150938&quot; &quot;GSM1150939&quot;</code></pre>
<p>With this we can tell that the first three listed IDs are from unexposed animals, and the last three IDs are from exposed animals</p>
<p>Let’s simplify the names of these columns to indicate exposure status and replicate number</p>
<div class="sourceCode" id="cb373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb373-1"><a href="loading-and-organizing-the-example-dataset.html#cb373-1"></a><span class="kw">colnames</span>(geodata) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Control_1&quot;</span>, <span class="st">&quot;Control_2&quot;</span>, <span class="st">&quot;Control_3&quot;</span>, <span class="st">&quot;Exposed_1&quot;</span>, <span class="st">&quot;Exposed_2&quot;</span>, <span class="st">&quot;Exposed_3&quot;</span>)</span></code></pre></div>
<p>And we’ll now need to re-define our ‘exposed’ vs ‘unexposed’ vectors for downstream script</p>
<div class="sourceCode" id="cb374"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb374-1"><a href="loading-and-organizing-the-example-dataset.html#cb374-1"></a>exposedIDs &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Exposed_1&quot;</span>, <span class="st">&quot;Exposed_2&quot;</span>, <span class="st">&quot;Exposed_3&quot;</span>)</span>
<span id="cb374-2"><a href="loading-and-organizing-the-example-dataset.html#cb374-2"></a>unexposedIDs &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Control_1&quot;</span>, <span class="st">&quot;Control_2&quot;</span>, <span class="st">&quot;Control_3&quot;</span>)</span></code></pre></div>
<p>Viewing the data again</p>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="loading-and-organizing-the-example-dataset.html#cb375-1"></a><span class="kw">head</span>(geodata)</span></code></pre></div>
<pre><code>##          Control_1 Control_2 Control_3 Exposed_1 Exposed_2 Exposed_3
## 10700001   5786.60   5830.08   5637.34   5313.33   5557.04   5469.90
## 10700002    192.92    206.86    220.83    183.12    177.16    198.64
## 10700003   1820.98   1795.79   1735.70   1578.02   1681.58   1632.20
## 10700004     66.95     65.61     64.41     60.19     60.41     60.67
## 10700005    770.07    753.41    731.20    684.53    657.25    667.66
## 10700006      5.80      5.35      5.48      5.58      5.39      5.35</code></pre>
<p>These data are now looking easier to interpret/analyze. Still, the row identifiers include 8 digit numbers starting with “107…”. We know that this dataset is a gene expression dataset, but these identifiers, in themselves, don’t tell us much about what genes these are referring to. These numeric IDs specifically represent microarray probesetIDs, that were produced by the Affymetrix platform used in the original study.</p>
<p><strong>But how can we tell which genes are represented by these data?!</strong></p>
<p><br><br></p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="script-preparations-6.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
