<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.9 Principal Component Analysis (PCA) | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="6.9 Principal Component Analysis (PCA) | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="6.9 Principal Component Analysis (PCA) | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.9 Principal Component Analysis (PCA) | The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  
  

<meta name="author" content="Kyle Roell, Lauren Koval, Rebecca Boyles, Grace Patlewicz, Caroline Ring, Cynthia Rider, Cavin Ward-Caviness, David M. Reif, Ilona Jaspers, Rebecca C. Fry, and Julia E. Rager" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="identifying-clusters-of-chemicals-through-k-means.html"/>
<link rel="next" href="concluding-remarks-5.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">TAME Toolkit</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#background"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#objectives"><i class="fa fa-check"></i>Objectives</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#module-development-overview"><i class="fa fa-check"></i>Module Development Overview</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#concluding-remarks"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#funding"><i class="fa fa-check"></i>Funding</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#author-affiliations"><i class="fa fa-check"></i>Author Affiliations</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="introduction-to-coding-in-r.html"><a href="introduction-to-coding-in-r.html"><i class="fa fa-check"></i><b>1</b> Introduction to Coding in R</a><ul>
<li class="chapter" data-level="1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html"><i class="fa fa-check"></i><b>1.1</b> General Introduction and Installation of R and RStudio</a><ul>
<li class="chapter" data-level="1.1.1" data-path="general-introduction-and-installation-of-r-and-rstudio.html"><a href="general-introduction-and-installation-of-r-and-rstudio.html#what-is-r"><i class="fa fa-check"></i><b>1.1.1</b> What is R?</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="downloading-and-installing-r.html"><a href="downloading-and-installing-r.html"><i class="fa fa-check"></i><b>1.2</b> Downloading and Installing R</a></li>
<li class="chapter" data-level="1.3" data-path="downloading-and-installing-r-studio.html"><a href="downloading-and-installing-r-studio.html"><i class="fa fa-check"></i><b>1.3</b> Downloading and Installing R Studio</a></li>
<li class="chapter" data-level="1.4" data-path="introduction-to-r-packages.html"><a href="introduction-to-r-packages.html"><i class="fa fa-check"></i><b>1.4</b> Introduction to R Packages</a><ul>
<li class="chapter" data-level="1.4.1" data-path="introduction-to-r-packages.html"><a href="introduction-to-r-packages.html#downloadinginstalling-r-packages"><i class="fa fa-check"></i><b>1.4.1</b> Downloading/Installing R Packages</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="scripting-basics.html"><a href="scripting-basics.html"><i class="fa fa-check"></i><b>1.5</b> Scripting Basics</a><ul>
<li class="chapter" data-level="1.5.1" data-path="scripting-basics.html"><a href="scripting-basics.html#comments"><i class="fa fa-check"></i><b>1.5.1</b> Comments</a></li>
<li class="chapter" data-level="1.5.2" data-path="scripting-basics.html"><a href="scripting-basics.html#starting-code"><i class="fa fa-check"></i><b>1.5.2</b> Starting Code</a></li>
<li class="chapter" data-level="1.5.3" data-path="scripting-basics.html"><a href="scripting-basics.html#setting-your-working-directory"><i class="fa fa-check"></i><b>1.5.3</b> Setting Your Working Directory</a></li>
<li class="chapter" data-level="1.5.4" data-path="scripting-basics.html"><a href="scripting-basics.html#importing-files"><i class="fa fa-check"></i><b>1.5.4</b> Importing Files</a></li>
<li class="chapter" data-level="1.5.5" data-path="scripting-basics.html"><a href="scripting-basics.html#viewing-data"><i class="fa fa-check"></i><b>1.5.5</b> Viewing Data</a></li>
<li class="chapter" data-level="1.5.6" data-path="scripting-basics.html"><a href="scripting-basics.html#exporting-data"><i class="fa fa-check"></i><b>1.5.6</b> Exporting Data</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="concluding-remarks-1.html"><a href="concluding-remarks-1.html"><i class="fa fa-check"></i><b>1.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-organization-basics.html"><a href="data-organization-basics.html"><i class="fa fa-check"></i><b>2</b> Data Organization Basics</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html"><i class="fa fa-check"></i><b>2.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="2.1.1" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#set-your-working-directory-1"><i class="fa fa-check"></i><b>2.1.1</b> Set your working directory</a></li>
<li class="chapter" data-level="2.1.2" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#importing-example-datasets"><i class="fa fa-check"></i><b>2.1.2</b> Importing example datasets</a></li>
<li class="chapter" data-level="2.1.3" data-path="introduction-to-training-module.html"><a href="introduction-to-training-module.html#viewing-example-datasets"><i class="fa fa-check"></i><b>2.1.3</b> Viewing example datasets</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="merging-data-using-base-r-syntax.html"><a href="merging-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.2</b> Merging Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.3" data-path="filtering-and-subsetting-data-using-base-r-syntax.html"><a href="filtering-and-subsetting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.3</b> Filtering and Subsetting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.4" data-path="melting-and-casting-data-using-base-r-syntax.html"><a href="melting-and-casting-data-using-base-r-syntax.html"><i class="fa fa-check"></i><b>2.4</b> Melting and Casting Data using Base R Syntax</a></li>
<li class="chapter" data-level="2.5" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html"><i class="fa fa-check"></i><b>2.5</b> Introduction to Tidyverse</a><ul>
<li class="chapter" data-level="2.5.1" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#downloading-and-loading-the-tidyverse-package"><i class="fa fa-check"></i><b>2.5.1</b> Downloading and Loading the Tidyverse Package</a></li>
<li class="chapter" data-level="2.5.2" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#merging-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.2</b> Merging Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.3" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#filtering-and-subsetting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.3</b> Filtering and Subsetting Data using Tidyverse Syntax</a></li>
<li class="chapter" data-level="2.5.4" data-path="introduction-to-tidyverse.html"><a href="introduction-to-tidyverse.html#melting-and-casting-data-using-tidyverse-syntax"><i class="fa fa-check"></i><b>2.5.4</b> Melting and Casting Data using Tidyverse Syntax</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="concluding-remarks-2.html"><a href="concluding-remarks-2.html"><i class="fa fa-check"></i><b>2.6</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="finding-and-visualizing-data-trend.html"><a href="finding-and-visualizing-data-trend.html"><i class="fa fa-check"></i><b>3</b> Finding and Visualizing Data Trend</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html"><i class="fa fa-check"></i><b>3.1</b> Introduction to Training Module</a><ul>
<li class="chapter" data-level="3.1.1" data-path="introduction-to-training-module-1.html"><a href="introduction-to-training-module-1.html#script-preparations"><i class="fa fa-check"></i><b>3.1.1</b> Script Preparations</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html"><i class="fa fa-check"></i><b>3.2</b> Basic Statistical Tests and Visualizations of Data Trends</a><ul>
<li class="chapter" data-level="3.2.1" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#visualize-and-test-data-for-normality"><i class="fa fa-check"></i><b>3.2.1</b> Visualize and Test Data for Normality</a></li>
<li class="chapter" data-level="3.2.2" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i><b>3.2.2</b> Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="3.2.3" data-path="basic-statistical-tests-and-visualizations-of-data-trends.html"><a href="basic-statistical-tests-and-visualizations-of-data-trends.html#two-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i><b>3.2.3</b> Two-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><i class="fa fa-check"></i><b>3.3</b> Regression Modeling and Visualization: Linear and Logistic Regressions</a><ul>
<li class="chapter" data-level="3.3.1" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#linear-regression"><i class="fa fa-check"></i><b>3.3.1</b> Linear Regression</a></li>
<li class="chapter" data-level="3.3.2" data-path="regression-modeling-and-visualization-linear-and-logistic-regressions.html"><a href="regression-modeling-and-visualization-linear-and-logistic-regressions.html#logistic-regression"><i class="fa fa-check"></i><b>3.3.2</b> Logistic Regression</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="concluding-remarks-3.html"><a href="concluding-remarks-3.html"><i class="fa fa-check"></i><b>3.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="high-dimensional-data-visualizations.html"><a href="high-dimensional-data-visualizations.html"><i class="fa fa-check"></i><b>4</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.1" data-path="introduction-to-training-module-2.html"><a href="introduction-to-training-module-2.html"><i class="fa fa-check"></i><b>4.1</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="4.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html"><i class="fa fa-check"></i><b>4.2</b> Script Preparations</a><ul>
<li class="chapter" data-level="4.2.1" data-path="script-preparations-1.html"><a href="script-preparations-1.html#formatting-dataframes-for-downstream-visualization-code"><i class="fa fa-check"></i><b>4.2.1</b> Formatting dataframes for downstream visualization code</a></li>
<li class="chapter" data-level="4.2.2" data-path="script-preparations-1.html"><a href="script-preparations-1.html#scaling-dataframes-for-downstream-data-visualizations"><i class="fa fa-check"></i><b>4.2.2</b> Scaling dataframes for downstream data visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html"><i class="fa fa-check"></i><b>4.3</b> High-Dimensional Data Visualizations</a><ul>
<li class="chapter" data-level="4.3.1" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#density-plot-visualizations"><i class="fa fa-check"></i><b>4.3.1</b> Density Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.2" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#ggally-visualizations"><i class="fa fa-check"></i><b>4.3.2</b> GGally Visualizations</a></li>
<li class="chapter" data-level="4.3.3" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#boxplot-visualizations"><i class="fa fa-check"></i><b>4.3.3</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="4.3.4" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#correlation-plot-visualizations"><i class="fa fa-check"></i><b>4.3.4</b> Correlation Plot Visualizations</a></li>
<li class="chapter" data-level="4.3.5" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#hierarchical-clustering-visualizations"><i class="fa fa-check"></i><b>4.3.5</b> Hierarchical Clustering Visualizations</a></li>
<li class="chapter" data-level="4.3.6" data-path="high-dimensional-data-visualizations-1.html"><a href="high-dimensional-data-visualizations-1.html#heatmap-visualizations"><i class="fa fa-check"></i><b>4.3.6</b> Heatmap Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="concluding-remarks-4.html"><a href="concluding-remarks-4.html"><i class="fa fa-check"></i><b>4.4</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i><b>5</b> FAIR Data Management Practices</a><ul>
<li class="chapter" data-level="5.1" data-path="background-on-training-module.html"><a href="background-on-training-module.html"><i class="fa fa-check"></i><b>5.1</b> Background on Training Module</a></li>
<li class="chapter" data-level="5.2" data-path="training-modules-questions.html"><a href="training-modules-questions.html"><i class="fa fa-check"></i><b>5.2</b> Training Module’s <strong>Questions</strong></a></li>
<li class="chapter" data-level="5.3" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html"><i class="fa fa-check"></i><b>5.3</b> Introduction to FAIR</a><ul>
<li class="chapter" data-level="5.3.1" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#with-this-background-we-can-now-answer-question-1"><i class="fa fa-check"></i><b>5.3.1</b> With this background, we can now answer <strong>Question 1</strong>:</a></li>
<li class="chapter" data-level="5.3.2" data-path="introduction-to-fair.html"><a href="introduction-to-fair.html#we-can-also-answer-question-2"><i class="fa fa-check"></i><b>5.3.2</b> We can also answer <strong>Question 2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="breaking-down-fair-letter-by-letter.html"><a href="breaking-down-fair-letter-by-letter.html"><i class="fa fa-check"></i><b>5.4</b> Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="5.5" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html"><i class="fa fa-check"></i><b>5.5</b> F (Findable) in FAIR</a><ul>
<li class="chapter" data-level="5.5.1" data-path="f-findable-in-fair.html"><a href="f-findable-in-fair.html#with-this-information-can-can-now-answer-question-3"><i class="fa fa-check"></i><b>5.5.1</b> With this information, can can now answer <strong>Question 3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="a-accessible-in-fair.html"><a href="a-accessible-in-fair.html"><i class="fa fa-check"></i><b>5.6</b> A (Accessible) in FAIR</a></li>
<li class="chapter" data-level="5.7" data-path="i-interoperable-in-fair.html"><a href="i-interoperable-in-fair.html"><i class="fa fa-check"></i><b>5.7</b> I (Interoperable) in FAIR</a></li>
<li class="chapter" data-level="5.8" data-path="r-reusable-in-fair.html"><a href="r-reusable-in-fair.html"><i class="fa fa-check"></i><b>5.8</b> R (Reusable) in FAIR</a></li>
<li class="chapter" data-level="5.9" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html"><i class="fa fa-check"></i><b>5.9</b> What does this mean to you?</a><ul>
<li class="chapter" data-level="5.9.1" data-path="what-does-this-mean-to-you.html"><a href="what-does-this-mean-to-you.html#with-this-information-can-can-now-answer-question-4"><i class="fa fa-check"></i><b>5.9.1</b> With this information, can can now answer <strong>Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.10" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html"><i class="fa fa-check"></i><b>5.10</b> Data Repositories for Sharing of Data</a><ul>
<li class="chapter" data-level="5.10.1" data-path="data-repositories-for-sharing-of-data.html"><a href="data-repositories-for-sharing-of-data.html#with-this-information-can-can-now-answer-question-5"><i class="fa fa-check"></i><b>5.10.1</b> With this information, can can now answer <strong>Question 5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="5.11" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html"><i class="fa fa-check"></i><b>5.11</b> Helpful Resources on FAIR</a><ul>
<li class="chapter" data-level="5.11.1" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-training-resources-on-fair"><i class="fa fa-check"></i><b>5.11.1</b> Additional Training Resources on FAIR</a></li>
<li class="chapter" data-level="5.11.2" data-path="helpful-resources-on-fair.html"><a href="helpful-resources-on-fair.html#additional-examples-and-documents-on-fair"><i class="fa fa-check"></i><b>5.11.2</b> Additional Examples and Documents on FAIR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="6" data-path="machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i><b>6</b> Machine Learning and Predictive Modeling for Environmental Health</a><ul>
<li class="chapter" data-level="6.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html"><i class="fa fa-check"></i><b>6.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="6.1.1" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#introduction-to-machine-learning-ml-and-predictive-modeling"><i class="fa fa-check"></i><b>6.1.1</b> Introduction to Machine Learning (ML) and Predictive Modeling</a></li>
<li class="chapter" data-level="6.1.2" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#defining-predictive-modeling-in-the-context-of-toxicology-and-environmental-health"><i class="fa fa-check"></i><b>6.1.2</b> Defining predictive modeling in the context of toxicology and environmental health</a></li>
<li class="chapter" data-level="6.1.3" data-path="background-on-training-module-1.html"><a href="background-on-training-module-1.html#distinguish-between-machine-learning-ml-and-traditional-statistical-methods"><i class="fa fa-check"></i><b>6.1.3</b> Distinguish between machine learning (ML) and traditional statistical methods</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="introduction-to-training-module-3.html"><a href="introduction-to-training-module-3.html"><i class="fa fa-check"></i><b>6.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="6.3" data-path="training-modules-environmental-health-questions.html"><a href="training-modules-environmental-health-questions.html"><i class="fa fa-check"></i><b>6.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="6.4" data-path="script-preparations-2.html"><a href="script-preparations-2.html"><i class="fa fa-check"></i><b>6.4</b> Script Preparations</a></li>
<li class="chapter" data-level="6.5" data-path="loading-the-example-dataset.html"><a href="loading-the-example-dataset.html"><i class="fa fa-check"></i><b>6.5</b> Loading the Example Dataset</a></li>
<li class="chapter" data-level="6.6" data-path="data-viewing.html"><a href="data-viewing.html"><i class="fa fa-check"></i><b>6.6</b> Data Viewing</a></li>
<li class="chapter" data-level="6.7" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html"><i class="fa fa-check"></i><b>6.7</b> Evaluating the Original Physicochemical Property Data across Substances</a><ul>
<li class="chapter" data-level="6.7.1" data-path="evaluating-the-original-physicochemical-property-data-across-substances.html"><a href="evaluating-the-original-physicochemical-property-data-across-substances.html#with-these-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>6.7.1</b> With these, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html"><i class="fa fa-check"></i><b>6.8</b> Identifying Clusters of Chemicals through k-means</a><ul>
<li class="chapter" data-level="6.8.1" data-path="identifying-clusters-of-chemicals-through-k-means.html"><a href="identifying-clusters-of-chemicals-through-k-means.html#with-this-we-can-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>6.8.1</b> With this, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html"><i class="fa fa-check"></i><b>6.9</b> Principal Component Analysis (PCA)</a><ul>
<li class="chapter" data-level="6.9.1" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-these-results-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>6.9.1</b> With these results, we can answer <strong>Environmental Health Question #3</strong>:</a></li>
<li class="chapter" data-level="6.9.2" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>6.9.2</b> With this, we can answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="6.9.3" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#with-this-we-can-answer-environmental-health-question-5"><i class="fa fa-check"></i><b>6.9.3</b> With this, we can answer <strong>Environmental Health Question #5</strong>:</a></li>
<li class="chapter" data-level="6.9.4" data-path="principal-component-analysis-pca.html"><a href="principal-component-analysis-pca.html#we-can-also-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>6.9.4</b> We can also answer <strong>Environmental Health Question #6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="concluding-remarks-5.html"><a href="concluding-remarks-5.html"><i class="fa fa-check"></i><b>6.10</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mixtures-analysis.html"><a href="mixtures-analysis.html"><i class="fa fa-check"></i><b>7</b> Mixtures Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html"><i class="fa fa-check"></i><b>7.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="7.1.1" data-path="background-on-training-module-2.html"><a href="background-on-training-module-2.html#introduction-to-mixtures-toxicology-and-in-silico-modeling-to-address-mixtures"><i class="fa fa-check"></i><b>7.1.1</b> Introduction to Mixtures Toxicology and <em>In Silico</em> Modeling to Address Mixtures</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="introduction-to-training-module-4.html"><a href="introduction-to-training-module-4.html"><i class="fa fa-check"></i><b>7.2</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="7.3" data-path="training-modules-environmental-health-questions-1.html"><a href="training-modules-environmental-health-questions-1.html"><i class="fa fa-check"></i><b>7.3</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="7.4" data-path="script-preparations-3.html"><a href="script-preparations-3.html"><i class="fa fa-check"></i><b>7.4</b> Script Preparations</a></li>
<li class="chapter" data-level="7.5" data-path="loading-the-example-ginkgo-biloba-datasets.html"><a href="loading-the-example-ginkgo-biloba-datasets.html"><i class="fa fa-check"></i><b>7.5</b> Loading the Example <em>Ginkgo biloba</em> Datasets</a></li>
<li class="chapter" data-level="7.6" data-path="data-viewing-1.html"><a href="data-viewing-1.html"><i class="fa fa-check"></i><b>7.6</b> Data Viewing</a><ul>
<li class="chapter" data-level="7.6.1" data-path="data-viewing-1.html"><a href="data-viewing-1.html#lets-also-see-what-kind-of-data-are-organized-within-the-datasets"><i class="fa fa-check"></i><b>7.6.1</b> Let’s also see what kind of data are organized within the datasets</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>7.7</b> Chemistry-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="7.7.1" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-chemical-profiles"><i class="fa fa-check"></i><b>7.7.1</b> This plot tells us a lot about sample groupings based on chemical profiles!</a></li>
<li class="chapter" data-level="7.7.2" data-path="chemistry-based-sufficient-similarity-analysis.html"><a href="chemistry-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-questions-1-2"><i class="fa fa-check"></i><b>7.7.2</b> With this, we can answer <strong>Environmental Health Questions 1-2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><i class="fa fa-check"></i><b>7.8</b> Creating a Heat Map of the Mixtures Chemistry Data</a><ul>
<li class="chapter" data-level="7.8.1" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#this-plot-tells-us-a-lot-about-the-individual-chemicals-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>7.8.1</b> This plot tells us a lot about the individual chemicals that differentiate the sample groupings</a></li>
<li class="chapter" data-level="7.8.2" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-this-we-can-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>7.8.2</b> With this, we can answer <strong>Environmental Health Question 3</strong>:</a></li>
<li class="chapter" data-level="7.8.3" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#lets-now-revisit-the-pca-plot"><i class="fa fa-check"></i><b>7.8.3</b> Let’s now revisit the PCA plot</a></li>
<li class="chapter" data-level="7.8.4" data-path="creating-a-heat-map-of-the-mixtures-chemistry-data.html"><a href="creating-a-heat-map-of-the-mixtures-chemistry-data.html#with-these-plots-side-by-side-we-can-now-answer-environmental-health-question-4"><i class="fa fa-check"></i><b>7.8.4</b> With these plots, side-by-side, we can now answer <strong>Environmental Health Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html"><i class="fa fa-check"></i><b>7.9</b> Toxicity-based Sufficient Similarity Analysis</a><ul>
<li class="chapter" data-level="7.9.1" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#this-plot-tells-us-a-lot-about-sample-groupings-based-on-toxicity-profiles"><i class="fa fa-check"></i><b>7.9.1</b> This plot tells us a lot about sample groupings based on toxicity profiles!</a></li>
<li class="chapter" data-level="7.9.2" data-path="toxicity-based-sufficient-similarity-analysis.html"><a href="toxicity-based-sufficient-similarity-analysis.html#with-this-we-can-answer-environmental-health-question-5-1"><i class="fa fa-check"></i><b>7.9.2</b> With this, we can answer <em>Environmental Health Question 5</em>:</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><i class="fa fa-check"></i><b>7.10</b> Creating a Heat Map of the Mixtures Toxicity Data</a><ul>
<li class="chapter" data-level="7.10.1" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#this-plot-tells-us-a-lot-about-the-individual-genes-that-differentiate-the-sample-groupings"><i class="fa fa-check"></i><b>7.10.1</b> This plot tells us a lot about the individual genes that differentiate the sample groupings</a></li>
<li class="chapter" data-level="7.10.2" data-path="creating-a-heat-map-of-the-mixtures-toxicity-data.html"><a href="creating-a-heat-map-of-the-mixtures-toxicity-data.html#with-this-we-can-answer-environmental-health-question-6"><i class="fa fa-check"></i><b>7.10.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><i class="fa fa-check"></i><b>7.11</b> Comparing Results from the Chemistry vs. Toxicity Sufficient Similarity Analyses</a><ul>
<li class="chapter" data-level="7.11.1" data-path="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html"><a href="comparing-results-from-the-chemistry-vs-toxicity-sufficient-similarity-analyses.html#visual-representation-of-some-of-the-major-grouping-similarities-vs-differences"><i class="fa fa-check"></i><b>7.11.1</b> Visual representation of some of the major grouping similarities vs differences</a></li>
</ul></li>
<li class="chapter" data-level="7.12" data-path="these-plots-can-help-us-answer-environmental-health-question-7.html"><a href="these-plots-can-help-us-answer-environmental-health-question-7.html"><i class="fa fa-check"></i><b>7.12</b> These plots can help us answer <strong>Environmental Health Question 7</strong>:</a></li>
<li class="chapter" data-level="7.13" data-path="concluding-remarks-6.html"><a href="concluding-remarks-6.html"><i class="fa fa-check"></i><b>7.13</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="omics-analyses-and-systems-biology.html"><a href="omics-analyses-and-systems-biology.html"><i class="fa fa-check"></i><b>8</b> -Omics Analyses and Systems Biology</a><ul>
<li class="chapter" data-level="8.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html"><i class="fa fa-check"></i><b>8.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="8.1.1" data-path="background-on-training-module-3.html"><a href="background-on-training-module-3.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i><b>8.1.1</b> Introduction to the Field of “-Omics”</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="introduction-to-the-omics-endpoint-transcriptomics.html"><a href="introduction-to-the-omics-endpoint-transcriptomics.html"><i class="fa fa-check"></i><b>8.2</b> Introduction to the -Omics Endpoint, Transcriptomics</a></li>
<li class="chapter" data-level="8.3" data-path="introduction-to-training-module-5.html"><a href="introduction-to-training-module-5.html"><i class="fa fa-check"></i><b>8.3</b> Introduction to Training Module</a></li>
<li class="chapter" data-level="8.4" data-path="training-modules-environmental-health-questions-2.html"><a href="training-modules-environmental-health-questions-2.html"><i class="fa fa-check"></i><b>8.4</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="8.5" data-path="script-preparations-4.html"><a href="script-preparations-4.html"><i class="fa fa-check"></i><b>8.5</b> Script Preparations</a></li>
<li class="chapter" data-level="8.6" data-path="loading-the-example-transcriptomic-dataset-and-metadata.html"><a href="loading-the-example-transcriptomic-dataset-and-metadata.html"><i class="fa fa-check"></i><b>8.6</b> Loading the Example Transcriptomic Dataset and Metadata</a></li>
<li class="chapter" data-level="8.7" data-path="data-viewing-2.html"><a href="data-viewing-2.html"><i class="fa fa-check"></i><b>8.7</b> Data Viewing</a><ul>
<li class="chapter" data-level="8.7.1" data-path="data-viewing-2.html"><a href="data-viewing-2.html#with-this-we-can-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>8.7.1</b> With this, we can answer <strong>Environmental Health Question 1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="formatting-data-for-downstream-statistics.html"><a href="formatting-data-for-downstream-statistics.html"><i class="fa fa-check"></i><b>8.8</b> Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="8.9" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html"><i class="fa fa-check"></i><b>8.9</b> Transcriptomics Data QA/QC</a><ul>
<li class="chapter" data-level="8.9.1" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#background-filter"><i class="fa fa-check"></i><b>8.9.1</b> Background Filter</a></li>
<li class="chapter" data-level="8.9.2" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#sample-filtering"><i class="fa fa-check"></i><b>8.9.2</b> Sample Filtering</a></li>
<li class="chapter" data-level="8.9.3" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#identifying-removing-sample-outliers"><i class="fa fa-check"></i><b>8.9.3</b> Identifying &amp; Removing Sample Outliers</a></li>
<li class="chapter" data-level="8.9.4" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#with-this-we-can-answer-environmental-health-question-2-1"><i class="fa fa-check"></i><b>8.9.4</b> With this, we can answer <strong>Environmental Health Question 2</strong>:</a></li>
<li class="chapter" data-level="8.9.5" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#we-can-also-answer-environmental-health-question-3"><i class="fa fa-check"></i><b>8.9.5</b> We can also answer <strong>Environmental Health Question 3</strong>:</a></li>
<li class="chapter" data-level="8.9.6" data-path="transcriptomics-data-qaqc.html"><a href="transcriptomics-data-qaqc.html#with-this-we-can-answer-environmental-health-question-4-1"><i class="fa fa-check"></i><b>8.9.6</b> With this, we can answer <strong>Environmental Health Question 4</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><i class="fa fa-check"></i><b>8.10</b> Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a><ul>
<li class="chapter" data-level="8.10.1" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-5-2"><i class="fa fa-check"></i><b>8.10.1</b> With this, we can answer <strong>Environmental Health Question 5</strong>:</a></li>
<li class="chapter" data-level="8.10.2" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-6-1"><i class="fa fa-check"></i><b>8.10.2</b> With this, we can answer <strong>Environmental Health Question 6</strong>:</a></li>
<li class="chapter" data-level="8.10.3" data-path="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html"><a href="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e-g-biomass-smoke-exposure.html#with-this-we-can-answer-environmental-health-question-7"><i class="fa fa-check"></i><b>8.10.3</b> With this, we can answer <strong>Environmental Health Question 7</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.11" data-path="visualizing-statistical-results-using-ma-plots.html"><a href="visualizing-statistical-results-using-ma-plots.html"><i class="fa fa-check"></i><b>8.11</b> Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="8.12" data-path="visualizing-statistical-results-using-volcano-plots.html"><a href="visualizing-statistical-results-using-volcano-plots.html"><i class="fa fa-check"></i><b>8.12</b> Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="8.13" data-path="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><i class="fa fa-check"></i><b>8.13</b> Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a><ul>
<li class="chapter" data-level="8.13.1" data-path="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html"><a href="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis.html#with-this-we-can-answer-environmental-health-question-8"><i class="fa fa-check"></i><b>8.13.1</b> With this, we can answer <strong>Environmental Health Question 8</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="8.14" data-path="concluding-remarks-7.html"><a href="concluding-remarks-7.html"><i class="fa fa-check"></i><b>8.14</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i><b>9</b> Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="10" data-path="read-across-toxicity-predictions.html"><a href="read-across-toxicity-predictions.html"><i class="fa fa-check"></i><b>10</b> Read-Across Toxicity Predictions</a></li>
<li class="chapter" data-level="11" data-path="environmental-health-databases-comparative-toxicogenomics-database.html"><a href="environmental-health-databases-comparative-toxicogenomics-database.html"><i class="fa fa-check"></i><b>11</b> Environmental Health Databases: Comparative Toxicogenomics Database</a><ul>
<li class="chapter" data-level="11.1" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html"><i class="fa fa-check"></i><b>11.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="11.1.1" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html#introduction-to-comparative-toxicogenomics-database-ctd"><i class="fa fa-check"></i><b>11.1.1</b> Introduction to Comparative Toxicogenomics Database (CTD)</a></li>
<li class="chapter" data-level="11.1.2" data-path="background-on-training-module-4.html"><a href="background-on-training-module-4.html#introduction-to-training-module-6"><i class="fa fa-check"></i><b>11.1.2</b> Introduction to Training Module</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="training-modules-environmental-health-questions-3.html"><a href="training-modules-environmental-health-questions-3.html"><i class="fa fa-check"></i><b>11.2</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="11.3" data-path="script-preparations-5.html"><a href="script-preparations-5.html"><i class="fa fa-check"></i><b>11.3</b> Script Preparations</a></li>
<li class="chapter" data-level="11.4" data-path="organizing-example-dataset-from-ctd.html"><a href="organizing-example-dataset-from-ctd.html"><i class="fa fa-check"></i><b>11.4</b> Organizing Example Dataset from CTD</a></li>
<li class="chapter" data-level="11.5" data-path="loading-the-example-ctd-dataset-into-r.html"><a href="loading-the-example-ctd-dataset-into-r.html"><i class="fa fa-check"></i><b>11.5</b> Loading the Example CTD Dataset into R</a></li>
<li class="chapter" data-level="11.6" data-path="data-viewing-3.html"><a href="data-viewing-3.html"><i class="fa fa-check"></i><b>11.6</b> Data Viewing</a></li>
<li class="chapter" data-level="11.7" data-path="filtering-the-data-for-genes-with-altered-expression.html"><a href="filtering-the-data-for-genes-with-altered-expression.html"><i class="fa fa-check"></i><b>11.7</b> Filtering the Data for Genes with Altered Expression</a><ul>
<li class="chapter" data-level="11.7.1" data-path="filtering-the-data-for-genes-with-altered-expression.html"><a href="filtering-the-data-for-genes-with-altered-expression.html#with-this-we-can-answer-environmental-health-question-1-1"><i class="fa fa-check"></i><b>11.7.1</b> With this, we can answer <strong>Environmental Health Question #1</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="identifying-genes-under-epigenetic-control.html"><a href="identifying-genes-under-epigenetic-control.html"><i class="fa fa-check"></i><b>11.8</b> Identifying Genes under Epigenetic Control</a></li>
<li class="chapter" data-level="11.9" data-path="method-1-for-list-comparisons-merging.html"><a href="method-1-for-list-comparisons-merging.html"><i class="fa fa-check"></i><b>11.9</b> Method 1 for list comparisons: Merging</a><ul>
<li class="chapter" data-level="11.9.1" data-path="method-1-for-list-comparisons-merging.html"><a href="method-1-for-list-comparisons-merging.html#with-this-we-can-answer-environmental-health-question-2-2"><i class="fa fa-check"></i><b>11.9.1</b> With this, we can answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="11.10" data-path="method-2-for-list-comparisons-intersection.html"><a href="method-2-for-list-comparisons-intersection.html"><i class="fa fa-check"></i><b>11.10</b> Method 2 for list comparisons: Intersection</a></li>
<li class="chapter" data-level="11.11" data-path="concluding-remarks-8.html"><a href="concluding-remarks-8.html"><i class="fa fa-check"></i><b>11.11</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="environmental-health-databases-gene-expression-omnibus.html"><a href="environmental-health-databases-gene-expression-omnibus.html"><i class="fa fa-check"></i><b>12</b> Environmental Health Databases: Gene Expression Omnibus</a><ul>
<li class="chapter" data-level="12.1" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html"><i class="fa fa-check"></i><b>12.1</b> Background on Training Module</a><ul>
<li class="chapter" data-level="12.1.1" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html#introduction-to-the-environmental-health-database-gene-expression-omnibus-geo"><i class="fa fa-check"></i><b>12.1.1</b> Introduction to the Environmental Health Database, Gene Expression Omnibus (GEO)</a></li>
<li class="chapter" data-level="12.1.2" data-path="background-on-training-module-5.html"><a href="background-on-training-module-5.html#introduction-to-training-module-7"><i class="fa fa-check"></i><b>12.1.2</b> Introduction to Training Module</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="training-modules-environmental-health-questions-4.html"><a href="training-modules-environmental-health-questions-4.html"><i class="fa fa-check"></i><b>12.2</b> Training Module’s <strong>Environmental Health Questions</strong></a></li>
<li class="chapter" data-level="12.3" data-path="script-preparations-6.html"><a href="script-preparations-6.html"><i class="fa fa-check"></i><b>12.3</b> Script Preparations</a></li>
<li class="chapter" data-level="12.4" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html"><i class="fa fa-check"></i><b>12.4</b> Loading and Organizing the Example Dataset</a><ul>
<li class="chapter" data-level="12.4.1" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html#manually-downloading-and-uploading-geo-files"><i class="fa fa-check"></i><b>12.4.1</b> 1. Manually Downloading and Uploading GEO Files</a></li>
<li class="chapter" data-level="12.4.2" data-path="loading-and-organizing-the-example-dataset.html"><a href="loading-and-organizing-the-example-dataset.html#uploading-and-organizing-geo-files-through-the-geoquery-package"><i class="fa fa-check"></i><b>12.4.2</b> 2. Uploading and Organizing GEO Files through the GEOquery Package</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><i class="fa fa-check"></i><b>12.5</b> Adding Gene Symbol Information to -Omic Data Sets through Platform Annotation Files</a><ul>
<li class="chapter" data-level="12.5.1" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html#with-this-we-can-now-answer-environmental-health-question-1"><i class="fa fa-check"></i><b>12.5.1</b> With this, we can now answer <strong>Environmental Health Question #1</strong>:</a></li>
<li class="chapter" data-level="12.5.2" data-path="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html"><a href="adding-gene-symbol-information-to-omic-data-sets-through-platform-annotation-files.html#we-can-also-answer-environmental-health-question-2"><i class="fa fa-check"></i><b>12.5.2</b> We can also answer <strong>Environmental Health Question #2</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="visualizing-gene-expression-data-using-boxplots-and-heat-maps.html"><a href="visualizing-gene-expression-data-using-boxplots-and-heat-maps.html"><i class="fa fa-check"></i><b>12.6</b> Visualizing Gene Expression Data using Boxplots and Heat Maps</a></li>
<li class="chapter" data-level="12.7" data-path="boxplot-visualizations-1.html"><a href="boxplot-visualizations-1.html"><i class="fa fa-check"></i><b>12.7</b> Boxplot Visualizations</a></li>
<li class="chapter" data-level="12.8" data-path="heatmap-visualizations-1.html"><a href="heatmap-visualizations-1.html"><i class="fa fa-check"></i><b>12.8</b> Heatmap Visualizations</a><ul>
<li class="chapter" data-level="12.8.1" data-path="heatmap-visualizations-1.html"><a href="heatmap-visualizations-1.html#we-can-also-answer-environmental-health-question-3-1"><i class="fa fa-check"></i><b>12.8.1</b> We can also answer <strong>Environmental Health Question #3</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="statistical-analyses-to-identify-genes-altered-by-formaldehyde.html"><a href="statistical-analyses-to-identify-genes-altered-by-formaldehyde.html"><i class="fa fa-check"></i><b>12.9</b> Statistical Analyses to Identify Genes altered by Formaldehyde</a></li>
<li class="chapter" data-level="12.10" data-path="interpreting-results.html"><a href="interpreting-results.html"><i class="fa fa-check"></i><b>12.10</b> Interpreting Results</a><ul>
<li class="chapter" data-level="12.10.1" data-path="interpreting-results.html"><a href="interpreting-results.html#with-this-we-can-answer-environmental-health-question-4-2"><i class="fa fa-check"></i><b>12.10.1</b> With this, we can answer <strong>Environmental Health Question #4</strong>:</a></li>
<li class="chapter" data-level="12.10.2" data-path="interpreting-results.html"><a href="interpreting-results.html#with-this-we-can-answer-environmental-health-question-5-3"><i class="fa fa-check"></i><b>12.10.2</b> With this, we can answer <strong>Environmental Health Question #5</strong>:</a></li>
</ul></li>
<li class="chapter" data-level="12.11" data-path="concluding-remarks-9.html"><a href="concluding-remarks-9.html"><i class="fa fa-check"></i><b>12.11</b> Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="database-integration-air-quality-study-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-study-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i><b>13</b> Database Integration: Air Quality Study, Mortality, and Environmental Justice Data</a></li>
<li class="chapter" data-level="14" data-path="additional-resources.html"><a href="additional-resources.html"><i class="fa fa-check"></i><b>14</b> Additional Resources</a><ul>
<li class="chapter" data-level="14.1" data-path="r-programming-resources.html"><a href="r-programming-resources.html"><i class="fa fa-check"></i><b>14.1</b> R Programming Resources:</a></li>
<li class="chapter" data-level="14.2" data-path="r-packages-resources.html"><a href="r-packages-resources.html"><i class="fa fa-check"></i><b>14.2</b> R Packages Resources:</a></li>
<li class="chapter" data-level="14.3" data-path="community-discussions-on-r-and-r-packages.html"><a href="community-discussions-on-r-and-r-packages.html"><i class="fa fa-check"></i><b>14.3</b> Community Discussions on R and R Packages:</a></li>
<li class="chapter" data-level="14.4" data-path="data-science-and-statistical-analysis-resources.html"><a href="data-science-and-statistical-analysis-resources.html"><i class="fa fa-check"></i><b>14.4</b> Data Science and Statistical Analysis Resources:</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="principal-component-analysis-pca" class="section level2">
<h2><span class="header-section-number">6.9</span> Principal Component Analysis (PCA)</h2>
<p>Next, we will run through some example analyses applying the common data reduction technique of PCA.</p>
<p>We can calculate the principal components across ALL physicochemical data across all chemicals using the princomp function</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="principal-component-analysis-pca.html#cb34-1"></a>my.pca &lt;-<span class="st"> </span><span class="kw">princomp</span>(dat.x,   <span class="co"># input dataframe of physchem data</span></span>
<span id="cb34-2"><a href="principal-component-analysis-pca.html#cb34-2"></a>                   <span class="dt">cor =</span> T) <span class="co"># calculations will be based on the correlation matrix (as opposed to covariance) since we have all numeric values here (default PCA option)</span></span></code></pre></div>
<p>Here are the resulting scores for each chemical’s contribution towards each principal component (shown here as components 1-10)</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="principal-component-analysis-pca.html#cb35-1"></a><span class="kw">head</span>(my.pca<span class="op">$</span>scores)</span></code></pre></div>
<pre><code>##                Comp.1     Comp.2      Comp.3      Comp.4      Comp.5     Comp.6
## 93762-09-5 -2.0425355 -1.4875982 -1.29779776 -0.04882877  0.25393797 -0.6799177
## 60270-55-5 -1.2291769  2.2936873  0.24345932  0.40280922  0.63000240 -1.0186985
## 70225-15-9 -1.0982561  1.3963638  0.03352018  0.90707254  0.05756006  0.1438501
## 335-24-0   -1.1374460  1.0712815 -0.14349891  1.09092722  0.21246867 -0.9427527
## 647-29-0   -0.4847481  0.1264224  1.16553341 -1.11771990 -0.29674860  0.1924128
## 68259-12-1 -0.3276157  0.2377300  1.32445577 -0.47677888 -1.17966092  0.0593078
##                 Comp.7      Comp.8       Comp.9     Comp.10
## 93762-09-5  0.14597268  1.25959099  0.231742917 -0.14124625
## 60270-55-5  0.11356003 -0.34454904 -0.385021331 -0.09883538
## 70225-15-9 -0.38489641  0.01723486 -0.006725509  0.02725202
## 335-24-0    0.22957369  0.11497271 -0.108096107 -0.17762819
## 647-29-0    0.18292023 -0.48181130  0.075229509 -0.22829905
## 68259-12-1 -0.01404007  0.03803686  0.043460416  0.18095023</code></pre>
<p>And the resulting loading factors of each property’s contribution towards each principal component</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="principal-component-analysis-pca.html#cb37-1"></a><span class="kw">head</span>(my.pca<span class="op">$</span>loadings)</span></code></pre></div>
<pre><code>##                                                        Comp.1       Comp.2
## Molecular.Weight                                   0.09825313  0.108454961
## OPERA..Boiling.Point                               0.46350428  0.029650863
## OPERA..Henry.s.Law.Constant                       -0.17856542 -0.502116638
## OPERA..Melting.Point                               0.20645719  0.474473487
## OPERA..Negative.Log.of.Acid.Dissociation.Constant  0.32172963 -0.119465105
## OPERA..Octanol.Air.Partition.Coefficient           0.45329804 -0.008918089
##                                                       Comp.3      Comp.4
## Molecular.Weight                                   0.6797404  0.48432419
## OPERA..Boiling.Point                              -0.1993659 -0.03108544
## OPERA..Henry.s.Law.Constant                       -0.1798767  0.27695374
## OPERA..Melting.Point                               0.2148579  0.09449999
## OPERA..Negative.Log.of.Acid.Dissociation.Constant -0.2862395  0.58268278
## OPERA..Octanol.Air.Partition.Coefficient          -0.1321577 -0.04820475
##                                                        Comp.5      Comp.6
## Molecular.Weight                                   0.17351578  0.35736795
## OPERA..Boiling.Point                               0.22224554 -0.01850753
## OPERA..Henry.s.Law.Constant                        0.30566003 -0.47066669
## OPERA..Melting.Point                              -0.08063905 -0.68672356
## OPERA..Negative.Log.of.Acid.Dissociation.Constant -0.66998767  0.02924804
## OPERA..Octanol.Air.Partition.Coefficient           0.20778895  0.20575789
##                                                        Comp.7      Comp.8
## Molecular.Weight                                   0.11763362  0.32938640
## OPERA..Boiling.Point                               0.12503355  0.09718690
## OPERA..Henry.s.Law.Constant                        0.21138163  0.44526650
## OPERA..Melting.Point                               0.34342931 -0.10233816
## OPERA..Negative.Log.of.Acid.Dissociation.Constant -0.09083446  0.03113686
## OPERA..Octanol.Air.Partition.Coefficient           0.44434707 -0.29734602
##                                                        Comp.9     Comp.10
## Molecular.Weight                                   0.03295675  0.02698233
## OPERA..Boiling.Point                               0.03336277  0.81709497
## OPERA..Henry.s.Law.Constant                        0.19706729 -0.10099077
## OPERA..Melting.Point                              -0.24532148 -0.10229774
## OPERA..Negative.Log.of.Acid.Dissociation.Constant  0.02576652 -0.03380215
## OPERA..Octanol.Air.Partition.Coefficient           0.49672303 -0.39565984</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="principal-component-analysis-pca.html#cb39-1"></a>my.pca<span class="op">$</span>loadings</span></code></pre></div>
<pre><code>## 
## Loadings:
##                                                   Comp.1 Comp.2 Comp.3 Comp.4
## Molecular.Weight                                          0.108  0.680  0.484
## OPERA..Boiling.Point                               0.464        -0.199       
## OPERA..Henry.s.Law.Constant                       -0.179 -0.502 -0.180  0.277
## OPERA..Melting.Point                               0.206  0.474  0.215       
## OPERA..Negative.Log.of.Acid.Dissociation.Constant  0.322 -0.119 -0.286  0.583
## OPERA..Octanol.Air.Partition.Coefficient           0.453        -0.132       
## OPERA..Octanol.Water.Distribution.Coefficient      0.330 -0.437        -0.151
## OPERA..Octanol.Water.Partition.Coefficient         0.162 -0.343  0.467 -0.485
## OPERA..Vapor.Pressure                             -0.352 -0.350  0.195  0.250
## OPERA..Water.Solubility                           -0.365  0.255 -0.254 -0.130
##                                                   Comp.5 Comp.6 Comp.7 Comp.8
## Molecular.Weight                                   0.174  0.357  0.118  0.329
## OPERA..Boiling.Point                               0.222         0.125       
## OPERA..Henry.s.Law.Constant                        0.306 -0.471  0.211  0.445
## OPERA..Melting.Point                                     -0.687  0.343 -0.102
## OPERA..Negative.Log.of.Acid.Dissociation.Constant -0.670                     
## OPERA..Octanol.Air.Partition.Coefficient           0.208  0.206  0.444 -0.297
## OPERA..Octanol.Water.Distribution.Coefficient             0.169  0.220       
## OPERA..Octanol.Water.Partition.Coefficient        -0.485 -0.162         0.178
## OPERA..Vapor.Pressure                                            0.350 -0.654
## OPERA..Water.Solubility                           -0.291  0.284  0.652  0.350
##                                                   Comp.9 Comp.10
## Molecular.Weight                                                
## OPERA..Boiling.Point                                      0.817 
## OPERA..Henry.s.Law.Constant                        0.197 -0.101 
## OPERA..Melting.Point                              -0.245 -0.102 
## OPERA..Negative.Log.of.Acid.Dissociation.Constant               
## OPERA..Octanol.Air.Partition.Coefficient           0.497 -0.396 
## OPERA..Octanol.Water.Distribution.Coefficient     -0.744 -0.199 
## OPERA..Octanol.Water.Partition.Coefficient         0.306        
## OPERA..Vapor.Pressure                                     0.313 
## OPERA..Water.Solubility                                         
## 
##                Comp.1 Comp.2 Comp.3 Comp.4 Comp.5 Comp.6 Comp.7 Comp.8 Comp.9
## SS loadings       1.0    1.0    1.0    1.0    1.0    1.0    1.0    1.0    1.0
## Proportion Var    0.1    0.1    0.1    0.1    0.1    0.1    0.1    0.1    0.1
## Cumulative Var    0.1    0.2    0.3    0.4    0.5    0.6    0.7    0.8    0.9
##                Comp.10
## SS loadings        1.0
## Proportion Var     0.1
## Cumulative Var     1.0</code></pre>
<div id="with-these-results-we-can-answer-environmental-health-question-3" class="section level3">
<h3><span class="header-section-number">6.9.1</span> With these results, we can answer <strong>Environmental Health Question #3</strong>:</h3>
<div id="upon-reducing-the-data-through-pca-which-physicochemical-property-contributes-the-most-towards-informing-data-variance-captured-in-the-primary-principal-component-comp.1" class="section level4">
<h4><span class="header-section-number">6.9.1.1</span> (3) Upon reducing the data through PCA, which physicochemical property contributes the most towards informing data variance captured in the primary principal component (Comp.1)?</h4>
</div>
<div id="answer-boiling-point-contributes-the-most-towards-principal-component-1." class="section level4">
<h4><span class="header-section-number">6.9.1.2</span> <em>Answer: Boiling point contributes the most towards principal component #1.</em></h4>
<p><br></p>
</div>
<div id="calculating-of-variance-captured-by-each-principal-component" class="section level4">
<h4><span class="header-section-number">6.9.1.3</span> Calculating % of Variance Captured by each Principal Component</h4>
<p>We can view summary statistics describing how much of the variance from the original dataset was captured by each component, using the summary function.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="principal-component-analysis-pca.html#cb41-1"></a><span class="kw">summary</span>(my.pca)</span></code></pre></div>
<pre><code>## Importance of components:
##                           Comp.1    Comp.2    Comp.3     Comp.4    Comp.5
## Standard deviation     2.0230157 1.5440214 1.2415840 0.76635873 0.6905932
## Proportion of Variance 0.4092593 0.2384002 0.1541531 0.05873057 0.0476919
## Cumulative Proportion  0.4092593 0.6476595 0.8018125 0.86054312 0.9082350
##                            Comp.6     Comp.7     Comp.8     Comp.9     Comp.10
## Standard deviation     0.60491164 0.48939394 0.40589919 0.32548349 0.203793303
## Proportion of Variance 0.03659181 0.02395064 0.01647542 0.01059395 0.004153171
## Cumulative Proportion  0.94482682 0.96877746 0.98525288 0.99584683 1.000000000</code></pre>
<p>We can also calculate these values, and pull them into a dataframe for future use.</p>
<p>For example, to pull the percentage of variance explained by each principal component, we can run the following calculations, where first eigenvalues (eigs) are calculated and then used to calculate percent of variance, per principal component:</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="principal-component-analysis-pca.html#cb43-1"></a>eigs &lt;-<span class="st"> </span>my.pca<span class="op">$</span>sdev<span class="op">^</span><span class="dv">2</span></span>
<span id="cb43-2"><a href="principal-component-analysis-pca.html#cb43-2"></a>Comp.stats &lt;-<span class="st"> </span><span class="kw">data.frame</span>(eigs, eigs<span class="op">/</span><span class="kw">sum</span>(eigs), <span class="dt">row.names=</span><span class="kw">names</span>(eigs))</span>
<span id="cb43-3"><a href="principal-component-analysis-pca.html#cb43-3"></a><span class="kw">colnames</span>(Comp.stats) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Eigen_Values&quot;</span>, <span class="st">&quot;Percent_of_Variance&quot;</span>)</span>
<span id="cb43-4"><a href="principal-component-analysis-pca.html#cb43-4"></a><span class="kw">head</span>(Comp.stats)</span></code></pre></div>
<pre><code>##        Eigen_Values Percent_of_Variance
## Comp.1    4.0925925          0.40925925
## Comp.2    2.3840022          0.23840022
## Comp.3    1.5415308          0.15415308
## Comp.4    0.5873057          0.05873057
## Comp.5    0.4769190          0.04769190
## Comp.6    0.3659181          0.03659181</code></pre>
<p>Here, we can see that Principal Component #1 (Comp.1) captures ~41% of the variance across all physicochemical property values, across all chemicals. Principal Component #2 captures ~24%, etc.</p>
<p><br></p>
</div>
<div id="visualizing-pca-results" class="section level4">
<h4><span class="header-section-number">6.9.1.4</span> Visualizing PCA Results</h4>
<p>Let’s now view the results of this PCA, focusing on the first two principal components, and coloring each chemical according to class (i.e. PFAS vs statins).</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="principal-component-analysis-pca.html#cb45-1"></a><span class="kw">ggplot</span>(<span class="kw">as.data.frame</span>(my.pca<span class="op">$</span>scores), <span class="kw">aes</span>(<span class="dt">x=</span>Comp<span class="fl">.1</span>, <span class="dt">y=</span>Comp<span class="fl">.2</span>, <span class="dt">color=</span><span class="kw">as.factor</span>(dat<span class="op">$</span>List))) <span class="op">+</span><span class="st"> </span></span>
<span id="cb45-2"><a href="principal-component-analysis-pca.html#cb45-2"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">4</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb45-3"><a href="principal-component-analysis-pca.html#cb45-3"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&#39;Version C: PCA Plot of the First 2 PCs, colored by Chemical Class&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb45-4"><a href="principal-component-analysis-pca.html#cb45-4"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Principal Component 1&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Principal Component 2&quot;</span>)</span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
</div>
</div>
<div id="with-this-we-can-answer-environmental-health-question-4" class="section level3">
<h3><span class="header-section-number">6.9.2</span> With this, we can answer <strong>Environmental Health Question #4</strong>:</h3>
<div id="how-do-the-data-compare-when-physicochemical-properties-are-reduced-using-pca" class="section level4">
<h4><span class="header-section-number">6.9.2.1</span> (4) How do the data compare when physicochemical properties are reduced using PCA?</h4>
</div>
<div id="answer-data-become-more-compressed-and-variables-reduce-across-principal-components-capturing-the-majority-of-variance.-this-results-in-improved-data-visualizations-where-all-dimensions-of-the-physiochemical-dataset-are-compressed-and-captured-across-the-displayed-components." class="section level4">
<h4><span class="header-section-number">6.9.2.2</span> <em>Answer: Data become more compressed, and variables reduce across principal components capturing the majority of variance. This results in improved data visualizations, where all dimensions of the physiochemical dataset are compressed and captured across the displayed components.</em></h4>
<p><br></p>
</div>
<div id="incorporating-k-means-into-pca-for-predictive-modeling" class="section level4">
<h4><span class="header-section-number">6.9.2.3</span> Incorporating K-Means into PCA for Predictive Modeling</h4>
<p>We can also identify cluster-based trends within data that are reduced, after running PCA. This example analysis does so, expanding upon the previously generated PCA results.</p>
</div>
<div id="estimate-k-means-clusters-from-pca-results" class="section level4">
<h4><span class="header-section-number">6.9.2.4</span> Estimate k-means clusters from PCA results</h4>
<p>Let’s first run script, similar to the previous k-means analysis and associated parameters, though instead here we will use data reduced values from the PCA analysis. Specifically, clusters across PCA “scores” values will be derived, where scores represent the relative amount each chemical contributed to each principal component.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="principal-component-analysis-pca.html#cb46-1"></a>clusters_PCA &lt;-<span class="st"> </span><span class="kw">kmeans</span>(my.pca<span class="op">$</span>scores, <span class="dt">centers =</span> num.centers, <span class="dt">iter.max =</span> <span class="dv">1000</span>, <span class="dt">nstart =</span> <span class="dv">50</span>)</span></code></pre></div>
<p>The resulting PCA score values that were derived as the final cluster centers can be pulled using:</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="principal-component-analysis-pca.html#cb47-1"></a>clusters_PCA<span class="op">$</span>centers</span></code></pre></div>
<pre><code>##      Comp.1     Comp.2     Comp.3       Comp.4      Comp.5      Comp.6
## 1  2.621672 -0.3065303 -0.6241284  0.008849520  0.18554906  0.05058781
## 2 -1.191669  0.1393319  0.2836947 -0.004022509 -0.08434048 -0.02299446
##        Comp.7      Comp.8       Comp.9      Comp.10
## 1  0.03429111 -0.05288759 -0.018394982  0.011945853
## 2 -0.01558687  0.02403981  0.008361355 -0.005429933</code></pre>
<p>Viewing the final cluster assignment, per chemical:</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="principal-component-analysis-pca.html#cb49-1"></a><span class="kw">head</span>(<span class="kw">cbind</span>(<span class="kw">rownames</span>(dat.x),clusters_PCA<span class="op">$</span>cluster))</span></code></pre></div>
<pre><code>##            [,1]         [,2]
## 93762-09-5 &quot;93762-09-5&quot; &quot;2&quot; 
## 60270-55-5 &quot;60270-55-5&quot; &quot;2&quot; 
## 70225-15-9 &quot;70225-15-9&quot; &quot;2&quot; 
## 335-24-0   &quot;335-24-0&quot;   &quot;2&quot; 
## 647-29-0   &quot;647-29-0&quot;   &quot;2&quot; 
## 68259-12-1 &quot;68259-12-1&quot; &quot;2&quot;</code></pre>
</div>
<div id="visualizing-k-means-clusters-from-pca-results" class="section level4">
<h4><span class="header-section-number">6.9.2.5</span> Visualizing k-means clusters from PCA results</h4>
<p>Let’s now view, again, the results of the main PCA, focusing on the first two principal components; though this time let’s color each chemical according to k-means cluster.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="principal-component-analysis-pca.html#cb51-1"></a><span class="kw">ggplot</span>(<span class="kw">as.data.frame</span>(my.pca<span class="op">$</span>scores), <span class="kw">aes</span>(<span class="dt">x=</span>Comp<span class="fl">.1</span>, <span class="dt">y=</span>Comp<span class="fl">.2</span>, <span class="dt">color=</span><span class="kw">as.factor</span>(clusters_PCA<span class="op">$</span>cluster))) <span class="op">+</span><span class="st"> </span></span>
<span id="cb51-2"><a href="principal-component-analysis-pca.html#cb51-2"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">4</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb51-3"><a href="principal-component-analysis-pca.html#cb51-3"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&#39;Version D: PCA Plot of the First 2 PCs, colored by k-means Clustering&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb51-4"><a href="principal-component-analysis-pca.html#cb51-4"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Principal Component 1&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Principal Component 2&quot;</span>)</span></code></pre></div>
<p><img src="TAME_Toolkit_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
</div>
</div>
<div id="with-this-we-can-answer-environmental-health-question-5" class="section level3">
<h3><span class="header-section-number">6.9.3</span> With this, we can answer <strong>Environmental Health Question #5</strong>:</h3>
<div id="if-we-did-not-have-information-telling-us-which-chemical-belonged-to-which-class-could-we-use-pca-and-k-means-to-accurately-predict-whether-a-chemical-is-a-pfas-vs-statin" class="section level4">
<h4><span class="header-section-number">6.9.3.1</span> (5) If we did not have information telling us which chemical belonged to which class, could we use PCA and k-means to accurately predict whether a chemical is a PFAS vs statin?</h4>
</div>
<div id="answer-yes-groupings-derived-from-k-means-displayed-in-this-pca-plot-line-up-almost-exactly-with-the-grouping-of-chemical-classes-see-version-c-of-this-plot-as-the-direct-comparison." class="section level4">
<h4><span class="header-section-number">6.9.3.2</span> <em>Answer: Yes!! Groupings derived from k-means, displayed in this PCA plot, line up almost exactly with the grouping of chemical classes (see Version C of this plot as the direct comparison).</em></h4>
</div>
</div>
<div id="we-can-also-answer-environmental-health-question-6" class="section level3">
<h3><span class="header-section-number">6.9.4</span> We can also answer <strong>Environmental Health Question #6</strong>:</h3>
<div id="what-kinds-of-applicationsendpoints-can-be-better-understood-andor-predicted-because-of-these-derived-chemical-groupings" class="section level4">
<h4><span class="header-section-number">6.9.4.1</span> (6) What kinds of applications/endpoints can be better understood and/or predicted, because of these derived chemical groupings?</h4>
</div>
<div id="answers" class="section level4">
<h4><span class="header-section-number">6.9.4.2</span> <em>Answers</em>:</h4>
<ul>
<li><em>With these well-informed chemical groupings, we can now better understand the variables that attribute to the chemical classifications.</em><br />
</li>
<li><em>We can also use this information to better understand data trends, and predict environmental fate and transport for these chemicals.</em><br />
</li>
<li><em>The reduced variables derived through PCA, and/or k-means clustering patterns can also be used as input variables to predict toxicological outcomes.</em></li>
</ul>
<p><br></p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="identifying-clusters-of-chemicals-through-k-means.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="concluding-remarks-5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["TAME_Toolkit.pdf", "TAME_Toolkit.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
